diff -ru src/mercurial/scmutil.py b/mercurial/scmutil.py
--- src/mercurial/scmutil.py	2012-06-02 06:48:20.000000000 +0200
+++ b/mercurial/scmutil.py	2012-06-28 12:08:03.313383534 +0200
@@ -453,7 +453,10 @@
         if len(getattr(sys, 'argv', [])) > 0:
             p = os.path.dirname(os.path.dirname(sys.argv[0]))
             path.extend(rcfiles(os.path.join(p, root)))
-        path.extend(rcfiles('/' + root))
+        # Disable /etc/mercurial for Sage's mercurial (because it is
+        # likely to refer to extensions that are not installed in
+        # Sage's mercurial).
+        #path.extend(rcfiles('/' + root))
         return path
 
     def userrcpath():
