--- ../src/mercurial/posix.py	2009-09-08 15:47:01.000000000 -0700
+++ posix.py	2009-09-08 15:47:31.000000000 -0700
@@ -39,7 +39,10 @@
     if len(getattr(sys, 'argv', [])) > 0:
         path.extend(rcfiles(os.path.dirname(sys.argv[0]) +
                               '/../etc/mercurial'))
-    path.extend(rcfiles('/etc/mercurial'))
+    # Disable /etc/mercurial for Sage's mercurial (because it is
+    # likely to refer to extensions that are not installed in
+    # Sage's mercurial).
+    # path.extend(rcfiles('/etc/mercurial'))
     return path
 
 def user_rcpath():
