--- ../src/mercurial/posix.py	2011-01-11 16:02:49.306666670 -0600
+++ posix.py	2011-01-11 16:16:15.443333336 -0600
@@ -40,7 +40,10 @@
     if len(getattr(sys, 'argv', [])) > 0:
         path.extend(rcfiles(os.path.dirname(sys.argv[0]) +
                               '/../etc/mercurial'))
-    path.extend(rcfiles('/etc/mercurial'))
+    # Disable /etc/mercurial for Sage's mercurial (because it is
+    # likely to refer to extensions that are not installed in
+    # Sage's mercurial).
+    # path.extend(rcfiles('/etc/mercurial'))
     return path
 
 def user_rcpath():
