diff --git a/../src/mercurial/posix.py b/posix.py
index 04e2527..da548e5 100644
--- a/../src/mercurial/posix.py
+++ b/posix.py
@@ -46,7 +46,10 @@ def system_rcpath():
     if len(getattr(sys, 'argv', [])) > 0:
         path.extend(rcfiles(os.path.dirname(sys.argv[0]) +
                               '/../etc/mercurial'))
-    path.extend(rcfiles('/etc/mercurial'))
+    # Disable /etc/mercurial for Sage's mercurial (because it is
+    # likely to refer to extensions that are not installed in
+    # Sage's mercurial).
+    # path.extend(rcfiles('/etc/mercurial'))
     return path
 
 def user_rcpath():
