--- ../src/setup.py	2009-07-23 10:58:06.000000000 -0700
+++ setup.py	2009-09-08 16:04:06.000000000 -0700
@@ -236,14 +236,14 @@
 packages = ['mercurial', 'mercurial.hgweb', 'hgext', 'hgext.convert',
             'hgext.highlight', 'hgext.zeroconf', ]
 
-if sys.platform == 'linux2' and os.uname()[2] > '2.6':
-    # The inotify extension is only usable with Linux 2.6 kernels.
-    # You also need a reasonably recent C library.
-    cc = new_compiler()
-    if has_function(cc, 'inotify_add_watch'):
-        ext_modules.append(Extension('hgext.inotify.linux._inotify',
-                                     ['hgext/inotify/linux/_inotify.c']))
-        packages.extend(['hgext.inotify', 'hgext.inotify.linux'])
+## if sys.platform == 'linux2' and os.uname()[2] > '2.6':
+##     # The inotify extension is only usable with Linux 2.6 kernels.
+##     # You also need a reasonably recent C library.
+##     cc = new_compiler()
+##     if has_function(cc, 'inotify_add_watch'):
+##         ext_modules.append(Extension('hgext.inotify.linux._inotify',
+##                                      ['hgext/inotify/linux/_inotify.c']))
+##         packages.extend(['hgext.inotify', 'hgext.inotify.linux'])
 
 datafiles = []
 for root in ('templates', 'i18n'):
