--- setup.py.orig	2008-08-15 06:50:35.676109000 -0400
+++ setup.py	2008-08-15 06:50:27.242722000 -0400
@@ -96,21 +96,6 @@
 
 packages = ['mercurial', 'mercurial.hgweb', 'hgext', 'hgext.convert']
 
-try:
-    import posix
-    ext_modules.append(Extension('mercurial.osutil', ['mercurial/osutil.c']))
-
-    if sys.platform == 'linux2' and os.uname()[2] > '2.6':
-        # The inotify extension is only usable with Linux 2.6 kernels.
-        # You also need a reasonably recent C library.
-        cc = new_compiler()
-        if has_function(cc, 'inotify_add_watch'):
-            ext_modules.append(Extension('hgext.inotify.linux._inotify',
-                                         ['hgext/inotify/linux/_inotify.c']))
-            packages.extend(['hgext.inotify', 'hgext.inotify.linux'])
-except ImportError:
-    pass
-
 setup(name='mercurial',
       version=mercurial.version.get_version(),
       author='Matt Mackall',
