diff -r 50c4de43aa87 mercurial/util.py
--- a/mercurial/util.py	Tue Jan 13 15:02:03 2009 -0800
+++ b/mercurial/util.py	Tue Jan 13 15:04:10 2009 -0800
@@ -1325,7 +1325,10 @@
         if len(getattr(sys, 'argv', [])) > 0:
             path.extend(rcfiles(os.path.dirname(sys.argv[0]) +
                                   '/../etc/mercurial'))
-        path.extend(rcfiles('/etc/mercurial'))
+        # Disable /etc/mercurial for Sage's mercurial (because it is
+        # likely to refer to extensions that are not installed in
+        # Sage's mercurial).
+        # path.extend(rcfiles('/etc/mercurial'))
         return path
 
     def user_rcpath():
