diff -ur src/Configure src.new/Configure
--- src/Configure	2007-08-21 06:57:01.000000000 +0200
+++ src.new/Configure	2011-10-19 22:20:50.000000000 +0200
@@ -1,23 +1,26 @@
 #! /bin/sh
+# Despite making some small changes to this rather strange shell script
+# I did not write 99% of it. David Kirkby, 21st August 2010. 
 
 if [ "$1" != "" ]; then
   echo "**ERROR**: Configure does not take any options for SYMPOW"; exit 1;
 fi
 
 whichexe() {
-    if [ -f /bin/$1 ]; then
-        echo /bin/$1
-        return;
-    fi;
-    if [ -f /usr/bin/$1 ]; then
-        echo /usr/bin/$1
-        return;
-    fi;
-    if [ -f /usr/local/bin/$1 ]; then
-        echo /usr/local/bin/$1
-        return;
-    fi;
-    echo `which $1`
+#    if [ -f /bin/$1 ]; then
+#        echo /bin/$1
+#        return;
+#    fi;
+#    if [ -f /usr/bin/$1 ]; then
+#        echo /usr/bin/$1
+#        return;
+#    fi;
+#    if [ -f /usr/local/bin/$1 ]; then
+#        echo /usr/local/bin/$1
+#        return;
+#    fi;
+#    echo `which $1`
+    echo $1
 }
 
 FILE="Makefile.new"
@@ -70,13 +73,6 @@
   echo "SH = $SH"
 fi
 
-CC=`whichexe cc` && echo "CC = $CC" >> $FILE
-if [ -z "$CC" ]; then
-  echo "**ERROR**: Could not find cc"; exit 1;
-else
-  echo "CC = $CC"
-fi
-
 UNAME=`whichexe uname`
 if [ -z "$UNAME" ];then
   echo "**ERROR**: Could not find uname"; exit 1;
@@ -84,18 +80,11 @@
   echo "UNAME = $UNAME"
 fi
 
-MACH=`"$UNAME" -m`
-for x in ix86 i386 i486 i586 i686 x86_64 ia64 
-do
-  if [ "$MACH" = "$x" -a `uname` = "Linux" ]; then
-     echo "You appear to have a $x based Linux system --- using fpu.c"
-     DEFS="-Dx86"
-  fi
-done
-if [ -z "$DEFS" ]; then
-  echo "You do not appear to have an x86 based system --- not using fpu.c"
-fi
+echo "CC=$CC" >> $FILE
 
+# All x86 extended precision FPU-control word stuff: moved to spkg-install
+# -- Jeroen Demeyer
+DEFS=""
 echo "DEFS = $DEFS" >> $FILE
 
 OPT="-O3" && echo "OPT = $OPT" >> $FILE
