--- ctypes__init__.py.orig	2009-02-15 21:38:41.000000000 -0800
+++ ctypes__init__.py	2009-02-15 21:39:26.000000000 -0800
@@ -23,8 +23,10 @@
     from _ctypes import FormatError
 
 DEFAULT_MODE = RTLD_LOCAL
-if _os.name == "posix" and _sys.platform == "darwin":
-    import gestalt
+
+# We don't care about this, since SAGE always uses Python >= 2.5. 
+#if _os.name == "posix" and _sys.platform == "darwin":
+#    import gestalt
 
     # gestalt.gestalt("sysv") returns the version number of the
     # currently active system file as BCD.
@@ -36,8 +38,8 @@
     # because RTLD_LOCAL does not work at least on some
     # libraries.
 
-    if gestalt.gestalt("sysv") < 0x1040:
-        DEFAULT_MODE = RTLD_GLOBAL
+#    if gestalt.gestalt("sysv") < 0x1040:
+#        DEFAULT_MODE = RTLD_GLOBAL
 
 from _ctypes import FUNCFLAG_CDECL as _FUNCFLAG_CDECL, \
      FUNCFLAG_PYTHONAPI as _FUNCFLAG_PYTHONAPI
