--- ../src/Lib/ctypes/__init__.py	2008-08-14 12:10:48.000000000 -0700
+++ ctypes__init__.py	2009-06-16 17:38:00.000000000 -0700
@@ -23,14 +23,14 @@
     from _ctypes import FormatError
 
 DEFAULT_MODE = RTLD_LOCAL
-if _os.name == "posix" and _sys.platform == "darwin":
-    # On OS X 10.3, we use RTLD_GLOBAL as default mode
-    # because RTLD_LOCAL does not work at least on some
-    # libraries.  OS X 10.3 is Darwin 7, so we check for
-    # that.
+## if _os.name == "posix" and _sys.platform == "darwin":
+##     # On OS X 10.3, we use RTLD_GLOBAL as default mode
+##     # because RTLD_LOCAL does not work at least on some
+##     # libraries.  OS X 10.3 is Darwin 7, so we check for
+##     # that.
 
-    if int(_os.uname()[2].split('.')[0]) < 8:
-        DEFAULT_MODE = RTLD_GLOBAL
+##     if int(_os.uname()[2].split('.')[0]) < 8:
+##         DEFAULT_MODE = RTLD_GLOBAL
 
 from _ctypes import FUNCFLAG_CDECL as _FUNCFLAG_CDECL, \
      FUNCFLAG_PYTHONAPI as _FUNCFLAG_PYTHONAPI, \
