--- src/Modules/cPickle.c.orig	2010-05-10 02:46:46.000000000 +1200
+++ src/Modules/cPickle.c	2011-05-21 08:26:22.244500967 +1200
@@ -2697,11 +2697,6 @@
         }
     }
 
-    if (PyType_IsSubtype(type, &PyType_Type)) {
-        res = save_global(self, args, NULL);
-        goto finally;
-    }
-
     /* Get a reduction callable, and call it.  This may come from
      * copy_reg.dispatch_table, the object's __reduce_ex__ method,
      * or the object's __reduce__ method.
@@ -2717,6 +2712,11 @@
         }
     }
     else {
+        if (PyType_IsSubtype(type, &PyType_Type)) {
+                res = save_global(self, args, NULL);
+                goto finally;
+        }
+
         /* Check for a __reduce_ex__ method. */
         __reduce__ = PyObject_GetAttr(args, __reduce_ex___str);
         if (__reduce__ != NULL) {
