--- ../src/Lib/locale.py	2009-03-18 10:11:06.000000000 -0700
+++ locale.py	2009-06-16 17:33:03.000000000 -0700
@@ -29,10 +29,12 @@
 try:
 
     from _locale import *
+    EMULATE = False
 
 except ImportError:
 
     # Locale emulation
+    EMULATE = True
 
     CHAR_MAX = 127
     LC_ALL = 6
@@ -522,7 +524,7 @@
     """
     _setlocale(category, _build_localename(getdefaultlocale()))
 
-if sys.platform in ('win32', 'darwin', 'mac'):
+if not EMULATE and sys.platform in ('win32', 'darwin', 'mac'):
     # On Win32, this will return the ANSI code page
     # On the Mac, it should return the system encoding;
     # it might return "ascii" instead
@@ -546,7 +548,10 @@
             according to the system configuration."""
             if do_setlocale:
                 oldloc = setlocale(LC_CTYPE)
-                setlocale(LC_CTYPE, "")
+                try:
+                 setlocale(LC_CTYPE, "")
+                except:
+                 return nl_langinfo(CODESET)
                 result = nl_langinfo(CODESET)
                 setlocale(LC_CTYPE, oldloc)
                 return result
