--- src/Modules/socketmodule.c	Sat Sep 19 08:46:24 2009
+++ patches/Modules.socketmodule.c	Sun May 30 07:38:45 2010
@@ -1092,7 +1092,14 @@
 		}
 #endif
 
-#ifdef HAVE_NETPACKET_PACKET_H
+/* The check for SIOCGIFNAME was added by David Kirkby on 
+30th May 2010 - see #9041, #9022 and http://bugs.python.org/issue8852
+It is safe to include this on any platform, as SIOCGIFNAME is used below
+so it is sensible to check for SIOCGIFNAME before using it. 
+*/
+
+#if defined(HAVE_NETPACKET_PACKET_H) && defined(SIOCGIFNAME) 
+
 	case AF_PACKET:
 	{
 		struct sockaddr_ll *a = (struct sockaddr_ll *)addr;
@@ -1382,7 +1389,14 @@
 	}
 #endif
 
-#ifdef HAVE_NETPACKET_PACKET_H
+/* The check for SIOCGIFINDEX was added by David Kirkby on 
+30th May 2010 - see #9041, #9022 and http://bugs.python.org/issue8852
+It is safe to include this on any platform, as SIOCGIFINDEX is used below
+so it is sensible to check for SIOCGIFINDEX before using it. 
+*/
+
+#if defined(HAVE_NETPACKET_PACKET_H) && defined(SIOCGIFINDEX) 
+
 	case AF_PACKET:
 	{
 		struct sockaddr_ll* addr;
@@ -4578,7 +4592,17 @@
 	PyModule_AddStringConstant(m, "BDADDR_LOCAL", "00:00:00:FF:FF:FF");
 #endif
 
-#ifdef HAVE_NETPACKET_PACKET_H
+/* The checks for PACKET_LOOPBACK and PACKET_FASTROUTE was added 
+by David Kirkby on 30th May 2010 - see #9041, #9022 and 
+http://bugs.python.org/issue8852
+It is safe to include patch on any platform, as 
+PACKET_LOOPBACK and PACKET_FASTROUTE are both used below, so unless
+they are defined, Python will not build this bit of code. So although the 
+patch was added for OpenSolaris, it is safe (and desirable) on any platform. 
+*/
+
+#if defined(HAVE_NETPACKET_PACKET_H) && defined(PACKET_LOOPBACK) && defined(PACKET_FASTROUTE)
+
 	PyModule_AddIntConstant(m, "AF_PACKET", AF_PACKET);
 	PyModule_AddIntConstant(m, "PF_PACKET", PF_PACKET);
 	PyModule_AddIntConstant(m, "PACKET_HOST", PACKET_HOST);
