diff -Naur src/Modules/Setup.dist src.new/Modules/Setup.dist
--- src/Modules/Setup.dist	2010-05-25 23:40:38.000000000 +0100
+++ src.new/Modules/Setup.dist	2011-04-24 09:21:56.521446467 +0100
@@ -221,7 +221,7 @@
 #
 # First, look at Setup.config; configure may have set this for you.
 
-#crypt cryptmodule.c # -lcrypt	# crypt(3); needs -lcrypt on some systems
+crypt cryptmodule.c -lcrypt	# crypt(3); needs -lcrypt on some systems like Ubuntu 11.04
 
 
 # Some more UNIX dependent modules -- off by default, since these
