diff -ur src.orig/configure.ac src/configure.ac
--- src.orig/configure.ac	2012-09-30 19:16:16.208133911 +0200
+++ src/configure.ac	2012-09-30 19:15:12.800599068 +0200
@@ -1113,6 +1113,9 @@
 	OSF*)
 	    BASECFLAGS="$BASECFLAGS -mieee"
 	    ;;
+	CYGWIN*)
+	    BASECFLAGS="-I/usr/include/ncurses $BASECFLAGS"
+	    ;;
     esac
     ;;

diff -ur src.orig/configure src/configure 
--- src.orig/configure  2012-09-30 19:16:16.216133852 +0200 
+++ src/configure       2012-09-30 19:15:44.876363826 +0200 
@@ -5613,6 +5613,9 @@
 	OSF*)
 	    BASECFLAGS="$BASECFLAGS -mieee"
 	    ;;
+	CYGWIN*)
+	    BASECFLAGS="-I/usr/include/ncurses $BASECFLAGS"
+	    ;;
     esac
     ;;
 
