--- local/include/python2.5/pymactoolbox.h.orig	2008-05-28 01:40:13.000000000 -0700
+++ local/include/python2.5/pymactoolbox.h	2008-05-28 02:00:00.000000000 -0700
@@ -134,18 +134,23 @@
 extern int GWorldObj_Convert(PyObject *, GWorldPtr *);
 
 /* Qt exports */
-extern PyObject *TrackObj_New(Track);
-extern int TrackObj_Convert(PyObject *, Track *);
-extern PyObject *MovieObj_New(Movie);
-extern int MovieObj_Convert(PyObject *, Movie *);
-extern PyObject *MovieCtlObj_New(MovieController);
-extern int MovieCtlObj_Convert(PyObject *, MovieController *);
-extern PyObject *TimeBaseObj_New(TimeBase);
-extern int TimeBaseObj_Convert(PyObject *, TimeBase *);
-extern PyObject *UserDataObj_New(UserData);
-extern int UserDataObj_Convert(PyObject *, UserData *);
-extern PyObject *MediaObj_New(Media);
-extern int MediaObj_Convert(PyObject *, Media *);
+// This file has been edited since in 64 bit OSX mode the 
+// includes below lead to compilation failures with 
+// twisted-8.0.1. Since we have to disable all OSX 
+// specific glue code anyway this seems to have no
+// consequences 
+//extern PyObject *TrackObj_New(Track);
+//extern int TrackObj_Convert(PyObject *, Track *);
+//extern PyObject *MovieObj_New(Movie);
+//extern int MovieObj_Convert(PyObject *, Movie *);
+//extern PyObject *MovieCtlObj_New(MovieController);
+//extern int MovieCtlObj_Convert(PyObject *, MovieController *);
+//extern PyObject *TimeBaseObj_New(TimeBase);
+//extern int TimeBaseObj_Convert(PyObject *, TimeBase *);
+//extern PyObject *UserDataObj_New(UserData);
+//extern int UserDataObj_Convert(PyObject *, UserData *);
+//extern PyObject *MediaObj_New(Media);
+//extern int MediaObj_Convert(PyObject *, Media *);
 
 /* Res exports */
 extern PyObject *ResObj_New(Handle);
