--- readline.c	2009-09-22 13:48:53.000000000 -0500
+++ readline.c-Itanium-fix	2009-09-22 13:49:58.000000000 -0500
@@ -31,16 +31,11 @@
 #include <readline/readline.h>
 #include <readline/history.h>
 
-#ifdef HAVE_RL_COMPLETION_MATCHES
-#define completion_matches(x, y) \
-	rl_completion_matches((x), ((rl_compentry_func_t *)(y)))
-#else
 #if defined(_RL_FUNCTION_TYPEDEF)
 extern char **completion_matches(char *, rl_compentry_func_t *);
 #else
 extern char **completion_matches(char *, CPFunction *);
 #endif
-#endif
 
 static void
 on_completion_display_matches_hook(char **matches,
@@ -759,11 +754,18 @@
 static char **
 flex_complete(char *text, int start, int end)
 {
+#ifdef HAVE_RL_COMPLETION_APPEND_CHARACTER
+	rl_completion_append_character ='\0';
+	rl_completion_suppress_append = 0;
+#endif
 	Py_XDECREF(begidx);
 	Py_XDECREF(endidx);
 	begidx = PyInt_FromLong((long) start);
 	endidx = PyInt_FromLong((long) end);
-	return completion_matches(text, *on_completion);
+	/* return completion_matches(text, *on_completion); */
+	/* SAGE: Fix by Kate Minola <kate01123@gmail.com> */
+	return (char **) completion_matches(text, (rl_compentry_func_t *)*on_completion);
+
 }
 
 
@@ -801,9 +803,6 @@
 	rl_completer_word_break_characters =
 		strdup(" \t\n`~!@#$%^&*()-=+[{]}\\|;:'\",<>/?");
 		/* All nonalphanums except '.' */
-#ifdef HAVE_RL_COMPLETION_APPEND_CHARACTER
-	rl_completion_append_character ='\0';
-#endif
 
 	begidx = PyInt_FromLong(0L);
 	endidx = PyInt_FromLong(0L);
@@ -854,15 +853,10 @@
 		while (!has_input)
 		{	struct timeval timeout = {0, 100000}; /* 0.1 seconds */
 
-			/* [Bug #1552726] Only limit the pause if an input hook has been 
-			   defined.  */
-		 	struct timeval *timeoutp = NULL;
-			if (PyOS_InputHook) 
-				timeoutp = &timeout;
 			FD_SET(fileno(rl_instream), &selectset);
 			/* select resets selectset if no input was available */
 			has_input = select(fileno(rl_instream) + 1, &selectset,
-					   NULL, NULL, timeoutp);
+					   NULL, NULL, &timeout);
 			if(PyOS_InputHook) PyOS_InputHook();
 		}
 
