diff -ru src/pexpect.py src.del/pexpect.py
--- src/pexpect.py	2005-11-17 15:36:09.000000000 +0100
+++ src.del/pexpect.py	2012-01-13 10:24:01.000000000 +0100
@@ -341,7 +341,10 @@
         """
         if self.closed:
             return
-        self.close()
+        try:
+            self.close()
+        except:
+            pass
 
     def __str__(self):
         """This returns the current state of the pexpect object as a string.
