--- a/pexpect.py	2007-04-16 07:08:24.000000000 -0700
+++ b/pexpect.py	2009-01-23 01:49:18.000000000 -0800
@@ -1130,7 +1130,7 @@
     """
     # Special case where filename already contains a path.
     if os.path.dirname(filename) != '':
-        if os.access (filename, os.X_OK):
+        if os.access (filename, os.X_OK) and not os.path.isdir(f):
             return filename
 
     if not os.environ.has_key('PATH') or os.environ['PATH'] == '':
@@ -1145,7 +1145,7 @@
 
     for path in pathlist:
         f = os.path.join(path, filename)
-        if os.access(f, os.X_OK):
+        if os.access(f, os.X_OK) and not os.path.isdir(f):
             return f
     return None
 
