From 021ee6b49d675d5e9b337d7ce075ec3d4ce5342f Mon Sep 17 00:00:00 2001
From: Moritz Firsching <moritz@math.fu-berlin.de>
Date: Sun, 8 Apr 2018 18:54:57 +0200
Subject: [PATCH 2/5] fix degree of empty expression

---
 src/pyscipopt/expr.pxi | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/pyscipopt/expr.pxi b/src/pyscipopt/expr.pxi
index 822ae99..4a3f9e8 100644
--- a/src/pyscipopt/expr.pxi
+++ b/src/pyscipopt/expr.pxi
@@ -272,7 +272,10 @@ cdef class Expr:
 
     def degree(self):
         '''computes highest degree of terms'''
-        return max(len(v) for v in self.terms)
+        if len(self.terms) == 0:
+            return 0
+        else:
+            return max(len(v) for v in self.terms)
 
 
 cdef class ExprCons:
-- 
2.17.0

