From e0abf475352ba80e3139b210e7cf4c54d5712f04 Mon Sep 17 00:00:00 2001
From: Moritz Firsching <moritz@math.fu-berlin.de>
Date: Tue, 10 Apr 2018 23:06:07 +0200
Subject: [PATCH 8/8] warn if problem was not solved before

---
 src/pyscipopt/scip.pyx | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/pyscipopt/scip.pyx b/src/pyscipopt/scip.pyx
index 407ef8d..9faacdf 100644
--- a/src/pyscipopt/scip.pyx
+++ b/src/pyscipopt/scip.pyx
@@ -1484,6 +1484,7 @@ cdef class Model:
 
     def getActivity(self, Constraint cons, Solution sol = None):
         """Retrieve activity of given constraint.
+        Can only be called after solving is completed.
 
         :param Constraint cons: linear or quadratic constraint
         :param Solution sol: solution to compute activity of, None to use current node's solution (Default value = None)
@@ -1492,6 +1493,9 @@ cdef class Model:
         cdef SCIP_Real activity
         cdef SCIP_SOL* scip_sol
 
+        if not self.getStage() >= SCIP_STAGE_SOLVING:
+            raise Warning("method cannot be called before problem is solved")
+
         if isinstance(sol, Solution):
             scip_sol = sol.sol
         else:
-- 
2.17.0

