diff -druN rpy2-2.3.8.orig/rpy/rinterface/_rinterface.c rpy2-2.3.8/rpy/rinterface/_rinterface.c
--- rpy2-2.3.8.orig/rpy/rinterface/_rinterface.c	2013-09-28 17:34:40.000000000 +0200
+++ rpy2-2.3.8/rpy/rinterface/_rinterface.c	2014-04-10 12:03:37.915617700 +0200
@@ -3389,7 +3389,7 @@
   }
 
   /* NA types */
-#if defined(Win32) || defined(Win64)
+#if defined(Win32) || defined(Win64) || defined(__CYGWIN__)
   NAInteger_Type.tp_base=&PyLong_Type;
 #endif
   if (PyType_Ready(&NAInteger_Type) < 0) {
@@ -3399,7 +3399,7 @@
     return NULL;
 #endif
   }
-#if defined(Win32) || defined(Win64)
+#if defined(Win32) || defined(Win64) || defined(__CYGWIN__)
   NALogical_Type.tp_base=&PyLong_Type;
 #endif
   if (PyType_Ready(&NALogical_Type) < 0) {
@@ -3409,7 +3409,7 @@
     return NULL;
 #endif
   }
-#if defined(Win32) || defined(Win64)
+#if defined(Win32) || defined(Win64) || defined(__CYGWIN__)
   NAReal_Type.tp_base=&PyFloat_Type;
 #endif
   if (PyType_Ready(&NAReal_Type) < 0) {
@@ -3419,7 +3419,7 @@
     return NULL;
 #endif
   }
-#if defined(Win32) || defined(Win64)
+#if defined(Win32) || defined(Win64) || defined(__CYGWIN__)
   NAComplex_Type.tp_base=&PyComplex_Type;
 #endif
   if (PyType_Ready(&NAComplex_Type) < 0) {
@@ -3430,7 +3430,7 @@
 #endif
   }
   
-#if (defined(Win32) || defined(Win64)) & PY_VERSION_HEX < 0x03010000
+#if (defined(Win32) || defined(Win64) || defined(__CYGWIN__)) & PY_VERSION_HEX < 0x03010000
   NACharacter_Type.tp_base=&PyString_Type;
 #elif defined(Win32)
   NACharacter_Type.tp_base=&PyUnicode_Type;
diff -druN rpy2-2.3.8.orig/rpy/rinterface/na_values.c rpy2-2.3.8/rpy/rinterface/na_values.c
--- rpy2-2.3.8.orig/rpy/rinterface/na_values.c	2013-09-28 17:33:27.000000000 +0200
+++ rpy2-2.3.8/rpy/rinterface/na_values.c	2014-04-10 12:01:04.134367700 +0200
@@ -203,7 +203,7 @@
         0,                      /*tp_methods*/
         0,                      /*tp_members*/
         0,                      /*tp_getset*/
-#if defined(Win32) || defined(Win64)
+#if defined(Win32) || defined(Win64) || defined(__CYGWIN__)
 	NULL,
 #else	
         &PyLong_Type,           /*tp_base*/
@@ -401,7 +401,7 @@
         0, //NAInteger_methods,           /*tp_methods*/
         0,                      /*tp_members*/
         0,                      /*tp_getset*/
-#if defined(Win32) || defined(Win64)
+#if defined(Win32) || defined(Win64) || defined(__CYGWIN__)
 	NULL,
 #else
         &PyLong_Type,             /*tp_base*/
@@ -587,7 +587,7 @@
         0, //NAInteger_methods,           /*tp_methods*/
         0,                      /*tp_members*/
         0,                      /*tp_getset*/
-#if defined(Win32) || defined(Win64)
+#if defined(Win32) || defined(Win64) || defined(__CYGWIN__)
 	NULL,
 #else
         &PyFloat_Type,             /*tp_base*/
@@ -689,7 +689,7 @@
         0, //NAInteger_methods,           /*tp_methods*/
         0,                      /*tp_members*/
         0,                      /*tp_getset*/
-#if defined(Win32) || defined(Win64)
+#if defined(Win32) || defined(Win64) || defined(__CYGWIN__)
 	NULL,
 #elif (PY_VERSION_HEX < 0x03010000)
         &PyString_Type,             /*tp_base*/
@@ -856,7 +856,7 @@
         0, //NAInteger_methods,           /*tp_methods*/
         0,                      /*tp_members*/
         0,                      /*tp_getset*/
-#if defined(Win32) || defined(Win64)
+#if defined(Win32) || defined(Win64) || defined(__CYGWIN__)
 	NULL,
 #else
         &PyComplex_Type,             /*tp_base*/
diff -druN rpy2-2.3.8.orig/setup.py rpy2-2.3.8/setup.py
--- rpy2-2.3.8.orig/setup.py	2013-09-28 17:34:18.000000000 +0200
+++ rpy2-2.3.8/setup.py	2014-04-10 12:06:38.774992700 +0200
@@ -416,7 +416,7 @@
                        ],
             include_dirs = [os.path.join(package_prefix,
                                          'rpy', 'rinterface'),] + include_dirs,
-            libraries = ['R', ],
+            libraries = ['R', 'readline'],
             library_dirs = r_libs,
             define_macros = define_macros,
             runtime_library_dirs = r_libs,
