diff --git a/setup.py b/setup.py
--- a/setup.py
+++ b/setup.py
@@ -279,7 +279,7 @@ class RConfig(object):
         rc = RConfig()
 
         for substring in re.split('(?<!-framework) ', string):
-            ok = False
+            ok = True # work program around some parsing issues on some systems
             for pattern in pp:
                 rconfig_m = pattern.match(substring)
                 if rconfig_m is not None:
