diff -ur src/Makefile src.patched/Makefile
--- src/Makefile	2010-02-16 05:26:57.000000000 +0100
+++ src.patched/Makefile	2011-04-26 09:46:41.000000000 +0200
@@ -1,14 +1,17 @@
 
 ##### Configurable options:
 
+# Don't need to set any of these compiler variables. They have already been
+# set when running SAGE_ROOT/local/bin/sage-env as part of installing a
+# package.
 ## Compiler:
-CC=gcc
+#CC=gcc
 #CC=cc
 
 ## Compiler flags:
 
 # GCC:  (also -march=pentium etc, for machine-dependent optimizing)
-CFLAGS=-Wall -O3 -fomit-frame-pointer -funroll-loops
+#CFLAGS=-Wall -O3 -fomit-frame-pointer -funroll-loops
 
 # GCC w/ debugging:
 #CFLAGS=-Wall -g -DINLINE=
@@ -29,14 +32,14 @@
 ##### End of configurable options
 
 
-all: cl
+all: libcliquer.so
 
 
 testcases: testcases.o cliquer.o graph.o reorder.o
 	$(CC) $(LDFLAGS) -o $@ testcases.o cliquer.o graph.o reorder.o
 
-cl: cl.o cliquer.o graph.o reorder.o
-	$(CC) $(LDFLAGS) -o $@ cl.o cliquer.o graph.o reorder.o
+libcliquer.so: cl.o cliquer.o graph.o reorder.o
+	$(CC) $(LDFLAGS) $(SAGESOFLAGS) -o $@ cl.o cliquer.o graph.o reorder.o
 
 
 cl.o testcases.o cliquer.o graph.o reorder.o: cliquer.h set.h graph.h misc.h reorder.h Makefile cliquerconf.h
