diff -ru src/Makefile b/Makefile
--- src/Makefile	2010-01-22 08:53:21.000000000 +0100
+++ b/Makefile	2014-01-16 14:55:51.977047191 +0100
@@ -1,24 +1,3 @@
-
-##### Configurable options:
-
-## Compiler:
-CC=gcc
-#CC=cc
-
-## Compiler flags:
-
-# GCC:  (also -march=pentium etc, for machine-dependent optimizing)
-CFLAGS=-Wall -O3 -fomit-frame-pointer -funroll-loops
-
-# GCC w/ debugging:
-#CFLAGS=-Wall -g -DINLINE=
-
-# Compaq C / Digital C:
-#CFLAGS=-arch=host -fast
-
-# SunOS:
-#CFLAGS=-fast
-
 ## Program options:
 
 # Enable long options for cl (eg. "cl --help"), comment out to disable.
@@ -29,14 +8,14 @@
 ##### End of configurable options
 
 
-all: cl
+all: libcliquer.so
 
 
 testcases: testcases.o cliquer.o graph.o reorder.o
 	$(CC) $(LDFLAGS) -o $@ testcases.o cliquer.o graph.o reorder.o
 
-cl: cl.o cliquer.o graph.o reorder.o
-	$(CC) $(LDFLAGS) -o $@ cl.o cliquer.o graph.o reorder.o
+libcliquer.so: cl.o cliquer.o graph.o reorder.o
+	$(CC) $(LDFLAGS) $(SAGESOFLAGS) -o $@ cl.o cliquer.o graph.o reorder.o
 
 
 cl.o testcases.o cliquer.o graph.o reorder.o: cliquer.h set.h graph.h misc.h reorder.h Makefile cliquerconf.h
