--- src/setup.py	2013-10-14 14:38:10.000000000 -0700
+++ src/setup.py	2014-04-02 12:46:32.000000000 -0700
@@ -271,7 +271,7 @@
         if not os.path.exists("config.status"):
             if os.system("chmod 0755 configure") != 0:
                 raise RuntimeError("chmod error")
-            cmd = "sh configure"    # we use "sh" here so that it'll work on mingw32 with standard python.org binaries
+            cmd = "sh %s" % (os.path.join(os.curdir, 'configure'))    # we use "sh" here so that it'll work on mingw32 with standard python.org binaries
             if self.verbose < 1:
                 cmd += " -q"
             if os.system(cmd) != 0:
