diff --git a/src/lisp/Makefile.in b/src/lisp/Makefile.in
index 30f615096..f0b3f0bd4 100644
--- a/src/lisp/Makefile.in
+++ b/src/lisp/Makefile.in
@@ -118,6 +118,8 @@ do_it.ecl: fricas-lisp.lisp fricas-package.lisp fricas-config.lisp \
 		  fricas-lisp.o primitives.o) ")))" \
 	            >> fricas-ecl.lisp
 	echo "(defvar *fricas-initial-lisp-forms* nil)" >> fricas-ecl.lisp
+	echo "(require :cmp)"
+	echo "(setf c::*user-cc-flags* (concatenate 'string c::*user-cc-flags* \" -I$(BASE)/$(fricas_src_srcdir)/include/ -I$(BASE)/$(fricas_src_srcdir)/../config/\"))" >> fricas-ecl.lisp
 	echo '(load "fricas-package.lisp")' \
 	     '(load "fricas-config.lisp")' \
 	     '(load "fricas-ecl.lisp")' \
diff --git a/src/lisp/fricas-lisp.lisp b/src/lisp/fricas-lisp.lisp
index d6c7484df..f99e2e75b 100644
--- a/src/lisp/fricas-lisp.lisp
+++ b/src/lisp/fricas-lisp.lisp
@@ -609,6 +609,13 @@ with this hack and will try to convince the GCL crowd to fix this.
     #+(and :clisp :ffi) `(defun clisp-init-foreign-calls () ,@arguments)
 )
 
+#+:ecl
+(ext:with-backend :c/c++
+  (ffi:clines
+   "#include <com.h>"
+   "#include <cfuns-c.H1>"
+   "#include <sockio-c.H1>"))
+
 (foreign-defs
 
 (fricas-foreign-call |writeablep| "writeablep" int
