From c56ab3fbeea95c232372ba1bafb8f09bedd4c010 Mon Sep 17 00:00:00 2001
From: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date: Tue, 26 Jul 2016 11:32:35 -0700
Subject: [PATCH 1/5] Add dependency of libscip on liblpi*

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index d9835fc..b276aa7 100644
--- a/Makefile
+++ b/Makefile
@@ -422,7 +422,7 @@ SCIPLIBEXTLIBS	+=	$(READLINE_LDFLAGS)
 endif
 SCIPLIBEXTLIBS	+=	$(ZIMPLLIB) $(LINKRPATH)$(realpath $(LIBDIR)/$(LIBTYPE))
 endif
-
+SCIPLIBEXTLIBS  +=      $(LPILIBFILE)
 
 #-----------------------------------------------------------------------------
 # SCIP Library
-- 
2.7.4 (Apple Git-66)

