From 99b9db6b9e1db1a048e4bade34e1c5d86d3055c8 Mon Sep 17 00:00:00 2001
From: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date: Tue, 26 Jul 2016 12:55:23 -0700
Subject: [PATCH 2/5] Library dependencies

---
 Makefile | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index b276aa7..90a0eb1 100644
--- a/Makefile
+++ b/Makefile
@@ -422,7 +422,7 @@ SCIPLIBEXTLIBS	+=	$(READLINE_LDFLAGS)
 endif
 SCIPLIBEXTLIBS	+=	$(ZIMPLLIB) $(LINKRPATH)$(realpath $(LIBDIR)/$(LIBTYPE))
 endif
-SCIPLIBEXTLIBS  +=      $(LPILIBFILE)
+SCIPLIBEXTLIBS  +=      $(LPILIBFILE) $(NLPILIBFILE)
 
 #-----------------------------------------------------------------------------
 # SCIP Library
@@ -1104,10 +1104,11 @@ endif
 libscip:		preprocess
 		@$(MAKE) $(SCIPLIBFILE) $(SCIPLIBLINK) $(SCIPLIBSHORTLINK)
 
-$(SCIPLIBFILE):	$(SCIPLIBOBJFILES) | $(LIBDIR)/$(LIBTYPE) $(LIBOBJSUBDIRS)
+$(SCIPLIBFILE):	$(SCIPLIBOBJFILES) $(LPILIBFILE) $(NLPILIBFILE) | $(LIBDIR)/$(LIBTYPE) $(LIBOBJSUBDIRS)
 		@echo "-> generating library $@"
 		-rm -f $@
 		$(LIBBUILD) $(LIBBUILDFLAGS) $(LIBBUILD_o)$@ $(SCIPLIBOBJFILES) $(SCIPLIBEXTLIBS)
+
 ifneq ($(RANLIB),)
 		$(RANLIB) $@
 endif
-- 
2.7.4 (Apple Git-66)

