From 0b208a25652550b3afd49ddd35e3879e82652a5b Mon Sep 17 00:00:00 2001
From: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date: Fri, 29 Jul 2016 14:18:12 -0700
Subject: [PATCH 3/5] Link SCIP against libscip

---
 Makefile | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 90a0eb1..d551556 100644
--- a/Makefile
+++ b/Makefile
@@ -1096,7 +1096,9 @@ ifeq ($(LINKER),C)
 		|| ($(MAKE) errorhints && false)
 endif
 ifeq ($(LINKER),CPP)
-		-$(LINKCXX) $(MAINOBJFILES) $(LINKCCSCIPALL) $(LINKCXX_o)$@ \
+		-$(LINKCXX) $(MAINOBJFILES) \
+		$(LINKCXX_L)$(LIBDIR) $(SCIPLIBFILE) $(LPILIBFILE) $(NLPILIBFILE) \
+		$(OFLAGS) $(LPSLDFLAGS) $(LDFLAGS) $(LINKCXX_o)$@ \
 		|| ($(MAKE) errorhints && false)
 endif
 
-- 
2.7.4 (Apple Git-66)

