From 274f5d7ebbda7080d45f4193c57dedb9e38e0bb6 Mon Sep 17 00:00:00 2001
From: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date: Sun, 31 Jul 2016 12:58:52 -0700
Subject: [PATCH 5/5] Use new variables SCIP_LIBBUILDFLAGS etc.

---
 Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 7ee691a..ecf7eff 100644
--- a/Makefile
+++ b/Makefile
@@ -949,7 +949,7 @@ makesciplibfile: preprocess
 $(SCIPLIBFILE):	$(SCIPLIBOBJFILES) $(LPILIBFILE) $(NLPILIBFILE) | $(LIBDIR) $(LIBOBJSUBDIRS)
 		@echo "-> generating library $@"
 		-rm -f $@
-		$(LIBBUILD) $(LIBBUILDFLAGS) $(LIBBUILD_o)$@ $(SCIPLIBOBJFILES) $(LPILIBFILE) $(NLPILIBFILE)
+		$(LIBBUILD) $(LIBBUILDFLAGS) $(SCIP_LIBBUILDFLAGS) $(LIBBUILD_o)$@ $(SCIPLIBOBJFILES) $(LPILIBFILE) $(NLPILIBFILE)
 ifneq ($(RANLIB),)
 		$(RANLIB) $@
 endif
@@ -957,7 +957,7 @@ endif
 $(OBJSCIPLIBFILE):	$(OBJSCIPLIBOBJFILES) | $(LIBOBJSUBDIRS) $(LIBDIR)
 		@echo "-> generating library $@"
 		-rm -f $@
-		$(LIBBUILD) $(LIBBUILDFLAGS) $(LIBBUILD_o)$@ $(OBJSCIPLIBOBJFILES)
+		$(LIBBUILD) $(LIBBUILDFLAGS) $(OBJSCIP_LIBBUILDFLAGS) $(LIBBUILD_o)$@ $(OBJSCIPLIBOBJFILES)
 ifneq ($(RANLIB),)
 		$(RANLIB) $@
 endif
@@ -965,7 +965,7 @@ endif
 $(LPILIBFILE):	$(LPILIBOBJFILES) | $(LIBOBJSUBDIRS) $(LIBDIR)
 		@echo "-> generating library $@"
 		-rm -f $@
-		$(LIBBUILD) $(LIBBUILDFLAGS) $(LIBBUILD_o)$@ $(LPILIBOBJFILES) $(LPILIBS)
+		$(LIBBUILD) $(LIBBUILDFLAGS) $(LPI_LIBBUILDFLAGS) $(LIBBUILD_o)$@ $(LPILIBOBJFILES) $(LPILIBS)
 ifneq ($(RANLIB),)
 		$(RANLIB) $@
 endif
@@ -973,7 +973,7 @@ endif
 $(NLPILIBFILE):	$(NLPILIBOBJFILES) $(NLPILIBSCIPOBJFILES) | $(LIBOBJSUBDIRS) $(LIBDIR)
 		@echo "-> generating library $@"
 		-rm -f $@
-		$(LIBBUILD) $(LIBBUILDFLAGS) $(LIBBUILD_o)$@ $(NLPILIBOBJFILES) $(NLPILIBSCIPOBJFILES)
+		$(LIBBUILD) $(LIBBUILDFLAGS) $(NLPI_LIBBUILDFLAGS) $(LIBBUILD_o)$@ $(NLPILIBOBJFILES) $(NLPILIBSCIPOBJFILES)
 ifneq ($(RANLIB),)
 		$(RANLIB) $@
 endif
-- 
2.7.4 (Apple Git-66)

