diff --git a/src/lsp/config.lsp.in b/src/lsp/config.lsp.in
index e8bed39..957d7ba 100644
--- a/src/lsp/config.lsp.in
+++ b/src/lsp/config.lsp.in
@@ -7,12 +7,12 @@
 ;;
 (in-package "LISP")
 
-#+(and (not ecl-min) (not nacl) (not :mingw32) (not :msvc) (not :cygwin) uname)
+#+(and (not ecl-min) (not nacl) (not :mingw32) (not :msvc) uname)
 (ffi:clines "
 #include <sys/utsname.h>
 ")
 
-#+(and (not ecl-min) (not nacl) (not :mingw32) (not :msvc) (not :cygwin) uname)
+#+(and (not ecl-min) (not nacl) (not :mingw32) (not :msvc)  uname)
 (defun uname ()
  (ffi:c-inline () () :object "{
         cl_object output;
@@ -28,7 +28,7 @@
         @(return) = output;
 }" :one-liner nil))
 
-#+(and ecl-min uname)
+#+ecl-min
 (defun uname ()
   "A poor man's uname"
   (list "@SOFTWARE_TYPE@"
