--- a/src/lsp/format.lsp
+++ b/src/lsp/format.lsp
@@ -196,6 +196,20 @@
       ;; Note that we have to compute the exponential _every_ _time_ in the loop
       ;; because multiplying just by 10.0l0 every time would lead to a greater
       ;; loss of precission.
+      (let ((ex (round (* exponent #.(log 2l0 10)))))
+      (declare (fixnum ex))
+      (if (minusp ex)
+          (loop for y of-type long-float
+	   	  = (* x (the long-float (expt 10.0l0 (- ex))))
+		         while (<= y 0.1l0)
+			        do (decf ex)
+				       finally (return (values y (the fixnum (+ delta ex)))))
+				           (loop for y of-type long-float
+					   	  = (/ x (the long-float (expt 10.0l0 ex)))
+						         while (> y 1.0l0)
+							        do (incf ex)
+								       finally (return (values y (the fixnum (+ delta ex)))))))
+      #+(or)
       (loop with ex of-type fixnum
 	   = (round (* exponent #.(log 2l0 10)))
 	 for y of-type long-float
