diff -r -u src/src/c/ffi/libraries.d src.patched/src/c/ffi/libraries.d
--- src/src/c/ffi/libraries.d	2011-11-07 21:16:18.000000000 +0100
+++ src.patched/src/c/ffi/libraries.d	2011-11-15 11:13:42.000000000 +0100
@@ -70,6 +70,7 @@
 #endif /* ENABLE_DLOPEN */
 #include <ecl/ecl-inl.h>
 #include <ecl/internal.h>
+#include <sys/stat.h>
 
 cl_object
 ecl_make_codeblock()
Only in src.patched/src/c/ffi: libraries.d.orig
