diff --git a/src/configure b/src/configure
index beca5e5b..103f4102 100755
--- a/src/configure
+++ b/src/configure
@@ -5304,8 +5304,8 @@ fi
     elif test "${enable_manual}" = "info"; then
       as_fn_error $? "Unable to build the manual: install-info not found." "$LINENO" 5
     fi
-  fi
-  # Extract the first word of "makeinfo", so it can be a program name with args.
+  else
+    # Extract the first word of "makeinfo", so it can be a program name with args.
 set dummy makeinfo; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
 $as_echo_n "checking for $ac_word... " >&6; }
@@ -5345,11 +5345,12 @@ $as_echo "no" >&6; }
 fi
 
 
-  if test "x${MAKEINFO}" = "x"; then
-    if test "${enable_manual}" = "auto"; then
-      enable_manual=no
-    else
-      as_fn_error $? "Unable to build the manual: makeinfo not found." "$LINENO" 5
+    if test "x${MAKEINFO}" = "x"; then
+      if test "${enable_manual}" = "auto"; then
+        enable_manual=no
+      else
+        as_fn_error $? "Unable to build the manual: makeinfo not found." "$LINENO" 5
+      fi
     fi
   fi
 fi
diff --git a/src/configure.ac b/src/configure.ac
index 0184a182..4e615408 100644
--- a/src/configure.ac
+++ b/src/configure.ac
@@ -359,13 +359,14 @@ if test "${enable_manual}" != "no"; then
     elif test "${enable_manual}" = "info"; then
       AC_MSG_ERROR([Unable to build the manual: install-info not found.])
     fi
-  fi
-  AC_PATH_PROG([MAKEINFO], [makeinfo], [])
-  if test "x${MAKEINFO}" = "x"; then
-    if test "${enable_manual}" = "auto"; then
-      enable_manual=no
-    else
-      AC_MSG_ERROR([Unable to build the manual: makeinfo not found.])
+  else
+    AC_PATH_PROG([MAKEINFO], [makeinfo], [])
+    if test "x${MAKEINFO}" = "x"; then
+      if test "${enable_manual}" = "auto"; then
+        enable_manual=no
+      else
+        AC_MSG_ERROR([Unable to build the manual: makeinfo not found.])
+      fi
     fi
   fi
 fi
