--- ../src/networkx/readwrite/gml.py	2010-08-10 10:02:08.530308126 -0600
+++ gml.py	2010-08-10 10:07:55.590293054 -0600
@@ -113,8 +113,11 @@
     try:
         from pyparsing import ParseException
     except ImportError:
-        raise ImportError, \
-          "Import Error: not able to import pyparsing: http://pyparsing.wikispaces.com/"
+        try:
+            from matplotlib.pyparsing import ParseException
+        except:
+            raise ImportError, \
+                "Import Error: not able to import pyparsing: http://pyparsing.wikispaces.com/"
 
     try:
         data = "".join(lines)
@@ -196,8 +199,15 @@
              ParseException, restOfLine, White, alphas, alphanums, nums,\
              OneOrMore,quotedString,removeQuotes,dblQuotedString
     except ImportError:
-        raise ImportError, \
-          "Import Error: not able to import pyparsing: http://pyparsing.wikispaces.com/"
+        try:
+            from matplotlib.pyparsing import \
+             Literal, CaselessLiteral, Word, Forward,\
+             ZeroOrMore, Group, Dict, Optional, Combine,\
+             ParseException, restOfLine, White, alphas, alphanums, nums,\
+             OneOrMore,quotedString,removeQuotes,dblQuotedString
+        except:
+            raise ImportError, \
+              "Import Error: not able to import pyparsing: http://pyparsing.wikispaces.com/"
 
     if not graph:
         lbrack = Literal("[").suppress()
@@ -350,4 +360,7 @@
     try:
         import pyparsing
     except:
-        raise SkipTest("pyparsing not available")
+        try:
+            import matplotlib.pyparsing
+        except:
+            raise SkipTest("pyparsing not available")
