--- ../src/networkx/readwrite/tests/test_gml.py	2010-08-10 10:02:08.530308126 -0600
+++ test_gml.py	2010-08-10 10:08:30.750848663 -0600
@@ -12,7 +12,10 @@
         try:
             import pyparsing
         except ImportError:
-            raise SkipTest('gml test: pyparsing not available.')
+            try:
+                import matplotlib.pyparsing as pyparsing
+            except:
+                raise SkipTest('gml test: pyparsing not available.')
 
     def setUp(self):
         self.simple_data="""Creator me
