--- src/src/dpe.h	2007-11-28 04:44:47.000000000 -0500
+++ ../libfplll-2.1.6-20071129_dirt/src/src/dpe.h	2008-01-27 21:14:37.282113000 -0500
@@ -65,8 +64,12 @@
 # define DPE_TRUNC trunc
 /* some versions of math.h do not include a prototype for round() */
 #ifndef round
+#if defined (__sun)
+#define round(a) ((fmod(a,1)<0.5)?floor(a):ceil(a))
+#else
 double round(double);
 #endif
+#endif
 #ifndef trunc
 double trunc(double);
 #endif
