commit 0ac824f6a5fb746c4dfbc554b9f7495e66485f10
Author: Martin Kroeker <martin@ruby.chemie.uni-freiburg.de>
Date:   Wed Jan 31 22:02:00 2018 +0100

    Also #define SPARC in config.h when autodetecting

    Fixes #1442

diff --git a/cpuid_sparc.c b/cpuid_sparc.c
index b1e212b..f390f0d 100644
--- a/cpuid_sparc.c
+++ b/cpuid_sparc.c
@@ -49,6 +49,7 @@ void get_subdirname(void){
 }
 
 void get_cpuconfig(void){
+  printf("#define SPARC\n");
   printf("#define V9\n");
   printf("#define DTB_DEFAULT_ENTRIES 32\n");
 }
commit d1b512a01a19d4b9b389a40af1ad2436c4398874
Author: Martin Kroeker <martin@ruby.chemie.uni-freiburg.de>
Date:   Thu Feb 1 18:15:15 2018 +0100

    Return a corename for SPARC

diff --git a/cpuid_sparc.c b/cpuid_sparc.c
index f390f0d..ab8f773 100644
--- a/cpuid_sparc.c
+++ b/cpuid_sparc.c
@@ -57,3 +57,8 @@ void get_cpuconfig(void){
 void get_libname(void){
   printf("v9\n");
 }
+
+void get_corename(void){
+  printf("sparc\n");
+}
+
commit efa84afd00b83dbe3ce922efe3a1e021312001c0
Author: Martin Kroeker <martin@ruby.chemie.uni-freiburg.de>
Date:   Thu Feb 1 18:20:38 2018 +0100

    Use get_corename for SPARC as well

diff --git a/getarch.c b/getarch.c
index c0fff04..24ea5fe 100644
--- a/getarch.c
+++ b/getarch.c
@@ -1116,7 +1116,7 @@ int main(int argc, char *argv[]){
 #ifdef FORCE
     printf("CORE=%s\n", CORENAME);
 #else
-#if defined(INTEL_AMD) || defined(POWER) || defined(__mips__) || defined(__arm__) || defined(__aarch64__) || defined(ZARCH)
+#if defined(INTEL_AMD) || defined(POWER) || defined(__mips__) || defined(__arm__) || defined(__aarch64__) || defined(ZARCH) || defined(sparc)
     printf("CORE=%s\n", get_corename());
 #endif
 #endif
@@ -1224,7 +1224,7 @@ int main(int argc, char *argv[]){
 #ifdef FORCE
     printf("#define CHAR_CORENAME \"%s\"\n", CORENAME);
 #else
-#if defined(INTEL_AMD) || defined(POWER) || defined(__mips__) || defined(__arm__) || defined(__aarch64__) || defined(ZARCH)
+#if defined(INTEL_AMD) || defined(POWER) || defined(__mips__) || defined(__arm__) || defined(__aarch64__) || defined(ZARCH) || defined(sparc)
     printf("#define CHAR_CORENAME \"%s\"\n", get_corename());
 #endif
 #endif
commit 1766cc833e4ca74fe71e83baf84d9b0ab0177d6e
Author: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date:   Thu Feb 1 21:33:03 2018 +0100

    Fix get_corename() for SPARC

diff --git a/cpuid_sparc.c b/cpuid_sparc.c
index ab8f773..6902a8f 100644
--- a/cpuid_sparc.c
+++ b/cpuid_sparc.c
@@ -58,7 +58,7 @@ void get_libname(void){
   printf("v9\n");
 }
 
-void get_corename(void){
-  printf("sparc\n");
+char *get_corename(void){
+  return "sparc";
 }
 
