--- ../src/polybori-0.6.4/polybori/include/pbori_routines_misc.h	2010-01-29 09:02:02.000000000 +0000
+++ pbori_routines_misc.h	2010-08-10 13:05:31.684597593 +0100
@@ -1081,7 +1081,8 @@
 PolyType
 substitute_variables(const MapType& idx2poly, const PolyType& poly) {
 
-  return substitute_variables__(idx2poly, poly.navigation(), poly*0);
+  assert(!idx2poly.empty());
+  return substitute_variables__(idx2poly, poly.navigation(), idx2poly[0]*0);
 } 
 
 
