--- ../src/polybori-0.7/SConstruct	2011-04-27 13:00:31.000000000 +0200
+++ SConstruct	2011-10-08 21:18:19.720087957 +0200
@@ -117,7 +117,7 @@
     def DefaultBuild(arg):
         return arg
 
-defaultenv = Environment()
+defaultenv = Environment(ENV = os.environ)
 if defaultenv['PLATFORM'] == "darwin":
     defaultenv['SHCCFLAGS'] += ["-fvisibility=hidden"]
     
@@ -169,6 +169,8 @@
 opts.Add('CXXFLAGS', "C++ compiler flags", "-std=c++98 -ftemplate-depth-100",
          converter = Split)
 
+opts.Add('M4RI_CFLAGS', "C compiler flags for M4RI", converter = Split) 
+
 opts.Add('LINKFLAGS', "Linker flags", defaultenv['LINKFLAGS'] + ['-s'])
 opts.Add('LIBS', 'custom libraries needed for build', [], converter = Split)
 
@@ -238,7 +240,7 @@
 
 if defaultenv['PLATFORM'] == "sunos":  # forcing gcc, keeping linker
     def is_gcc():
-        compilerenv = Environment(options = opts)
+        compilerenv = Environment(ENV = os.environ, options = opts)
         return compilerenv['CC']  == 'gcc'
     
     if is_gcc():
@@ -247,7 +249,7 @@
             if arg in tools:
                 tools.remove(arg)
         tools +=  [ 'gcc', 'g++', 'ar']
-        defaultenv = Environment(tools=tools)
+        defaultenv = Environment(ENV = os.environ, tools=tools)
 
 for var in Split("""CCCOM CXXCOM SHCCCOM SHCXXCOM SHLINKCOM LINKCOM LINK SHLINK
 SHLIBPREFIX LIBPREFIX SHLIBSUFFIX LIBSUFFIX"""):
@@ -270,17 +272,10 @@
 if not GetOption('clean'):
     tools +=  ["disttar", "doxygen"]
 
-# Get paths an related things from current environment
-# todo: Are these settings sane in any case?
-getenv = dict()
-for key in ['PATH', 'HOME', 'LD_LIBRARY_PATH'] :
-    try:
-        getenv[key] = os.environ[key]
-    except KeyError:
-        pass
-
-
-env = Environment(ENV = getenv, options = opts, tools = tools, toolpath = '.')
+# Get paths and related things from current environment os.environ
+# note: We cannot avoid those due to non-standard system setups,
+#       also we do not know which variables are used in general
+env = Environment(ENV = os.environ, options = opts, tools = tools, toolpath = '.')
 
 if 'dump' in COMMAND_LINE_TARGETS:
   print env.Dump()
@@ -370,9 +365,40 @@
 external_m4ri = False
 
 if not env.GetOption('clean'):
-    conf = Configure(env)
+    def GuessM4RIFlags(context):
+        context.Message('Guessing m4ri compile flags... ')
+        test_src =  """
+        #include <m4ri/%s>
+        #include <stdio.h>
+        int main(int argc, char **argv) {
+          /* we test for some possible current and future configurations */
+          %s
+          return 0;
+        }
+        """  %  \
+        ("%s", ''.join(["""
+        #if defined(__M4RI_HAVE_%(macro)s)
+          if (__M4RI_HAVE_%(macro)s)
+            printf("-m%(option)s ");
+        #endif""" % \
+        dict(macro=opt.replace('.','_').upper(), option=opt) for opt in \
+            Split("sse sse2 sse3 sse4 sse4.1 sse4.2 sse4a ssse3 mmx 3dnow") ]) )
+        (result, values) = context.TryRun(test_src % "m4ri_config.h", '.c')
+	if result != 1:
+            (result, values) = context.TryRun(test_src % "config.h", '.c')
+
+        result = (result == 1)
+        if result:
+            context.Display(values)
+            env.Append(M4RI_CFLAGS=Split(values))
+
+        context.Result(result)
+        return result
+
+    conf = Configure(env, custom_tests = {'GuessM4RIFlags': GuessM4RIFlags})
+
     if conf.CheckCHeader("gd.h") and conf.CheckLib("gd"):
-        env.Append(LIBS=["gd"])
+        env.Append(LIBS=["gd","png12","z"])
         env.Append(CPPDEFINES=["HAVE_GD"])
 
     if env['FORCE_HASH_MAP']:
@@ -436,7 +462,7 @@
                 print "Tutorial will not be installed"
     external_m4ri = conf.CheckLib('m4ri')
     if conf.CheckCHeader("gd.h") and conf.CheckLib("gd"):
-        env.Append(LIBS=["gd"])
+        env.Append(LIBS=["gd","png12","z"])
         env.Append(CPPDEFINES=["HAVE_GD"])
     if external_m4ri:
        env['LIBS'] += ['m4ri']
@@ -448,6 +474,8 @@
            print "Symlinking to", m4ri_inc, "..."
            os.symlink('.', m4ri_inc)
 
+    conf.GuessM4RIFlags()
+
     env = conf.Finish()
 
 
@@ -493,6 +521,12 @@
 
 env.Append(BUILDERS={'SymLink' : symlinkbld})
 
+env.Append(CCFLAGS="$M4RI_CFLAGS")
+env.Append(CXXFLAGS="$M4RI_CFLAGS")
+env.Append(SHCCFLAGS="$M4RI_CFLAGS")
+env.Append(SHCXXFLAGS="$M4RI_CFLAGS")
+
+
 ######################################################################
 # Stuff for building Cudd library
 ######################################################################
@@ -505,8 +539,7 @@
 
 env.Append(LIBPATH=[CuddPath()])
 
-cudd_resources = [CuddPath('obj/cuddObj.cc')]
-cudd_resources += glob(CuddPath('util/*.c'))
+cudd_resources = glob(CuddPath('util/*.c'))
 
 cudd_headers = [ CuddPath(fname) for fname in ['obj/cuddObj.hh', 'util/util.h',
                                                'cudd/cuddInt.h'] ] 
@@ -565,9 +598,11 @@
 slib = env.SharedLibrary
 if env['SHLIBVERSIONING']:
     slib = VersionatedSharedLibrary
-#if env['PLATFORM']=="darwin":
-#    slib=env.LoadableModule
-
+# since "slib = VersionatedSharedLibrary" is broken on OSX 10.4 we use the 
+# following. We are currently not using shared libraries anyway due to
+# bugs in the deallocation of the various PolyBoRi libraries
+if env['PLATFORM']=="darwin":
+    slib=env.LoadableModule
 
 libCuddShared = slib(CuddPath(cudd_name), list(shared_resources))
 
@@ -1319,7 +1354,7 @@
 
     if HAVE_PYTHON_EXTENSION or extern_python_ext:
         cmdline = """$PYTHON -c "import compileall; compileall.compile_dir('"""
-        cmdline += InstPyPath() + """', ddir = ''); """ 
+        cmdline += InstPyPath('polybori') + """', ddir = ''); """ 
         cmdline += """compileall.compile_dir('"""+ InstPath(GUIPath())
         cmdline += """', ddir='')" """
         FinalizeNonExecs(env.Command([file.path + 'c' for file in pyfiles],
