--- src/polybori-0.3/SConstruct	2008-03-13 10:32:54.000000000 -0400
+++ patches/SConstruct	2008-04-09 21:04:05.000000000 -0400
@@ -102,7 +102,9 @@
 opts.Add('CCFLAGS', "C compiler flags", "-O3 -ansi", converter = Split)
 opts.Add('CXXFLAGS', "C++ compiler flags", "$CCFLAGS -ftemplate-depth-100",
          converter = Split)
-opts.Add('LINKFLAGS', "Linker flags", ['-s'], converter = Split)
+#The following linker flags causes the build to fail on OSX 10.4. but works on 10.5
+#opts.Add('LINKFLAGS', "Linker flags", ['-s'], converter = Split)
+#End modifications
 opts.Add('LIBS', 'custom libraries needed for build', [], converter = Split)
 
 opts.Add('PREFIX', 'installation prefix directory', '/usr/local')
