--- ../src/polybori-0.8.0/SConstruct	2011-09-07 22:25:22.000000000 +0200
+++ SConstruct	2011-09-28 23:10:26.333657937 +0200
@@ -529,7 +529,8 @@
         }
         """  %  \
         ("%s", ''.join(["""
-        #if defined(__M4RI_HAVE_%(macro)s) || defined(HAVE_%(macro)s )
+        #if (defined(__M4RI_HAVE_%(macro)s) && (__M4RI_HAVE_%(macro)s)) || \
+          defined(HAVE_%(macro)s)
           printf("-m%(option)s ");
         #endif""" % \
         dict(macro=opt.replace('.','_').upper(), option=opt) for opt in \
