diff -ur src/Python/pythonrun.c new/Python/pythonrun.c
--- src/Python/pythonrun.c	2012-12-04 20:43:32.000000000 +0000
+++ new/Python/pythonrun.c	2012-12-27 16:33:24.552894299 +0000
@@ -37,13 +37,7 @@
 #include "windows.h"
 #endif
 
-#ifndef Py_REF_DEBUG
 #define PRINT_TOTAL_REFS()
-#else /* Py_REF_DEBUG */
-#define PRINT_TOTAL_REFS() fprintf(stderr,                              \
-                   "[%" PY_FORMAT_SIZE_T "d refs]\n",                   \
-                   _Py_GetRefTotal())
-#endif
 
 #ifdef __cplusplus
 extern "C" {
Only in new/Python: pythonrun.c~
