diff --git a/Lib/distutils/unixccompiler.py b/Lib/distutils/unixccompiler.py
index 3af540e..625f155 100644
--- a/Lib/distutils/unixccompiler.py
+++ b/Lib/distutils/unixccompiler.py
@@ -238,7 +238,7 @@ class UnixCCompiler(CCompiler):
             return ["+s", "-L" + dir]
         elif sys.platform[:7] == "irix646" or sys.platform[:6] == "osf1V5":
             return ["-rpath", dir]
-        elif self._is_gcc(compiler):
+        elif sys.platform[:5] == "linux" or self._is_gcc(compiler):
             return "-Wl,-R" + dir
         else:
             return "-R" + dir
