--- src.orig/Makefile.pre.in	2013-04-06 15:02:34.000000000 +0100
+++ src/Makefile.pre.in	2013-04-07 12:59:46.675111329 +0100
@@ -59,7 +59,7 @@
 # Shared libraries must be installed with executable mode on some systems;
 # rather than figuring out exactly which, we always give them executable mode.
 # Also, making them read-only seems to be a good idea...
-INSTALL_SHARED= ${INSTALL} -m 555
+INSTALL_SHARED= ${INSTALL} -m 755
 
 MKDIR_P=	@MKDIR_P@
 
