--- src/Lib/socket.py.orig	2011-05-20 15:28:49.980525832 +1200
+++ src/Lib/socket.py	2011-05-20 15:28:53.943299694 +1200
@@ -51,6 +51,9 @@
 
 try:
     import _ssl
+    from _ssl import SSLError as sslerror
+    # we try this second line since sometimes the first
+    # passes even though the module isn't there
 except ImportError:
     # no SSL support
     pass
@@ -64,7 +67,6 @@
         return _realssl.sslwrap_simple(sock, keyfile, certfile)
 
     # we need to import the same constants we used to...
-    from _ssl import SSLError as sslerror
     from _ssl import \
          RAND_add, \
          RAND_egd, \
