diff --git a/setup.py b/setup.py
index d594e02..e34994d 100644
--- a/setup.py
+++ b/setup.py
@@ -1136,7 +1136,7 @@ class PyBuildExt(build_ext):
         dbm_setup_debug = False   # verbose debug prints from this script?
         dbm_order = ['gdbm']
         # The standard Unix dbm module:
-        if host_platform not in ['cygwin']:
+        if host_platform not in ['win32']:
             config_args = [arg.strip("'")
                            for arg in sysconfig.get_config_var("CONFIG_ARGS").split()]
             dbm_args = [arg for arg in config_args
@@ -1191,6 +1191,15 @@ class PyBuildExt(build_ext):
                                     ],
                                 libraries = gdbm_libs)
                             break
+                        if find_file("ndbm.h", inc_dirs, []) is not None:
+                            print("building dbm using gdbm")
+                            dbmext = Extension(
+                                '_dbm', ['_dbmmodule.c'],
+                                define_macros=[
+                                    ('HAVE_NDBM_H', None),
+                                    ],
+                                libraries = gdbm_libs)
+                            break
                 elif cand == "bdb":
                     if db_incs is not None:
                         if dbm_setup_debug: print("building dbm using bdb")
