diff -ru a/Lib/distutils/unixccompiler.py b/Lib/distutils/unixccompiler.py
--- a/Lib/distutils/unixccompiler.py	2018-06-27 05:07:35.000000000 +0200
+++ b/Lib/distutils/unixccompiler.py	2018-06-28 09:32:52.719006731 +0200
@@ -234,7 +234,7 @@
                 return ["-Wl,+s", "-L" + dir]
             return ["+s", "-L" + dir]
         else:
-            if self._is_gcc(compiler):
+            if sys.platform[:5] == "linux" or self._is_gcc(compiler):
                 # gcc on non-GNU systems does not need -Wl, but can
                 # use it anyway.  Since distutils has always passed in
                 # -Wl whenever gcc was used in the past it is probably
