diff --git a/SuiteSparse_config/SuiteSparse_config.mk b/SuiteSparse_config/SuiteSparse_config.mk
index 2fe1ab09..352ff593 100644
--- a/SuiteSparse_config/SuiteSparse_config.mk
+++ b/SuiteSparse_config/SuiteSparse_config.mk
@@ -370,8 +370,8 @@ SUITESPARSE_VERSION = 5.10.1
         # command line in the Terminal, before doing 'make':
         # xcode-select --install
         CF += -fno-common
-        BLAS ?= -framework Accelerate
-        LAPACK ?= -framework Accelerate
+        #BLAS ?= -framework Accelerate
+        #LAPACK ?= -framework Accelerate
         # OpenMP is not yet supported by default in clang
         CFOPENMP =
         LDLIBS += -rpath $(INSTALL_LIB)
