diff --git a/SuiteSparse_config/SuiteSparse_config.mk b/SuiteSparse_config/SuiteSparse_config.mk
index 352ff593..61f0076e 100644
--- a/SuiteSparse_config/SuiteSparse_config.mk
+++ b/SuiteSparse_config/SuiteSparse_config.mk
@@ -72,9 +72,9 @@ SUITESPARSE_VERSION = 5.10.1
     # which puts the libraries in /yada/mylibs, include files in /yoda/myinc,
     # and documentation in /solo/mydox.
     INSTALL ?= $(SUITESPARSE)
-    INSTALL_LIB ?= $(INSTALL)/lib
-    INSTALL_INCLUDE ?= $(INSTALL)/include
-    INSTALL_DOC ?= $(INSTALL)/share/doc/suitesparse-$(SUITESPARSE_VERSION)
+    INSTALL_LIB ?= $(DESTDIR)$(INSTALL)/lib
+    INSTALL_INCLUDE ?= $(DESTDIR)$(INSTALL)/include
+    INSTALL_DOC ?= $(DESTDIR)$(INSTALL)/share/doc/suitesparse-$(SUITESPARSE_VERSION)
 
     #---------------------------------------------------------------------------
     # parallel make
@@ -358,7 +358,7 @@ SUITESPARSE_VERSION = 5.10.1
 
     ifeq ($(UNAME),Linux)
         # add the realtime library, librt, and SuiteSparse/lib
-        LDLIBS += -lrt -Wl,-rpath=$(INSTALL_LIB)
+        LDLIBS += -lrt
     endif
 
     #---------------------------------------------------------------------------
@@ -464,7 +464,7 @@ else
         SO_TARGET = $(LIBRARY).$(VERSION).dylib
         SO_OPTS  += -dynamiclib -compatibility_version $(SO_VERSION) \
                     -current_version $(VERSION) \
-                    -Wl,-install_name -Wl,@rpath/$(SO_MAIN) \
+                    -Wl,-install_name -Wl,$(INSTALL)/lib/$(SO_MAIN) \
                     -shared -undefined dynamic_lookup
         # When a Mac *.dylib file is moved, this command is required
         # to change its internal name to match its location in the filesystem:
