diff --git a/AMD/Lib/Makefile b/AMD/Lib/Makefile
index 3b92f6a0..582bef2a 100644
--- a/AMD/Lib/Makefile
+++ b/AMD/Lib/Makefile
@@ -2,7 +2,7 @@
 # AMD Lib/Makefile
 #-------------------------------------------------------------------------------
 
-LIBRARY = libamd
+LIBRARY = amd
 VERSION = 2.4.6
 SO_VERSION = 2
 
@@ -81,19 +81,22 @@ libamdf77.a: $(AMDF77)
 #-------------------------------------------------------------------------------
 
 # install AMD
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(OBJ)
+$(INSTALL_SO): $(OBJ)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CC) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
+	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
 	$(CP) ../Include/amd.h $(INSTALL_INCLUDE)
 	$(CP) ../Doc/AMD_UserGuide.pdf $(INSTALL_DOC)
 	$(CP) ../README.txt $(INSTALL_DOC)/AMD_README.txt
-	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
 	chmod 644 $(INSTALL_INCLUDE)/amd.h
 	chmod 644 $(INSTALL_DOC)/AMD_UserGuide.pdf
 	chmod 644 $(INSTALL_DOC)/AMD_README.txt
diff --git a/BTF/Lib/Makefile b/BTF/Lib/Makefile
index 85b7a264..6e4e2ec4 100644
--- a/BTF/Lib/Makefile
+++ b/BTF/Lib/Makefile
@@ -2,7 +2,7 @@
 # BTF Lib/Makefile
 #-------------------------------------------------------------------------------
 
-LIBRARY = libbtf
+LIBRARY = btf
 VERSION = 1.2.6
 SO_VERSION = 1
 
@@ -66,18 +66,21 @@ btf_l_strongcomp.o: ../Source/btf_strongcomp.c
 #-------------------------------------------------------------------------------
 
 # install BTF
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(OBJ)
+$(INSTALL_SO): $(OBJ)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CC) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
+	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
 	$(CP) ../Include/btf.h $(INSTALL_INCLUDE)
 	$(CP) ../README.txt $(INSTALL_DOC)/BTF_README.txt
-	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
 	chmod 644 $(INSTALL_INCLUDE)/btf.h
 	chmod 644 $(INSTALL_DOC)/BTF_README.txt
 
diff --git a/CAMD/Lib/Makefile b/CAMD/Lib/Makefile
index b3ae159e..27cb072a 100644
--- a/CAMD/Lib/Makefile
+++ b/CAMD/Lib/Makefile
@@ -2,7 +2,7 @@
 # CAMD Lib/Makefile
 #-------------------------------------------------------------------------------
 
-LIBRARY = libcamd
+LIBRARY = camd
 VERSION = 2.4.6
 SO_VERSION = 2
 
@@ -62,19 +62,22 @@ $(AR_TARGET): $(OBJ)
 #-------------------------------------------------------------------------------
 
 # install CAMD
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(OBJ)
+$(INSTALL_SO): $(OBJ)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CC) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
+	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
 	$(CP) ../Include/camd.h $(INSTALL_INCLUDE)
 	$(CP) ../Doc/CAMD_UserGuide.pdf $(INSTALL_DOC)
 	$(CP) ../README.txt $(INSTALL_DOC)/CAMD_README.txt
-	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
 	chmod 644 $(INSTALL_INCLUDE)/camd.h
 	chmod 644 $(INSTALL_DOC)/CAMD_UserGuide.pdf
 	chmod 644 $(INSTALL_DOC)/CAMD_README.txt
diff --git a/CCOLAMD/Lib/Makefile b/CCOLAMD/Lib/Makefile
index 52c52eb9..18190be0 100644
--- a/CCOLAMD/Lib/Makefile
+++ b/CCOLAMD/Lib/Makefile
@@ -2,7 +2,7 @@
 # CCOLAMD Lib/Makefile
 #-------------------------------------------------------------------------------
 
-LIBRARY = libccolamd
+LIBRARY = ccolamd
 VERSION = 2.9.6
 SO_VERSION = 2
 
@@ -49,18 +49,21 @@ distclean: clean
 	- $(RM) -r $(PURGE)
 
 # install CCOLAMD
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(OBJ)
+$(INSTALL_SO): $(OBJ)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CC) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
+	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
 	$(CP) ../Include/ccolamd.h $(INSTALL_INCLUDE)
 	$(CP) ../README.txt $(INSTALL_DOC)/CCOLAMD_README.txt
-	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
 	chmod 644 $(INSTALL_INCLUDE)/ccolamd.h
 	chmod 644 $(INSTALL_DOC)/CCOLAMD_README.txt
 
diff --git a/CHOLMOD/Lib/Makefile b/CHOLMOD/Lib/Makefile
index bc3eab69..d240b4cd 100644
--- a/CHOLMOD/Lib/Makefile
+++ b/CHOLMOD/Lib/Makefile
@@ -2,7 +2,7 @@
 # CHOLMOD/Lib/Makefile: for compiling the CHOLMOD library
 #===============================================================================
 
-LIBRARY = libcholmod
+LIBRARY = cholmod
 VERSION = 3.0.14
 SO_VERSION = 3
 
@@ -535,20 +535,23 @@ cholmod_l_gpu.o: ../GPU/cholmod_gpu.c
 #-------------------------------------------------------------------------------
 
 # install CHOLMOD
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(OBJ)
+$(INSTALL_SO): $(OBJ)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CXX) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
+	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
 	$(CP) ../Include/cholmod*.h $(INSTALL_INCLUDE)
 	$(RM) $(INSTALL_INCLUDE)/cholmod_internal.h
 	$(CP) ../Doc/CHOLMOD_UserGuide.pdf $(INSTALL_DOC)
 	$(CP) ../README.txt $(INSTALL_DOC)/CHOLMOD_README.txt
-	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
 	chmod 644 $(INSTALL_INCLUDE)/cholmod*.h
 	chmod 644 $(INSTALL_DOC)/CHOLMOD_UserGuide.pdf
 	chmod 644 $(INSTALL_DOC)/CHOLMOD_README.txt
diff --git a/COLAMD/Lib/Makefile b/COLAMD/Lib/Makefile
index 2d0c5aa6..26720b62 100644
--- a/COLAMD/Lib/Makefile
+++ b/COLAMD/Lib/Makefile
@@ -2,7 +2,7 @@
 # COLAMD Lib/Makefile
 #-------------------------------------------------------------------------------
 
-LIBRARY = libcolamd
+LIBRARY = colamd
 VERSION = 2.9.6
 SO_VERSION = 2
 
@@ -49,18 +49,21 @@ distclean: clean
 	- $(RM) -r $(PURGE)
 
 # install COLAMD
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(OBJ)
+$(INSTALL_SO): $(OBJ)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CC) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
+	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
 	$(CP) ../Include/colamd.h $(INSTALL_INCLUDE)
 	$(CP) ../README.txt $(INSTALL_DOC)/COLAMD_README.txt
-	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
 	chmod 644 $(INSTALL_INCLUDE)/colamd.h
 	chmod 644 $(INSTALL_DOC)/COLAMD_README.txt
 
diff --git a/CSparse/Lib/Makefile b/CSparse/Lib/Makefile
index e3f32cec..468f7a47 100644
--- a/CSparse/Lib/Makefile
+++ b/CSparse/Lib/Makefile
@@ -14,7 +14,7 @@
 # install' in this Makefile only installs a static compiled library in
 # CSparse/Lib.  It does not install it for system-wide usage.
 
-LIBRARY = libcsparse
+LIBRARY = csparse
 CF = $(CFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -O
 
 I = -I../Include
diff --git a/CXSparse/Lib/Makefile b/CXSparse/Lib/Makefile
index a21e3607..270b8027 100644
--- a/CXSparse/Lib/Makefile
+++ b/CXSparse/Lib/Makefile
@@ -2,7 +2,7 @@
 # CXSparse Lib/Makefile
 #-------------------------------------------------------------------------------
 
-LIBRARY = libcxsparse
+LIBRARY = cxsparse
 VERSION = 3.2.0
 SO_VERSION = 3
 
@@ -113,18 +113,21 @@ distclean: clean
 	- $(RM) -r $(PURGE)
 
 # install CXSparse
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(CS)
+$(INSTALL_SO): $(CS)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CC) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
+	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
 	$(CP) ../Include/cs.h $(INSTALL_INCLUDE)
 	$(CP) ../README.txt $(INSTALL_DOC)/CXSPARSE_README.txt
-	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
 	chmod 644 $(INSTALL_INCLUDE)/cs.h
 	chmod 644 $(INSTALL_DOC)/CXSPARSE_README.txt
 
diff --git a/CXSparse_newfiles/Lib/Makefile b/CXSparse_newfiles/Lib/Makefile
index a21e3607..270b8027 100644
--- a/CXSparse_newfiles/Lib/Makefile
+++ b/CXSparse_newfiles/Lib/Makefile
@@ -2,7 +2,7 @@
 # CXSparse Lib/Makefile
 #-------------------------------------------------------------------------------
 
-LIBRARY = libcxsparse
+LIBRARY = cxsparse
 VERSION = 3.2.0
 SO_VERSION = 3
 
@@ -113,18 +113,21 @@ distclean: clean
 	- $(RM) -r $(PURGE)
 
 # install CXSparse
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(CS)
+$(INSTALL_SO): $(CS)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CC) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
+	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
 	$(CP) ../Include/cs.h $(INSTALL_INCLUDE)
 	$(CP) ../README.txt $(INSTALL_DOC)/CXSPARSE_README.txt
-	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
 	chmod 644 $(INSTALL_INCLUDE)/cs.h
 	chmod 644 $(INSTALL_DOC)/CXSPARSE_README.txt
 
diff --git a/GPUQREngine/Lib/Makefile b/GPUQREngine/Lib/Makefile
index adf46039..7bb34807 100644
--- a/GPUQREngine/Lib/Makefile
+++ b/GPUQREngine/Lib/Makefile
@@ -2,7 +2,7 @@
 # GPUQREngine/Lib/Makefile: for compiling the GPUQREngine library
 #-------------------------------------------------------------------------------
 
-LIBRARY = libGPUQREngine
+LIBRARY = GPUQREngine
 VERSION = 1.0.5
 SO_VERSION = 1
 
@@ -129,17 +129,20 @@ $(AR_TARGET): $(OBJS)
 #-------------------------------------------------------------------------------
 
 # install GPUQREngine.  Note that the include files are not installed.
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(OBJS)
+$(INSTALL_SO): $(OBJS)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CXX) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
-	$(CP) ../README.txt $(INSTALL_DOC)/GPUQRENGINE_README.txt
 	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
+	$(CP) ../README.txt $(INSTALL_DOC)/GPUQRENGINE_README.txt
 	chmod 644 $(INSTALL_DOC)/GPUQRENGINE_README.txt
 
 # uninstall GPUQREngine
diff --git a/KLU/Lib/Makefile b/KLU/Lib/Makefile
index 2dc62cb4..35e2ed3a 100644
--- a/KLU/Lib/Makefile
+++ b/KLU/Lib/Makefile
@@ -2,7 +2,7 @@
 # KLU Lib/Makefile
 #-------------------------------------------------------------------------------
 
-LIBRARY = libklu
+LIBRARY = klu
 VERSION = 1.3.8
 SO_VERSION = 1
 
@@ -263,19 +263,22 @@ klu_l_memory.o: ../Source/klu_memory.c
 #-------------------------------------------------------------------------------
 
 # install KLU
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(OBJ)
+$(INSTALL_SO): $(OBJ)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CC) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
+	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
 	$(CP) ../Include/klu.h $(INSTALL_INCLUDE)
 	$(CP) ../Doc/KLU_UserGuide.pdf $(INSTALL_DOC)
 	$(CP) ../README.txt $(INSTALL_DOC)/KLU_README.txt
-	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
 	chmod 644 $(INSTALL_INCLUDE)/klu.h
 	chmod 644 $(INSTALL_DOC)/KLU_UserGuide.pdf
 	chmod 644 $(INSTALL_DOC)/KLU_README.txt
diff --git a/LDL/Lib/Makefile b/LDL/Lib/Makefile
index 604ffa27..da5ecd53 100644
--- a/LDL/Lib/Makefile
+++ b/LDL/Lib/Makefile
@@ -2,7 +2,7 @@
 # LDL Lib/Makefile 
 #-------------------------------------------------------------------------------
 
-LIBRARY = libldl
+LIBRARY = ldl
 VERSION = 2.2.6
 SO_VERSION = 2
 
@@ -46,19 +46,22 @@ clean:
 	- $(RM) -r $(CLEAN)
 
 # install LDL
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(OBJ)
+$(INSTALL_SO): $(OBJ)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CC) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
+	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
 	$(CP) ../Include/ldl.h $(INSTALL_INCLUDE)
 	$(CP) ../Doc/ldl_userguide.pdf $(INSTALL_DOC)
 	$(CP) ../README.txt $(INSTALL_DOC)/LDL_README.txt
-	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
 	chmod 644 $(INSTALL_INCLUDE)/ldl.h
 	chmod 644 $(INSTALL_DOC)/ldl_userguide.pdf
 	chmod 644 $(INSTALL_DOC)/LDL_README.txt
diff --git a/RBio/Lib/Makefile b/RBio/Lib/Makefile
index 056715de..4ba42157 100644
--- a/RBio/Lib/Makefile
+++ b/RBio/Lib/Makefile
@@ -2,7 +2,7 @@
 # RBio/Lib/Makefile: for compiling the RBio library
 #===============================================================================
 
-LIBRARY = librbio
+LIBRARY = rbio
 VERSION = 2.2.6
 SO_VERSION = 2
 
@@ -60,18 +60,21 @@ RBio_i.o: ../Source/RBio.c
 #-------------------------------------------------------------------------------
 
 # install RBio
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(OBJ)
+$(INSTALL_SO): $(OBJ)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CC) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
+	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
 	$(CP) ../Include/RBio.h $(INSTALL_INCLUDE)
 	$(CP) ../README.txt $(INSTALL_DOC)/RBIO_README.txt
-	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
 	chmod 644 $(INSTALL_INCLUDE)/RBio.h
 	chmod 644 $(INSTALL_DOC)/RBIO_README.txt
 
diff --git a/SLIP_LU/Lib/Makefile b/SLIP_LU/Lib/Makefile
index 8b2221f1..b2153824 100644
--- a/SLIP_LU/Lib/Makefile
+++ b/SLIP_LU/Lib/Makefile
@@ -8,7 +8,7 @@
 # To run a demo using the library
 #       cd ../Demo ; make
 
-LIBRARY = libsliplu
+LIBRARY = sliplu
 VERSION = 1.0.2
 SO_VERSION = 1
 
@@ -63,19 +63,22 @@ $(AR_TARGET): $(OBJ)
 #-------------------------------------------------------------------------------
 
 # install SLIP_LU
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(OBJ)
+$(INSTALL_SO): $(OBJ)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CC) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
+	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
 	$(CP) ../Include/SLIP_LU.h $(INSTALL_INCLUDE)
 	$(CP) ../Doc/SLIP_LU_UserGuide.pdf $(INSTALL_DOC)
 	$(CP) ../README.md $(INSTALL_DOC)/SLIP_LU_README.md
-	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
 	chmod 644 $(INSTALL_INCLUDE)/SLIP_LU.h
 	chmod 644 $(INSTALL_DOC)/SLIP_LU_UserGuide.pdf
 	chmod 644 $(INSTALL_DOC)/SLIP_LU_README.md
diff --git a/SPQR/Lib/Makefile b/SPQR/Lib/Makefile
index 06c142ae..10e32007 100644
--- a/SPQR/Lib/Makefile
+++ b/SPQR/Lib/Makefile
@@ -2,7 +2,7 @@
 # SuiteSparseQR/Lib/Makefile
 #===============================================================================
 
-LIBRARY = libspqr
+LIBRARY = spqr
 VERSION = 2.0.9
 SO_VERSION = 2
 
@@ -242,22 +242,25 @@ spqrgpu_computeFrontStaging.o: ../SPQRGPU/spqrgpu_computeFrontStaging.cpp
 #-------------------------------------------------------------------------------
 
 # install SPQR
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(OBJ)
+$(INSTALL_SO): $(OBJ)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CXX) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
+	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
 	$(CP) ../Include/SuiteSparseQR.hpp $(INSTALL_INCLUDE)
 	$(CP) ../Include/SuiteSparseQR_C.h $(INSTALL_INCLUDE)
 	$(CP) ../Include/SuiteSparseQR_definitions.h $(INSTALL_INCLUDE)
 	$(CP) ../Include/spqr.hpp $(INSTALL_INCLUDE)
 	$(CP) ../Doc/spqr_user_guide.pdf $(INSTALL_DOC)
 	$(CP) ../README.txt $(INSTALL_DOC)/SPQR_README.txt
-	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
 	chmod 644 $(INSTALL_INCLUDE)/SuiteSparseQR.hpp
 	chmod 644 $(INSTALL_INCLUDE)/SuiteSparseQR_C.h
 	chmod 644 $(INSTALL_INCLUDE)/SuiteSparseQR_definitions.h
diff --git a/SuiteSparse_GPURuntime/Lib/Makefile b/SuiteSparse_GPURuntime/Lib/Makefile
index 31f536a5..2f6c54d0 100644
--- a/SuiteSparse_GPURuntime/Lib/Makefile
+++ b/SuiteSparse_GPURuntime/Lib/Makefile
@@ -2,7 +2,7 @@
 # SuiteSparse_GPURuntime/Lib/Makfile
 #-------------------------------------------------------------------------------
 
-LIBRARY = libSuiteSparse_GPURuntime
+LIBRARY = SuiteSparse_GPURuntime
 VERSION = 1.0.5
 SO_VERSION = 1
 
@@ -70,17 +70,20 @@ SuiteSparseGPU_Workspace_transfer.o: ../Source/SuiteSparseGPU_Workspace_transfer
 #-------------------------------------------------------------------------------
 
 # install SuiteSparse_GPURuntime (just the library, not the include files)
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(OBJS)
+$(INSTALL_SO): $(OBJS)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CXX) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
-	$(CP) ../README.txt $(INSTALL_DOC)/GPURUNTIME_README.txt
 	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
+	$(CP) ../README.txt $(INSTALL_DOC)/GPURUNTIME_README.txt
 	chmod 644 $(INSTALL_DOC)/GPURUNTIME_README.txt
 
 # uninstall SuiteSparse_GPURuntime
diff --git a/SuiteSparse_config/Makefile b/SuiteSparse_config/Makefile
index 6a5814d4..e2acc53e 100644
--- a/SuiteSparse_config/Makefile
+++ b/SuiteSparse_config/Makefile
@@ -6,7 +6,7 @@ SUITESPARSE ?= $(realpath $(CURDIR)/..)
 export SUITESPARSE
 
 # version of SuiteSparse_config is also version of SuiteSparse meta-package
-LIBRARY = libsuitesparseconfig
+LIBRARY = suitesparseconfig
 VERSION = 5.10.1
 SO_VERSION = 5
 
@@ -44,19 +44,23 @@ clean:
 	- $(RM) -r $(CLEAN)
 
 # install SuiteSparse_config
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(OBJ)
+$(INSTALL_SO): $(OBJ)
 	@mkdir -p $(INSTALL_LIB)
+# Likely redundant with the above but not on Cygwin
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CC) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
-	$(CP) SuiteSparse_config.h $(INSTALL_INCLUDE)
-	$(CP) README.txt $(INSTALL_DOC)/SUITESPARSECONFIG_README.txt
 	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
 	chmod 755 $(INSTALL_LIB)/$(SO_PLAIN)
+endif
+	$(CP) SuiteSparse_config.h $(INSTALL_INCLUDE)
+	$(CP) README.txt $(INSTALL_DOC)/SUITESPARSECONFIG_README.txt
 	chmod 644 $(INSTALL_INCLUDE)/SuiteSparse_config.h
 	chmod 644 $(INSTALL_DOC)/SUITESPARSECONFIG_README.txt
 
diff --git a/SuiteSparse_config/SuiteSparse_config.mk b/SuiteSparse_config/SuiteSparse_config.mk
index 61f0076e..5caa89dd 100644
--- a/SuiteSparse_config/SuiteSparse_config.mk
+++ b/SuiteSparse_config/SuiteSparse_config.mk
@@ -16,14 +16,14 @@ SUITESPARSE_VERSION = 5.10.1
     # To disable these auto configurations, use 'make UNAME=custom'
 
     ifndef UNAME
-        ifeq ($(OS),Windows_NT)
-            # Cygwin Make on Windows has an $(OS) variable, but not uname.
-            # Note that this option is untested.
-            UNAME = Windows
-        else
-            # Linux and Darwin (Mac OSX) have been tested.
-            UNAME := $(shell uname)
-        endif
+        # Linux and Darwin (Mac OSX) have been tested.
+        UNAME := $(shell uname)
+    endif
+
+    # On Cygwin we'll typically have UNAME=CYGWIN<something> but just normalize
+    # to "CYGWIN"
+    ifeq ($(findstring CYGWIN,$(UNAME)),CYGWIN)
+        UNAME := Cygwin
     endif
 
 #===============================================================================
@@ -170,6 +170,11 @@ SUITESPARSE_VERSION = 5.10.1
     # It places its shared *.so libraries in SuiteSparse/lib.
     # Linux also requires the -lrt library (see below)
     LDLIBS ?= -lm
+    # Note: Because suitesparse doesn't really do install staging properly
+    # (it just outputs final build artifacts directly to their install paths)
+    # we must add that path to our linker flags in order to link properly
+    # against built libraries; it might be better to fix the whole build
+    # system though).
     LDFLAGS += -L$(INSTALL_LIB)
 
     # NOTE: Use of the Intel MKL BLAS is strongly recommended.  The OpenBLAS can
@@ -447,21 +452,23 @@ SUITESPARSE_VERSION = 5.10.1
 
 SO_OPTS = $(LDFLAGS)
 
-ifeq ($(UNAME),Windows)
-    # Cygwin Make on Windows (untested)
-    AR_TARGET = $(LIBRARY).lib
-    SO_PLAIN  = $(LIBRARY).dll
-    SO_MAIN   = $(LIBRARY).$(SO_VERSION).dll
-    SO_TARGET = $(LIBRARY).$(VERSION).dll
-    SO_INSTALL_NAME = echo
+ifeq ($(UNAME),Cygwin)
+    # Cygwin Make on Windows
+    AR_TARGET = lib$(LIBRARY).a
+    SO_TARGET = cyg$(LIBRARY)-$(SO_VERSION).dll
+    IMPLIB = lib$(LIBRARY).dll.a
+    SO_OPTS += -shared -Wl,--no-undefined -Wl,--out-implib -Wl,$(INSTALL_LIB)/$(IMPLIB)
+    INSTALL_DLL := $(DESTDIR)$(INSTALL)/bin
+    INSTALL_SO = $(INSTALL_DLL)/$(SO_TARGET)
 else
     # Mac or Linux/Unix
-    AR_TARGET = $(LIBRARY).a
+    AR_TARGET = lib$(LIBRARY).a
+    INSTALL_SO = $(INSTALL_LIB)/$(SO_TARGET)
     ifeq ($(UNAME),Darwin)
         # Mac
-        SO_PLAIN  = $(LIBRARY).dylib
-        SO_MAIN   = $(LIBRARY).$(SO_VERSION).dylib
-        SO_TARGET = $(LIBRARY).$(VERSION).dylib
+        SO_PLAIN  = lib$(LIBRARY).dylib
+        SO_MAIN   = lib$(LIBRARY).$(SO_VERSION).dylib
+        SO_TARGET = lib$(LIBRARY).$(VERSION).dylib
         SO_OPTS  += -dynamiclib -compatibility_version $(SO_VERSION) \
                     -current_version $(VERSION) \
                     -Wl,-install_name -Wl,$(INSTALL)/lib/$(SO_MAIN) \
@@ -471,9 +478,9 @@ else
         SO_INSTALL_NAME = install_name_tool -id
     else
         # Linux and other variants of Unix
-        SO_PLAIN  = $(LIBRARY).so
-        SO_MAIN   = $(LIBRARY).so.$(SO_VERSION)
-        SO_TARGET = $(LIBRARY).so.$(VERSION)
+        SO_PLAIN  = lib$(LIBRARY).so
+        SO_MAIN   = lib$(LIBRARY).so.$(SO_VERSION)
+        SO_TARGET = lib$(LIBRARY).so.$(VERSION)
         SO_OPTS  += -shared -Wl,-soname -Wl,$(SO_MAIN) -Wl,--no-undefined
         # Linux/Unix *.so files can be moved without modification:
         SO_INSTALL_NAME = echo
diff --git a/SuiteSparse_config/xerbla/Makefile b/SuiteSparse_config/xerbla/Makefile
index db68a2ea..c59820c2 100644
--- a/SuiteSparse_config/xerbla/Makefile
+++ b/SuiteSparse_config/xerbla/Makefile
@@ -16,9 +16,9 @@ library:
 all: library
 
 ifeq ($(USE_FORTRAN),0)
-    LIBRARY = libcerbla
+    LIBRARY = cerbla
 else
-    LIBRARY = libxerbla
+    LIBRARY = xerbla
 endif
 
 include ../SuiteSparse_config.mk
@@ -44,19 +44,22 @@ $(AR_TARGET): $(DEPENDS)
 	- $(RM) xerbla.o
 
 # install libcerbla / libxerbla
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(DEPENDS)
+$(INSTALL_SO): $(DEPENDS)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(COMPILE)
 	$(CC) $(SO_OPTS) xerbla.o -o $@
 	- $(RM) xerbla.o
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
-	$(CP) xerbla.h $(INSTALL_INCLUDE)
 	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
+	$(CP) xerbla.h $(INSTALL_INCLUDE)
 	chmod 644 $(INSTALL_INCLUDE)/xerbla.h
 
 # uninstall libcerbla / libxerbla
diff --git a/UMFPACK/Lib/Makefile b/UMFPACK/Lib/Makefile
index 78436de1..0f291068 100644
--- a/UMFPACK/Lib/Makefile
+++ b/UMFPACK/Lib/Makefile
@@ -2,7 +2,7 @@
 # UMFPACK Makefile for compiling on Unix systems
 #-------------------------------------------------------------------------------
 
-LIBRARY = libumfpack
+LIBRARY = umfpack
 VERSION = 5.7.9
 SO_VERSION = 5
 
@@ -288,20 +288,23 @@ clean:
 
 #-------------------------------------------------------------------------------
 # install UMFPACK
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) $(INSTALL_SO)
 
-$(INSTALL_LIB)/$(SO_TARGET): $(OBJ)
+$(INSTALL_SO): $(OBJ)
 	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(dir $(INSTALL_SO))
 	@mkdir -p $(INSTALL_INCLUDE)
 	@mkdir -p $(INSTALL_DOC)
 	$(CC) $(SO_OPTS) $^ -o $@ $(LDLIBS)
+ifneq ($(UNAME),Cygwin)
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_PLAIN) )
 	( cd $(INSTALL_LIB) ; ln -sf $(SO_TARGET) $(SO_MAIN) )
+	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
+endif
 	$(CP) ../Include/umfpack*.h $(INSTALL_INCLUDE)
 	$(CP) ../Doc/UMFPACK_UserGuide.pdf $(INSTALL_DOC)
 	$(CP) ../Doc/UMFPACK_QuickStart.pdf $(INSTALL_DOC)
 	$(CP) ../README.txt $(INSTALL_DOC)/UMFPACK_README.txt
-	chmod 755 $(INSTALL_LIB)/$(SO_TARGET)
 	chmod 644 $(INSTALL_INCLUDE)/umfpack*.h
 	chmod 644 $(INSTALL_DOC)/UMFPACK_UserGuide.pdf
 	chmod 644 $(INSTALL_DOC)/UMFPACK_QuickStart.pdf
