--- a/Makefile	2011-09-23 21:09:12.000000000 +0100
+++ b/Makefile	2016-06-30 07:47:38.314034517 +0100
@@ -70,8 +70,8 @@
   BIN_INSTALL_DIR = "/usr/local/bin/"
 endif
 
-cflags = $(CFLAGS) $(CPPFLAGS) -Wall -ansi -pedantic -I $(GMP_INC_DIR)	\
-         -Wno-uninitialized -Wno-unused-parameter
+cflags = $(CFLAGS) $(CPPFLAGS) -Wall -I $(GMP_INC_DIR)	\
+         -Wno-uninitialized -Wno-unused-parameter -std=c++11
 program = frobby
 library = libfrobby.a
 benchArgs = $(FROBBYARGS)
@@ -93,7 +93,7 @@
 ifeq ($(MODE), debug)
   rawSources := $(rawSources) $(rawTests)
   outdir = bin/debug/
-  cflags += -g -D DEBUG -fno-inline -Werror -Wextra -Wno-uninitialized \
+  cflags += -g -D DEBUG -fno-inline -Wextra -Wno-uninitialized \
             -Wno-unused-parameter
   MATCH=true
 endif
@@ -217,6 +217,7 @@
 	  $(patsubst $(outdir)main.o,,$(objs))
 else
 	ar crs bin/$(library) $(patsubst $(outdir)main.o,,$(objs))
+	$(RANLIB) bin/$(library)
 endif
 
 # Compile and output object files.
