diff -druN ntl.orig/src/DoConfig ntl/src/DoConfig
--- ntl.orig/src/DoConfig	2014-08-21 20:06:48.000000000 -0700
+++ ntl/src/DoConfig	2014-08-26 07:21:25.985701655 -0700
@@ -22,6 +22,7 @@
 'LDFLAGS'     => '',
 'LDLIBS'      => '-lm',
 'CPPFLAGS'    => '',
+'LIBTOOL_LINK_FLAGS' => '',
 
 'DEF_PREFIX'  => '/usr/local',
 
diff -druN ntl.orig/src/mfile ntl/src/mfile
--- ntl.orig/src/mfile	2014-08-21 20:06:48.000000000 -0700
+++ ntl/src/mfile	2014-08-26 07:22:53.585710415 -0700
@@ -40,6 +40,9 @@
 LIBTOOL=@{LIBTOOL}
 # libtool command
 
+LIBTOOL_LINK_FLAGS=@{LIBTOOL_LINK_FLAGS}
+# flags to pass to libtool when linking
+
 DEF_PREFIX=@{DEF_PREFIX}
 
 PREFIX=@{PREFIX}
@@ -361,7 +364,7 @@
 ntl.a:	$(OBJ) 
 @{LSTAT}	$(AR) $(ARFLAGS) ntl.a $(OBJ) #LSTAT
 @{LSTAT}	- $(RANLIB) ntl.a #LSTAT
-@{LSHAR}	$(LIBTOOL) --mode=link $(LINK) -o libntl.la $(OBJ:.o=.lo) $(GMP_OPT_LIBDIR) $(GMP_OPT_LIB) $(GF2X_OPT_LIBDIR) $(GF2X_OPT_LIB) $(LDLIBS) -rpath $(LIBDIR) -version-info `cat VERSION_INFO` #LSHAR
+@{LSHAR}	$(LIBTOOL) --mode=link $(LINK) $(LIBTOOL_LINK_FLAGS) -o libntl.la $(OBJ:.o=.lo) $(GMP_OPT_LIBDIR) $(GMP_OPT_LIB) $(GF2X_OPT_LIBDIR) $(GF2X_OPT_LIB) $(LDLIBS) -rpath $(LIBDIR) -version-info `cat VERSION_INFO` #LSHAR
 
 @{LSTAT}LCOMP= #LSTAT
 @{LSHAR}LCOMP=$(LIBTOOL) --mode=compile #LSHAR
