--- src.orig/src/DoConfig	2012-08-07 12:15:52.820361339 +0200
+++ src/src/DoConfig	2012-08-07 12:17:12.828358360 +0200
@@ -21,11 +21,12 @@
 'RANLIB'      => 'ranlib',
 'LIBTOOL'     => 'libtool',
 
-'LDFLAGS'     => '',
-'LDFLAGS_CXX' => '$(LDFLAGS)',
-'LDLIBS'      => '-lm',
-'LDLIBS_CXX'  => '$(LDLIBS)',
-'CPPFLAGS'    => '',
+'LDFLAGS'            => '',
+'LDFLAGS_CXX'        => '$(LDFLAGS)',
+'LDLIBS'             => '-lm -lstdc++',
+'LDLIBS_CXX'         => '$(LDLIBS)',
+'CPPFLAGS'           => '',
+'LIBTOOL_LINK_FLAGS' => '',
 
 'DEF_PREFIX'  => '/usr/local',
 
--- src/src/mfile	2009-08-14 14:53:11.000000000 +0200
+++ src/src/mfile	2013-06-05 20:43:25.845240069 +0200
@@ -68,6 +68,9 @@
 LIBTOOL=@{LIBTOOL}
 # libtool command
 
+LIBTOOL_LINK_FLAGS=@{LIBTOOL_LINK_FLAGS}
+# flags to pass to libtool when linking
+
 DEF_PREFIX=@{DEF_PREFIX}
 
 PREFIX=@{PREFIX}
@@ -370,7 +373,7 @@ 
 ntl.a:	$(OBJ) 
 @{LSTAT}	$(AR) $(ARFLAGS) ntl.a $(OBJ) #LSTAT
 @{LSTAT}	- $(RANLIB) ntl.a #LSTAT
-@{LSHAR}	$(LIBTOOL) --mode=link $(LINK_CXX) -o libntl.la $(OBJ:.o=.lo) $(GMP_OPT_LIBDIR) $(GMP_OPT_LIB) $(GF2X_OPT_LIBDIR) $(GF2X_OPT_LIB) $(LDLIBS_CXX) -rpath $(LIBDIR) -version-info `cat VERSION_INFO` #LSHAR
+@{LSHAR}	$(LIBTOOL) --mode=link $(LINK_CXX) $(LIBTOOL_LINK_FLAGS) -o libntl.la $(OBJ:.o=.lo) $(GMP_OPT_LIBDIR) $(GMP_OPT_LIB) $(GF2X_OPT_LIBDIR) $(GF2X_OPT_LIB) $(LDLIBS_CXX) -rpath $(LIBDIR) -version-info `cat VERSION_INFO` #LSHAR
 
 @{LSTAT}LCOMP= #LSTAT
 @{LSHAR}LCOMP=$(LIBTOOL) --mode=compile #LSHAR
