Let libtool build a shared lib on Cygwin.

diff -druN a/ntl/src/DoConfig b/ntl/src/DoConfig
--- a/ntl/src/DoConfig	2015-05-23 07:44:51.000000000 -0700
+++ b/ntl/src/DoConfig	2015-07-03 01:25:35.820272305 -0700
@@ -39,6 +39,7 @@
 'LDFLAGS'     => '',
 'LDLIBS'      => '-lm',
 'CPPFLAGS'    => '',
+'LIBTOOL_LINK_FLAGS' => '',
 
 'DEF_PREFIX'  => '/usr/local',
 
diff -druN a/ntl/src/mfile b/ntl/src/mfile
--- a/ntl/src/mfile	2015-05-23 07:44:51.000000000 -0700
+++ b/ntl/src/mfile	2015-07-03 01:26:13.120365725 -0700
@@ -36,6 +36,9 @@
 LIBTOOL=@{LIBTOOL}
 # libtool command
 
+LIBTOOL_LINK_FLAGS=@{LIBTOOL_LINK_FLAGS}
+# flags to pass to libtool when linking
+
 DEF_PREFIX=@{DEF_PREFIX}
 
 PREFIX=@{PREFIX}
@@ -365,7 +368,7 @@
 ntl.a:	$(OBJ) 
 @{LSTAT}	$(AR) $(ARFLAGS) ntl.a $(OBJ) #LSTAT
 @{LSTAT}	- $(RANLIB) ntl.a #LSTAT
-@{LSHAR}	$(LIBTOOL) --tag=CXX --mode=link $(LINK) -o libntl.la $(OBJ:.o=.lo) $(GMP_OPT_LIBDIR) $(GMP_OPT_LIB) $(GF2X_OPT_LIBDIR) $(GF2X_OPT_LIB) $(LDLIBS) -rpath $(LIBDIR) -version-info `cat VERSION_INFO` #LSHAR
+@{LSHAR}	$(LIBTOOL) --tag=CXX --mode=link $(LINK) $(LIBTOOL_LINK_FLAGS) -o libntl.la $(OBJ:.o=.lo) $(GMP_OPT_LIBDIR) $(GMP_OPT_LIB) $(GF2X_OPT_LIBDIR) $(GF2X_OPT_LIB) $(LDLIBS) -rpath $(LIBDIR) -version-info `cat VERSION_INFO` #LSHAR
 
 @{LSTAT}LCOMP= #LSTAT
 @{LSHAR}LCOMP=$(LIBTOOL) --tag=CXX --mode=compile #LSHAR
