--- mfile.orig	2008-05-11 14:05:47.000000000 -0700
+++ mfile	2008-05-11 14:07:29.000000000 -0700
@@ -282,8 +282,8 @@
 NTL_INCLUDE = -I../include -I.
 # NTL needs this to find its include files
 
-COMPILE = $(CC) $(NTL_INCLUDE) $(CPPFLAGS) $(CFLAGS) -c
-COMPILE_CXX = $(CXX) $(NTL_INCLUDE) $(CPPFLAGS) $(CXXFLAGS) -c
+COMPILE = $(CC) $(NTL_INCLUDE) $(CPPFLAGS) $(CFLAGS) $(FPICFLAG) -c
+COMPILE_CXX = $(CXX) $(NTL_INCLUDE) $(CPPFLAGS) $(CXXFLAGS) $(FPICFLAG) -c
 
 LINK = $(CC) $(NTL_INCLUDE) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS)
 LINK_CXX = $(CXX) $(NTL_INCLUDE) $(CPPFLAGS) $(CXXFLAGS) $(LDFLAGS_CXX)
@@ -413,12 +413,11 @@
 #
 #################################################################
 
-clobber:	
+clobber: clean
 	- rm ntl.a mach_desc.h ../include/NTL/mach_desc.h  GetTime.c 
 	- rm lip_gmp_aux_impl.h ../include/NTL/gmp_aux.h
-	- sh RemoveProg $(PROGS) MakeDesc TestGetTime gen_lip_gmp_aux gen_gmp_aux
-	- rm *.o
-	- rm -r small
+	- sh RemoveProg $(PROGS)
+	- rm libntl*.so*
 
 clean:
 	- rm *.o 
@@ -519,3 +518,26 @@
 ################################################################
 
 
+shobj: $(OBJ)
+shobj: FPICFLAG = -fPIC
+
+libntl.dylib:
+	- rm -f $(OBJ) # clean, preserving tuning parameters
+	make shobj
+	$(LINK_CXX) -fPIC -dynamiclib -undefined dynamic_lookup -o libntl.dylib $(OBJ) $(GMP_LIBDIR) $(GMP_LIB)
+
+libntl.dylib64:
+	- rm -f $(OBJ) # clean, preserving tuning parameters
+	make shobj
+	$(LINK_CXX) -fPIC -m64 -dynamiclib -undefined dynamic_lookup -o libntl.dylib $(OBJ) $(GMP_LIBDIR) $(GMP_LIB)
+
+libntl.dll:
+	- rm -f $(OBJ) # clean, preserving tuning parameters
+	make shobj
+	$(LINK_CXX) -fPIC -shared -o libntl.dll $(OBJ) $(GMP_LIBDIR) $(GMP_LIB)
+
+libntl.so: DIRNAME
+	- rm -f $(OBJ) # clean, preserving tuning parameters
+	make shobj
+	$(LINK_CXX) -fPIC -shared -Wl,-soname,lib`cat DIRNAME`.so -o lib`cat DIRNAME`.so $(OBJ) $(GMP_LIBDIR) $(GMP_LIB)
+	ln -s lib`cat DIRNAME`.so libntl.so
