diff --git a/lib-src-gmp/Makefile.am b/lib-src-gmp/Makefile.am
index b3ab11a..a9673b8 100644
--- a/lib-src-gmp/Makefile.am
+++ b/lib-src-gmp/Makefile.am
@@ -13,7 +13,8 @@ cddmp_f.c \
 cddio_f.c \
 cddlib_f.c \
 cddproj_f.c \
-setoper.c
+setoper.c \
+random.c
 
 include_HEADERS = \
 cdd.h \
@@ -24,5 +25,5 @@
 cddtypes_f.h \
 setoper.h
 
-AM_CPPFLAGS = -I$(gmpincludedir) -DGMPRATIONAL
+AM_CPPFLAGS = -DGMPRATIONAL
 AM_LDFLAGS = -version-info 0:0:0
