--- src/pkg_resources.py	2013-12-13 08:31:50.000000000 -0800
+++ src/pkg_resources.py	2014-02-04 15:24:58.000000000 -0800
@@ -959,19 +959,7 @@
 
         See Distribute #375 for more details.
         """
-        if os.name == 'nt' and not path.startswith(os.environ['windir']):
-            # On Windows, permissions are generally restrictive by default
-            #  and temp directories are not writable by other users, so
-            #  bypass the warning.
-            return
-        mode = os.stat(path).st_mode
-        if mode & stat.S_IWOTH or mode & stat.S_IWGRP:
-            msg = ("%s is writable by group/others and vulnerable to attack "
-                "when "
-                "used with get_resource_filename. Consider a more secure "
-                "location (set with .set_extraction_path or the "
-                "PYTHON_EGG_CACHE environment variable)." % path)
-            warnings.warn(msg, UserWarning)
+        return
 
     def postprocess(self, tempname, filename):
         """Perform any platform-specific postprocessing of `tempname`
