diff --git a/pkg/cohomolo-1.6.7/Makefile.in b/pkg/cohomolo-1.6.7/Makefile.in
index 96a6c75..ed054b6 100644
--- a/pkg/cohomolo-1.6.7/Makefile.in
+++ b/pkg/cohomolo-1.6.7/Makefile.in
@@ -2,8 +2,7 @@
 # all executables are put into the bin directory.
 BIN = bin/@GAPARCH@
 
-CC=gcc
-CFLAGS = -O
+CFLAGS += -O
 
 all:
 	mkdir -p bin/@GAPARCH@;
diff --git a/pkg/cohomolo-1.6.7/standalone/progs.d/findpres/makefile b/pkg/cohomolo-1.6.7/standalone/progs.d/findpres/makefile
index be60c44..0e6e662 100644
--- a/pkg/cohomolo-1.6.7/standalone/progs.d/findpres/makefile
+++ b/pkg/cohomolo-1.6.7/standalone/progs.d/findpres/makefile
@@ -1,5 +1,4 @@
-CC = gcc
-CFLAGS= -O2
+CFLAGS += -O2
 all:  makegp gprun conrun grrun
 .c.o:
 	${CC} -c $(CFLAGS) $*.c 
diff --git a/pkg/cohomolo-1.6.7/standalone/progs.d/makefile b/pkg/cohomolo-1.6.7/standalone/progs.d/makefile
index 799bcdb..b22d7aa 100644
--- a/pkg/cohomolo-1.6.7/standalone/progs.d/makefile
+++ b/pkg/cohomolo-1.6.7/standalone/progs.d/makefile
@@ -1,5 +1,4 @@
-CC = gcc
-CFLAGS= -O -fno-builtin
+CFLAGS += -O -fno-builtin
 BIN= ../exec.d
 gap:    ${BIN}/gprun ${BIN}/egrun ${BIN}/sylrun ${BIN}/normrun ${BIN}/conrun\
 	${BIN}/pcrun ${BIN}/scrun ${BIN}/selgen ${BIN}/matcalc ${BIN}/nqmrun\
diff --git a/pkg/cohomolo-1.6.7/standalone/progs.d/makefile.gap b/pkg/cohomolo-1.6.7/standalone/progs.d/makefile.gap
index 7258a2c..b8950a6 100644
--- a/pkg/cohomolo-1.6.7/standalone/progs.d/makefile.gap
+++ b/pkg/cohomolo-1.6.7/standalone/progs.d/makefile.gap
@@ -1,5 +1,4 @@
-CC = gcc
-CFLAGS= -O2
+CFLAGS += -O2
 EXEC= ../exec.d
 all:  ${EXEC}/makegp ${EXEC}/gprun ${EXEC}/egrun ${EXEC}/sylrun ${EXEC}/normrun\
 	${EXEC}/sylnorm ${EXEC}/conrun ${EXEC}/pcrun ${EXEC}/scrun\
