diff --git a/Makefile.in b/Makefile.in
index 23d4648..0859ac3 100644
--- a/pkg/guava-3.14/Makefile.in
+++ b/pkg/guava-3.14/Makefile.in
@@ -1,7 +1,6 @@
 GAPPATH=@GAPPATH@
 GAPARCH=@GAPARCH@
-CC      = gcc
-CFLAGS  = -O2
+CFLAGS  += -O2
 SRCDIR  = ./src/leon
 CJSRCDIR= ./src/ctjhai
 BINDIR = bin/$(GAPARCH)
diff --git a/src/leon/Makefile b/src/leon/Makefile
index 75d89e7..923cd5f 100644
--- a/pkg/guava-3.14/src/leon/Makefile
+++ b/pkg/guava-3.14/src/leon/Makefile
@@ -1,4 +1,4 @@
-COMPILE = gcc
+COMPILE = $(CC)
 CFLAGS  += -O2
 SRCDIR  = ./src
 COMPOPT = $(CPPFLAGS) $(CFLAGS) -c
diff --git a/src/leon/Makefile.in b/src/leon/Makefile.in
index 75d89e7..923cd5f 100644
--- a/pkg/guava-3.14/src/leon/Makefile.in
+++ b/pkg/guava-3.14/src/leon/Makefile.in
@@ -1,4 +1,4 @@
-COMPILE = gcc
+COMPILE = $(CC)
 CFLAGS  += -O2
 SRCDIR  = ./src
 COMPOPT = $(CPPFLAGS) $(CFLAGS) -c
