diff --git b/guava-3.12/src/leon/src/group.h a/guava-3.12/src/leon/src/group.h
index 1e86e00..0f6c021 100644
--- b/guava-3.12/src/leon/src/group.h
+++ a/guava-3.12/src/leon/src/group.h
@@ -103,11 +103,11 @@ that special symbol. */
 #define SCANF_Int_FORMAT  "%u"
 
 #ifndef MAX_NAME_LENGTH
-#define MAX_NAME_LENGTH      64
+#define MAX_NAME_LENGTH      256
 #endif
 
 #ifndef MAX_FILE_NAME_LENGTH
-#define MAX_FILE_NAME_LENGTH 60
+#define MAX_FILE_NAME_LENGTH 256 
 #endif
 
 #ifndef DEFAULT_MAX_BASE_SIZE
