Index: ntl-5.4.1.p6/src/DoConfig
===================================================================
--- ntl-5.4.1.p6.orig/src/DoConfig	2007-11-15 01:06:06.000000000 -0500
+++ ntl-5.4.1.p6/src/DoConfig	2007-11-15 01:06:19.000000000 -0500
@@ -26,7 +26,7 @@
 'PREFIX'      => '/usr/local',
 'LIBDIR'      => '$(PREFIX)/lib',
 'INCLUDEDIR'  => '$(PREFIX)/include',
-'DOCDIR'      => '$(PREFIX)/doc',
+'DOCDIR'      => '$(PREFIX)/share/doc',
 
 'GMP_PREFIX'  => undef,
 'GMP_INCDIR'  => undef,
