From 50ce55eea115b74eae9421de43a4405e254ef579 Mon Sep 17 00:00:00 2001
From: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date: Fri, 10 May 2019 14:59:44 +0200
Subject: [PATCH 3/3] Shell::Mock: Add compile_scope

---
 perllib/Polymake/Core/ShellMock.pm | 1 +
 1 file changed, 1 insertion(+)

diff --git a/perllib/Polymake/Core/ShellMock.pm b/perllib/Polymake/Core/ShellMock.pm
index 46d59654..137f0b24 100644
--- a/perllib/Polymake/Core/ShellMock.pm
+++ b/perllib/Polymake/Core/ShellMock.pm
@@ -34,6 +34,7 @@ use Polymake::Struct (
 
 sub term { shift }
 sub interactive { 0 }
+sub compile_scope { undef }
 
 sub complete {
    my ($self, $string)=@_;
-- 
2.19.0

