diff --git a/support/configure.pl b/support/configure.pl
index 334c0b4..796ae1e 100644
--- a/support/configure.pl
+++ b/support/configure.pl
@@ -503,7 +503,7 @@ if ($^O eq "darwin") {
       }
       print "ok\n";
       if ( $NeedsArchFlag ) {
-         $ARCHFLAGS="-arch $Platform";
+         $ARCHFLAGS="";
       } else {
          $ARCHFLAGS="";
       }
@@ -525,7 +525,7 @@ if ($^O eq "darwin") {
          }
       }
       if ( $NeedsArchFlag ) {
-         $ARCHFLAGS="-arch $Platform";
+         $ARCHFLAGS="";
       } else {
          $ARCHFLAGS="";
       }
@@ -738,7 +738,14 @@ Otherwise, reconfigure and reinstall your perl.
       # We also build a test program for libperl since e.g. on Debian based systems the
       # check for Config::Config{libperl} will not detect a missing libperl-dev package
       chomp(my $perlldflags = `$PERL -MExtUtils::Embed -e ldopts`);
-      my $build_error=build_test_program(<<'---', CXXflags => `$PERL -MExtUtils::Embed -e ccopts`, LDflags => "$ARCHFLAGS $perlldflags" );
+
+      my $allarch=qr/ -arch \s+ \S+ (?: \s+ -arch \s+ \S+)* /x;
+      $CXXflags =  `$PERL -MExtUtils::Embed -e ccopts` ;
+      $CXXflags =~ s/$allarch//o;
+      $LDflags = $perlldflags;
+      $LDflags =~ s/$allarch//o;
+
+      my $build_error=build_test_program(<<'---', CXXflags => $CXXflags , LDflags => $LDflags );
 #include <EXTERN.h>
 #include <perl.h>
 
