--- ../src/makefile	2005-03-03 12:07:30.000000000 -0500
+++ ./makefile	2013-02-14 12:20:29.174766134 -0500
@@ -12,6 +12,7 @@
 gflags = -c $(includedirs) -g
 
 cflags = $(gflags) # the default setting
+cflags = -c $(includedirs) -O2 -fPIC -g -DFAST -DALLTRUE
 
 ifdef optimize
 	NDEBUG = true
@@ -24,11 +25,35 @@
 
 cc = g++
 
-all: coxeter #clean
+all: coxeter executable
 
 coxeter: $(objects)
+ifeq ($(UNAME),Darwin)
+		$(CC) -dynamiclib -Wl,-headerpad_max_install_names,-undefined,dynamic_lookup,-compatibility_version,3.0,-current_version,3.0,-install_name,$(SAGE_LOCAL)/lib/libcoxeter3.dylib -o libcoxeter3.dylib $(objects)
+else
+		$(CC) -shared -Wl,-soname,libcoxeter3.so -o libcoxeter3.so $(objects) -lc
+endif
+
+
+executable: $(objects)
 	$(cc) -o coxeter $(objects)
 
+DATADIR="$$SAGE_LOCAL/coxeter/"
+INCLUDEDIR="$$SAGE_LOCAL/include/coxeter/"
+
+install: coxeter executable
+	cp coxeter "$$SAGE_LOCAL/bin/"
+	if [ $UNAME = "Darwin" ]; then                 \
+	    cp libcoxeter3.dylib "$$SAGE_LOCAL/lib/";  \
+	else                                           \
+	    cp libcoxeter3.so    "$$SAGE_LOCAL/lib/";  \
+	fi
+
+	mkdir -p $(DATADIR)
+	cp -r coxeter_matrices headers messages $(DATADIR)
+	mkdir -p $(INCLUDEDIR)
+	cp -r *.h *.hpp $(INCLUDEDIR)
+
 clean:
 	rm -f $(objects)
 
@@ -152,6 +177,7 @@
 posets.o: posets.cpp posets.h globals.h bits.h list.h memory.h \
   constants.h list.hpp error.h io.h wgraph.h interface.h automata.h \
   coxtypes.h minroots.h dotval.h graph.h type.h transducer.h
+sage.o: sage.cpp sage.h globals.h coxtypes.h coxgroup.h list.h schubert.h
 schubert.o: schubert.cpp schubert.h globals.h coxtypes.h io.h list.h \
   memory.h constants.h list.hpp error.h bits.h interface.h automata.h \
   minroots.h dotval.h graph.h type.h transducer.h stack.h stack.hpp
