diff -druN a/error.cpp b/error.cpp
--- a/error.cpp
+++ b/error.cpp
@@ -250,7 +250,7 @@ void error::Error(int number, ... )
       notCoxelt();
       break;
     case NOT_DESCENT: {
-      const char *const str = va_arg(ap,const char *const);
+      const char *str = va_arg(ap,const char *);
       notDescent(str);
       break;
     }
@@ -280,7 +280,7 @@ void error::Error(int number, ... )
       parNbrOverflow();
       break;
     case PARSE_ERROR: {
-      const char *const str = va_arg(ap,const char *const);
+      const char *str = va_arg(ap,const char *);
       parseError(str);
     }
       break;
diff -druN a/version.h b/version.h
--- a/version.h
+++ b/version.h
@@ -9,8 +10,8 @@
 #define VERSION_H
 
 namespace version {
-  char* const NAME = "Coxeter";
-  char* const VERSION = "3.0_beta2";
+  const char* const NAME = "Coxeter";
+  const char* const VERSION = "3.1.sage";
 };
 
 #endif

