Pass -no-undefined to libtool on Cygwin.
--- zeromq-4.0.4.old/configure.ac	2013-11-13 09:57:50.000000000 -0800
+++ zeromq-4.0.4.new/configure.ac	2014-11-24 06:35:03.896022989 -0800
@@ -248,6 +248,7 @@
         if test "x$enable_static" = "xyes"; then
             AC_MSG_ERROR([Building static libraries is not supported under Cygwin])
         fi
+        libzmq_on_cygwin="yes"
         ;;
     *)
         AC_MSG_ERROR([unsupported system: ${host_os}.])
@@ -429,6 +430,7 @@
 
 AM_CONDITIONAL(BUILD_PGM, test "x$libzmq_pgm_ext" = "xyes")
 AM_CONDITIONAL(ON_MINGW, test "x$libzmq_on_mingw32" = "xyes")
+AM_CONDITIONAL(ON_CYGWIN, test "x$libzmq_on_cygwin" = "xyes")
 AM_CONDITIONAL(ON_ANDROID, test "x$libzmq_on_android" = "xyes")
 AM_CONDITIONAL(ON_LINUX, test "x$libzmq_on_linux" = "xyes")
 
--- zeromq-4.0.4.old/src/Makefile.am	2014-03-10 14:41:29.000000000 -0700
+++ zeromq-4.0.4.new/src/Makefile.am	2014-11-24 06:36:21.106125358 -0800
@@ -175,6 +175,9 @@
 else
 if ON_LINUX
 libzmq_la_LDFLAGS = -version-info @LTVER@ @LIBZMQ_EXTRA_LDFLAGS@ -Wl,--version-script=$(srcdir)/libzmq.vers
+if ON_CYGWIN
+libzmq_la_LDFLAGS += -no-undefined
+endif
 else
 libzmq_la_LDFLAGS = -version-info @LTVER@ @LIBZMQ_EXTRA_LDFLAGS@ -Wl
 endif
