--- ../src/scipy/stats/mstats_basic.py	2009-02-11 09:55:44.000000000 +0700
+++ mstats_basic.py	2009-11-12 12:42:59.000000000 +0700
@@ -103,7 +103,7 @@
     """
     if len(args) == 1 and not isinstance(args[0], ndarray):
         output = ma.asarray(args[0])
-        assert(output.ndim == 2, "The input should be 2D!")
+        assert output.ndim == 2, "The input should be 2D!"
     else:
         n = len(args)
         m = max([len(k) for k in args])
