--- a/scipy/spatial/setup.py	2021-11-23 14:19:51.100109800 -0300
+++ b/scipy/spatial/setup.py	2022-03-06 23:01:17.458072945 -0300
@@ -40,7 +40,7 @@
     inc_dirs = [get_python_inc()]
     if inc_dirs[0] != get_python_inc(plat_specific=1):
         inc_dirs.append(get_python_inc(plat_specific=1))
-    inc_dirs.append(get_numpy_include_dirs())
+    inc_dirs.insert(0, get_numpy_include_dirs())
     inc_dirs.append(join(dirname(dirname(__file__)), '_lib'))
     inc_dirs.append(join(dirname(dirname(__file__)), '_build_utils', 'src'))
 
