From 28e5fcaeabe26d46167e75815e0d9a4fdfa1a973 Mon Sep 17 00:00:00 2001
From: "Martin R. Albrecht" <martinralbrecht@googlemail.com>
Date: Wed, 5 Dec 2018 11:47:33 +0000
Subject: [PATCH] set language_version

fixes #127
---
 setup.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 8f916a9..a5125f8 100755
--- a/setup.py
+++ b/setup.py
@@ -110,7 +110,9 @@ def run(self):
                 self.extensions,
                 include_path=["src"],
                 build_dir=self.cythonize_dir,
-                compiler_directives={'binding': True, "embedsignature": True})
+                compiler_directives={'binding': True,
+                                     'embedsignature': True,
+                                     'language_level': 2})
         super(build_ext, self).run()
 
     def _generate_config_pxi(self):
