diff -Naur singular-4.0.3/xalloc/configure.ac latest/xalloc/configure.ac
--- singular-4.0.3/xalloc/configure.ac	2016-08-20 02:56:16.000000000 +1200
+++ latest/xalloc/configure.ac	2016-09-12 14:26:25.761034425 +1200
@@ -99,6 +99,6 @@
 dnl lllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllllll
 dnl wrap it up
 dnl
-AC_CONFIG_FILES([Makefile])
+AC_CONFIG_FILES([Makefile omalloc.pc])
 AC_OUTPUT
 
diff -Naur singular-4.0.3/xalloc/Makefile.am latest/xalloc/Makefile.am
--- singular-4.0.3/xalloc/Makefile.am	2016-08-20 02:56:16.000000000 +1200
+++ latest/xalloc/Makefile.am	2016-09-12 14:28:44.561417546 +1200
@@ -21,3 +21,6 @@
 CLEANFILES = $(BUILT_SOURCES)
 DISTCLEANFILES = omConfig.h
 
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = omalloc.pc
+
diff -Naur singular-4.0.3/xalloc/omalloc.pc.in latest/xalloc/omalloc.pc.in
--- singular-4.0.3/xalloc/omalloc.pc.in	1970-01-01 12:00:00.000000000 +1200
+++ latest/xalloc/omalloc.pc.in	2016-09-12 14:25:21.020855411 +1200
@@ -0,0 +1,17 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: @PACKAGE@
+Description: The Singular memory manager
+Version: @PACKAGE_VERSION@
+URL: https://github.com/Singular/Sources/tree/spielwiese/@PACKAGE@
+
+# Requires:
+# Conflicts:
+
+Cflags: -I${includedir}
+Libs: -L${libdir} -lomalloc
+# Libs.private:
+
