--- patches/dietz-cu2-Makefile	2009-09-28 18:39:39.327006842 -0700
+++ src/dietz/cu2/Makefile	2008-02-17 17:02:11.000000000 -0800
@@ -1,5 +1,10 @@
 # makefile for cu2 with gcc on *nix by Eric
 
+CC=gcc
+CPP=g++
+LINK=g++
+CFLAGS=-O2
+LFLAGS=
 RM=rm -f
 INCLUDES=cu2.h config.h
 OBJS=cu2.o main.o
@@ -14,10 +19,10 @@
 debug: build-debug
 
 build: $(OBJS)
-	$(CXX) $(CXXFLAGS)  -o cu2  $(OBJS) $(LIBS)
+	$(LINK) $(LFLAGS)  -o cu2  $(OBJS) $(LIBS)
 
 build-debug: $(DBGOBJS)
-	$(CXX) $(DBGLFLAGS)  -o cu2  $(OBJS) $(DBGLIBS)
+	$(LINK) $(DBGLFLAGS)  -o cu2  $(OBJS) $(DBGLIBS)
 
 clean:
 	$(RM) $(OBJS)
@@ -26,15 +31,15 @@
 	$(RM) cu2
 
 main.o: main.cpp $(INCLUDES)
-	$(CXX) $(CXXFLAGS) -c main.cpp
+	$(CPP) $(CFLAGS) -c main.cpp
 
 cu2.o: cu2.cpp $(INCLUDES)
-	$(CXX) $(CXXFLAGS) -c cu2.cpp
+	$(CPP) $(CFLAGS) -c cu2.cpp
 
 main.o.dbg: main.cpp $(INCLUDES)
-	$(CXX) $(DBGCFLAGS) -c main.cpp
+	$(CPP) $(DBGCFLAGS) -c main.cpp
 
 cu2.o.dbg: cu2.cpp $(INCLUDES)
-	$(CXX) $(DBGCFLAGS) -c cu2.cpp
+	$(CPP) $(DBGCFLAGS) -c cu2.cpp
 
 dummy:
