--- rubiks-20070912/dietz/cu2/Makefile	2008-02-18 02:02:11.000000000 +0100
+++ dietz-cu2-Makefile	2015-12-21 20:13:22.404335387 +0100
@@ -1,10 +1,5 @@
 # makefile for cu2 with gcc on *nix by Eric
 
-CC=gcc
-CPP=g++
-LINK=g++
-CFLAGS=-O2
-LFLAGS=
 RM=rm -f
 INCLUDES=cu2.h config.h
 OBJS=cu2.o main.o
@@ -19,10 +14,10 @@
 debug: build-debug
 
 build: $(OBJS)
-	$(LINK) $(LFLAGS)  -o cu2  $(OBJS) $(LIBS)
+	$(CXX) $(CXXFLAGS)  -o cu2  $(OBJS) $(LIBS) $(LDFLAGS)
 
 build-debug: $(DBGOBJS)
-	$(LINK) $(DBGLFLAGS)  -o cu2  $(OBJS) $(DBGLIBS)
+	$(CXX) $(DBGLFLAGS)  -o cu2  $(OBJS) $(DBGLIBS) $(LDFLAGS)
 
 clean:
 	$(RM) $(OBJS)
@@ -31,15 +26,15 @@
 	$(RM) cu2
 
 main.o: main.cpp $(INCLUDES)
-	$(CPP) $(CFLAGS) -c main.cpp
+	$(CXX) $(CXXFLAGS) -c main.cpp
 
 cu2.o: cu2.cpp $(INCLUDES)
-	$(CPP) $(CFLAGS) -c cu2.cpp
+	$(CXX) $(CXXFLAGS) -c cu2.cpp
 
 main.o.dbg: main.cpp $(INCLUDES)
-	$(CPP) $(DBGCFLAGS) -c main.cpp
+	$(CXX) $(DBGCFLAGS) -c main.cpp
 
 cu2.o.dbg: cu2.cpp $(INCLUDES)
-	$(CPP) $(DBGCFLAGS) -c cu2.cpp
+	$(CXX) $(DBGCFLAGS) -c cu2.cpp
 
 dummy:
