--- Make-arch.orig	2009-01-20 03:56:18.000000000 -0800
+++ Make-arch	2009-01-20 03:52:23.000000000 -0800
@@ -869,6 +869,17 @@
 	"RANLIB = ranlib" \
 	"LIBS = -L. -ltachyon $(MISCLIB)"
 
+macosx-64:
+	$(MAKE) all \
+	"ARCH = macosx" \
+	"CC = cc" \
+	"CFLAGS = -Os -m64 -ffast-math -DBsd $(MISCFLAGS)" \
+	"AR = ar" \
+	"ARFLAGS = r" \
+	"STRIP = strip" \
+	"RANLIB = ranlib" \
+	"LIBS = -L. -ltachyon $(MISCLIB)"
+
 macosx-thr:
 	$(MAKE) all \
 	"ARCH = macosx-thr" \
@@ -1131,7 +1142,7 @@
 	$(MAKE) all \
 	"ARCH = linux-thr" \
 	"CC = gcc" \
-	"CFLAGS = -m32 -Wall -O6 -fomit-frame-pointer -ffast-math -DLinux -DTHR -D_REENTRANT $(MISCFLAGS)" \
+	"CFLAGS = -m32 -Wall -O6 -fomit-frame-pointer -ffast-math -DLinux -DTHR -D_REENTRANT $(MISCFLAGS) $(GCCFIX)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = strip" \
