--- ../src/unix/Make-arch	2007-05-06 12:12:48.000000000 +0700
+++ Make-arch	2009-11-06 12:01:57.000000000 +0700
@@ -869,6 +869,17 @@
 	"RANLIB = ranlib" \
 	"LIBS = -L. -ltachyon $(MISCLIB)"
 
+macosx-64:
+	$(MAKE) all \
+	"ARCH = macosx" \
+	"CC = cc" \
+	"CFLAGS = -Os -m64 -ffast-math -DBsd $(MISCFLAGS)" \
+	"AR = ar" \
+	"ARFLAGS = r" \
+	"STRIP = strip" \
+	"RANLIB = ranlib" \
+	"LIBS = -L. -ltachyon $(MISCLIB)"
+
 macosx-thr:
 	$(MAKE) all \
 	"ARCH = macosx-thr" \
@@ -956,7 +967,7 @@
 	$(MAKE) all \
 	"ARCH = win32" \
 	"CC = gcc" \
-	"CFLAGS = -mpentium -Wall -O6 -fomit-frame-pointer -ffast-math -DWIN32 $(MISCFLAGS)" \
+	"CFLAGS = -Wall -O6 -fomit-frame-pointer -ffast-math -DWIN32 $(MISCFLAGS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = echo" \
@@ -967,7 +978,7 @@
 	$(MAKE) all \
 	"ARCH = win32-mpi" \
 	"CC = gcc" \
-	"CFLAGS = -mpentium -Wall -O6 -fomit-frame-pointer -ffast-math -DWIN32 $(MISCFLAGS) -I'/Program files/MPIPro/INCLUDE' -DMPI" \
+	"CFLAGS = -Wall -O6 -fomit-frame-pointer -ffast-math -DWIN32 $(MISCFLAGS) -I'/Program files/MPIPro/INCLUDE' -DMPI" \
 	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = echo" \
@@ -1131,7 +1142,7 @@
 	$(MAKE) all \
 	"ARCH = linux-thr" \
 	"CC = gcc" \
-	"CFLAGS = -m32 -Wall -O6 -fomit-frame-pointer -ffast-math -DLinux -DTHR -D_REENTRANT $(MISCFLAGS)" \
+	"CFLAGS = -m32 -Wall -O6 -fomit-frame-pointer -ffast-math -DLinux -DTHR -D_REENTRANT $(MISCFLAGS) $(GCCFIX)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = strip" \
