--- a/unix/Makefile
+++ b/unix/Makefile
@@ -121,35 +121,35 @@ ${MGFLIB} : ../libmgf/Makefile
 	};
 
 ${ARCHDIR}/tachyon : ${RAYLIB} ${PARSELIB} ${OBJDIR}/main.o ${OBJDIR}/getargs.o ${OBJDIR}/parse.o ${OBJDIR}/nffparse.o ${OBJDIR}/glwin.o ${OBJDIR}/spaceball.o ${OBJDIR}/trackball.o ${PARSEOBJS} 
-	${CC} ${CFLAGS} ${DEMOINC} -o ${ARCHDIR}/tachyon ${OBJDIR}/main.o ${OBJDIR}/getargs.o ${OBJDIR}/parse.o ${OBJDIR}/nffparse.o ${OBJDIR}/glwin.o ${OBJDIR}/spaceball.o ${OBJDIR}/trackball.o ${PARSEOBJS} -L${RAYLIBDIR} ${PARSELIBS} ${LIBS}
+	${CC} ${CFLAGS} ${DEMOINC} -o ${ARCHDIR}/tachyon ${LDFLAGS} ${OBJDIR}/main.o ${OBJDIR}/getargs.o ${OBJDIR}/parse.o ${OBJDIR}/nffparse.o ${OBJDIR}/glwin.o ${OBJDIR}/spaceball.o ${OBJDIR}/trackball.o ${PARSEOBJS} -L${RAYLIBDIR} ${PARSELIBS} ${LIBS}
 	${STRIP} ${ARCHDIR}/tachyon
 
 ${ARCHDIR}/animray : ${RAYLIB} ${OBJDIR}/mainanim.o
-	${CC} ${CFLAGS} ${DEMOINC} -o ${ARCHDIR}/animray ${OBJDIR}/mainanim.o -L${RAYLIBDIR} ${LIBS}
+	${CC} ${CFLAGS} ${DEMOINC} -o ${ARCHDIR}/animray ${LDFLAGS} ${OBJDIR}/mainanim.o -L${RAYLIBDIR} ${LIBS}
 	${STRIP} ${ARCHDIR}/animray 
 
 ${ARCHDIR}/animspheres : ${RAYLIB} ${OBJDIR}/animspheres.o 
-	${CC} ${CFLAGS} ${DEMOINC} -o ${ARCHDIR}/animspheres ${OBJDIR}/animspheres.o -L${RAYLIBDIR} ${LIBS}
+	${CC} ${CFLAGS} ${DEMOINC} -o ${ARCHDIR}/animspheres ${LDFLAGS} ${OBJDIR}/animspheres.o -L${RAYLIBDIR} ${LIBS}
 	${STRIP} ${ARCHDIR}/animspheres
 
 ${ARCHDIR}/animspheres2 : ${RAYLIB} ${OBJDIR}/animspheres2.o 
-	${CC} ${CFLAGS} ${DEMOINC} -o ${ARCHDIR}/animspheres2 ${OBJDIR}/animspheres2.o -L${RAYLIBDIR} ${LIBS}
+	${CC} ${CFLAGS} ${DEMOINC} -o ${ARCHDIR}/animspheres2 ${LDFLAGS} ${OBJDIR}/animspheres2.o -L${RAYLIBDIR} ${LIBS}
 	${STRIP} ${ARCHDIR}/animspheres2
 
 ${ARCHDIR}/hypertex : ${RAYLIB} ${OBJDIR}/hypertex.o ${DEMOSRC}/hypertex.c 
-	${CC} ${CFLAGS} ${DEMOINC} -o ${ARCHDIR}/hypertex ${OBJDIR}/hypertex.o -L${RAYLIBDIR} ${LIBS}
+	${CC} ${CFLAGS} ${DEMOINC} -o ${ARCHDIR}/hypertex ${LDFLAGS} ${OBJDIR}/hypertex.o -L${RAYLIBDIR} ${LIBS}
 	${STRIP} ${ARCHDIR}/hypertex 
 
 ${ARCHDIR}/fire : ${RAYLIB} ${OBJDIR}/fire.o ${DEMOSRC}/fire.c 
-	${CC} ${CFLAGS} ${DEMOINC} -o ${ARCHDIR}/fire ${OBJDIR}/fire.o -L${RAYLIBDIR} ${LIBS}
+	${CC} ${CFLAGS} ${DEMOINC} -o ${ARCHDIR}/fire ${LDFLAGS} ${OBJDIR}/fire.o -L${RAYLIBDIR} ${LIBS}
 	${STRIP} ${ARCHDIR}/fire
 
 ${ARCHDIR}/animskull : ${RAYLIB} ${OBJDIR}/animskull.o 
-	${CC} ${CFLAGS} ${DEMOINC} -o ${ARCHDIR}/animskull ${OBJDIR}/animskull.o -L${RAYLIBDIR} ${LIBS}
+	${CC} ${CFLAGS} ${DEMOINC} -o ${ARCHDIR}/animskull ${LDFLAGS} ${OBJDIR}/animskull.o -L${RAYLIBDIR} ${LIBS}
 	${STRIP} ${ARCHDIR}/animskull
 
 ${ARCHDIR}/tgatoyuv : ${RAYLIB} ${DEMOSRC}/tgatoyuv.c 
-	${CC} ${CFLAGS} ${DEMOINC} -o ${ARCHDIR}/tgatoyuv ${DEMOSRC}/tgatoyuv.c -L${RAYLIBDIR} ${LIBS}
+	${CC} ${CFLAGS} ${DEMOINC} -o ${ARCHDIR}/tgatoyuv ${LDFLAGS} ${DEMOSRC}/tgatoyuv.c -L${RAYLIBDIR} ${LIBS}
 	${STRIP} ${ARCHDIR}/tgatoyuv
 
 #
