--- ../src/unix/Make-config	2007-01-24 00:35:44.000000000 -0800
+++ Make-config	2010-09-03 06:32:11.000000000 -0700
@@ -128,14 +128,10 @@
 #   http://www.libpng.org/
 ##########################################################################
 # Uncomment the following lines to disable PNG support
-USEPNG=
-PNGINC=
-PNGLIB=
-
-# Uncomment the following lines to enable PNG support
-#USEPNG= -DUSEPNG
-#PNGINC= -I/usr/local/include
-#PNGLIB= -L/usr/local/lib -lpng -lz
+USEPNG= -DUSEPNG
+PNGINC= -I$(SAGE_LOCAL)/include
+PNGLIB= -L$(SAGE_LOCAL)/lib -lpng12 -lz $(LDFLAGS)
+
 
 
 ##########################################################################
