--- a/unix/Make-config	2011-03-13 11:01:07.000000000 +0000
+++ b/unix/Make-config	2016-07-04 15:29:57.982923000 +0000
@@ -9,7 +9,7 @@
 #
 # Raytracer configuration variables
 #   Edit any of these settings as needed to configure directories
-#   and source areas for your system.  
+#   and source areas for your system.
 #   Important items to edit are the X11 configuration and MPI
 #
 
@@ -62,7 +62,7 @@
 # UMR CS Dept
 #MPIDIR=/software/all/mpi
 
-# Default Paragon XP/S location 
+# Default Paragon XP/S location
 #MPIDIR=/usr
 
 # Cornell Theory Center SP-2  splogin.tc.cornell.edu
@@ -72,7 +72,7 @@
 ##########################################################################
 # Floating point configuration:
 #   Leaving this blank will cause the library to use double precision floats
-#   Setting -DUSESINGLEFLT causes the library to use single precision floats 
+#   Setting -DUSESINGLEFLT causes the library to use single precision floats
 ##########################################################################
 # Uncomment the following line for double precision floating point math
 # uses about twice as much memory per object as single precision math.
@@ -86,7 +86,7 @@
 # Object mailbox storage configuration:
 #   Leaving this blank will cause the library to use auxiliary mailbox data
 #   structures for improving the effectiveness of grid-based ray tracing.
-#   Setting -DDISABLEMBOX will cause the library to disable this feature. 
+#   Setting -DDISABLEMBOX will cause the library to disable this feature.
 ##########################################################################
 # Uncomment the following line for full mailbox data structure use, this
 # uses a per-thread mailbox array, or either 4 or 8 bytes per scene object,
@@ -94,15 +94,15 @@
 MBOX=
 # Uncomment the following line to disable the use of mailbox data structures,
 # this eliminates per-thread storage normally allocated for the mailbox
-# data structure, but may incur a rendering speed penalty. 
+# data structure, but may incur a rendering speed penalty.
 #MBOX=-DDISABLEMBOX
 
 
 ##########################################################################
 # JPEG support configuration:
 #   JPEGINC is the directory where your Independent JPEG Group include files
-#   are made available.  JPEGLIB is the directory where your Independent 
-#   JPEG Group libraries are made available. 
+#   are made available.  JPEGLIB is the directory where your Independent
+#   JPEG Group libraries are made available.
 #
 # IJG JPEG library version 6b can be downloaded from:
 #   http://www.ijg.org/files/
@@ -122,29 +122,25 @@
 # PNG support configuration:
 #   PNGINC is the directory where your libpng and libz include files
 #   are made available.  PNGLIB is the directory where your libpng
-#   and libz libraries are made available. 
+#   and libz libraries are made available.
 #
 # LibPNG can be downlaoded from:
 #   http://www.libpng.org/
 ##########################################################################
 # Uncomment the following lines to disable PNG support
-USEPNG=
-PNGINC=
-PNGLIB=
+USEPNG= -DUSEPNG
+PNGINC= -I$(SAGE_LOCAL)/include
+PNGLIB= -L$(SAGE_LOCAL)/lib -lpng -lz
 
-# Uncomment the following lines to enable PNG support
-#USEPNG= -DUSEPNG
-#PNGINC= -I/usr/local/include
-#PNGLIB= -L/usr/local/lib -lpng -lz
 
 
 ##########################################################################
-# OMF (Open Media Framework) configuration 
+# OMF (Open Media Framework) configuration
 #   Requires OMF Toolkit version 2.x
 ##########################################################################
 #OMFDIR = /disk5/users/johns/graphics/OMFKT202/Toolkit
 #OMFINC = -I$(OMFDIR)/include -I$(OMFDIR)/kitomfi -I$(OMFDIR)/bento -I$(OMFDIR)/jpeg -I$(OMFDIR)/portinc -I$(OMFDIR)/avidjpg
-#OMFLIB = -L$(OMFDIR)/DO_sun5_opt/usr/lib -lAJPG -lOMFI -lbento -ljpeg 
+#OMFLIB = -L$(OMFDIR)/DO_sun5_opt/usr/lib -lAJPG -lOMFI -lbento -ljpeg
 #OMFDEF = -DUSEOMF
 
 
@@ -159,7 +155,7 @@
 
 ##########################################################################
 # Spaceball I/O library configuration:
-#   A spaceball can used for fly-throughs of scenes when running on  
+#   A spaceball can used for fly-throughs of scenes when running on
 #   a fast multiprocessor, parallel machine, or PC cluster.
 #
 # Libsball can be downloaded from:
@@ -172,7 +168,7 @@
 
 ##########################################################################
 # MGF Materials and Geometry Format scene parser library
-#   If enabled, this allows Tachyon to read MGF scene files using 
+#   If enabled, this allows Tachyon to read MGF scene files using
 #   compiled-in MGF scene parser code.
 ##########################################################################
 #MGFDIR=../../libmgf
