--- a/src/basemath/base2.c
+++ b/src/basemath/base2.c
@@ -3344,10 +3344,7 @@ polcompositum0(GEN A, GEN B, long flall)
     for (i=1; i<l; i++)
     { /* invmod possibly very costly */
       GEN D = gel(C,i), invH1 = QXQ_inv(H1, D);
-      if (typ(H0) == t_POL)
-        a = RgXQ_mul(RgX_neg(H0), invH1, D);
-      else /* scalar */
-        a = RgX_Rg_mul(invH1, gneg(H0));
+      a = RgXQ_mul(RgX_neg(H0), invH1, D);
       b = gadd(pol_x(v), gmulsg(k,a));
       gel(C,i) = mkvec4(D, mkpolmod(a,D), mkpolmod(b,D), stoi(-k));
     }
