commit d02e043959bd90e7354e7675633abb8de5f27857
Author: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date:   Mon Jan 9 16:06:02 2017 +0100

    Simplify handling of --graphic option and check validity

diff --git a/config/Makefile.SH b/config/Makefile.SH
index def1a9c..bba22ff 100644
--- a/config/Makefile.SH
+++ b/config/Makefile.SH
@@ -69,6 +69,7 @@ esac
 PLOTCFLAGS=
 PLOTLIBS=
 postconfig=:
+plotrunpath=
 case "$which_graphic_lib" in
 none)
   graph=plotnull;;
@@ -90,20 +91,18 @@ fltk)
 win32)
   PLOTLIBS="-lgdi32"
   graph=plotWin32;;
+X11)
+  PLOTCFLAGS="$PLOTCFLAGS $X11_INC"
+  PLOTLIBS="$PLOTLIBS $X11_LIBS"
+  plotrunpath=$X11
+  graph=plotX;;
+*)
+  echo >&2 "### Unrecognized graphic library '$which_graphic_lib'."
+  exit 1;;
 esac
 graph="plotport $graph"
 libgraph="plottty"
 
-plotrunpath=
-case "$which_graphic_lib" in
-  *X11*)
-    PLOTCFLAGS="$PLOTCFLAGS $X11_INC"
-    PLOTLIBS="$PLOTLIBS $X11_LIBS"
-    plotrunpath=$X11
-    graph="plotX $graph"
-  ;;
-esac
-
 KERNOBJS=
 for f in $kernel; do
   KERNOBJS="$KERNOBJS $f\$(_O)"
