diff -ur src.orig/config/Makefile.SH src/config/Makefile.SH
--- src.orig/config/Makefile.SH	2012-09-30 19:47:06.594410247 +0200
+++ src/config/Makefile.SH	2012-09-30 19:46:42.422591885 +0200
@@ -53,6 +53,12 @@
   darwin)
     ld_install_name="-Wl,-install_name -Wl,\"$libdir\"/\$(LIBPARI_DYN)";;
 esac
+case "$osname" in
+  mingw|cygwin)
+    install_implib="\$(INSTALL_DATA) \$(LIBPARI_SO)\$(_A) \$(LIBDIR)/\$(LIBPARI_SO)\$(_A)";;
+  *)
+    install_implib="";;
+esac
 
 PLOTCFLAGS=
 PLOTLIBS=
@@ -470,6 +476,8 @@
 	-mkdir -p \$(LIBDIR)
 	-\$(RM) \$(LIBDIR)/\$(LIBPARI_DYN) \$(LIBDIR)/\$(LIBPARI_SONAME) \$(LIBDIR)/\$(LIBPARI_SO)
 	\$(DLLD_IGNORE)\$(DLLD) -o \$(LIBDIR)/\$(LIBPARI_DYN) \$(DLLDFLAGS) \$(OBJS) \$(EXTRADLLDFLAGS) \$(GMPLIBS) \$(EXPORT_FILE) \$(LD_INSTALL_NAME)
+	$install_implib
+
 install-lib-dyn-link: install-lib-dyn-base
 	-if test "\$(LIBPARI_SONAME)" != "\$(LIBPARI_SO)"; then \
 	  \$(LN) \$(LIBPARI_DYN) \$(LIBDIR)/\$(LIBPARI_SONAME);\
