commit c8ddd6e4952a8731f5e99ff17cfa37cb430b47aa
Author: Bill Allombert <Bill.Allombert@math.u-bordeaux1.fr>
Date:   Tue Mar 28 10:42:54 2017 +0200

    GP was incorrectly issuing fflush(NULL) (From Erik Bray)

diff --git a/src/language/es.c b/src/language/es.c
index 38f3737..77bcf8f 100644
--- a/src/language/es.c
+++ b/src/language/es.c
@@ -532,7 +532,8 @@ _puts_log(const char *s)
   fputs(s, f);
 }
 static void
-_flush_log(void) { (void)fflush(pari_logfile); }
+_flush_log(void)
+{ if (pari_logfile != NULL) (void)fflush(pari_logfile); }
 
 static void
 normalOutC(char c) { putc(c, pari_outfile); _putc_log(c); }
