commit e87d56e7d48081c0f137af4838f887300e6ba344
Author: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date:   Wed Jan 2 14:35:24 2013 +0100

    galoisanalysis: only copy O[4] if n >= 4

diff --git a/src/basemath/galconj.c b/src/basemath/galconj.c
index ed1fdfb..8a2f071 100644
--- a/src/basemath/galconj.c
+++ b/src/basemath/galconj.c
@@ -1288,7 +1288,7 @@ galoisanalysis(GEN T, struct galois_analysis *ga, long calcul_l)
   ga->deg = deg;
   ga->ord = order;
   ga->l  = O[1];
-  ga->p4 = O[4];
+  ga->p4 = (n >= 4 ? O[4] : 0);
   if (DEBUGLEVEL >= 4)
     err_printf("GaloisAnalysis:p=%ld l=%ld group=%ld deg=%ld ord=%ld\n",
                plift, O[1], group, deg, order);
