diff -ru src/src/language/init.c b/src/language/init.c
--- src/src/language/init.c	2014-02-01 16:29:48.955825926 +0100
+++ b/src/language/init.c	2014-02-01 16:31:24.612818588 +0100
@@ -612,14 +612,16 @@
 pari_init_functions(void)
 {
   pari_stack_init(&s_MODULES, sizeof(*MODULES),(void**)&MODULES);
-  pari_stack_pushp(&s_MODULES,functions_basic);
   pari_stack_init(&s_OLDMODULES, sizeof(*OLDMODULES),(void**)&OLDMODULES);
-  pari_stack_pushp(&s_OLDMODULES,oldfonctions);
   functions_hash = (entree**) pari_calloc(sizeof(entree*)*functions_tblsz);
   pari_fill_hashtable(functions_hash,
                       new_fun_set? functions_basic: oldfonctions);
   defaults_hash = (entree**) pari_calloc(sizeof(entree*)*functions_tblsz);
   pari_add_defaults_module(functions_default);
+  /* Move this down to work around GCC bug
+   * http://gcc.gnu.org/bugzilla/show_bug.cgi?id=49330 */
+  pari_stack_pushp(&s_MODULES,functions_basic);
+  pari_stack_pushp(&s_OLDMODULES,oldfonctions);
 }
 
 void
