diff -ru src/config/get_fltk b/config/get_fltk
--- src/config/get_fltk	2014-02-01 16:36:02.492702307 +0100
+++ b/config/get_fltk	2014-02-01 16:43:22.183289684 +0100
@@ -16,8 +16,15 @@
   fi;;
 esac
 if test -n "$FLTKDIR"; then
+  if test -d "$FLTKDIR/include/FL"; then
   echo "Using FLTK library, FLTKDIR = $FLTKDIR"
   FLTK_LIBS="`fltk-config --ldflags` -lstdc++"
+  else
+    echo "### Found the FLTK library, but not the necessary headers."
+    echo "### Building without FLTK support."
+    FLTKDIR=""
+    FLTK_LIBS=""
+  fi
 else
   echo "### FLTK not found. Building without FLTK support"
   FLTK_LIBS=
