--- config/get_dlld	(revision 12541)
+++ config/get_dlld	(working copy)
@@ -67,7 +67,7 @@
   else # DLLD != GNU ld
     case "$osname" in
       aix)     DLLDFLAGS='-r' ;;
-      darwin)  DLLDFLAGS='-flat_namespace -undefined suppress' ;;
+      darwin)  DLLD=gcc; DLLDFLAGS='-flat_namespace -undefined suppress' ;;
       freebsd) DLLDFLAGS='-Bshareable -x' ;;
       hpux)    DLLDFLAGS='-b' ;;
       irix)    DLLDFLAGS='-shared -elf -no_unresolved -all' ;;
