--- ../src/config/get_kernel	2007-03-09 15:49:59.000000000 -0800
+++ get_kernel	2010-01-03 23:07:55.000000000 -0800
@@ -12,7 +12,7 @@
            4) asmarch=ix86;;
            8) asmarch=x86_64;;
            esac;                        pretty=$arch ;;
-  x86_64)  case "$sizeof_long" in   
+  x86_64|amd64)  case "$sizeof_long" in   
            4) asmarch=ix86;;
            8) asmarch=x86_64;;
            esac;                        pretty='amd64';;
