--- config/get_kernel	(revision 12541)
+++ config/get_kernel	(working copy)
@@ -12,7 +12,7 @@
            4) asmarch=ix86;;
            8) asmarch=x86_64;;
            esac;                        prettya=$arch ;;
-  x86_64)  case "$sizeof_long" in
+  x86_64|amd64)  case "$sizeof_long" in
            4) asmarch=ix86;;
            8) asmarch=x86_64;;
            esac;                        prettya='amd64';;
@@ -25,8 +25,8 @@
            8) asmarch=hppa64;           prettya='PA-RISC2.0';;
            esac;;
   alpha)         asmarch=$arch;         prettya=Alpha ;;
-  ppc|ppc64)  case "$sizeof_long" in
-        4) asmarch=ppc;;
+  ppc)  case "$sizeof_long" in
+        4) asmarch=$arch;;
         8) asmarch=ppc64;;
         esac;                           prettya='PowerPC' ;;
   arm*)          asmarch=none;          prettya=$arch ;;
