--- a/config/get_X11	2010-08-22 21:25:54.000000000 +0200
+++ b/config/get_X11	2010-09-06 11:05:14.000000000 +0200
@@ -28,20 +28,44 @@
   pth=`echo $x11pth | sed 's,/lib,/include,g'`
   x=`./locate X11/Xos.h '' $pth`
   case $x in
-   /*) Xincroot=`echo $x | sed 's,/X11/Xos.h,,'`;;
-   *)  Xincroot=;;
+   /*) Xincroot=`echo $x | sed 's,/X11/Xos.h,,'`
+       # echo "...Found X11 header files in \"$Xincroot/X11\"."
+       ;;
+   *)  Xincroot=""
+       echo "### Could not find X11 header files."
+       ;;
   esac
 fi
+
 # X11 -- Lib
-pth=$x11pth
-lib=X11; . ./locatelib
+# leif: ALSO search */lib64/* directories if we're doing a 64-bit build,
+#       and do this FIRST (since we might find a 32-bit library in */lib/*).
+#       E.g. on 64-bit Fedora 13, libX11.so is located in /usr/lib64, and
+#       /usr/lib is NOT a symbolic link to /usr/lib64.
+#       This still fails if ONLY a 32-bit X11 library is installed on a
+#       64-bit system, and the */lib/* directories aren't the 64-bit ones.
+if test "$sizeof_long" -eq 8; then
+  # 64-bit build
+  pth=`echo $x11pth | sed 's,/lib,/lib64,g'` 
+  lib=X11; . ./locatelib
+fi
+if test -z "$X11"; then
+  # X11 library not yet found, search */lib/* directories
+  pth=$x11pth
+  lib=X11; . ./locatelib
+fi
+if test -z "$X11"; then
+  echo "### Could not find X11 library."
+fi 
+ 
 if test -f $Xincroot/X11/Xos.h -a -n "$X11"; then
   echo ..."Found X11 header files in $Xincroot/X11"
+  echo ..."Found X11 library in $X11"
   test -n "$extralib" && echo ..."Extra Libraries are $extralib"
   X11_LIBS="-L$X11 -lX11 $extralib"
   X11_INC="-I$Xincroot"
 else
-  echo "### X11 not found"
+  echo "### Not all necessary parts of X11 found."
   X11=
 fi
 
