--- ../src/config/get_dlld	2009-09-19 05:01:14.000000000 -0700
+++ get_dlld	2010-03-05 20:36:54.000000000 -0800
@@ -65,7 +65,7 @@
   else # DLLD != GNU ld
     case "$osname" in
       aix)     DLLDFLAGS='-r' ;;
-      darwin)  DLLD=; DLLDFLAGS='-dylib -flat_namespace -undefined suppress' ;;
+      darwin)  DLLD=gcc; DLLDFLAGS='-flat_namespace -undefined suppress' ;;
       freebsd) DLLDFLAGS='-Bshareable -x' ;;
       hpux)    DLLDFLAGS='-b' ;;
       irix)    DLLDFLAGS='-shared -elf -no_unresolved -all' ;;
@@ -88,10 +88,14 @@
       sparc*) extra='-mimpure-text';;
       *) extra=;;
     esac
-    DLLDFLAGS="-shared $extra \$(CFLAGS) \$(DLCFLAGS) $DLLDFLAGS"
+    if test "$osname" = darwin; then
+        DLLDFLAGS="-dynamiclib $extra \$(CFLAGS) \$(DLCFLAGS) $DLLDFLAGS"
+    else
+        DLLDFLAGS="-shared $extra \$(CFLAGS) \$(DLCFLAGS) $DLLDFLAGS"
+    fi
   fi
   case "$osname" in
-    gnu*|cygwin|osf1|freebsd|linux|sunos|solaris) EXTRADLLDFLAGS='-lc ${LIBS}';;
+    gnu*|osf1|freebsd|linux|sunos|solaris) EXTRADLLDFLAGS='-lc ${LIBS}';;
   esac
 
   if test "$fastread" != yes; then
