--- a/config/get_config_options	2010-08-22 21:25:54.000000000 +0200
+++ b/config/get_config_options	2010-08-31 05:18:55.000000000 +0200
@@ -58,9 +58,20 @@
   -h|-help|--help|-\?)error=true;;
   -v|-verbhelp|--verbhelp)error=verb;;
   -s|--static) static=y;;
-  -graphic|--graphic) shift; which_graphic_lib=$1;;
+  -graphic|--graphic)
+      shift; which_graphic_lib="$1"
+      case "$which_graphic_lib" in
+        auto|none|X11|qt|Qt|fltk|win32) ;;
+        *) echo "*** Unrecognized graphic lib \"$which_graphic_lib.\"" >&2
+           error=true;;
+      esac;;
   --graphic=*|--graphi=*|--graph=*|--grap=*|--gra=*|--gr=*|--g=*)
-	       which_graphic_lib=`echo "$1" | sed -e 's/[-a-z]*=//'`;;
+      which_graphic_lib=`echo "$1" | sed -e 's/[-a-z]*=//'`
+      case "$which_graphic_lib" in
+        auto|none|X11|qt|Qt|fltk|win32) ;;
+        *) echo "*** Unrecognized graphic lib \"$which_graphic_lib.\"" >&2
+           error=true;;
+      esac;;
 
   --without-readline|--with-readline=no) without_readline=yes ;;
   --with-readline) with_readline=yes ;;
@@ -126,7 +137,7 @@
 Build Options:
   --host=<arch-osname>  target achitecture
   --kernel=<kern>       kernel used
-  --graphic=<gr>        graphic library used (none X11 Qt fltk win32)
+  --graphic=<gr>        graphic library used (none X11 qt Qt fltk win32 auto)
   --time=<fun>          timing function to use (getrusage times ftime)
   --builddir=<dir>      directory where the object files will be created
   --tune                tune the binary for compiling host (slow)
