--- a/config/get_fltk	2010-11-04 14:51:50.910510930 +0100
+++ b/config/get_fltk	2010-11-04 14:59:12.158786073 +0100
@@ -10,18 +10,24 @@
    fi
    ;;
 *) if test ! -d "$with_fltk"; then
-    echo "### FLTK directory '$with_fltk' not found"
+    echo "### Specified FLTK directory '$with_fltk' not found"
   else
     FLTKDIR=$with_fltk
   fi;;
 esac
 if test -n "$FLTKDIR"; then
-  echo "Using FLTK library, FLTKDIR = $FLTKDIR"
+  if test -d "$FLTKDIR/include/FL"; then
+    echo "Using FLTK library, FLTKDIR = \"$FLTKDIR\""
+  else
+    echo "### Found the FLTK library, but not the necessary headers."
+    echo "### Building without FLTK support."
+    FLTKDIR=""
+  fi
 else
   echo "### FLTK not found. Building without FLTK support"
 fi
 case "$osname" in
   cygwin) FLTK_LIBS="-lgdi32 -lole32 -luuid -lwsock32 -lsupc++" ;;
   darwin) FLTK_LIBS="-framework Carbon -lsupc++" ;;
-  *)      FLTK_LIBS="$X11_LIBS" ;;
+  *)      FLTK_LIBS="$X11_LIBS -lstdc++" ;;
 esac
