--- ../src/doc/gphelp.in	2006-03-06 10:05:00.000000000 -0800
+++ gphelp.in	2007-12-26 21:35:54.000000000 -0800
@@ -51,6 +51,8 @@
 $xdviref = $ENV{GPXDVIREF} || "$xdvi -paper 29.7x21cm";
 $gzip = "gzip";
 $zcat = "$gzip -dc";
+$bzip = "bzip2";
+$bzcat = "$bzip -dc";
 $docdir = &get_docdir();
 
 $refcard = (@ARGV and $ARGV[-1] =~ /refcard/i);
@@ -115,6 +117,8 @@
   $cu .= $cu ? "\e[1m": "\e[4m";
   $cb .= "\e[1m";
   $detex = 1 if (!$detex && !$ENV{DISPLAY});
+  # Disable TeX for Sage.
+  $detex = 1;
 }
 
 sub get_docdir {
@@ -175,8 +179,14 @@
     die "Cannot find $docfile"
       if (! -r "$docfile.z" &&
           ! -r "$docfile.gz" &&
-          ! -r "$docfile.Z");
-    $pipe = $zcat;
+          ! -r "$docfile.Z" &&
+	  ! -r "$docfile.bz2");
+    if (-r "$docfile.bz2") {
+	$pipe = $bzcat;
+	$docfile = "$docfile.bz2";
+    } else {
+	$pipe = $zcat;
+    }
   }
 }
 
