--- doc/gphelp.in	(revision 12541)
+++ doc/gphelp.in	(working copy)
@@ -51,6 +51,8 @@
 $xdviref = $ENV{GPXDVIREF} || "$xdvi -paper 29.7x21cm";
 $gzip = "gzip";
 $zcat = "$gzip -dc";
+$bzip = "bzip2";
+$bzcat = "$bzip -dc";
 $docdir = &get_docdir();
 
 $refcard = (@ARGV and $ARGV[-1] =~ /refcard/i);
@@ -112,6 +114,8 @@
   $cu .= $cu ? "\e[1m": "\e[4m";
   $cb .= "\e[1m";
   $detex = 1 if (!$ENV{DISPLAY});
+  # Disable TeX for Sage.
+  $detex = 1;
 }
 
 sub get_docdir {
@@ -171,8 +175,14 @@
     die "Cannot find $docfile"
       if (! -r "$docfile.z" &&
           ! -r "$docfile.gz" &&
-          ! -r "$docfile.Z");
-    $pipe = $zcat;
+          ! -r "$docfile.Z" &&
+	  ! -r "$docfile.bz2");
+    if (-r "$docfile.bz2") {
+	$pipe = $bzcat;
+	$docfile = "$docfile.bz2";
+    } else {
+	$pipe = $zcat;
+    }
   }
 }
 
