--- src/language/init.c	(revision 12541)
+++ src/language/init.c	(working copy)
@@ -19,6 +19,7 @@
 /*                                                                 */
 /*******************************************************************/
 #include <string.h>
+#include <stdlib.h>  /* patch for Sage */
 #include "pari.h"
 #include "paripriv.h"
 #include "anal.h"
@@ -684,6 +685,7 @@
 static void
 pari_exit(void)
 {
+  abort(); /* patch for Sage */
   fprintferr("  ***   Error in the PARI system. End of program.\n");
   exit(1);
 }
