diff -ru src/config/get_cc b/config/get_cc
--- src/config/get_cc	2014-02-01 21:41:54.534348273 +0100
+++ b/config/get_cc	2014-02-01 21:42:50.850930971 +0100
@@ -94,7 +94,11 @@
     OPTFLAGS="$OPTFLAGS -fno-strict-aliasing"
   fi
   rm -f $exe $exe$exe_suff
-  KERNELCFLAGS=-funroll-loops
+  if [ "$SAGE_DEBUG" = yes ]; then
+    KERNELCFLAGS=-O1
+  else
+    KERNELCFLAGS=-funroll-loops
+  fi
 
   DBGFLAGS=${DBGFLAGS:-"-g $warn"}
   # Specific optimisations for some architectures
