Index: config/Makefile.SH
===================================================================
--- config/Makefile.SH	(revision 12583)
+++ config/Makefile.SH	(working copy)
@@ -599,9 +599,9 @@
 EOT
 fi
 
-if test -e $TOP/.svn/entries; then
+if test -f "$TOP/.svn/entries"; then
   vcfile="../.svn/entries"
-else if test -e $TOP/.git/index; then
+else if test -f "$TOP/.git/index"; then
   vcfile="../.git/index"
   fi
 fi
