--- src/headers/paripriv.h	(revision 12541)
+++ src/headers/paripriv.h	(working copy)
@@ -333,7 +333,11 @@
   pari_timer *T;
 } gp_data;
   /* GP_DATA->flags */
-enum { QUIET=1, TEST=2, SIMPLIFY=4, CHRONO=8, ECHO=16, STRICTMATCH=32,
+
+/* The ECHO symbol is already defined in Sage when building on OS X. Since this is paripriv.h, 
+   it's very unlikely that anything in here is used, so this should be safe. */
+
+enum { QUIET=1, TEST=2, SIMPLIFY=4, CHRONO=8, xxECHOxx=16, STRICTMATCH=32,
        USE_READLINE=64, SECURE=128, EMACS=256, TEXMACS=512, BREAKLOOP=1024,
        RECOVER=2048};
 
