--- paripriv.h.orig	Wed Oct 14 11:08:45 2009
+++ paripriv-Solaris.h	Wed Oct 14 12:31:52 2009
@@ -255,8 +255,13 @@
 GEN  ZX_disc_all(GEN,ulong);
 long ZX_get_prec(GEN x);
 GEN  ZX_resultant_all(GEN A, GEN B, GEN dB, ulong bound);
+/* 
+Coment out the following two lines, as they cause problems on 
+Solaris. I was originally going to do this on all platforms, not just 
+Solaris, but I noticed there was already an altered partpriv-osx.h, 
+so this modified version will only be copied over if using Solaris. 
 GEN  ZY_ZXY_resultant_all(GEN A, GEN B0, long *lambda, GEN *LPRS);
-GEN  ZY_ZXY_rnfequation(GEN A, GEN B0, long *lambda);
+GEN  ZY_ZXY_rnfequation(GEN A, GEN B0, long *lambda); */
 GEN  RgXQ_u_pow(GEN x, ulong n, GEN T);
 GEN  RgX_gcd_simple(GEN x, GEN y);
 GEN  RgX_extgcd_simple(GEN a, GEN b, GEN *pu, GEN *pv);
@@ -425,8 +430,15 @@
   pari_timer *T;
 } gp_data;
   /* GP_DATA->flags */
+/* 
+
+Coment out the following two enum, as it causes problems on 
+Solaris. I was originally going to do this on all platforms, not just 
+Solaris, but I noticed there was already an altered partpriv-osx.h, 
+so this modified version will only be copied over if using Solaris. 
 enum { QUIET=1, TEST=2, SIMPLIFY=4, CHRONO=8, ECHO=16, STRICTMATCH=32,
        USE_READLINE=64, SECURE=128, EMACS=256, TEXMACS=512};
+*/
 
 extern gp_data *GP_DATA;
 
