Index: src/basemath/buch4.c
===================================================================
--- src/basemath/buch4.c	(revision 12768)
+++ src/basemath/buch4.c	(working copy)
@@ -618,8 +618,9 @@
   *S2 = shallowconcat(*S2, idealprimedec(rel,p));
 }
 
+/* N a t_INT */
 static void
-fa_pr_append(GEN nf,GEN rel,GEN N,GEN *prod,GEN *S1,GEN *S2)
+Zfa_pr_append(GEN nf,GEN rel,GEN N,GEN *prod,GEN *S1,GEN *S2)
 {
   if (!is_pm1(N))
   {
@@ -628,6 +629,18 @@
     for (i=1; i<l; i++) pr_append(nf,rel,gel(v,i),prod,S1,S2);
   }
 }
+/* N a t_INT or t_FRAC */
+static void
+fa_pr_append(GEN nf,GEN rel,GEN N,GEN *prod,GEN *S1,GEN *S2)
+{
+  if (typ(N) == t_FRAC)
+  {
+    Zfa_pr_append(nf,rel,gel(N,1),prod,S1,S2);
+    Zfa_pr_append(nf,rel,gel(N,2),prod,S1,S2);
+  }
+  else
+    Zfa_pr_append(nf,rel,N,prod,S1,S2);
+}
 
 static GEN
 pol_up(GEN rnfeq, GEN x, long v)
