diff -ru src/doc/gphelp.in b/doc/gphelp.in
--- src/doc/gphelp.in	2013-05-06 16:01:56.000000000 +0200
+++ b/doc/gphelp.in	2013-05-06 16:41:15.522405496 +0200
@@ -1,4 +1,4 @@
-#!@perl@
+#!/usr/bin/env perl
 #
 # $Id$
 #
diff -ru src/misc/tex2mail.in b/misc/tex2mail.in
--- src/misc/tex2mail.in	2013-05-06 16:01:47.000000000 +0200
+++ b/misc/tex2mail.in	2013-05-06 16:41:43.523232305 +0200
@@ -1,4 +1,4 @@
-#!@perl@
+#!/usr/bin/env perl
 # Original version by Ilya Zakharevich. Minor updates by the PARI group
 # Features:
 #      % at the end of a line followed by \n\n is recognized as end of 
