From 1420fa40a7d7b43cf2d21c530a0a1d394bac245d Mon Sep 17 00:00:00 2001
From: Karim Belabas <Karim.Belabas@math.u-bordeaux.fr>
Date: Sun, 18 Jun 2017 17:30:46 +0200
Subject: [PATCH] 82- polredabs(non-monic t_POL, 1) => potentially wrong
 transformation map

---
 src/basemath/base1.c | 2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/basemath/base1.c b/src/basemath/base1.c
index 47c6ba1..9f1c736 100644
--- a/src/basemath/base1.c
+++ b/src/basemath/base1.c
@@ -2484,7 +2484,7 @@ polredabs0(GEN x, long flag)
         GEN w = gel(vw,2);
         for (i = 1; i < l; i++)
           w = ZV_union_shallow(w, gel(Z_factor(gel(v,i)),1));
-        nfinit_basic(&S, mkvec2(x,w));
+        nfinit_basic(&S, mkvec2(S.T0,w));
       }
     }
     v = polredabs_aux(&S, &u);
