Index: src/language/init.c
===================================================================
--- a/src/language/init.c	(revision 13228)
+++ b/src/language/init.c	(working copy)
@@ -731,8 +731,8 @@
   pari_stackcheck_init(&u);
   if ((init_opts&INIT_DFTm)) {
     GP_DATA = default_gp_data();
+    pari_init_defaults();
     gp_expand_path(GP_DATA->path);
-    pari_init_defaults();
   }
 
   if ((init_opts&INIT_SIGm)) pari_sig_init(pari_sighandler);
