commit e5f592696f1cceab9d07e1b22b86067ea0fa8d09
Author: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date:   Fri Nov 4 12:37:38 2016 +0100

    Use DEBUGMEM for stack size warnings

diff -ru src/src/gp/gp.c b/src/gp/gp.c
--- src/src/gp/gp.c	2017-01-30 17:36:54.045921458 +0100
+++ b/src/gp/gp.c	2017-01-30 17:37:07.225921164 +0100
@@ -559,6 +559,7 @@
 #endif
   stdin_isatty = pari_stdin_isatty();
   pari_init_defaults();
+  DEBUGMEM = 1;  /* Use \gm1 by default in gp */
   pari_library_path = DL_DFLT_NAME;
   pari_stack_init(&s_A,sizeof(*A),(void**)&A);
   pari_init_opts(1000000 * sizeof(long), 0, INIT_SIGm | INIT_noPRIMEm | INIT_noIMTm);
diff -ru src/src/language/init.c b/src/language/init.c
--- src/src/language/init.c	2017-01-30 17:36:54.046921458 +0100
+++ b/src/language/init.c	2017-01-30 17:37:07.227921164 +0100
@@ -793,7 +793,8 @@
   evalstate_reset();
   paristack_setsize(pari_mainstack->rsize, newsize);
   s = pari_mainstack->vsize ? pari_mainstack->vsize : pari_mainstack->rsize;
-  pari_warn(warner,"new maximum stack size = %lu (%.3f Mbytes)", s, s/1048576.);
+  if (DEBUGMEM > 0)
+    pari_warn(warner,"new maximum stack size = %lu (%.3f Mbytes)", s, s/1048576.);
   pari_init_errcatch();
   cb_pari_err_recover(-1);
 }
@@ -807,7 +808,8 @@
     pari_mainstack_resize(pari_mainstack, newsize, vsize);
   evalstate_reset();
   s = pari_mainstack->rsize;
-  pari_warn(warner,"new stack size = %lu (%.3f Mbytes)", s, s/1048576.);
+  if (DEBUGMEM > 0)
+    pari_warn(warner,"new stack size = %lu (%.3f Mbytes)", s, s/1048576.);
   pari_init_errcatch();
   cb_pari_err_recover(-1);
 }
@@ -821,7 +823,8 @@
   if (newsize <= pari_mainstack->size) return;
   if (pari_mainstack_setsize(pari_mainstack, newsize))
   {
-    pari_warn(warner, "increasing stack size to %lu", pari_mainstack->size);
+    if (DEBUGMEM > 0)
+      pari_warn(warner, "increasing stack size to %lu", pari_mainstack->size);
   }
 }
 
