diff -druN ppl-1.1.orig/src/assert.hh ppl-1.1/src/assert.hh
--- ppl-1.1.orig/src/assert.hh	2013-10-28 13:38:33.000000000 +0100
+++ ppl-1.1/src/assert.hh	2014-04-08 11:40:48.924491100 +0200
@@ -98,7 +98,7 @@
 
 namespace Parma_Polyhedra_Library {
 
-#if PPL_CXX_SUPPORTS_ATTRIBUTE_WEAK
+#if PPL_CXX_SUPPORTS_ATTRIBUTE_WEAK && ! (defined(__CYGWIN__) && defined(__x86_64__))
 #define PPL_WEAK_NORETURN __attribute__((weak, noreturn))
 #else
 #define PPL_WEAK_NORETURN __attribute__((noreturn))
diff -druN ppl-1.1.orig/src/ppl.hh.dist ppl-1.1/src/ppl.hh.dist
--- ppl-1.1.orig/src/ppl.hh.dist	2013-10-28 13:47:20.000000000 +0100
+++ ppl-1.1/src/ppl.hh.dist	2014-04-08 11:01:13.361991100 +0200
@@ -1739,7 +1739,7 @@
 
 namespace Parma_Polyhedra_Library {
 
-#if PPL_CXX_SUPPORTS_ATTRIBUTE_WEAK
+#if PPL_CXX_SUPPORTS_ATTRIBUTE_WEAK && ! (defined(__CYGWIN__) && defined(__x86_64__))
 #define PPL_WEAK_NORETURN __attribute__((weak, noreturn))
 #else
 #define PPL_WEAK_NORETURN __attribute__((noreturn))
