From c7894260aaf36c447dd091bf622c7c2bcdb15a28 Mon Sep 17 00:00:00 2001
From: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date: Thu, 30 Jun 2022 13:18:46 -0700
Subject: [PATCH] setup.py: Remove check for SETUPTOOLS_USE_DISTUTILS

---
 setup.py | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/setup.py b/setup.py
index 353b6c09693..2bc70173eca 100755
--- a/setup.py
+++ b/setup.py
@@ -85,17 +85,6 @@
 # so that it is in sys.modules
 import numpy.distutils.command.sdist
 import setuptools
-if int(setuptools.__version__.split('.')[0]) >= 60:
-    # setuptools >= 60 switches to vendored distutils by default; this
-    # may break the numpy build, so make sure the stdlib version is used
-    try:
-        setuptools_use_distutils = os.environ['SETUPTOOLS_USE_DISTUTILS']
-    except KeyError:
-        os.environ['SETUPTOOLS_USE_DISTUTILS'] = "stdlib"
-    else:
-        if setuptools_use_distutils != "stdlib":
-            raise RuntimeError("setuptools versions >= '60.0.0' require "
-                    "SETUPTOOLS_USE_DISTUTILS=stdlib in the environment")
 
 # Initialize cmdclass from versioneer
 from numpy.distutils.core import numpy_cmdclass
