--- arrayobject.c.orig	2008-09-27 14:46:06.000000000 -0700
+++ arrayobject.c	2008-09-27 14:59:35.000000000 -0700
@@ -1318,9 +1318,9 @@
     intp newd[MAX_DIMS];
     char msg[] = "PyArray_FromDimsAndDataAndDescr: use PyArray_NewFromDescr.";
 
-    if (DEPRECATE(msg) < 0) {
-        return NULL;
-    }
+    /*if (DEPRECATE(msg) < 0) {*/
+    /*    return NULL;*/
+    /*}*/
     if (!PyArray_ISNBO(descr->byteorder))
         descr->byteorder = '=';
     for(i = 0; i < nd; i++) {
@@ -1342,9 +1342,9 @@
     PyObject *ret;
     char msg[] = "PyArray_FromDims: use PyArray_SimpleNew.";
 
-    if (DEPRECATE(msg) < 0) {
-        return NULL;
-    }
+    /*if (DEPRECATE(msg) < 0) { */
+    /*    return NULL; */
+    /*} */
     ret = PyArray_FromDimsAndDataAndDescr(nd, d,
                                           PyArray_DescrFromType(type),
                                           NULL);
