--- a/numpy/linalg/setup.py	2009-03-29 06:24:21.000000000 -0500
+++ b/numpy/linalg/setup.py	2008-09-25 22:43:28.000000000 -0500
@@ -27,7 +27,8 @@
                                     'zlapack_lite.c', 'dlapack_lite.c',
                                     'blas_lite.c', 'dlamch.c',
                                     'f2c_lite.c','f2c.h'],
-                         extra_info = lapack_info
+                         extra_info = lapack_info,
+                         extra_link_args = ['-shared']
                          )
 
     return config
