--- a/numpy/linalg/setup.py	2014-03-25 12:43:23.000000000 -0700
+++ b/numpy/linalg/setup.py 	2014-09-20 09:19:56.805175830 -0700
@@ -46,6 +46,7 @@
                          sources = [get_lapack_lite_sources],
                          depends =  ['umath_linalg.c.src'] + lapack_lite_src,
                          extra_info = lapack_info,
+                         extra_link_args = ['-shared'],
                          libraries = ['npymath'],
                          )

