diff -Naur numpy-1.10.1.orig/numpy/core/function_base.py numpy-1.10.1/numpy/core/function_base.py
--- numpy-1.10.1.orig/numpy/core/function_base.py	2015-11-11 10:12:45.583322683 +1300
+++ numpy-1.10.1/numpy/core/function_base.py	2015-11-11 10:14:05.813343880 +1300
@@ -3,7 +3,7 @@
 __all__ = ['logspace', 'linspace']
 
 from . import numeric as _nx
-from .numeric import result_type, NaN
+from .numeric import result_type, NaN, asanyarray
 
 
 def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None):
@@ -87,8 +87,8 @@
     div = (num - 1) if endpoint else num
 
     # Convert float/complex array scalars to float, gh-3504
-    start = start * 1.
-    stop = stop * 1.
+    start = asanyarray(start) * 1.
+    stop = asanyarray(stop) * 1.
 
     dt = result_type(start, stop, float(num))
     if dtype is None:
