--- numpytest.py.orig	2008-09-27 14:43:18.000000000 -0700
+++ numpytest.py	2008-09-27 14:44:36.000000000 -0700
@@ -36,8 +36,8 @@
 
       restore_path()
     """
-    warnings.warn("set_package_path will be removed in NumPy 1.3; please "
-                  "update your code", DeprecationWarning, stacklevel=2)
+    #warnings.warn("set_package_path will be removed in NumPy 1.3; please "
+    #              "update your code", DeprecationWarning, stacklevel=2)
 
     from distutils.util import get_platform
     f = get_frame(level)
@@ -62,8 +62,8 @@
 
       restore_path()
     """
-    warnings.warn("set_local_path will be removed in NumPy 1.3; please "
-                  "update your code", DeprecationWarning, stacklevel=2)
+    #warnings.warn("set_local_path will be removed in NumPy 1.3; please "
+    #              "update your code", DeprecationWarning, stacklevel=2)
 
     f = get_frame(level)
     if f.f_locals['__name__']=='__main__':
@@ -77,8 +77,8 @@
     return
 
 def restore_path():
-    warnings.warn("restore_path will be removed in NumPy 1.3; please "
-                  "update your code", DeprecationWarning, stacklevel=2)
+    #warnings.warn("restore_path will be removed in NumPy 1.3; please "
+    #              "update your code", DeprecationWarning, stacklevel=2)
 
     if DEBUG:
         print 'Removing %r from sys.path' % (sys.path[0])
@@ -118,8 +118,8 @@
 
 class NumpyTestCase (unittest.TestCase):
     def __init__(self, *args, **kwds):
-        warnings.warn("NumpyTestCase will be removed in the next release; please update your code to use nose or unittest",
-                         DeprecationWarning, stacklevel=2)
+        #warnings.warn("NumpyTestCase will be removed in the next release; please update your code to use nose or unittest",
+        #                 DeprecationWarning, stacklevel=2)
         unittest.TestCase.__init__(self, *args, **kwds)
 
     def measure(self,code_str,times=1):
@@ -289,8 +289,8 @@
         return
 
     def __init__(self, package=None):
-        warnings.warn("NumpyTest will be removed in the next release; please update your code to use nose or unittest",
-                         DeprecationWarning, stacklevel=2)
+        #warnings.warn("NumpyTest will be removed in the next release; please update your code to use nose or unittest",
+        #                 DeprecationWarning, stacklevel=2)
         if package is None:
             from numpy.distutils.misc_util import get_frame
             f = get_frame(1)
