--- ../src/numpy/core/setup.py	2010-08-25 08:32:15.000000000 -0400
+++ setup.py	2010-10-05 22:29:44.475225000 -0400
@@ -215,12 +215,15 @@
     _macros = ["isnan", "isinf", "signbit", "isfinite"]
     if sys.version_info[:2] >= (2, 6):
         for f in _macros:
-            st = config.check_decl(fname2def("decl_%s" % f),
+            py_symbol = fname2def("decl_%s" % f)
+            already_declared = config.check_decl(py_symbol,
                     headers=["Python.h", "math.h"])
-            if not st:
-                macros.append(f)
+            if already_declared:
+                if config.check_macro_true(py_symbol,
+                        headers=["Python.h", "math.h"]):
+                    pub.append('NPY_%s' % fname2def("decl_%s" % f))
             else:
-                _add_decl(f)
+                macros.append(f)
     else:
         macros = _macros[:]
     # Normally, isnan and isinf are macro (C99), but some platforms only have
