diff --git a/src.bak/lib-src-gmp/Makefile.in b/src/lib-src-gmp/Makefile.in
index 4ab9342..93ff335 100644
--- a/src.bak/lib-src-gmp/Makefile.in
+++ b/src/lib-src-gmp/Makefile.in
@@ -54,7 +54,7 @@ LTLIBRARIES = $(lib_LTLIBRARIES)
 libcddgmp_la_LIBADD =
 am_libcddgmp_la_OBJECTS = cddcore.lo cddlp.lo cddmp.lo cddio.lo \
 	cddlib.lo cddproj.lo cddcore_f.lo cddlp_f.lo cddmp_f.lo \
-	cddio_f.lo cddlib_f.lo cddproj_f.lo setoper.lo
+	cddio_f.lo cddlib_f.lo cddproj_f.lo setoper.lo random.lo
 libcddgmp_la_OBJECTS = $(am_libcddgmp_la_OBJECTS)
 DEFAULT_INCLUDES = -I.@am__isrc@
 depcomp = $(SHELL) $(top_srcdir)/depcomp
@@ -199,7 +199,8 @@ cddmp_f.c \
 cddio_f.c \
 cddlib_f.c \
 cddproj_f.c \
-setoper.c
+setoper.c \
+random.c
 
 include_HEADERS = \
 cdd.h \
@@ -293,6 +294,7 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/cddmp_f.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/cddproj.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/cddproj_f.Plo@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/random.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/setoper.Plo@am__quote@
 
 .c.o:
