diff --git a/src.bak/src-gmp/Makefile.am b/src/src-gmp/Makefile.am
index e4dbb86..e579d71 100644
--- a/src.bak/src-gmp/Makefile.am
+++ b/src/src-gmp/Makefile.am
@@ -11,7 +11,8 @@ testcdd1_gmp \
 testcdd2_gmp \
 testlp1_gmp \
 testlp2_gmp \
-testlp3_gmp
+testlp3_gmp \
+cdd_both_reps_gmp
 #cddmathlink
 
 scdd_gmp_SOURCES                   = simplecdd.c
@@ -27,6 +28,7 @@ testcdd2_gmp_SOURCES               = testcdd2.c
 testlp1_gmp_SOURCES                = testlp1.c
 testlp2_gmp_SOURCES                = testlp2.c
 testlp3_gmp_SOURCES                = testlp3.c
+cdd_both_reps_gmp_SOURCES          = cdd_both_reps.c
 # cddmathlink_SOURCES                = cddmathlink.c cddmlio.h cddmlio.c
 
 LDADD        = ../lib-src-gmp/libcddgmp.la
