commit 615b4bf8b13d55a576bc60ad04f7b17d75f49021
Author: Yasuaki Honda <yasuaki.honda@gmail.com>
Date:   Sun Apr 26 12:15:14 2020 +0900

    Fix for Bug #3629, to compile with ECL 20.4.24

diff --git a/lisp-utils/defsystem.lisp b/lisp-utils/defsystem.lisp
index dda669d26..b8e96eebe 100644
--- a/lisp-utils/defsystem.lisp
+++ b/lisp-utils/defsystem.lisp
@@ -4152,7 +4152,7 @@ the system definition, if provided."
       #+:ecl
       (progn
         (ext:package-lock "CL" nil)
-        (setf (symbol-function 'lisp:require)
+        (setf (symbol-function 'cl:require)
               (symbol-function 'new-require))
         (ext:package-lock "CL" t))
       #+:lispworks
