diff --git a/src/init-cl.lisp b/src/init-cl.lisp
index 0e1f18b..125f449 100644
--- a/src/init-cl.lisp
+++ b/src/init-cl.lisp
@@ -134,7 +134,7 @@ When one changes, the other does too."
 	  (setq libdir     (combine-path maxima-prefix-env "lib"))
 	  (setq libexecdir (combine-path maxima-prefix-env "libexec"))
 	  (setq datadir    (combine-path maxima-prefix-env "share"))
-	  (setq infodir    (combine-path maxima-prefix-env #+(or cygwin windows win32 win64) "share" "info")))
+	  (setq infodir    (combine-path maxima-prefix-env "share" "info")))
 	(progn
 	  (setq libdir     (maxima-parse-dirstring *autoconf-libdir*))
 	  (setq libexecdir (maxima-parse-dirstring *autoconf-libexecdir*))
