diff --git a/src/grind.lisp b/src/grind.lisp
index 4ae73a3..86b80e6 100644
--- a/src/grind.lisp
+++ b/src/grind.lisp
@@ -295,12 +295,12 @@
       (msz nil l r)
       (do ((nl) (w 0))
           ((null (cdr x))
-           (setq nl (cons (if (stringp (car x))
+           (setq nl (cons (if (atom (car x))
                               (msz (makestring (car x)) l r)
                               (msize (car x) l r lop rop))
                           nl))
            (cons (+ w (caar nl)) (nreverse nl)))
-        (setq nl (cons (if (stringp (car x))
+        (setq nl (cons (if (atom (car x))
                            (msz (makestring (car x)) l r)
                            (msize (car x) l r lop rop))
                        nl)
