--- ../src/src/python/__init__.py	2010-09-16 12:52:50.000000000 +0800
+++ __init__.py	2010-10-12 06:11:48.000000000 +0800
@@ -50,14 +50,7 @@
     std       standard deviation of the distribution
     '''
 
-    try:    
-        from cvxopt import gsl
-    except:
-        from cvxopt.base import matrix
-        from random import gauss
-        return matrix([gauss(mean, std) for k in xrange(nrows*ncols)],
-                      (nrows,ncols), 'd' )
-        
+    from cvxopt import gsl
     return gsl.normal(nrows, ncols, mean, std)
 
 def uniform(nrows, ncols=1, a=0, b=1):
@@ -81,14 +74,7 @@
     b         upper bound
     '''
 
-    try:    
-        from cvxopt import gsl
-    except:
-        from cvxopt.base import matrix
-        from random import uniform
-        return matrix([uniform(a, b) for k in xrange(nrows*ncols)],
-                      (nrows,ncols), 'd' )
-
+    from cvxopt import gsl
     return gsl.uniform(nrows, ncols, a, b)
 
 def setseed(val = 0):
@@ -102,14 +88,9 @@
               is used. 
     '''    
 
-    try:    
-        from cvxopt import gsl
-        gsl.setseed(val)
-    except:
-        from random import seed
-        if val is 0: val = None
-        seed(val)
-        
+    from cvxopt import gsl
+    gsl.setseed(val)
+
  
 def getseed():
     '''
@@ -118,12 +99,8 @@
     getseed()
     '''
 
-    try:    
-        from cvxopt import gsl
-        return gsl.getseed()
-    except:
-        raise NotImplementedError, "getseed() not installed (requires GSL) "
-    
+    from cvxopt import gsl
+    return gsl.getseed()
 
 
 import __builtin__
