diff -ru src/src/C/cvxopt.h src.patched/src/C/cvxopt.h
--- src/src/C/cvxopt.h	2010-09-16 06:52:51.000000000 +0200
+++ src.patched/src/C/cvxopt.h	2010-10-12 00:11:27.000000000 +0200
@@ -27,7 +27,21 @@
 /* ANSI99 complex is disabled during build of CHOLMOD */
 
 #ifndef NO_ANSI99_COMPLEX
-#include "complex.h"
+
+#include <complex.h>
+/* work around Solaris 10 specific problem in complex.h */
+ #if defined(__sun__) && defined(__GNUC__)
+    #if __GNUC__ < 4  || ( __GNUC__ == 4 && __GNUC_MINOR__ < 5   )
+ 
+       #undef  _Complex_I
+       #define _Complex_I (__extension__ 1.0iF)
+ 
+       #undef I
+       #define I _Complex_I
+ 
+    #endif
+ #endif
+
 #define MAT_BUFZ(O)  ((complex *)((matrix *)O)->buffer)
 #endif
 
@@ -55,7 +69,7 @@
 typedef struct {
   PyObject_HEAD
   void *buffer;          /* in column-major-mode array of type 'id' */
-  int_t nrows, ncols;    /* number of rows and columns */
+  int nrows, ncols;    /* number of rows and columns */
   int   id;              /* DOUBLE, INT, COMPLEX */
 } matrix;
 
