--- src/src/setup.py.orig	2012-06-25 22:04:22.846029845 +1200
+++ src/src/setup.py	2012-06-25 22:02:20.505936660 +1200
@@ -2,29 +2,39 @@
 from glob import glob
 
 # Modifiy this if BLAS and LAPACK libraries are not in /usr/lib.
-BLAS_LIB_DIR = '/usr/lib'
+
+import os
+SAGE_LIB = os.environ['SAGE_LOCAL']+'/lib'
+SAGE_INCLUDE = os.environ['SAGE_LOCAL']+'/include'
+BLAS_LIB_DIR = SAGE_LIB
 
 # Default names of BLAS and LAPACK libraries
-BLAS_LIB = ['blas']
 LAPACK_LIB = ['lapack']
 BLAS_EXTRA_LINK_ARGS = []
 
+if os.environ['UNAME'] == 'CYGWIN':
+  BLAS_LIB =['blas', 'gfortran']
+elif os.environ['UNAME'] == 'Darwin':
+  BLAS_LIB =['blas','gfortran','m']
+else:
+  BLAS_LIB =['f77blas','cblas','gfortran','atlas','m']
+
 # Set environment variable BLAS_NOUNDERSCORES=1 if your BLAS/LAPACK do
 # not use trailing underscores
 BLAS_NOUNDERSCORES = False
 
 # Set to 1 if you are using the random number generators in the GNU
 # Scientific Library.
-BUILD_GSL = 0
+BUILD_GSL = 1
 
 # Directory containing libgsl (used only when BUILD_GSL = 1).
-GSL_LIB_DIR = '/usr/lib'
+GSL_LIB_DIR = SAGE_LIB
 
 # Directory containing the GSL header files (used only when BUILD_GSL = 1).
-GSL_INC_DIR = '/usr/include/gsl'
+GSL_INC_DIR = SAGE_INCLUDE
 
 # Set to 1 if you are installing the fftw module.
-BUILD_FFTW = 0 
+BUILD_FFTW = 0
 
 # Directory containing libfftw3 (used only when BUILD_FFTW = 1).
 FFTW_LIB_DIR = '/usr/lib'
@@ -33,13 +43,13 @@
 FFTW_INC_DIR = '/usr/include'
 
 # Set to 1 if you are installing the glpk module.
-BUILD_GLPK = 0 
+BUILD_GLPK = 1 
 
 # Directory containing libglpk (used only when BUILD_GLPK = 1).
-GLPK_LIB_DIR = '/usr/lib'
+GLPK_LIB_DIR = SAGE_LIB
 
 # Directory containing glpk.h (used only when BUILD_GLPK = 1).
-GLPK_INC_DIR = '/usr/include'
+GLPK_INC_DIR = SAGE_INCLUDE
 
 # Set to 1 if you are installing the DSDP module.
 BUILD_DSDP = 0
