diff --git a/setup.py b/setup.py
index 031ad53..b243eac 100644
--- a/setup.py
+++ b/setup.py
@@ -6,26 +6,38 @@ from glob import glob
 import os
 
 # Modifiy this if BLAS and LAPACK libraries are not in /usr/lib.
-BLAS_LIB_DIR = '/usr/lib'
+
+import os
+SAGE_LIB = os.environ['SAGE_LOCAL']+'/lib'
+SAGE_INCLUDE = os.environ['SAGE_LOCAL']+'/include'
 
 # Default names of BLAS and LAPACK libraries
-BLAS_LIB = ['blas']
 LAPACK_LIB = ['lapack']
 BLAS_EXTRA_LINK_ARGS = []
 
+if os.environ['UNAME'] == 'CYGWIN':
+    BLAS_LIB =['blas', 'gfortran']
+    BLAS_LIB_DIR = '/usr/lib'
+elif os.environ['UNAME'] == 'Darwin':
+    BLAS_LIB =['blas','gfortran','m']
+    BLAS_LIB_DIR = '/System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/'
+else:
+    BLAS_LIB =['f77blas','cblas','gfortran','atlas','m']
+    BLAS_LIB_DIR = SAGE_LIB
+
 # Set environment variable BLAS_NOUNDERSCORES=1 if your BLAS/LAPACK do
 # not use trailing underscores
 BLAS_NOUNDERSCORES = False
 
 # Set to 1 if you are using the random number generators in the GNU
 # Scientific Library.
-BUILD_GSL = 0
+BUILD_GSL = 1
 
 # Directory containing libgsl (used only when BUILD_GSL = 1).
-GSL_LIB_DIR = '/usr/lib'
+GSL_LIB_DIR = SAGE_LIB
 
 # Directory containing the GSL header files (used only when BUILD_GSL = 1).
-GSL_INC_DIR = '/usr/include/gsl'
+GSL_INC_DIR = SAGE_INCLUDE
 
 # Set to 1 if you are installing the fftw module.
 BUILD_FFTW = 0 
@@ -37,13 +49,13 @@ FFTW_LIB_DIR = '/usr/lib'
 FFTW_INC_DIR = '/usr/include'
 
 # Set to 1 if you are installing the glpk module.
-BUILD_GLPK = 0 
+BUILD_GLPK = 1
 
 # Directory containing libglpk (used only when BUILD_GLPK = 1).
-GLPK_LIB_DIR = '/usr/lib'
+GLPK_LIB_DIR = SAGE_LIB
 
 # Directory containing glpk.h (used only when BUILD_GLPK = 1).
-GLPK_INC_DIR = '/usr/include'
+GLPK_INC_DIR = SAGE_INCLUDE
 
 # Set to 1 if you are installing the DSDP module.
 BUILD_DSDP = 0
@@ -114,7 +126,7 @@ if BUILD_FFTW:
     extmods += [fftw];
 
 if BUILD_GLPK:
-    glpk = Extension('glpk', libraries = ['glpk'],
+    glpk = Extension('glpk', libraries = ['glpk', 'gmp', 'z'],
         include_dirs = [ GLPK_INC_DIR ],
         library_dirs = [ GLPK_LIB_DIR ],
         sources = ['src/C/glpk.c'] )
@@ -213,7 +225,7 @@ else:
     amd = Extension('amd',
         include_dirs = [ 'src/C/SuiteSparse/AMD/Include',
             'src/C/SuiteSparse/SuiteSparse_config' ],
-        define_macros = MACROS,
+        define_macros = MACROS  + [('NTIMER', '1')],  # this will be in upstream at some point
         sources = [ 'src/C/amd.c', 'src/C/SuiteSparse/SuiteSparse_config/SuiteSparse_config.c'] +
         glob('src/C/SuiteSparse/AMD/Source/*.c') )
 
