diff --git a/cvxopt-1.1.2.p1/src/src/python/__init__.py b/cvxopt-1.1.2.p1/src/src/python/__init__.py.orig
index 6a3719a..6aedfaa 100644
--- a/cvxopt-1.1.2.p1/src/src/python/__init__.py
+++ b/cvxopt-1.1.2.p1/src/src/python/__init__.py.orig
@@ -53,7 +53,7 @@ def normal(nrows, ncols=1, mean=0.0, std=1.0):
         from cvxopt import gsl
     except:
         from cvxopt.base import matrix
-        from sage.misc.prandom import gauss
+        from random import gauss
         return matrix([gauss(mean, std) for k in xrange(nrows*ncols)],
                       (nrows,ncols), 'd' )
         
@@ -84,7 +84,7 @@ def uniform(nrows, ncols=1, a=0, b=1):
         from cvxopt import gsl
     except:
         from cvxopt.base import matrix
-        from sage.misc.prandom import uniform
+        from random import uniform
         return matrix([uniform(a, b) for k in xrange(nrows*ncols)],
                       (nrows,ncols), 'd' )
 
@@ -105,7 +105,7 @@ def setseed(val = 0):
         from cvxopt import gsl
         gsl.setseed(val)
     except:
-        from sage.misc.prandom import seed
+        from random import seed
         if val is 0: val = None
         seed(val)
         
