diff -ru src/config b/config
--- src/config	2019-02-26 14:15:30.000000000 +0000
+++ b/config	2019-03-05 10:53:04.281408906 +0000
@@ -371,8 +371,10 @@
 # this is where the translation occurs into SSLeay terms
 # ---------------------------------------------------------------------------
 
-# Only set CC if not supplied already
-if [ -z "$CROSS_COMPILE$CC" ]; then
+# Save given CC
+userCC="$CC"
+
+if true; then
   GCCVER=`sh -c "gcc -dumpversion" 2>/dev/null`
   if [ "$GCCVER" != "" ]; then
     # then strip off whatever prefix egcs prepends the number with...
@@ -433,6 +435,11 @@
 
 CCVER=${CCVER:-0}
 
+# Restore given $CC
+if [ -n "$userCC" ]; then
+    CC="$userCC"
+fi
+
 # read the output of the embedded GuessOS
 read GUESSOS
 
