From 09e79cf75146ca34c637f919d1f4527a63743300 Mon Sep 17 00:00:00 2001
From: Marc Pfetsch <pfetsch@mathematik.tu-darmstadt.de>
Date: Fri, 9 Dec 2022 18:52:09 +0100
Subject: [PATCH] use cmake FindLAPACK

---
 CMakeLists.txt     | 14 +++++---------
 src/CMakeLists.txt |  6 +++---
 2 files changed, 8 insertions(+), 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 15e4edc..d4f14f5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -79,16 +79,12 @@ add_custom_target(scip_update_githash
                   COMMAND ${CMAKE_COMMAND} -DDST=${PROJECT_SOURCE_DIR}/scipsdpgithash.c
                                            -P ${CMAKE_BINARY_DIR}/scip_update_githash.cmake)
 
-
 # find lapack and blas
-find_library(LAPACK_LIBRARY lapack)
-if(LAPACK_LIBRARY)
-  message(STATUS "Found lapack library: " ${LAPACK_LIBRARY})
-endif()
-
-find_library(BLAS_LIBRARY blas)
-if(BLAS_LIBRARY)
-  message(STATUS "Found blas library: " ${BLAS_LIBRARY})
+find_package(LAPACK REQUIRED)
+if(LAPACK_FOUND)
+  message(STATUS "Found lapack library: " ${LAPACK_LIBRARIES})
+else()
+  message(FATAL_ERROR "Lapack not found")
 endif()
 
 # search the selected symmetry computation program
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 30c66b7..5e5d7d3 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -125,11 +125,11 @@ setLibProperties(libscipsdp "scipsdp")
 #put binary in bin directory
 set (CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 if(SHARED)
-    target_link_libraries(libscipsdp PRIVATE ${ZIMPL_PIC_LIBRARIES} ${SDPS_PIC_LIBRARIES} ${SYM_LIBRARIES} ${LAPACK_LIBRARY} ${BLAS_LIBRARY} m)
+    target_link_libraries(libscipsdp PRIVATE ${ZIMPL_PIC_LIBRARIES} ${SDPS_PIC_LIBRARIES} ${SYM_LIBRARIES} ${LAPACK_LIBRARIES} m)
     add_executable(scipsdp scipsdp/main.c ${scipsdpsources} ${objscipsdpsources} ${sdpisources} ${sym})
-    target_link_libraries(scipsdp ${ZLIB_LIBRARIES} ${Readline_LIBRARY} ${GMP_LIBRARIES} ${ZIMPL_LIBRARIES} ${SDPS_LIBRARIES} ${SCIP_LIBRARIES} ${SYM_LIBRARIES} ${LAPACK_LIBRARY} ${BLAS_LIBRARY} m)
+    target_link_libraries(scipsdp ${ZLIB_LIBRARIES} ${Readline_LIBRARY} ${GMP_LIBRARIES} ${ZIMPL_LIBRARIES} ${SDPS_LIBRARIES} ${SCIP_LIBRARIES} ${SYM_LIBRARIES} ${LAPACK_LIBRARIES} m)
 else()
-    target_link_libraries(libscipsdp PRIVATE ${ZIMPL_LIBRARIES} ${SDPS_LIBRARIES} ${SYM_LIBRARIES} ${LAPACK_LIBRARY} ${BLAS_LIBRARY} m)
+    target_link_libraries(libscipsdp PRIVATE ${ZIMPL_LIBRARIES} ${SDPS_LIBRARIES} ${SYM_LIBRARIES} ${LAPACK_LIBRARIES} m)
     add_executable(scipsdp scipsdp/main.c ${scipsdpsources} ${objscipsdpsources} ${sdpisources} ${sym})
     target_link_libraries(scipsdp libscipsdp ${SCIP_LIBRARIES})
 endif()
