commit b12db4b95295fec9b32131039a6ee4a746c6f832
Author: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date:   Wed Dec 7 15:03:48 2022 -0800

    src/CMakeLists.txt: Remote hardcoded RPATH settings

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f293115..64ac3d1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -5,8 +5,7 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 #
 function(setLibProperties targetname outputname)
     set_target_properties(${targetname} PROPERTIES
-        OUTPUT_NAME ${outputname}
-        MACOSX_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+        OUTPUT_NAME ${outputname})
 endfunction(setLibProperties)
 
 set(CMAKE_C_STANDARD 99)
@@ -141,13 +140,7 @@ target_compile_definitions(scipsdp PRIVATE EXTERN=extern)
 
 set_target_properties(libscipsdp PROPERTIES
     VERSION ${SCIPSDP_VERSION_MAJOR}.${SCIPSDP_VERSION_MINOR}.${SCIPSDP_VERSION_PATCH}
-    SOVERSION ${SCIPSDP_VERSION_MAJOR}.${SCIPSDP_VERSION_MINOR}
-    INSTALL_RPATH_USE_LINK_PATH TRUE)
-
-# set the install rpath to the installed destination
-set_target_properties(scipsdp PROPERTIES
-    INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib"
-    INSTALL_RPATH_USE_LINK_PATH TRUE)
+    SOVERSION ${SCIPSDP_VERSION_MAJOR}.${SCIPSDP_VERSION_MINOR})
 
 # install the header files of scip
 install(FILES ${sdpiheaders} DESTINATION include/sdpi)
