--- ../src/support/shobj-conf	2009-01-04 11:32:42.000000000 -0800
+++ shobj-conf	2010-06-22 15:06:53.000000000 -0700
@@ -132,11 +132,12 @@
 	SHOBJ_CFLAGS=-fPIC
 	SHOBJ_LD='${CC}'
 
-	if [ -x /usr/bin/objformat ] && [ "`/usr/bin/objformat`" = "elf" ]; then
+	if [ ! -x /usr/bin/objformat ] || [ "`/usr/bin/objformat`" = "elf" ]; then
 		SHOBJ_LDFLAGS='-shared -Wl,-soname,$@'
 
 		SHLIB_XLDFLAGS='-Wl,-rpath,$(libdir)'
 		SHLIB_LIBVERSION='$(SHLIB_LIBSUFF).$(SHLIB_MAJOR)'
+                SHLIB_LIBS='-ltermcap'
 	else
 		SHOBJ_LDFLAGS='-shared'
 
