diff -ru src/configure src.configure/configure
--- src/configure	2011-10-24 00:05:54.000000000 +0200
+++ src.configure/configure	2012-03-30 16:31:51.409247321 +0200
@@ -22492,7 +22492,7 @@
 if ac_fn_f77_try_compile "$LINENO"; then :
   ac_cv_prog_f77_v=
 # Try some options frequently used verbose output
-for ac_verb in -v -verbose --verbose -V -\#\#\#; do
+for ac_verb in -\#\#\# -v -verbose --verbose -V ; do
   cat > conftest.$ac_ext <<_ACEOF
       program main
 
@@ -22808,7 +22808,7 @@
 if ac_fn_c_try_compile "$LINENO"; then :
   r_cv_prog_c_v=
 # Try some options frequently used verbose output
-for r_verb in -v -verbose --verbose -V -\#\#\#; do
+for r_verb in -\#\#\# -v -verbose --verbose -V ; do
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
