--- R-3.3.1_original/src/library/parallel/src/fork.c	2016-03-17 00:02:50.000000000 +0100
+++ R-3.3.1_patched/src/library/parallel/src/fork.c	2016-08-20 10:54:42.799029599 +0200
@@ -227,15 +227,16 @@
 #else
 /* sigaction is not viable, so use the "dumb" way
    to clean up anything that comes our way */
-static void setup_sig_handler() {
-    signal(SIGCHLD, parent_sig_handler);
-}
 
 static void parent_sig_handler(int sig) {
     /* clean up when a child terminates */
     if (sig == SIGCHLD)
 	clean_zombies();
 }
+
+static void setup_sig_handler() {
+    signal(SIGCHLD, parent_sig_handler);
+}
 #endif
 
 SEXP mc_fork(SEXP sEstranged)
