diff -ru src/src/scripts/R.sh.in src.new/src/scripts/R.sh.in
--- src/src/scripts/R.sh.in	2011-10-03 00:02:35.000000000 +0200
+++ src.new/src/scripts/R.sh.in	2012-03-30 17:17:12.409254535 +0200
@@ -26,6 +26,14 @@
   esac
 fi
 
+# R in the Sage installation, we set this to allow moving the Sage
+# install tree.
+# This script is also used during R's spkg-install, do not change
+# R_HOME_DIR then (spkg-install sets SAGE_BUILDING_R=yes).
+if test x$SAGE_BUILDING_R = x; then
+    R_HOME_DIR="$SAGE_LOCAL/lib/R/"
+fi
+
 if test -n "${R_HOME}" && \
    test "${R_HOME}" != "${R_HOME_DIR}"; then
   echo "WARNING: ignoring environment value of R_HOME"
