--- R-3.3.1_original/src/scripts/Makefile.in	2013-08-05 00:05:05.000000000 +0200
+++ R-3.3.1_patched/src/scripts/Makefile.in	2016-08-20 11:06:19.103189078 +0200
@@ -90,10 +90,7 @@
 install: installdirs install-cmds
 	@rm -f $(DESTDIR)$(bindir)/R
 	@(d=`$(ECHO) '$(rhome)' | sed 's,/,\\\/,g';`; \
-	  d2=`$(ECHO) '$(rsharedir)' | sed 's,/,\\\/,g';`; \
-	  d3=`$(ECHO) '$(rincludedir)' | sed 's,/,\\\/,g';`; \
-	  d4=`$(ECHO) '$(rdocdir)' | sed 's,/,\\\/,g';`; \
-	  sed -e "1,/R_HOME_DIR=/s/\\(R_HOME_DIR=\\).*/\\1$${d}/;" -e "s/\\(R_SHARE_DIR=\\).*/\\1$${d2}/;" -e  "s/\\(R_INCLUDE_DIR=\\).*/\\1$${d3}/;" -e "s/\\(R_DOC_DIR=\\).*/\\1$${d4}/;"\
+	  sed -e "1,/R_HOME_DIR=/s/\\(R_HOME_DIR=\\).*/\\1$${d}/;"\
 	  < R.fe > "$(DESTDIR)$(Rexecbindir)/R")
 	@$(INSTALL_SCRIPT)  "$(DESTDIR)$(Rexecbindir)/R" "$(DESTDIR)$(bindir)/R"
 	@chmod 755 "$(DESTDIR)$(bindir)/R" "$(DESTDIR)$(Rexecbindir)/R"
