Index: R/src/library/stats/src/cov.c
===================================================================
--- R/src/library/stats/src/cov.c	(révision 60268)
+++ R/src/library/stats/src/cov.c	(copie de travail)
@@ -1,6 +1,6 @@
 /*
  *  R : A Computer Language for Statistical Data Analysis
- *  Copyright (C) 1995-2012	The R Core Team
+ *  Copyright (C) 1995-2013	The R Core Team
  *  Copyright (C) 2003		The R Foundation
  *
  *  This program is free software; you can redistribute it and/or modify
@@ -22,6 +22,12 @@
 #include <config.h>
 #endif
 
+#ifdef HAVE_LONG_DOUBLE
+# define SQRTL sqrtl
+#else
+# define SQRTL sqrt
+#endif
+
 #include <Defn.h>
 #include <Rmath.h>
 
@@ -113,7 +119,7 @@
 			    ysd /= n1;					\
 			    sum /= n1;					\
 			}						\
-			sum /= (sqrtl(xsd) * sqrtl(ysd));	       	\
+			sum /= (SQRTL(xsd) * SQRTL(ysd));	       	\
 			if(sum > 1.) sum = 1.;				\
 		    }							\
 		}							\
@@ -424,7 +430,7 @@
 			    if (ind[n1] != 0 &&	 xx[k] != xx[n1])	\
 				sum ++; /* = sign(. - .)^2 */		\
 	    }								\
-	    _X_##m [i] = (double)sqrtl(sum);				\
+	    _X_##m [i] = (double)SQRTL(sum);				\
 	}
 
 	COV_SDEV(x); /* -> xm[.] */
@@ -513,7 +519,7 @@
 			    if (xx[k] != xx[n1])			\
 				sum ++; /* = sign(. - .)^2 */		\
 		}							\
-		_X_##m [i] = (double) sqrtl(sum);			\
+		_X_##m [i] = (double) SQRTL(sum);			\
 	    }
 
 	COV_SDEV(x); /* -> xm[.] */
