--- sys-std.c.orig	2008-09-30 07:14:13.000000000 -0700
+++ sys-std.c	2008-09-30 07:14:50.000000000 -0700
@@ -398,7 +398,7 @@
 #ifdef HAVE_LIBREADLINE
 
 # ifdef HAVE_READLINE_READLINE_H
-#  include <readline/readline.h>
+#  include "readline/readline.h"
 /* For compatibility with pre-readline4.2 systems: */
 #  if !defined (_RL_FUNCTION_TYPEDEF)
 typedef void rl_vcpfunc_t (char *);
@@ -428,7 +428,7 @@
 
 
 # ifdef HAVE_READLINE_HISTORY_H
-#  include <readline/history.h>
+#  include "readline/history.h"
 # endif
 
 
