diff -u -r Pygments-1.3.1/pygments/lexers/agile.py src/pygments/lexers/agile.py
--- Pygments-1.3.1/pygments/lexers/agile.py	2010-03-01 18:46:36.000000000 +0100
+++ src/pygments/lexers/agile.py	2010-11-18 20:41:11.468650551 +0100
@@ -315,6 +315,11 @@
                 insertions.append((len(curcode),
                                    [(0, Generic.Prompt, line[:4])]))
                 curcode += line[4:]
+            elif line.startswith('sage: '):
+                tb = 0
+                insertions.append((len(curcode),
+                                   [(0, Generic.Prompt, line[:6])]))
+                curcode += line[6:]
             elif line.rstrip() == '...' and not tb:
                 # only a new >>> prompt can end an exception block
                 # otherwise an ellipsis in place of the traceback frames
