From d5743eb30e21233d8adeedab5a73dc9444e367ff Mon Sep 17 00:00:00 2001
From: "R. Andrew Ohana" <andrew.ohana@gmail.com>
Date: Sat, 11 May 2013 23:15:39 -0700
Subject: [PATCH] fully fix build_dir copying

---
 Cython/Build/Dependencies.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Cython/Build/Dependencies.py b/Cython/Build/Dependencies.py
index a2676e1..5e73782 100644
--- a/Cython/Build/Dependencies.py
+++ b/Cython/Build/Dependencies.py
@@ -671,10 +671,10 @@ def cythonize(module_list, exclude=[], nthreads=0, aliases=None, quiet=False, fo
         if build_dir:
             root = os.path.realpath(os.path.abspath(find_root_package_dir(m.sources[0])))
             def copy_to_build_dir(filepath, root=root):
-                filepath = os.path.abspath(filepath)
-                if os.path.realpath(filepath).startswith(root):
-                    mod_dir = os.path.join(
-                        build_dir, os.path.dirname(_relpath(filepath)))
+                filepath = os.path.realpath(os.path.abspath(filepath))
+                if filepath.startswith(root):
+                    mod_dir = os.path.join(build_dir,
+                            os.path.dirname(_relpath(filepath, root)))
                     if not os.path.isdir(mod_dir):
                         os.makedirs(mod_dir)
                     shutil.copy(filepath, mod_dir)
-- 
1.8.2.1

