From dd9eb10bc5657beed1f66221511c16ec6f4e517d Mon Sep 17 00:00:00 2001
From: Robert Bradshaw <robertwb@gmail.com>
Date: Tue, 24 Jan 2017 16:46:31 -0800
Subject: [PATCH] More numpy struct format normalization.

---
 tests/run/numpy_test.pyx | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tests/run/numpy_test.pyx b/tests/run/numpy_test.pyx
index 5fca12574..39fe1ffea 100644
--- a/tests/run/numpy_test.pyx
+++ b/tests/run/numpy_test.pyx
@@ -520,7 +520,8 @@ def test_point_record():
         test[i].x = i
         test[i].y = -i
     print re.sub(
-        r'\.0+\b', '.', repr(test).replace('<', '!').replace('>', '!'))
+        r'\.0+\b', '.', repr(test).replace('<', '!').replace('>', '!')
+                                  .replace('( ', '(').replace(',  ', ', '))
 
 # Test fused np.ndarray dtypes and runtime dispatch
 @testcase
