--- a/src/Cython/Compiler/ExprNodes.py
+++ b/src/Cython/Compiler/ExprNodes.py
@@ -3011,7 +3011,7 @@
                     self.args[i] = arg = arg.coerce_to(arg_ctype, env)
             if arg.is_temp and i > 0:
                 some_args_in_temps = True
-        if some_args_in_temps:
+        if some_args_in_temps and False:
             # if some args are temps and others are not, they may get
             # constructed in the wrong order (temps first) => make
             # sure they are either all temps or all not temps (except
