commit fc728e4a0b32d27567c0a2f31c7bf57084bcc6df
Author: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date:   Tue Apr 28 14:34:52 2015 +0200

    PyTypeObject members tp_compare, tp_richcompare, tp_base

diff --git a/Cython/Includes/cpython/object.pxd b/Cython/Includes/cpython/object.pxd
index 6dc3022..ebd1acd 100644
--- a/Cython/Includes/cpython/object.pxd
+++ b/Cython/Includes/cpython/object.pxd
@@ -22,6 +22,9 @@ cdef extern from "Python.h":
     ctypedef Py_hash_t (*hashfunc)(object)
     ctypedef object (*reprfunc)(object)
 
+    ctypedef int (*cmpfunc)(object, object)
+    ctypedef object (*richcmpfunc)(object, object, int)
+
     # The following functions use 'PyObject*' as first argument instead of 'object' to prevent
     # accidental reference counting when calling them during a garbage collection run.
     ctypedef void (*destructor)(PyObject*)
@@ -46,6 +49,11 @@ cdef extern from "Python.h":
         reprfunc tp_str
         reprfunc tp_repr
 
+        cmpfunc tp_compare
+        richcmpfunc tp_richcompare
+
+        PyTypeObject* tp_base
+
     ctypedef struct PyObject:
         Py_ssize_t ob_refcnt
         PyTypeObject *ob_type
