diff -ru src/Cython/Utility/ObjectHandling.c b/Cython/Utility/ObjectHandling.c
--- src/Cython/Utility/ObjectHandling.c	2013-04-19 15:06:41.000000000 +0200
+++ b/Cython/Utility/ObjectHandling.c	2013-04-21 16:23:25.072392813 +0200
@@ -621,7 +621,9 @@
     *_stop = stop;
 }
 
-#if defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
+#if defined(__GNUC__)
+static CYTHON_INLINE void __Pyx_copy_object_array(PyObject** __restrict__ src, PyObject** __restrict__ dest, Py_ssize_t length) {
+#elif defined (__STDC_VERSION__) && __STDC_VERSION__ >= 199901L
 static CYTHON_INLINE void __Pyx_copy_object_array(PyObject** restrict src, PyObject** restrict dest, Py_ssize_t length) {
 #else
 static CYTHON_INLINE void __Pyx_copy_object_array(PyObject** src, PyObject** dest, Py_ssize_t length) {
