commit bb4f0a9c656b52579ca7fabcc43fd72804e0e318
Author: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date:   Thu Feb 8 17:49:42 2018 +0100

    Use NamedTemporaryFile instead of mkstemp

diff --git a/Cython/Debugger/libpython.py b/Cython/Debugger/libpython.py
index 9e94168..dc3e44a 100644
--- a/Cython/Debugger/libpython.py
+++ b/Cython/Debugger/libpython.py
@@ -1941,7 +1941,6 @@ PyLocals()
 ##################################################################
 
 import re
-import atexit
 import warnings
 import tempfile
 import textwrap
@@ -2023,14 +2022,13 @@ class _LoggingState(object):
     """
 
     def __init__(self):
-        self.fd, self.filename = tempfile.mkstemp()
-        self.file = os.fdopen(self.fd, 'r+')
+        f = tempfile.NamedTemporaryFile('r+')
+        self.file = f
+        self.filename = f.name
+        self.fd = f.fileno()
         _execute("set logging file %s" % self.filename)
         self.file_position_stack = []
 
-        atexit.register(os.close, self.fd)
-        atexit.register(os.remove, self.filename)
-
     def __enter__(self):
         if not self.file_position_stack:
             _execute("set logging redirect on")
