--- matrix.h.orig	2009-10-22 01:25:58.000000000 +0200
+++ matrix.h	2010-04-28 18:21:29.000000000 +0200
@@ -107,7 +107,7 @@
     return ret;
   }
   Vektor<typ>& operator[](int n){assert(n>=0 && n<getHeight());return (rows[n]);}
-  const Vektor<typ>& operator[](int n)const{assert(n>=0 && n<getHeight());return (rows[n]);}
+  const Vektor<typ>& operator[](int n)const{/*assert(n>=0 && n<getHeight());*/return (rows[n]);}
 };
 
 typedef Matrix<int> IntegerMatrix;
