diff --git a/Makefile b/Makefile
index 0892969..f816b22 100644
--- a/Makefile
+++ b/Makefile
@@ -33,13 +33,8 @@ GMP_LINKOPTIONS = $(gmppath)/lib/libgmp.a
 GMP_INCLUDEOPTIONS = -I $(gmppath)/include
 endif
 
-ifeq ($(cddpath),)
-CDD_LINKOPTIONS = -L/usr/local -lcddgmp
+CDD_LINKOPTIONS = -L$SAGE_LOCAL/lib -lcddgmp -lgmp -lm
 CDD_INCLUDEOPTIONS =
-else
-CDD_LINKOPTIONS = $(cddpath)/lib/libcddgmp.a
-CDD_INCLUDEOPTIONS = -I $(cddpath)/include
-endif
 
 ifeq ($(soplex),)
 SOPLEX_PATH =
@@ -75,18 +70,10 @@ ADDITIONALINCLUDEOPTIONS = $(CDD_INCLUDEOPTIONS) $(GMP_INCLUDEOPTIONS) $(SOPLEX_
 MKDIR=mkdir -p
 
 
-SHELL       = /bin/sh
-#ARCH        = LINUX
-CC          = gcc
-CLINKER     = $(CC)
-CXX         = g++
-CCLINKER    = $(CXX)
-#OPTFLAGS    = -O2 -DGMPRATIONAL -DNDEBUG
 OPTFLAGS    = -O2 -DGMPRATIONAL -Wuninitialized
 
-CFLAGS	  = $(OPTFLAGS) $(GPROFFLAG) $(ADDITIONALINCLUDEOPTIONS) #-pedantic
-CCFLAGS	  = $(CFLAGS)
-FFLAGS	  = $(OPTFLAGS)
+CFLAGS_LOCAL  = $(CFLAGS) $(OPTFLAGS) $(GPROFFLAG) $(ADDITIONALINCLUDEOPTIONS) #-pedantic
+CXXFLAGS_LOCAL= $(CXXFLAGS) $(OPTFLAGS) $(GPROFFLAG) $(ADDITIONALINCLUDEOPTIONS) #-pedantic
 
 CATSOBJECTS =	$(GCATSPATH)lp_cdd.o \
 		$(SOPLEX_OBJECTS) \
@@ -317,7 +304,7 @@ $(PREFIX):
 default: $(OBJECTS) $(ADDITIONALOBJECTS) $(EXECS)
 
 $(MAIN): $(OBJECTS)
-	$(CCLINKER) $(OBJECTS) $(ADDITIONALLINKOPTIONS) $(GPROFFLAG) -o $(MAIN)
+	$(CXX) $(CXXFLAGS) -o $(MAIN) $(LDFLAGS) $(OBJECTS) $(ADDITIONALLINKOPTIONS) $(GPROFFLAG)
 
 release:
 	rm -f -r $(RELEASEDIR)/*
@@ -431,11 +418,11 @@ library:
 	cp gfanlib_* gfanlib/
 	tar zcf -  gfanlib > gfanlib.tar.gz
 .c.o:
-	$(CC) $(CFLAGS) -c $<
+	$(CC) $(CFLAGS_LOCAL) $(CPPFLAGS) -c $<
 .cc.o:
 	$(CXX) -c $<
 .cpp.o:
-	$(CXX) $(CFLAGS) -c $<
+	$(CXX) $(CXXFLAGS_LOCAL) $(CPPFLAGS) -c $<
 .C.o:
 	$(CXX) -c $<
 # wget http://ftp.sunet.se/pub/gnu/gmp/gmp-4.2.2.tar.gz
