diff --git a/Makefile b/Makefile
index 0892969..c626eb9 100644
--- a/Makefile
+++ b/Makefile
@@ -26,7 +26,7 @@ sage_link.so: sage_link.pyx setup.py
 endif
 
 ifeq ($(gmppath),)
-GMP_LINKOPTIONS = -lgmp
+GMP_LINKOPTIONS = -lgmp -lm
 GMP_INCLUDEOPTIONS =
 else
 GMP_LINKOPTIONS = $(gmppath)/lib/libgmp.a
@@ -34,7 +34,7 @@ GMP_INCLUDEOPTIONS = -I $(gmppath)/include
 endif
 
 ifeq ($(cddpath),)
-CDD_LINKOPTIONS = -L/usr/local -lcddgmp
+CDD_LINKOPTIONS = -lcddgmp
 CDD_INCLUDEOPTIONS =
 else
 CDD_LINKOPTIONS = $(cddpath)/lib/libcddgmp.a
@@ -75,18 +75,11 @@ ADDITIONALINCLUDEOPTIONS = $(CDD_INCLUDEOPTIONS) $(GMP_INCLUDEOPTIONS) $(SOPLEX_
 MKDIR=mkdir -p
 
 
-SHELL       = /bin/sh
-#ARCH        = LINUX
-CC          = gcc
-CLINKER     = $(CC)
-CXX         = g++
-CCLINKER    = $(CXX)
 #OPTFLAGS    = -O2 -DGMPRATIONAL -DNDEBUG
 OPTFLAGS    = -O2 -DGMPRATIONAL -Wuninitialized
 
-CFLAGS	  = $(OPTFLAGS) $(GPROFFLAG) $(ADDITIONALINCLUDEOPTIONS) #-pedantic
-CCFLAGS	  = $(CFLAGS)
-FFLAGS	  = $(OPTFLAGS)
+CFLAGS_LOCAL  = $(CFLAGS) $(OPTFLAGS) $(GPROFFLAG) $(ADDITIONALINCLUDEOPTIONS) #-pedantic
+CXXFLAGS_LOCAL= $(CXXFLAGS) $(OPTFLAGS) $(GPROFFLAG) $(ADDITIONALINCLUDEOPTIONS) #-pedantic
 
 CATSOBJECTS =	$(GCATSPATH)lp_cdd.o \
 		$(SOPLEX_OBJECTS) \
@@ -317,7 +310,7 @@ $(PREFIX):
 default: $(OBJECTS) $(ADDITIONALOBJECTS) $(EXECS)
 
 $(MAIN): $(OBJECTS)
-	$(CCLINKER) $(OBJECTS) $(ADDITIONALLINKOPTIONS) $(GPROFFLAG) -o $(MAIN)
+	$(CXX) $(CXXFLAGS) -o $(MAIN) $(LDFLAGS) $(OBJECTS) $(ADDITIONALLINKOPTIONS) $(GPROFFLAG)
 
 release:
 	rm -f -r $(RELEASEDIR)/*
@@ -431,11 +424,11 @@ library:
 	cp gfanlib_* gfanlib/
 	tar zcf -  gfanlib > gfanlib.tar.gz
 .c.o:
-	$(CC) $(CFLAGS) -c $<
+	$(CC) $(CFLAGS_LOCAL) $(CPPFLAGS) -c $<
 .cc.o:
 	$(CXX) -c $<
 .cpp.o:
-	$(CXX) $(CFLAGS) -c $<
+	$(CXX) $(CFLAGS_LOCAL) $(CPPFLAGS) -c $<
 .C.o:
 	$(CXX) -c $<
 # wget http://ftp.sunet.se/pub/gnu/gmp/gmp-4.2.2.tar.gz
