From cc535b88a2ac93fdd390a7ef8ab6005d7a3cd8a0 Mon Sep 17 00:00:00 2001
From: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date: Thu, 22 Aug 2019 22:18:17 -0400
Subject: [PATCH 1/2] Don't use timer_t on macOS

---
 source/main/MAIN.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/source/main/MAIN.c b/source/main/MAIN.c
index 723306d..73f8cef 100644
--- a/source/main/MAIN.c
+++ b/source/main/MAIN.c
@@ -81,6 +81,9 @@ static void init_SIGINT_handler()
 
 static int sendSignalAfterInterval(int seconds, int signum)
 {
+#ifdef __APPLE__
+  return 1;
+#else
   /* Create timer */
   timer_t timerid;
   struct sigevent sev;
@@ -100,6 +103,7 @@ static int sendSignalAfterInterval(int seconds, int signum)
     return 2;
 
   return 0;
+#endif
 }
 
 int main(int argc, char **argv)
-- 
2.19.0

