diff -Nurp orig/extensions/rend/PLOT_2D_CAD.cc qesource/extensions/rend/PLOT_2D_CAD.cc
--- orig/extensions/rend/PLOT_2D_CAD.cc	2007-03-13 11:27:29.000000000 -0500
+++ qesource/extensions/rend/PLOT_2D_CAD.cc	2008-08-22 00:07:36.000000000 -0500
@@ -23,7 +23,7 @@ extern "C" {
 #include <errno.h>
 #include <stdio.h>
 	   }
-#include <string>
+#include <cstring>
 #include <fstream>
 #include <cmath>
 #include "rend.h"
diff -Nurp orig/source/db/convenientstreams.h qesource/source/db/convenientstreams.h
--- orig/source/db/convenientstreams.h	2007-10-25 10:31:01.000000000 -0500
+++ qesource/source/db/convenientstreams.h	2008-08-22 00:11:40.000000000 -0500
@@ -11,6 +11,7 @@
 #ifndef _CONVENIENT_STREAMS_
 #define _CONVENIENT_STREAMS_
 #include <string>
+#include <cstring>
 #include <sstream>
 using namespace std;
 
diff -Nurp orig/source/db/unnamedpipe.h qesource/source/db/unnamedpipe.h
--- orig/source/db/unnamedpipe.h	2007-03-13 11:27:29.000000000 -0500
+++ qesource/source/db/unnamedpipe.h	2008-08-22 00:08:33.000000000 -0500
@@ -17,6 +17,7 @@
 #include <unistd.h>
 #include <iostream>
 #include <streambuf>
+#include <cstring>
 using namespace std;
 
 /**************************************************
