--- a/source/main/BEGINQEPCAD.c.orig	2018-01-30 14:04:58.000000000 -0700
+++ b/source/main/BEGINQEPCAD.c	2018-08-29 20:03:05.718540333 -0600
@@ -127,7 +127,8 @@ void BEGINQEPCAD(int &argc, char**& argv
 void QEPCAD_ProcessRC(int argc, char **argv)
 {
   char *qepath = getenv("qe");
-  if (qepath == NULL) { FAIL("QEPCAD_ProcessRC","Environment variable qe not defined!"); }
+  if (qepath == NULL) { setenv("qe", "/usr/share/qepcad", 1); qepath = getenv("qe"); }
+  if (getenv("SINGULARPATH") == NULL) { setenv("SINGULARPATH", "@LIBDIR@/Singular", 1); }
   string rcFileName = qepath + string("/default.qepcadrc");
   ifstream rcin(rcFileName.c_str());
   if (!rcin) { return; }
