diff -ru mpc-1.0.0dev1126/m4/mpc.m4 mpc-1.0.0dev1126M/m4/mpc.m4
--- mpc-1.0.0dev1126/m4/mpc.m4	2012-03-22 15:02:41.000000000 +0100
+++ mpc-1.0.0dev1126M/m4/mpc.m4	2012-03-23 14:08:42.000000000 +0100
@@ -91,10 +91,6 @@
   AC_REQUIRE([AC_PROG_GREP])
   if echo $VERSION | grep -c dev >/dev/null 2>&1 ; then
     if test "x$GCC" = "xyes" -a "x$compiler" != "xicc" -a "x$compiler" != "xg++"; then
-      case $host in
-         *darwin*) ;;
-         *) MPC_C_CHECK_FLAG(-D_FORTIFY_SOURCE=2,$1) ;;
-      esac
       MPC_C_CHECK_FLAG(-g)
       MPC_C_CHECK_FLAG(-std=c99)
       MPC_C_CHECK_FLAG(-pedantic)
diff -ru mpc-1.0.0dev1126/configure mpc-1.0.0dev1126M/configure
--- mpc-1.0.0dev1126/configure	2012-03-23 10:05:22.000000000 +0100
+++ mpc-1.0.0dev1126M/configure	2012-03-23 14:08:55.000000000 +0100
@@ -11630,85 +11630,6 @@
 
   if echo $VERSION | grep -c dev >/dev/null 2>&1 ; then
     if test "x$GCC" = "xyes" -a "x$compiler" != "xicc" -a "x$compiler" != "xg++"; then
-      case $host in
-         *darwin*) ;;
-         *)
-
-
-
-
-
-  flag=`echo "-D_FORTIFY_SOURCE=2" | $SED 'y% .=/+-(){}<>:*,%_______________%'`
-
-  { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether the C compiler accepts the -D_FORTIFY_SOURCE=2 flag" >&5
-$as_echo_n "checking whether the C compiler accepts the -D_FORTIFY_SOURCE=2 flag... " >&6; }
-if eval \${ax_cv_c_check_flag_$flag+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-
-
-    ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
-ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
-ac_compiler_gnu=$ac_cv_c_compiler_gnu
-
-
-    save_CFLAGS="$CFLAGS"
-    CFLAGS="$CFLAGS -D_FORTIFY_SOURCE=2"
-    cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-
-int
-main ()
-{
-
-  ;
-  return 0;
-}
-
-_ACEOF
-if ac_fn_c_try_compile "$LINENO"; then :
-
-      eval "ax_cv_c_check_flag_$flag=yes"
-
-else
-
-      eval "ax_cv_c_check_flag_$flag=no"
-
-fi
-rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
-
-    CFLAGS="$save_CFLAGS"
-
-    ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
-ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
-ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
-ac_compiler_gnu=$ac_cv_c_compiler_gnu
-
-
-
-fi
-eval ac_res=\$ax_cv_c_check_flag_$flag
-	       { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_res" >&5
-$as_echo "$ac_res" >&6; }
-
-  if eval "test \"`echo '$ax_cv_c_check_flag_'$flag`\" = yes"; then :
-
-    :
-    CFLAGS="$CFLAGS -D_FORTIFY_SOURCE=2"
-
-else
-
-    :
-
-
-fi
-
- ;;
-      esac
 
 
 
