--- ../src/Setup.py	2012-05-05 15:14:14.000000000 -0700
+++ Setup.py	2012-05-05 15:15:00.000000000 -0700
@@ -41,7 +41,7 @@
 
 libdirs = dirtest([
     os.environ["SAGE_LOCAL"]+"/lib",
-    "/usr/local/lib", "/sw/lib", "/usr/lib",
+    "/usr/local/lib", "/usr/lib",
     "/usr/lib/X11", "/usr/X11R6/lib",
     "/opt/gnome/lib",
 ])
@@ -51,7 +51,7 @@
 
 incdirs = dirtest([
     os.environ["SAGE_LOCAL"]+"/include",
-    "/usr/local/include", "/sw/include", "/usr/include",
+    "/usr/local/include", "/usr/include",
     "/usr/include/X11", "/usr/X11R6/include",
     "/opt/gnome/include",
 ])
@@ -61,13 +61,12 @@
 import os
 if os.uname()[0][:6] == "CYGWIN":
     want_libs = [
-        "gd", "png", "z", "freetype", "libiconv"
+        "gd", "png12", "z", "freetype", "iconv", "fontconfig",
     ]
     libs = filetest(libdirs, want_libs)
-    libs.append('libiconv')
 else:
     want_libs = [
-        "gd", "png", "z", "freetype"
+        "gd", "png12", "z", "freetype"
     ]
     libs = filetest(libdirs, want_libs)
 
@@ -92,18 +91,22 @@
 if "Xpm" in libs and "X11" not in libs:
     remove("Xpm", libs)
 
-if "png" in libs and "z" not in libs:
-    remove("png", libs)
+if "png12" in libs and "z" not in libs:
+    remove("png12", libs)
 
-if "z" in libs and "png" not in libs:
-    remove("png", libs)
+if "z" in libs and "png12" not in libs:
+    remove("png12", libs)
 
 # build the macro list
 
 macros = []
 
 for l in libs:
-    macros.append(( "HAVE_LIB%s" % l.upper(), None ))
+    if l == "png12":
+       macros.append(( "HAVE_LIBPNG", None ))
+    else:
+       macros.append(( "HAVE_LIB%s" % l.upper(), None ))
+
 
 # OK, now do it!
 
