diff -ru src/Setup.py b/Setup.py
--- src/Setup.py	2005-03-11 05:29:59.000000000 +0100
+++ b/Setup.py	2012-05-06 00:54:31.893372094 +0200
@@ -40,7 +40,8 @@
 # Unixoid OS's I will just look for the usual suspects.
 
 libdirs = dirtest([
-    "/usr/local/lib", "/sw/lib", "/usr/lib",
+    os.environ["SAGE_LOCAL"]+"/lib",
+    "/usr/lib",
     "/usr/lib/X11", "/usr/X11R6/lib",
     "/opt/gnome/lib",
 ])
@@ -48,21 +49,25 @@
 # include_dirs are also non-portable; same trick here.
 
 incdirs = dirtest([
-    "/usr/local/include", "/sw/include", "/usr/include",
+    os.environ["SAGE_LOCAL"]+"/include",
+    "/usr/include",
     "/usr/include/X11", "/usr/X11R6/include",
     "/opt/gnome/include",
 ])
 
 # Try to identify our libraries
 
-want_libs = [
-    "gd",
-    "jpeg", "png", "gif", "z",
-    "X11", "Xpm",
-    "ttf", "freetype",
-]
-
-libs = filetest(libdirs, want_libs)
+import os
+if os.uname()[0][:6] == "CYGWIN":
+    want_libs = [
+        "gd", "png12", "z", "freetype", "iconv", "fontconfig",
+    ]
+    libs = filetest(libdirs, want_libs)
+else:
+    want_libs = [
+        "gd", "png12", "z", "freetype"
+    ]
+    libs = filetest(libdirs, want_libs)
 
 missing = []
 
@@ -85,18 +90,22 @@
 if "Xpm" in libs and "X11" not in libs:
     remove("Xpm", libs)
 
-if "png" in libs and "z" not in libs:
-    remove("png", libs)
+if "png12" in libs and "z" not in libs:
+    remove("png12", libs)
 
-if "z" in libs and "png" not in libs:
-    remove("png", libs)
+if "z" in libs and "png12" not in libs:
+    remove("png12", libs)
 
 # build the macro list
 
 macros = []
 
 for l in libs:
-    macros.append(( "HAVE_LIB%s" % l.upper(), None ))
+    if l == "png12":
+       macros.append(( "HAVE_LIBPNG", None ))
+    else:
+       macros.append(( "HAVE_LIB%s" % l.upper(), None ))
+
 
 # OK, now do it!
 
