Variables must be declared "extern" in header file

diff -ru planarity-2.2.0/c/planarity.h b/c/planarity.h
--- planarity-2.2.0/c/planarity.h	2015-05-05 14:20:57.000000000 +0200
+++ b/c/planarity.h	2015-05-12 12:10:49.780865449 +0200
@@ -77,7 +77,7 @@
 int legacyCommandLine(int argc, char *argv[]);
 int menu();
 
-char Mode,
+extern char Mode,
      OrigOut,
      EmbeddableOut,
      ObstructedOut,
@@ -88,7 +88,7 @@
 
 /* Low-level Utilities */
 #define MAXLINE 1024
-char Line[MAXLINE];
+extern char Line[MAXLINE];
 
 void Message(char *message);
 void ErrorMessage(char *message);
