--- a/check/TP16-sol.cas.out1	2016-07-16 14:21:09.000000000 +0200
+++ b/check/TP16-sol.cas.out1	2016-09-14 14:08:07.110527423 +0200
@@ -48,7 +48,7 @@
 "Done",
 [-sqrt(13)-1,sqrt(13)-1,4],
 y^2+6*sqrt(13)+18,y^2-6*sqrt(13)+18,y^2,
--sqrt(6)*sqrt(sqrt(13)-3),sqrt(6)*sqrt(sqrt(13)-3),-sqrt(6)*I*sqrt(sqrt(13)+3),sqrt(6)*I*sqrt(sqrt(13)+3),0,
+sqrt(6)*I*sqrt(sqrt(13)+3),-sqrt(6)*I*sqrt(sqrt(13)+3),-sqrt(6)*sqrt(sqrt(13)-3),sqrt(6)*sqrt(sqrt(13)-3),0,
 "No such variable u",
 x^2+1/4*y^2+1/9*z^2-1,
 x^2+y^2+z^2-u,
