diff --git a/giac-1.4.9.45-orig/src/src/index.h b/giac-1.4.9.45/src/src/index.h
index e74be63..8ed12f9 100755
--- a/src/index.h
+++ b/src/index.h
@@ -42,7 +42,7 @@
 #include <unordered_map>
 #endif
 
-#if defined UNORDERED_MAP  && !defined(__clang__) && !defined(VISUALC) // && !defined(__APPLE__)
+#if defined UNORDERED_MAP  && !defined(VISUALC) // && !defined(__APPLE__)
 #include <tr1/unordered_map>
 #define HASH_MAP_NAMESPACE std::tr1
 #define hash_map unordered_map
