--- a/src/ifactor.cc	2014-10-11 16:29:01.000000000 +0200
+++ b/src/ifactor.cc	2014-10-11 16:29:28.000000000 +0200
@@ -3708,7 +3708,7 @@
       return giac_ifactors(n0._VECTptr->front(),contextptr);
 #ifdef HAVE_LIBPARI
 #ifdef __APPLE__
-    return vecteur(1,gensizeerr(gettext("(Mac OS) Large number, you can try pari(); pari_factor(")+n0.print(contextptr)+")"));
+    //    return vecteur(1,gensizeerr(gettext("(Mac OS) Large number, you can try pari(); pari_factor(")+n0.print(contextptr)+")"));
 #endif
     if (!is_integer(n0) || is_zero(n0))
       return vecteur(1,gensizeerr(gettext("ifactors")));
