--- a/check/TP00-sol.cas.out1	2015-02-02 14:47:11.000000000 +0100
+++ b/check/TP00-sol.cas.out1	2015-02-13 12:02:57.000000000 +0100
@@ -66,7 +66,7 @@
 5,
 (sqrt(5)-1)/4+(I)*sqrt(2*sqrt(5)+10)/4,
 "Done",
- (i,j,k)->if k=1 then segment(a^i,a^j); else seq(arc(a^i,a^j,l/k),l=(1 .. k)); 
+ (i,j,k)->if k=1 then segment(a^i,a^j,'color'=56); else seq(arc(a^i,a^j,l/k,'color'=56),l=(1 .. k)); 
 fi ,
 "Done",
 pnt(pnt[(sqrt(5)-1)/4+(I)*sqrt(2*sqrt(5)+10)/4,1048580]),pnt(pnt[((sqrt(5)-1)/4+(I)*sqrt(2*sqrt(5)+10)/4)^2,1048580]),pnt(pnt[((sqrt(5)-1)/4+(I)*sqrt(2*sqrt(5)+10)/4)^3,1048580]),pnt(pnt[((sqrt(5)-1)/4+(I)*sqrt(2*sqrt(5)+10)/4)^4,1048580]),pnt(pnt[((sqrt(5)-1)/4+(I)*sqrt(2*sqrt(5)+10)/4)^5,1048580]),
