--- a/check/TP06-sol.cas	2012-11-06 14:17:56.000000000 +0100
+++ b/check/TP06-sol.cas	2014-08-14 11:05:28.000000000 +0200
@@ -15,7 +15,7 @@
 normal(monlagrange(A)-det(A-x*idn(k)));
 
  k:=50:A:=matrix(k,k,(i,j)->rand(21)-10):
- pari():;
+ pari_charpoly:=proc(AA,nn) local ttmp; ttmp:=pcar(AA,pmin); poly1[op(ttmp)] end: // emulate parioutput
  pari_charpoly(A,0);
  pppp3:=poly2symb(pari_charpoly(A,1),x);
  pppp2:=pcar(A,lagrange);
