diff -ruN check/cas.out check.new/cas.out
--- a/check/cas.out	2015-04-30 10:21:51.000000000 +0200
+++ b/check/cas.out	2015-05-31 14:18:05.209432192 +0200
@@ -252,8 +252,8 @@
 matrix[[-8.1,-2.1,6.9,-9.1,2.9],[-5.1,-2.1,9.9,9.9,2.9],[-2.1,8.9,7.9,-7.1,6.9],[0.9,-3.1,7.9,-9.1,0.9],[0.9,-7.1,-1.1,2.9,8.9]],
 matrix[[-8.1-X,-2.1,6.9,-9.1,2.9],[-5.1,-2.1-X,9.9,9.9,2.9],[-2.1,8.9,7.9-X,-7.1,6.9],[0.9,-3.1,7.9,-9.1-X,0.9],[0.9,-7.1,-1.1,2.9,8.9-X]],
 -1.00000000004*X^5-2.49999999953*X^4+110.499999998*X^3+1240.4*X^2-11443.7*X-228845.7,
-1/20736*(12*x+12*(-1/sqrt(2)+(i)/sqrt(2)))*(12*x+12*(-1/sqrt(2)-(i)/sqrt(2)))*(12*x+12*(1/sqrt(2)+(i)/sqrt(2)))*(12*x+12*(1/sqrt(2)-(i)/sqrt(2))),
-1/20736*(12*x+12*(-1/sqrt(2)+(i)/sqrt(2)))*(12*x+12*(-1/sqrt(2)-(i)/sqrt(2)))*(12*x+12*(1/sqrt(2)+(i)/sqrt(2)))*(12*x+12*(1/sqrt(2)-(i)/sqrt(2))),
+0,
+0,
 [[1,2],[3,4]],
 [[1.0,2.0],[3.0,4.0]],
 [[-2.0,1.0],[1.5,-0.5]],
diff -ruN check/testcas check.new/testcas
--- a/check/testcas	2008-10-16 10:46:06.000000000 +0200
+++ b/check/testcas	2015-05-31 14:18:35.233432678 +0200
@@ -252,8 +252,8 @@
 A:=matrix(k,k,(l,j)->rand(21)-10.1);
 B:=A-X*identity(k);
 simplify(det(B));
-factor(x^4+1,exp(i*pi/4));
-factor(x^4+1,exp(i*pi/4)+1);
+regroup(factor(x^4+1,exp(i*pi/4))-'1/20736*(12*x+12*(-1/sqrt(2)+(i)/sqrt(2)))*(12*x+12*(-1/sqrt(2)-(i)/sqrt(2)))*(12*x+12*(1/sqrt(2)+(i)/sqrt(2)))*(12*x+12*(1/sqrt(2)-(i)/sqrt(2)))');
+regroup((factor(x^4+1,exp(i*pi/4)+1))-'1/20736*(12*x+12*((i)/sqrt(2)-1/sqrt(2)))*(12*x+12*(-(i)/sqrt(2)-1/sqrt(2)))*(12*x+12*((i)/sqrt(2)+1/sqrt(2)))*(12*x+12*(-(i)/sqrt(2)+1/sqrt(2)))');
 A:=[[1,2],[3,4]];
 B:=approx(A);
 inv(B);
diff -ruN check/TP12-sol.cas check.new/TP12-sol.cas
--- a/check/TP12-sol.cas	2012-07-23 13:19:37.000000000 +0200
+++ b/check/TP12-sol.cas	2015-05-31 12:03:31.629446903 +0200
@@ -105,7 +105,7 @@
  m;
  end;
  ordre(-1,2^1000);
- pari();
+ pari():;
  if (pari_znorder(Mod(5,11^5*2^40*101)) == ordre(5,11^5*2^40*101)) then
  print("ca marche") else print("il y a une erreur") fi;
 
diff -ruN check/TP12-sol.cas.out1 check.new/TP12-sol.cas.out1
--- a/check/TP12-sol.cas.out1	2014-06-25 09:15:40.000000000 +0200
+++ b/check/TP12-sol.cas.out1	2015-05-31 12:05:59.525449300 +0200
@@ -126,10 +126,5 @@
  
 end;,
 2,
-"All PARI functions are now defined with the pari_ prefix.
-PARI functions are also defined without prefix except:
-% abs acos acosh apply arg asin asinh atan atanh binomial bitand bitor bitxor break ceil charpoly concat conj content cos cosh default divisors erfc eval exp factor factorial floor frac gcd global hilbert imag isprime kill lcm length local matrix max min next nextprime norm print real round select shift sign simplify sin sinh solve sqrt subst sum tan tanh taylor trace truncate type until valuation vector version write 
-When working with p-adic numbers use them in a pari() call
-Type ?pari for short help
-Inside xcas, try Help->Manuals->PARI for HTML help",
+"Done",
 0
