diff -ruN a/Makefile.am b/Makefile.am
--- a/Makefile.am	2017-02-06 16:01:16.484367300 +0100
+++ b/Makefile.am	2017-02-06 16:06:28.013569300 +0100
@@ -8,18 +8,18 @@
 ## FIXME: installing autoconf-generated config.h is just PLAIN WRONG.
 
 install-exec-hook:
-	cp config.h $(DESTDIR)/$(includedir)/giac
-	./mkinstalldirs $(DESTDIR)/$(prefix)/share/
-	./mkinstalldirs $(DESTDIR)/$(prefix)/share/pixmaps
-	cp xcas.xpm $(DESTDIR)/$(prefix)/share/pixmaps/xcas.xpm
-	./mkinstalldirs $(DESTDIR)/$(prefix)/share/applications
-	cp xcas.desktop $(DESTDIR)/$(prefix)/share/applications/xcas.desktop
-	./mkinstalldirs $(DESTDIR)/$(prefix)/share/application-registry
-	cp xcas.applications $(DESTDIR)/$(prefix)/share/application-registry/xcas.applications
-	./mkinstalldirs $(DESTDIR)/$(prefix)/share/icons/
-	./mkinstalldirs $(DESTDIR)/$(prefix)/share/icons/hicolor
-	for SIZE in 256 128 64 32 16; do ./mkinstalldirs $(DESTDIR)/$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE} && ./mkinstalldirs $(DESTDIR)/$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE}/apps && cp icons/xcas_$${SIZE}.png $(DESTDIR)/$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE}/apps/xcas.png; done
-	for SIZE in 256 128 64 32 16; do ./mkinstalldirs $(DESTDIR)/$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE}/mimetypes && cp icons/x-xcas_$${SIZE}.png $(DESTDIR)/$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE}/mimetypes/application-x-xcas.png;  done
+	cp config.h $(DESTDIR)$(includedir)/giac
+	./mkinstalldirs $(DESTDIR)$(prefix)/share/
+	./mkinstalldirs $(DESTDIR)$(prefix)/share/pixmaps
+	cp xcas.xpm $(DESTDIR)$(prefix)/share/pixmaps/xcas.xpm
+	./mkinstalldirs $(DESTDIR)$(prefix)/share/applications
+	cp xcas.desktop $(DESTDIR)$(prefix)/share/applications/xcas.desktop
+	./mkinstalldirs $(DESTDIR)$(prefix)/share/application-registry
+	cp xcas.applications $(DESTDIR)$(prefix)/share/application-registry/xcas.applications
+	./mkinstalldirs $(DESTDIR)$(prefix)/share/icons/
+	./mkinstalldirs $(DESTDIR)$(prefix)/share/icons/hicolor
+	for SIZE in 256 128 64 32 16; do ./mkinstalldirs $(DESTDIR)$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE} && ./mkinstalldirs $(DESTDIR)$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE}/apps && cp icons/xcas_$${SIZE}.png $(DESTDIR)$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE}/apps/xcas.png; done
+	for SIZE in 256 128 64 32 16; do ./mkinstalldirs $(DESTDIR)$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE}/mimetypes && cp icons/x-xcas_$${SIZE}.png $(DESTDIR)$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE}/mimetypes/application-x-xcas.png;  done
 
 ACLOCAL_AMFLAGS = -I m4
 
diff -ruN a/Makefile.in b/Makefile.in
--- a/Makefile.in	2017-02-06 16:01:16.279348700 +0100
+++ b/Makefile.in	2017-02-06 16:07:13.838103600 +0100
@@ -847,18 +847,18 @@
 $(PACKAGE)-$(VERSION).tar.gz: dist
 
 install-exec-hook:
-	cp config.h $(DESTDIR)/$(includedir)/giac
-	./mkinstalldirs $(DESTDIR)/$(prefix)/share/
-	./mkinstalldirs $(DESTDIR)/$(prefix)/share/pixmaps
-	cp xcas.xpm $(DESTDIR)/$(prefix)/share/pixmaps/xcas.xpm
-	./mkinstalldirs $(DESTDIR)/$(prefix)/share/applications
-	cp xcas.desktop $(DESTDIR)/$(prefix)/share/applications/xcas.desktop
-	./mkinstalldirs $(DESTDIR)/$(prefix)/share/application-registry
-	cp xcas.applications $(DESTDIR)/$(prefix)/share/application-registry/xcas.applications
-	./mkinstalldirs $(DESTDIR)/$(prefix)/share/icons/
-	./mkinstalldirs $(DESTDIR)/$(prefix)/share/icons/hicolor
-	for SIZE in 256 128 64 32 16; do ./mkinstalldirs $(DESTDIR)/$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE} && ./mkinstalldirs $(DESTDIR)/$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE}/apps && cp icons/xcas_$${SIZE}.png $(DESTDIR)/$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE}/apps/xcas.png; done
-	for SIZE in 256 128 64 32 16; do ./mkinstalldirs $(DESTDIR)/$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE}/mimetypes && cp icons/x-xcas_$${SIZE}.png $(DESTDIR)/$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE}/mimetypes/application-x-xcas.png;  done
+	cp config.h $(DESTDIR)$(includedir)/giac
+	./mkinstalldirs $(DESTDIR)$(prefix)/share/
+	./mkinstalldirs $(DESTDIR)$(prefix)/share/pixmaps
+	cp xcas.xpm $(DESTDIR)$(prefix)/share/pixmaps/xcas.xpm
+	./mkinstalldirs $(DESTDIR)$(prefix)/share/applications
+	cp xcas.desktop $(DESTDIR)$(prefix)/share/applications/xcas.desktop
+	./mkinstalldirs $(DESTDIR)$(prefix)/share/application-registry
+	cp xcas.applications $(DESTDIR)$(prefix)/share/application-registry/xcas.applications
+	./mkinstalldirs $(DESTDIR)$(prefix)/share/icons/
+	./mkinstalldirs $(DESTDIR)$(prefix)/share/icons/hicolor
+	for SIZE in 256 128 64 32 16; do ./mkinstalldirs $(DESTDIR)$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE} && ./mkinstalldirs $(DESTDIR)$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE}/apps && cp icons/xcas_$${SIZE}.png $(DESTDIR)$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE}/apps/xcas.png; done
+	for SIZE in 256 128 64 32 16; do ./mkinstalldirs $(DESTDIR)$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE}/mimetypes && cp icons/x-xcas_$${SIZE}.png $(DESTDIR)$(prefix)/share/icons/hicolor/$${SIZE}x$${SIZE}/mimetypes/application-x-xcas.png;  done
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
diff -ruN a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am	2017-02-06 16:01:51.663379800 +0100
+++ b/src/Makefile.am	2017-02-06 16:01:25.730657700 +0100
@@ -85,8 +85,8 @@
 bin_SCRIPTS = pgiac
 
 install-exec-hook:
-	rm -f $(DESTDIR)$(bindir)/cas_help
-	mv $(DESTDIR)/$(bindir)/aide$(EXEEXT) $(DESTDIR)$(bindir)/cas_help
+	rm -f $(DESTDIR)$(bindir)/cas_help$(EXEEXT)
+	mv $(DESTDIR)$(bindir)/aide$(EXEEXT) $(DESTDIR)$(bindir)/cas_help
 	ln -sf cas_help $(DESTDIR)$(bindir)/en_cas_help
 	ln -sf cas_help $(DESTDIR)$(bindir)/es_cas_help
 	ln -sf cas_help $(DESTDIR)$(bindir)/fr_cas_help
diff -ruN a/src/Makefile.in b/src/Makefile.in
--- a/src/Makefile.in	2017-02-06 16:01:52.110694800 +0100
+++ b/src/Makefile.in	2017-02-06 16:01:25.669644700 +0100
@@ -1030,7 +1030,7 @@
 
 install-exec-hook:
 	rm -f $(DESTDIR)$(bindir)/cas_help
-	mv $(DESTDIR)/$(bindir)/aide$(EXEEXT) $(DESTDIR)$(bindir)/cas_help
+	mv $(DESTDIR)$(bindir)/aide$(EXEEXT) $(DESTDIR)$(bindir)/cas_help
 	ln -sf cas_help $(DESTDIR)$(bindir)/en_cas_help
 	ln -sf cas_help $(DESTDIR)$(bindir)/es_cas_help
 	ln -sf cas_help $(DESTDIR)$(bindir)/fr_cas_help
