--- a/fflas-ffpack/fflas/fflas_level3.inl
+++ b/fflas-ffpack/fflas/fflas_level3.inl
@@ -338,6 +338,26 @@
 	       typename Field::Element_ptr C, const size_t ldc,
 	       const ParSeqHelper::Parallel<Cut,Param> par);
 
+	template<typename Field>
+	typename Field::Element_ptr
+	pfgemm (const Field& F,
+		const FFLAS_TRANSPOSE ta,
+		const FFLAS_TRANSPOSE tb,
+		const size_t m,
+		const size_t n,
+		const size_t k,
+		const typename Field::Element alpha,
+		typename Field::ConstElement_ptr A, const size_t lda,
+		typename Field::ConstElement_ptr B, const size_t ldb,
+		const typename Field::Element beta,
+		typename Field::Element_ptr C, const size_t ldc){
+		PAR_BLOCK{
+			ParSeqHelper::Parallel<CuttingStrategy::Block,StrategyParameter::Threads> par;
+			fgemm(F,ta,tb,m,n,k,alpha,A,lda,B,ldb,beta,C,ldc,par);
+		}
+		return C;
+	}
+
 	template<class Field>
         typename Field::Element*
         pfgemm_1D_rec( const Field& F,
