diff -u -r old/aclocal.m4 src/aclocal.m4
--- old/aclocal.m4	2011-04-01 00:35:38.000000000 +0100
+++ src/aclocal.m4	2013-04-05 21:00:44.179404568 +0100
@@ -5088,7 +5088,6 @@
 		chmod +x mk_shared_lib.sh
 		;;
 	darwin*) #(vi
-		EXTRA_CFLAGS="-no-cpp-precomp"
 		CC_SHARED_OPTS="-dynamic"
 		MK_SHARED_LIB='${CC} ${CFLAGS} -dynamiclib -install_name ${libdir}/`basename $[@]` -compatibility_version ${ABI_VERSION} -current_version ${ABI_VERSION} -o $[@]'
 		test "$cf_cv_shlib_version" = auto && cf_cv_shlib_version=abi
diff -u -r old/Ada95/aclocal.m4 src/Ada95/aclocal.m4
--- old/Ada95/aclocal.m4	2011-04-01 00:32:36.000000000 +0100
+++ src/Ada95/aclocal.m4	2013-04-05 21:02:04.550258173 +0100
@@ -2935,7 +2935,6 @@
 		chmod +x mk_shared_lib.sh
 		;;
 	darwin*) #(vi
-		EXTRA_CFLAGS="-no-cpp-precomp"
 		CC_SHARED_OPTS="-dynamic"
 		MK_SHARED_LIB='${CC} ${CFLAGS} -dynamiclib -install_name ${libdir}/`basename $[@]` -compatibility_version ${ABI_VERSION} -current_version ${ABI_VERSION} -o $[@]'
 		test "$cf_cv_shlib_version" = auto && cf_cv_shlib_version=abi
Only in src/Ada95: aclocal.m4~
diff -u -r old/Ada95/configure src/Ada95/configure
--- old/Ada95/configure	2011-04-01 00:34:47.000000000 +0100
+++ src/Ada95/configure	2013-04-05 21:01:57.871353745 +0100
@@ -7460,7 +7460,6 @@
 		chmod +x mk_shared_lib.sh
 		;;
 	darwin*) #(vi
-		EXTRA_CFLAGS="-no-cpp-precomp"
 		CC_SHARED_OPTS="-dynamic"
 		MK_SHARED_LIB='${CC} ${CFLAGS} -dynamiclib -install_name ${libdir}/`basename $@` -compatibility_version ${ABI_VERSION} -current_version ${ABI_VERSION} -o $@'
 		test "$cf_cv_shlib_version" = auto && cf_cv_shlib_version=abi
Only in src/Ada95: configure~
diff -u -r old/configure src/configure
--- old/configure	2011-04-01 00:35:51.000000000 +0100
+++ src/configure	2013-04-05 21:01:48.756484101 +0100
@@ -5584,7 +5584,6 @@
 		chmod +x mk_shared_lib.sh
 		;;
 	darwin*) #(vi
-		EXTRA_CFLAGS="-no-cpp-precomp"
 		CC_SHARED_OPTS="-dynamic"
 		MK_SHARED_LIB='${CC} ${CFLAGS} -dynamiclib -install_name ${libdir}/`basename $@` -compatibility_version ${ABI_VERSION} -current_version ${ABI_VERSION} -o $@'
 		test "$cf_cv_shlib_version" = auto && cf_cv_shlib_version=abi
Only in src: configure~
