diff -druN src.orig/configure src/configure
--- src.orig/configure	2011-04-01 01:35:51.000000000 +0200
+++ src/configure	2013-11-04 11:19:23.771479564 +0100
@@ -7864,7 +7864,7 @@
 echo "${ECHO_T}$cf_result" >&6
 
 if test "$cf_result" = yes ; then
-	CPPFLAGS="$CPPFLAGS -D_XOPEN_SOURCE_EXTENDED"
+	CFLAGS="$CFLAGS -D_XOPEN_SOURCE_EXTENDED"
 elif test "x" != "x" ; then
 	echo "$as_me:7869: checking checking for compatible value versus " >&5
 echo $ECHO_N "checking checking for compatible value versus ... $ECHO_C" >&6
@@ -7907,7 +7907,7 @@
 echo "${ECHO_T}$cf_result" >&6
 	if test "$cf_result" = no ; then
 		# perhaps we can override it - try...
-		CPPFLAGS="$CPPFLAGS -D_XOPEN_SOURCE_EXTENDED="
+		CFLAGS="$CFLAGS -D_XOPEN_SOURCE_EXTENDED="
 	fi
 fi
 
