--- src.orig/configure	2012-12-14 18:45:34.000000000 +0100
+++ src/configure	2013-01-15 16:09:19.962069400 +0100
@@ -4125,11 +4125,22 @@
   *)
     # user specified directory
     GMP_HOME="$with_gmp"
-	if test -d ${GMP_HOME}/include && test -f ${GMP_HOME}/lib/libgmp.a ; then
+    if test -d ${GMP_HOME}/include ; 
+    then
       GMP_CFLAGS="-I${GMP_HOME}/include"
-      GMP_LIBS="${GMP_HOME}/lib/libgmp.a"
+      if test -f ${GMP_HOME}/lib/libgmp.a ; then
+        GMP_LIBS="${GMP_HOME}/lib/libgmp.a"
+      elif test -f ${GMP_HOME}/lib/libgmp.so; then
+        GMP_LIBS="${GMP_HOME}/lib/libgmp.so"
+      elif test -f ${GMP_HOME}/lib/libgmp.dll.a; then
+        GMP_LIBS="${GMP_HOME}/lib/libgmp.dll.a"
+      elif test -f ${GMP_HOME}/lib/libgmp.dylib; then
+        GMP_LIBS="${GMP_HOME}/lib/libgmp.dylib"
+      else
+        as_fn_error $? "Could not locate GMP library in the specified location" "$LINENO" 5
+      fi;
     else
-      as_fn_error $? "Could not locate GMP in the specified location" "$LINENO" 5
+      as_fn_error $? "Could not locate GMP header in the specified location" "$LINENO" 5
     fi;
     MAKE_GMP=""
     ;;
