--- old/src/macfloat.h	2012-09-01 14:09:26.594406286 +0100
+++ src/src/macfloat.h	2012-09-13 14:17:08.192879267 +0100
@@ -30,8 +30,17 @@
 #define MATH(name) name
 #endif
 
+#if !defined(SYS_IS_64_BIT) && defined(SPARC)
+#include <string.h>
+static inline Double VAL_MACFLOAT(Obj obj)
+{ Double __val; memcpy(&__val,ADDR_OBJ(obj),sizeof(Double)); return __val; }
+static inline void SET_VAL_MACFLOAT(Obj obj, Double val)
+{ Double __val = (val); memcpy(ADDR_OBJ(obj),&__val,sizeof(Double)); }
+#else
 #define VAL_MACFLOAT(obj) (*(Double *)ADDR_OBJ(obj))
 #define SET_VAL_MACFLOAT(obj, val) (*(Double *)ADDR_OBJ(obj) = val)
+#endif
+
 #define IS_MACFLOAT(obj) (TNUM_OBJ(obj) == T_MACFLOAT)
 
 extern Obj NEW_MACFLOAT( Double val);
