--- old/src/macfloat.h	2012-09-01 14:09:26.594406286 +0100
+++ src/src/macfloat.h	2012-09-06 14:32:14.515929047 +0100
@@ -30,8 +30,11 @@
 #define MATH(name) name
 #endif
 
-#define VAL_MACFLOAT(obj) (*(Double *)ADDR_OBJ(obj))
-#define SET_VAL_MACFLOAT(obj, val) (*(Double *)ADDR_OBJ(obj) = val)
+#define VAL_MACFLOAT(obj) ({ Double __val;  \
+  memcpy(&__val,ADDR_OBJ(obj),sizeof(Double)); __val; })
+#define SET_VAL_MACFLOAT(obj, val) { Double __val = (val); \
+  memcpy(ADDR_OBJ(obj),&__val,sizeof(Double)); }
+
 #define IS_MACFLOAT(obj) (TNUM_OBJ(obj) == T_MACFLOAT)
 
 extern Obj NEW_MACFLOAT( Double val);
