--- saveload.c.orig	2010-04-28 17:24:27.000000000 +0200
+++ saveload.c	2010-01-25 18:32:37.000000000 +0100
@@ -702,6 +702,7 @@
     }
 }
 
+static Obj ProtectFname;
 
 Obj SaveWorkspace( Obj fname )
 {
@@ -721,9 +722,13 @@
         return Fail;
       }
   
+  /* For some reason itanium GC seems unable to spot fname */
+  ProtectFname = fname;
   /* Do a full garbage collection */
   CollectBags( 0, 1);
   
+  ProtectFname = (Obj)0L;
+
   /* Add indices in link words of all bags, for saving inter-bag references */
   NextSaveIndex = 0;
   CallbackForAllBags( AddSaveIndex );
@@ -1061,6 +1066,8 @@
 static Int InitKernel (
     StructInitInfo *    module )
 {
+    InitGlobalBag(&ProtectFname, "Protected Filename for SaveWorkspace");
+
     /* init filters and functions                                          */
     InitHdlrFuncsFromTable( GVarFuncs );
 
