--- saveload.c.orig	2010-04-28 17:26:05.000000000 +0200
+++ saveload.c	2010-04-28 17:26:58.000000000 +0200
@@ -437,20 +437,20 @@
 
 void SaveDouble( Double d)
 {
+  union { UInt1 buf[sizeof(Double)]; double d; } v;
   UInt i;
-  UInt1 buf[sizeof(Double)];
-  *(Double *)buf = d;
+  v.d = d;
   for (i = 0; i < sizeof(Double); i++)
-    SAVE_BYTE(buf[i]);
+    SAVE_BYTE(v.buf[i]);
 }
 
 Double LoadDouble( void)
 {
+  union { UInt1 buf[sizeof(Double)]; double d; } v;
   UInt i;
-  UInt1 buf[sizeof(Double)];
   for (i = 0; i < sizeof(Double); i++)
-    buf[i] = LOAD_BYTE();
-  return *(Double *)buf;
+    v.buf[i] = LOAD_BYTE();
+  return v.d;
 }
 
 /***************************************************************************
--- sysfiles.c.orig	2010-04-28 17:26:32.000000000 +0200
+++ sysfiles.c	2010-04-28 17:26:58.000000000 +0200
@@ -192,7 +192,8 @@
     Char *              filename,
     Int4                crc_gap,
     Char *              result, 
-    Int                 len )
+    Int                 len,
+    StructInitInfo **   info_result )
 {
     UInt4               crc_dyn = 0;
     UInt4               crc_sta = 0;
@@ -389,11 +390,11 @@
 		if (found_dyn)
 			syUnloadLastModule ();
 #endif        
-        *(StructInitInfo**)result = info_sta;
+		*info_result = info_sta;
         return 2;
     }
     if ( found_gap && found_dyn ) {
-        *(StructInitInfo**)result = info_dyn;
+		*info_result = info_dyn;
         return 1;
     }
     if ( found_gap ) {
@@ -404,11 +405,11 @@
 		if (found_dyn)
 			syUnloadLastModule ();
 #endif        
-        *(StructInitInfo**)result = info_sta;
+		*info_result = info_sta;
         return 2;
     }
     if ( found_dyn ) {
-        *(StructInitInfo**)result = info_dyn;
+		*info_result = info_dyn;
         return 1;
     }
     return 0;
--- gap.c.orig	2010-04-28 17:26:18.000000000 +0200
+++ gap.c	2010-04-28 17:26:58.000000000 +0200
@@ -1670,7 +1670,7 @@
 
     /* check if have a statically or dynamically loadable module           */
     crc1 = INT_INTOBJ(crc);
-    res  = SyFindOrLinkGapRootFile(CSTR_STRING(filename), crc1, result, 256);
+    res  = SyFindOrLinkGapRootFile(CSTR_STRING(filename), crc1, result, 256, &info);
 
     /* not found                                                           */
     if ( res == 0 ) {
@@ -1685,7 +1685,6 @@
             Pr( "#I  READ_GAP_ROOT: loading '%s' dynamically\n",
                 (Int)CSTR_STRING(filename), 0L );
         }
-        info = *(StructInitInfo**)result;
         res  = info->initKernel(info);
 	UpdateCopyFopyInfo();
         res  = res || info->initLibrary(info);
@@ -1703,7 +1702,6 @@
             Pr( "#I  READ_GAP_ROOT: loading '%s' statically\n",
                 (Int)CSTR_STRING(filename), 0L );
         }
-        info = *(StructInitInfo**)result;
         res  = info->initKernel(info);
 	UpdateCopyFopyInfo();
         res  = res || info->initLibrary(info);
--- sysfiles.h.orig	2010-04-28 17:27:16.000000000 +0200
+++ sysfiles.h	2010-04-28 17:26:58.000000000 +0200
@@ -47,7 +47,8 @@
             Char *          filename,
             Int4            crc_gap,
             Char *          result,
-            Int             len );
+            Int             len,
+        StructInitInfo **   info_result );
 
 
 /****************************************************************************
--- streams.c.orig	2010-04-28 17:26:25.000000000 +0200
+++ streams.c	2010-04-28 17:26:58.000000000 +0200
@@ -322,7 +322,7 @@
     StructInitInfo *    info;
 
     /* try to find the file                                                */
-    res = SyFindOrLinkGapRootFile( filename, 0L, result, 256 );
+    res = SyFindOrLinkGapRootFile( filename, 0L, result, 256, &info );
 
     /* not found                                                           */
     if ( res == 0 ) {
@@ -335,7 +335,6 @@
             Pr( "#I  READ_GAP_ROOT: loading '%s' dynamically\n",
                 (Int)filename, 0L );
         }
-        info = *(StructInitInfo**)result;
 	res  = info->initKernel(info);
 	if (!SyRestoring) {
 	  UpdateCopyFopyInfo();
@@ -356,7 +355,6 @@
             Pr( "#I  READ_GAP_ROOT: loading '%s' statically\n",
                 (Int)filename, 0L );
         }
-        info = *(StructInitInfo**)result;
 	res  = info->initKernel(info);
 	if (!SyRestoring) {
 	  UpdateCopyFopyInfo();
