diff -druN a/latest/Singular/Makefile.in b/latest/Singular/Makefile.in
--- a/latest/Singular/Makefile.in	2014-08-06 08:59:15.000000000 -0700
+++ b/latest/Singular/Makefile.in	2014-10-21 02:24:21.709753895 -0700
@@ -169,6 +169,13 @@
 LIBSINGULAR_FLAGS = -shared -single_module
 endif
 
+ifeq ($(SINGUNAME),x86_64-Win)
+SO_SUFFIX = dll
+MODULE_SUFFIX    = dll
+LIBSINGULAR_FLAGS = -shared
+LIBSINGULAR_LIBS = -lsingfac -lsingcf -lntl -lreadline @FLINT_LIBS@ -lgmp -lomalloc_ndebug
+endif
+
 ifeq ($(SINGUNAME),ix86-Win)
 SO_SUFFIX = dll
 MODULE_SUFFIX    = dll
diff -druN a/latest/singuname.sh b/latest/singuname.sh
--- a/latest/singuname.sh	2014-08-06 08:59:15.000000000 -0700
+++ b/latest/singuname.sh	2014-10-21 02:24:21.709753895 -0700
@@ -96,7 +96,12 @@
 elif (echo $uname_m | $egrep "x86_64" > $devnull)
 then
     prefix=x86_64
-    if (echo $uname_a | $egrep "Linux" > $devnull)
+    # Win ################
+    if (echo $uname_a | $egrep "CYGWIN" > $devnull)
+    then
+        echo ${prefix}-Win
+        exit 0
+    elif (echo $uname_a | $egrep "Linux" > $devnull)
     then
         echo ${prefix}-Linux
         exit 0
