diff -ru src/IntegerProgramming/Makefile.in b/IntegerProgramming/Makefile.in
--- src/IntegerProgramming/Makefile.in	2012-06-19 11:00:05.000000000 +0200
+++ b/IntegerProgramming/Makefile.in	2012-07-12 17:00:42.253780191 +0200
@@ -64,11 +64,11 @@
 
 install install-libsingular: $(MAIN1) $(MAIN2) $(MAIN3) $(MAIN4) $(LLL)
 	${MKINSTALLDIRS} ${bindir}
-	${INSTALL_PROGRAM} $(MAIN1) ${bindir}
-	${INSTALL_PROGRAM} $(MAIN2) ${bindir}
-	${INSTALL_PROGRAM} $(MAIN3) ${bindir}
-	${INSTALL_PROGRAM} $(MAIN4) ${bindir}
-	${INSTALL_PROGRAM} $(LLL) ${bindir}
+	${INSTALL_PROGRAM} $(MAIN1).exe ${bindir}
+	${INSTALL_PROGRAM} $(MAIN2).exe ${bindir}
+	${INSTALL_PROGRAM} $(MAIN3).exe ${bindir}
+	${INSTALL_PROGRAM} $(MAIN4).exe ${bindir}
+	${INSTALL_PROGRAM} $(LLL).exe ${bindir}
 
 install-bindist: $(MAIN1) $(MAIN2) $(MAIN3) $(MAIN4)
 	${MKINSTALLDIRS} ${install_bindir}
