diff -druN singular-3.1.6.orig/latest/Singular/Makefile.in singular-3.1.6/latest/Singular/Makefile.in
--- latest/Singular/Makefile.in	2012-12-19 13:01:16.000000000 -0800
+++ latest/Singular/Makefile.in	2014-01-18 13:16:43.947415952 -0800
@@ -167,6 +167,13 @@
 LIBSINGULAR_LD = $(LD)
 endif
 
+ifeq ($(SINGUNAME),x86_64-Win)
+SO_SUFFIX = dll
+MODULE_SUFFIX    = dll
+LIBSINGULAR_FLAGS = -shared
+LIBSINGULAR_LIBS = -lsingfac -lsingcf -lntl -lreadline -lgmp -lomalloc_ndebug
+endif
+
 ifeq ($(SINGUNAME),ix86-Win)
 SO_SUFFIX = dll
 MODULE_SUFFIX    = dll
diff -druN singular-3.1.6.orig/latest/singuname.sh singular-3.1.6/latest/singuname.sh
--- latest/singuname.sh	2012-12-19 13:01:16.000000000 -0800
+++ latest/singuname.sh	2014-01-18 13:15:48.907410448 -0800
@@ -96,7 +96,12 @@
 elif (echo $uname_m | $egrep "x86_64" > $devnull)
 then
     prefix=x86_64
-    if (echo $uname_a | $egrep "Linux" > $devnull)
+    # Win ################
+    if (echo $uname_a | $egrep "CYGWIN" > $devnull)
+    then
+        echo ${prefix}-Win
+        exit 0
+    elif (echo $uname_a | $egrep "Linux" > $devnull)
     then
         echo ${prefix}-Linux
         exit 0
