diff -ru src/Singular/Makefile.in src.debug/Singular/Makefile.in
--- src/Singular/Makefile.in	2012-07-11 12:00:13.000000000 +0200
+++ src.debug/Singular/Makefile.in	2012-07-12 16:51:29.923359810 +0200
@@ -82,10 +82,10 @@
 CPPFLAGS	= -I${srcdir} -I.. -I@prefix@ @CPPFLAGS@ @PYTHON_CPPFLAGS@
 ## -I/usr/include/python2.4 -I/usr/local/include -I../modules/python
 endif
-DEFS		= -DNDEBUG -DOM_NDEBUG -D@SING_UNAME@ @DEFS@
+DEFS		= -DOM_NDEBUG -D@SING_UNAME@ @DEFS@
 LDFLAGS2	= @LDFLAGS@
 ifneq ($(SINGUNAME),ix86-Win)
-LDFLAGS		= -L@prefix@/kernel -L../kernel -lkernel @LDFLAGS@ 
+LDFLAGS		= -L@prefix@/kernel -L../kernel -lkernel_g @LDFLAGS@ 
 LDFLAGSG	= -L@prefix@/kernel -L../kernel -lkernel_g @LD_DYN_FLAGS1@ @LDFLAGS@ 
 LDFLAGSP	= -L@prefix@/kernel -L../kernel -lkernel_p @LD_DYN_FLAGS1@ @LDFLAGS@ 
 else
diff -ru src/kernel/Makefile.in src.debug/kernel/Makefile.in
--- src/kernel/Makefile.in	2012-04-17 21:00:08.000000000 +0200
+++ src.debug/kernel/Makefile.in	2012-07-12 16:51:29.923359810 +0200
@@ -49,7 +49,7 @@
 CXXFLAGS	= @CXXFLAGS@ ${PIPE} 
 CXXTEMPLFLAGS	= @CXXTEMPLFLAGS@
 CPPFLAGS	= -I${srcdir} -I.. -I@prefix@  @CPPFLAGS@ 
-DEFS		= -DNDEBUG -DOM_NDEBUG -D@SING_UNAME@ @DEFS@
+DEFS		= -DOM_NDEBUG -D@SING_UNAME@ @DEFS@
 LDFLAGS		= @LDFLAGS@
 LD_DYN_FLAGS	= @LD_DYN_FLAGS@
 SFLAGS		= @SFLAGS@
@@ -64,7 +64,7 @@
 LIBS		= -lm @NEED_LIBS@ 
 else
 # for the 2-0-* versions under Windows, we don't need gdbm, readline and ncurses
-LIBS		= -lsingfac -lsingcf -lntl -lgmp -lreadline -lncurses -lomalloc_ndebug
+LIBS		= -lsingfac -lsingcf -lntl -lgmp -lreadline -lncurses -lomalloc
 #LIBS		= -lsingfac -lsingcf -lgmp
 endif
 MP_LIBS		= @MP_LIBS@
