diff -druN a/latest/kernel/Makefile.in b/latest/kernel/Makefile.in
--- a/latest/kernel/Makefile.in	2014-08-06 08:59:15.000000000 -0700
+++ b/latest/kernel/Makefile.in	2014-10-21 02:28:30.420231800 -0700
@@ -49,7 +49,7 @@
 CXXFLAGS	= @CXXFLAGS@ ${PIPE} 
 CXXTEMPLFLAGS	= @CXXTEMPLFLAGS@
 CPPFLAGS	= -I${srcdir} -I.. -I@prefix@  @CPPFLAGS@ 
-DEFS		= -DNDEBUG -DOM_NDEBUG -D@SING_UNAME@ @DEFS@
+DEFS		= -D@SING_UNAME@ @DEFS@
 LDFLAGS		= @LDFLAGS@
 LD_DYN_FLAGS	= @LD_DYN_FLAGS@
 SFLAGS		= @SFLAGS@
diff -druN a/latest/Singular/configure b/latest/Singular/configure
--- a/latest/Singular/configure	2014-10-21 02:27:51.270156368 -0700
+++ b/latest/Singular/configure	2014-10-21 02:33:39.470848773 -0700
@@ -10006,11 +10006,7 @@
 
 NEED_LIBSG="${NEED_LIBS}"
 
-if test "$ac_cv_sizeof_voidp" != 4; then
-  NEED_LIBS="${NEED_LIBS} -lomalloc"
-else
-  NEED_LIBS="${NEED_LIBS} -lomalloc_ndebug"
-fi
+NEED_LIBS="${NEED_LIBS} -lomalloc_ndebug"
 
 
 
diff -druN a/latest/Singular/Makefile.in b/latest/Singular/Makefile.in
--- a/latest/Singular/Makefile.in	2014-10-21 02:26:19.439979752 -0700
+++ b/latest/Singular/Makefile.in	2014-10-21 02:31:57.110639474 -0700
@@ -93,10 +93,10 @@
 CPPFLAGS	= -I${srcdir} -I.. -I@prefix@ @CPPFLAGS@ @PYTHON_CPPFLAGS@ @FLINT_CFLAGS@
 ## -I/usr/include/python2.4 -I/usr/local/include -I../modules/python
 endif
-DEFS		= -DNDEBUG -DOM_NDEBUG -D@SING_UNAME@ @DEFS@
+DEFS		= -D@SING_UNAME@ @DEFS@
 LDFLAGS2	= @LDFLAGS@
 ifneq ($(SINGUNAME),ix86-Win)
-LDFLAGS		= -L@prefix@/kernel -L../kernel -lkernel @LDFLAGS@ 
+LDFLAGS		= -L@prefix@/kernel -L../kernel -lkernel_g @LDFLAGS@ 
 LDFLAGSG	= -L@prefix@/kernel -L../kernel -lkernel_g @LD_DYN_FLAGS1@ @LDFLAGS@ 
 LDFLAGSP	= -L@prefix@/kernel -L../kernel -lkernel_p @LD_DYN_FLAGS1@ @LDFLAGS@ 
 else
@@ -140,7 +140,7 @@
 #
 # Handle libSINGULAR stuff
 #
-LIBSINGULAR_LIBS =-lsingfac -lsingcf -lntl -lreadline @FLINT_LIBS@ -lgmp -lomalloc
+LIBSINGULAR_LIBS =-lsingfac -lsingcf -lntl -lreadline @FLINT_LIBS@ -lgmp -lomalloc_ndebug
 
 SO_SUFFIX = so
 
@@ -180,14 +180,14 @@
 SO_SUFFIX = dll
 MODULE_SUFFIX    = dll
 LIBSINGULAR_FLAGS = -shared
-LIBSINGULAR_LIBS = -lsingfac -lsingcf -lntl -lreadline @FLINT_LIBS@ -lgmp -lomalloc  -lhtmlhelp
+LIBSINGULAR_LIBS = -lsingfac -lsingcf -lntl -lreadline @FLINT_LIBS@ -lgmp -lomalloc_ndebug  -lhtmlhelp
 endif
 
 ifeq ($(SINGUNAME),ix86-SunOS)
 SO_SUFFIX  = so
 MODULE_SUFFIX    = so
 LIBSINGULAR_FLAGS = -shared
-LIBSINGULAR_LIBS = -lsingfac -lsingcf -lntl -lreadline @FLINT_LIBS@ -lgmp -lomalloc
+LIBSINGULAR_LIBS = -lsingfac -lsingcf -lntl -lreadline @FLINT_LIBS@ -lgmp -lomalloc_ndebug
 endif
 
 #
@@ -452,7 +452,7 @@
 	echo "#define CXX \"" ${CXX} ${CXXFLAGS} ${DEFS} "\"" >>version.h
 
 libparse: libparse_main.o utils.o
-	${CXX}  ${CPPFLAGS} -DSTANDALONE_PARSER -o libparse libparse_main.o utils.o ../kernel/fegetopt.o ${LDFLAGS2} -lomalloc
+	${CXX}  ${CPPFLAGS} -DSTANDALONE_PARSER -o libparse libparse_main.o utils.o ../kernel/fegetopt.o ${LDFLAGS2} -lomalloc_ndebug
 
 libparse_main.o: libparse.cc
 	${CXX} ${CPPFLAGS} -c -o libparse_main.o libparse.cc  -DSTANDALONE_PARSER
@@ -784,7 +784,7 @@
 Singularg${S_EXT}: scanner.cc  $(OBJG)  $(DL_LIBSG)\
            iparith.og mpsr_Tok.og claptmpl.og tesths.cc version.h 
 	$(CXXG) ${CXXFLAGSG} ${CXXTEMPLFLAGSG} ${CPPFLAGS} ${DEFSG} -o Singularg${S_EXT} \
-	tesths.cc iparith.og mpsr_Tok.og claptmpl.og $(OBJG) ${LDFLAGSG} ${LIBSG} -lomalloc ../kernel/mmalloc.og \
+	tesths.cc iparith.og mpsr_Tok.og claptmpl.og $(OBJG) ${LDFLAGSG} ${LIBSG} -lomalloc_ndebug ../kernel/mmalloc.og \
 	${LD_DYN_FLAGS2} ${LD_LIBC}
 
 iparith.og: iparith.inc iparith.cc
@@ -799,7 +799,7 @@
 
 ESingularg: mod2.h feOpt.cc version.h emacs.cc \
            feOptES.inc
-	${CXXG} ${CXXFLAGSG} ${CPPFLAGS} -DESINGULAR -g ${DEFSG} -o ESingularg emacs.cc ../kernel/fegetopt.og ${LDFLAGS} ${LIBS} -lomalloc ../kernel/mmalloc.og
+	${CXXG} ${CXXFLAGSG} ${CPPFLAGS} -DESINGULAR -g ${DEFSG} -o ESingularg emacs.cc ../kernel/fegetopt.og ${LDFLAGS} ${LIBS} -lomalloc_ndebug ../kernel/mmalloc.og
 
 ##
 ## .op files for having -O and -g pg object versions available at the same time
@@ -864,7 +864,7 @@
            iparith.op mpsr_Tok.op claptmpl.op tesths.cc version.h 
 	(cd ../kernel;make libkernel_p.a;make install;cd ../Singular)
 	$(CXXP) ${CXXFLAGSP} ${CXXTEMPLFLAGSP} ${CPPFLAGS} ${DEFSP} -o Singularp${S_EXT} \
-	tesths.cc iparith.op mpsr_Tok.op claptmpl.op $(OBJP) ${LDFLAGSP} ${LIBSP} -lomalloc ../kernel/mmalloc.op \
+	tesths.cc iparith.op mpsr_Tok.op claptmpl.op $(OBJP) ${LDFLAGSP} ${LIBSP} -lomalloc_ndebug ../kernel/mmalloc.op \
 	${LD_DYN_FLAGS2}
 
 iparith.op: iparith.inc iparith.cc
