diff --git a/sympy/core/function.py b/sympy/core/function.py
index 17f7d40..b62da2b 100644
--- a/sympy/core/function.py
+++ b/sympy/core/function.py
@@ -723,13 +716,6 @@ def __new__(cls, *args, **options):
     def _eval_as_leading_term(self, x):
         return self
 
-    def _sage_(self):
-        import sage.all as sage
-        fname = str(self.func)
-        args = [arg._sage_() for arg in self.args]
-        func = sage.function(fname)(*args)
-        return func
-
 class UndefinedFunction(FunctionClass):
     """
     The (meta)class of undefined functions.
