diff --git a/sympy/core/numbers.py b/sympy/core/numbers.py
--- a/sympy/core/numbers.py
+++ b/sympy/core/numbers.py
@@ -2857,6 +2857,11 @@ def _eval_power(self, expt):
                     return S.ComplexInfinity
                 else:
                     return S.Zero
+    
+    def _sage_(self):
+        import sage.all as sage
+        return sage.UnsignedInfinityRing.gen()
+
 
 zoo = S.ComplexInfinity
 

