--- a/sympy/functions/combinatorial/factorials.py~        2014-02-22 20:13:32.000000000 +0100
+++ a/sympy/functions/combinatorial/factorials.py 2014-12-14 16:15:31.423994987 +0100
@@ -339,6 +339,10 @@
     def _eval_rewrite_as_gamma(self, x, k):
         return C.gamma(x + k) / C.gamma(x)
 
+    def _sage_(self):
+        import sage.all as sage
+        return sage.rising_factorial(self.args[0]._sage_(), self.args[1]._sage_())
+
 
 class FallingFactorial(CombinatorialFunction):
     """Falling factorial (related to rising factorial) is a double valued
@@ -403,6 +407,10 @@
     def _eval_rewrite_as_gamma(self, x, k):
         return (-1)**k * C.gamma(-x + k) / C.gamma(-x)
 
+    def _sage_(self):
+        import sage.all as sage
+        return sage.falling_factorial(self.args[0]._sage_(), self.args[1]._sage_())
+
 rf = RisingFactorial
 ff = FallingFactorial
