--- a/pyproject.toml	2021-12-01 10:23:30.000000000 -0300
+++ b/pyproject.toml	2021-12-13 18:01:36.239921204 -0300
@@ -3,7 +3,6 @@
 requires=[
   "setuptools",
   "wheel",
-  "debugpy",
   "ipython>=5",
   "jupyter_core>=4.2",
   "jupyter_client",
diff -ruN a/setup.py b/setup.py
--- a/setup.py	2021-12-01 10:23:09.000000000 -0300
+++ b/setup.py	2021-12-13 18:01:40.112873823 -0300
@@ -63,7 +63,6 @@
     install_requires=[
         'importlib-metadata<5;python_version<"3.8.0"',
         'argcomplete>=1.12.3;python_version<"3.8.0"',
-        'debugpy>=1.0.0,<2.0',
         'ipython>=7.23.1',
         'traitlets>=5.1.0,<6.0',
         'jupyter_client<8.0',
