diff --git a/ginac/mpoly-singular.cpp b/ginac/mpoly-singular.cpp
index 4ddb961..74b8815 100644
--- a/ginac/mpoly-singular.cpp
+++ b/ginac/mpoly-singular.cpp
@@ -255,7 +255,7 @@ static ex coeff_to_ex(const CanonicalForm& f, const exvector& revmap)
         if (f.isImm())
                 return numeric(f.intval());
         if (f.inZ())
-                throw std::runtime_error("can't happen in coeff_to_ex");
+                return can2num(f);
         if (f.inQ()) {
                 CanonicalForm num = f.num();
                 CanonicalForm den = f.den();
@@ -277,7 +277,7 @@ static ex canonical_to_ex(const CanonicalForm& f, const exvector& revmap)
                 if (f.isImm())
                         return numeric(f.intval());
                 if (f.inZ())
-                        throw std::runtime_error("can't happen in canonical_to_ex #1");
+                        return can2num(f);
                 if (f.inQ()) {
                         CanonicalForm num = f.num();
                         CanonicalForm den = f.den();
