diff --git a/ginac/numeric.cpp b/ginac/numeric.cpp
index c92dffa..0e1352c 100644
--- a/ginac/numeric.cpp
+++ b/ginac/numeric.cpp
@@ -1969,7 +1969,6 @@ numeric & operator+=(numeric & lh, const numeric & rh)
                         }
                         lh.hash = (long)PyObject_Hash(lh.v._pyobject);
                         Py_DECREF(p);
-                        Py_INCREF(lh.v._pyobject);
                         return lh;
                         }
                 default:
@@ -2049,7 +2048,6 @@ numeric & operator-=(numeric & lh, const numeric & rh)
                         }
                         lh.hash = (long)PyObject_Hash(lh.v._pyobject);
                         Py_DECREF(p);
-                        Py_INCREF(lh.v._pyobject);
                         return lh;
                         }
                 default:
@@ -2131,7 +2129,6 @@ numeric & operator*=(numeric & lh, const numeric & rh)
                         }
                         lh.hash = (long)PyObject_Hash(lh.v._pyobject);
                         Py_DECREF(p);
-                        Py_INCREF(lh.v._pyobject);
                         return lh;
                         }
                 default:
@@ -2265,7 +2262,6 @@ numeric & operator/=(numeric & lh, const numeric & rh)
                                         }
                                         lh.hash = (long)PyObject_Hash(lh.v._pyobject);
                                         Py_DECREF(p);
-                                        Py_INCREF(lh.v._pyobject);
                                         return lh;
                                 } else if (PyLong_Check(rh.v._pyobject)) {
                                         PyObject* d = py_funcs.py_integer_from_python_obj(rh.v._pyobject);
@@ -2277,7 +2273,6 @@ numeric & operator/=(numeric & lh, const numeric & rh)
                                         lh.hash = (long)PyObject_Hash(lh.v._pyobject);
                                         Py_DECREF(d);
                                         Py_DECREF(p);
-                                        Py_INCREF(lh.v._pyobject);
                                         return lh;
                                 }
                         } else if (PyLong_Check(p)) {
@@ -2290,7 +2285,6 @@ numeric & operator/=(numeric & lh, const numeric & rh)
                                 lh.hash = (long)PyObject_Hash(lh.v._pyobject);
                                 Py_DECREF(n);
                                 Py_DECREF(p);
-                                Py_INCREF(lh.v._pyobject);
                                 return lh;
                         }
                         }
@@ -2306,7 +2300,6 @@ numeric & operator/=(numeric & lh, const numeric & rh)
                                 lh.hash = (long)PyObject_Hash(lh.v._pyobject);
                                 Py_DECREF(n);
                                 Py_DECREF(p);
-                                Py_INCREF(lh.v._pyobject);
                                 return lh;
                         }
                         }
@@ -2318,7 +2311,6 @@ numeric & operator/=(numeric & lh, const numeric & rh)
                         }
                         lh.hash = (long)PyObject_Hash(lh.v._pyobject);
                         Py_DECREF(p);
-                        Py_INCREF(lh.v._pyobject);
                         return lh;
                         }
                 default:
