diff --git a/ginac/power.cpp b/ginac/power.cpp
index 74dc725..ca504f1 100644
--- a/ginac/power.cpp
+++ b/ginac/power.cpp
@@ -744,12 +744,15 @@ ex power::eval(int level) const
                                                 outex = outer[0];
                                         else
                                                 outex = mul(outer).hold();
-                                        if (inner.size() == 1)
-                                                innex = inner[0];
-                                        else
-                                                innex = mul(inner).hold();
-                                        ex p = power(innex, eexponent).hold();
-                                        return (new mul(outex, p))->setflag(status_flags::dynallocated | status_flags::evaluated);
+                                        if (inner.size() > 0) {
+                                                if (inner.size() == 1)
+                                                        innex = inner[0];
+                                                else
+                                                        innex = mul(inner).hold();
+                                                ex p = power(innex, eexponent).hold();
+                                                return (new mul(outex, p))->setflag(status_flags::dynallocated | status_flags::evaluated);
+                                        }
+                                        return outex;
                                 }
                         }
                 }
