diff --git a/ginac/numeric.cpp b/ginac/numeric.cpp
index bf43991..058599b 100644
--- a/ginac/numeric.cpp
+++ b/ginac/numeric.cpp
@@ -1536,8 +1536,12 @@ const ex numeric::power(const numeric &exponent) const {
                         return c;
                 else if (d.is_minus_one()
                                 and expo.denom().is_equal(*_num2_p)) {
-                        static numeric a[] = { *_num1_p, I, *_num_1_p, -I };
-                        return a[expo.numer().to_long() % 4] * c;
+                        switch (expo.numer().to_long() % 4) {
+                                case 0: return c;
+                                case 1: return I*c;
+                                case 2: return -c;
+                                case 3: return -I*c;
+                        };
                 }
                 else if (not c_unit)
                         return d.power(expo) * c;
