Index: src/kernel/Makefile.in
===================================================================
--- src.orig/kernel/Makefile.in	2008-02-08 17:15:37.000000000 -0800
+++ src/kernel/Makefile.in	2008-02-08 17:15:53.000000000 -0800
@@ -316,7 +316,9 @@
 
 install-libsingular:
 	${MKINSTALLDIRS} ${includedir}/singular
-	cp *.h ${includedir}/singular
+	-for file in *.h; do \
+	  ${INSTALL} $$file ${includedir}/singular; \
+	done
 
 install: all installbin
 
