--- ../src/IPython/Magic.py	2008-08-14 16:59:40.000000000 -0700
+++ Magic.py	2009-06-17 23:04:21.000000000 -0700
@@ -35,7 +35,6 @@
 from cStringIO import StringIO
 from getopt import getopt,GetoptError
 from pprint import pprint, pformat
-from sets import Set
 
 # cProfile was added in Python2.5
 try:
@@ -75,7 +74,7 @@
     head, tail = dh[:-10], dh[-10:]
 
     newhead = []
-    done = Set()
+    done = set()
     for h in head:
         if h in done:
             continue
@@ -149,7 +148,7 @@
                  filter(inst_magic,self.__dict__.keys()) + \
                  filter(inst_bound_magic,self.__class__.__dict__.keys())
         out = []
-        for fn in Set(magics):
+        for fn in set(magics):
             out.append(fn.replace('magic_','',1))
         out.sort()
         return out
