--- src/IPython/OInspect.py	2011-08-01 03:52:04.000000000 +0100
+++ src/IPython/OInspect.py	2011-12-17 19:12:48.000000000 +0000
@@ -429,7 +429,16 @@
                 fname = 'Dynamically generated function. No source code available.'
             if (fname.endswith('.so') or fname.endswith('.dll')):
                 binary_file = True
-            out.writeln(header('File:\t\t')+fname)
+            out.writeln(header('Loaded File:\t')+fname)
+            #this code is to cope with the use of two copy's of all files
+            import sage
+            import re
+            sageroot = sage.misc.sageinspect.SAGE_ROOT+'/'
+            runpathpattern = '^'+sageroot+'local/lib/python[^/]*/site-packages'
+            develbranch = sageroot+'devel/sage'
+            fname=re.sub(runpathpattern,develbranch,fname)
+            out.writeln(header('Source File:\t')+fname)
+            #end of sage specific code
         except:
             # if anything goes wrong, we don't want to show source, so it's as
             # if the file was binary
