--- src/IPython/rlineimpl.py	2011-07-31 19:52:04.000000000 -0700
+++ src/IPython/rlineimpl.py	2011-08-16 11:15:51.000000000 -0700
@@ -31,26 +31,29 @@
 # Test to see if libedit is being used instead of GNU readline.
 # Thanks to Boyd Waters for this patch.
 uses_libedit = False
-if sys.platform == 'darwin' and have_readline:
-    import commands
-    # Boyd's patch had a 'while True' here, I'm always a little worried about
-    # infinite loops with such code, so for now I'm taking a more conservative
-    # approach. See https://bugs.launchpad.net/ipython/+bug/411599.
-    for i in range(10):
-        try:
-            (status, result) = commands.getstatusoutput( "otool -L %s | grep libedit" % _rl.__file__ )
-            break
-        except IOError, (errno, strerror):
-            if errno == 4:
-                continue
-            else:
-                break
+# Sage under gdb throws an error when running commands.getstatusoutput(), so disable this check 
+# because we do not link against libedit anyway.
 
-    if status == 0 and len(result) > 0:
-        # we are bound to libedit - new in Leopard
-        _rl.parse_and_bind("bind ^I rl_complete")
-        print "Leopard libedit detected."
-        uses_libedit = True
+# if sys.platform == 'darwin' and have_readline:
+#     import commands
+#     # Boyd's patch had a 'while True' here, I'm always a little worried about
+#     # infinite loops with such code, so for now I'm taking a more conservative
+#     # approach. See https://bugs.launchpad.net/ipython/+bug/411599.
+#     for i in range(10):
+#         try:
+#             (status, result) = commands.getstatusoutput( "otool -L %s | grep libedit" % _rl.__file__ )
+#             break
+#         except IOError, (errno, strerror):
+#             if errno == 4:
+#                 continue
+#             else:
+#                 break
+
+#     if status == 0 and len(result) > 0:
+#         # we are bound to libedit - new in Leopard
+#         _rl.parse_and_bind("bind ^I rl_complete")
+#         print "Leopard libedit detected."
+#         uses_libedit = True
 
 
 # the clear_history() function was only introduced in Python 2.4 and is
