--- ../src/IPython/ultraTB.py	2008-08-23 18:25:35.000000000 -0700
+++ ultraTB.py	2009-01-24 02:53:15.000000000 -0800
@@ -279,6 +279,9 @@
             scheme = DEFAULT_SCHEME
     _line_format = _parser.format2
 
+    if all(line.strip() == '' for line in lines):
+        return []
+
     for line in lines:
         new_line, err = _line_format(line,'str',scheme)
         if not err: line = new_line
