Index: src/makefile
===================================================================
--- src.orig/makefile	2008-05-19 18:01:53.000000000 -0400
+++ src/makefile	2008-05-19 18:05:39.000000000 -0400
@@ -89,7 +89,8 @@
 	$(CC) -fPIC -shared -o libflint.dll $(FLINTOBJ) $(LIBS)
 
 libflint.so: $(FLINTOBJ)
-	$(CC) -fPIC -shared -o libflint.so $(FLINTOBJ) $(LIBS)
+	$(CC) -fPIC -shared -Wl,-soname,lib`cat DIRNAME`.so -o lib`cat DIRNAME`.so $(FLINTOBJ) $(LIBS)
+	ln -s lib`cat DIRNAME`.so libflint.so
 
 mpn_extras.o: mpn_extras.c $(HEADERS)
 	$(CC) $(CFLAGS) -c mpn_extras.c -o mpn_extras.o
Index: src/DIRNAME
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ src/DIRNAME	2008-05-19 18:01:58.000000000 -0400
@@ -0,0 +1 @@
+flint-1.06
\ No newline at end of file
