From 6923f95105c0fb71f471f89b7beabd6ccc2b7d52 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matthias=20K=C3=B6ppe?= <mkoeppe@math.ucdavis.edu>
Date: Thu, 12 Aug 2021 10:34:31 -0700
Subject: [PATCH 1/4] requirements.txt: Conditionalize async_generator on
 python_version

---
 requirements.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/requirements.txt b/requirements.txt
index 4f25058..fe1554d 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,5 +1,5 @@
 traitlets>=4.2
 jupyter_client>=6.1.5
 nbformat>=5.0
-async_generator
+async_generator; python_version<'3.7'
 nest_asyncio

From 45a9cfe70d3d6e8e9ad559d598a90c1615a67c92 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matthias=20K=C3=B6ppe?= <mkoeppe@math.ucdavis.edu>
Date: Thu, 12 Aug 2021 12:20:21 -0700
Subject: [PATCH 2/4] client.py: Import asynccontextmanager from contextlib on
 Python >= 3.7

---
 nbclient/client.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/nbclient/client.py b/nbclient/client.py
index 40ec273..c57ddd9 100644
--- a/nbclient/client.py
+++ b/nbclient/client.py
@@ -5,7 +5,12 @@
 import signal
 from textwrap import dedent
 
-from async_generator import asynccontextmanager
+try:
+    from contextlib import asynccontextmanager
+except ImportError:
+    # Use the backport package async-generator for Python < 3.7.
+    # This should be removed when nbclient drops support for Python 3.6
+    from async_generator import asynccontextmanager
 from contextlib import contextmanager
 
 from time import monotonic

From 53095d6aaba7fcd5492f234a503c689e6b05bdb1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matthias=20K=C3=B6ppe?= <mkoeppe@math.ucdavis.edu>
Date: Thu, 12 Aug 2021 12:31:06 -0700
Subject: [PATCH 3/4] mypy.ini: Set python_version = 3.9 to match tox.ini

---
 mypy.ini | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mypy.ini b/mypy.ini
index d5ebf80..c218c2d 100644
--- a/mypy.ini
+++ b/mypy.ini
@@ -1,7 +1,7 @@
 # Global options:
 
 [mypy]
-python_version = 3.6
+python_version = 3.9
 
 # Per-module options:
 

From 2dbbbc7f31b37b6c01a86939652fb05e01684399 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matthias=20K=C3=B6ppe?= <mkoeppe@math.ucdavis.edu>
Date: Thu, 12 Aug 2021 23:35:50 -0700
Subject: [PATCH 4/4] nbclient/client.py: Suppress a mypy error

Co-authored-by: David Brochart <david.brochart@gmail.com>
---
 nbclient/client.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nbclient/client.py b/nbclient/client.py
index c57ddd9..9910308 100644
--- a/nbclient/client.py
+++ b/nbclient/client.py
@@ -10,7 +10,7 @@
 except ImportError:
     # Use the backport package async-generator for Python < 3.7.
     # This should be removed when nbclient drops support for Python 3.6
-    from async_generator import asynccontextmanager
+    from async_generator import asynccontextmanager  # type: ignore
 from contextlib import contextmanager
 
 from time import monotonic
