diff -druN src/examples/Makefile.am src.patched/examples/Makefile.am
--- src/examples/Makefile.am	2014-08-21 23:00:00.000000000 -0700
+++ src.patched/examples/Makefile.am	2014-11-03 07:11:27.621234167 -0800
@@ -2,7 +2,7 @@
 
 AM_CPPFLAGS = -I$(srcdir)/../src
 
-LDADD = ../src/libglpk.la
+LDADD = ../src/libglpk.la -lz
 
 bin_PROGRAMS = glpsol
 
diff -druN src/src/Makefile.am src.patched/src/Makefile.am
--- src/src/Makefile.am	2014-11-03 07:11:04.331186723 -0800
+++ src.patched/src/Makefile.am	2014-11-03 07:11:27.601234126 -0800
@@ -13,8 +13,9 @@
 -I$(srcdir)/env \
 -I$(srcdir)/minisat \
 -I$(srcdir)/misc \
--I$(srcdir)/proxy \
--I$(srcdir)/zlib
+-I$(srcdir)/proxy
+
+libglpk_la_LIBADD = -lz
 
 libglpk_la_LDFLAGS = \
 -version-info 37:2:1 \
@@ -151,21 +152,6 @@
 misc/wclique.c \
 misc/wclique1.c \
 proxy/proxy.c \
-proxy/proxy1.c \
-zlib/adler32.c \
-zlib/compress.c \
-zlib/crc32.c \
-zlib/deflate.c \
-zlib/gzclose.c \
-zlib/gzlib.c \
-zlib/gzread.c \
-zlib/gzwrite.c \
-zlib/inffast.c \
-zlib/inflate.c \
-zlib/inftrees.c \
-zlib/trees.c \
-zlib/uncompr.c \
-zlib/zio.c \
-zlib/zutil.c
+proxy/proxy1.c
 
 ## eof ##
