diff -r -u ecm-6.4.4-orig/x86_64/mulredc10.asm ecm-6.4.4/x86_64/mulredc10.asm
--- ecm-6.4.4-orig/x86_64/mulredc10.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc10.asm	2015-09-23 21:02:14.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 bytes long
 	GLOBL GSYM_PREFIX`'mulredc10
 	TYPE(GSYM_PREFIX`'mulredc`'10,`function')
 
@@ -535,7 +535,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc11.asm ecm-6.4.4/x86_64/mulredc11.asm
--- ecm-6.4.4-orig/x86_64/mulredc11.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc11.asm	2015-09-23 21:05:49.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 bytes long
 	GLOBL GSYM_PREFIX`'mulredc11
 	TYPE(GSYM_PREFIX`'mulredc`'11,`function')
 
@@ -579,7 +579,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc12.asm ecm-6.4.4/x86_64/mulredc12.asm
--- ecm-6.4.4-orig/x86_64/mulredc12.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc12.asm	2015-09-23 21:05:55.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 bytes long
 	GLOBL GSYM_PREFIX`'mulredc12
 	TYPE(GSYM_PREFIX`'mulredc`'12,`function')
 
@@ -623,7 +623,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc13.asm ecm-6.4.4/x86_64/mulredc13.asm
--- ecm-6.4.4-orig/x86_64/mulredc13.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc13.asm	2015-09-23 21:05:57.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 bytes long
 	GLOBL GSYM_PREFIX`'mulredc13
 	TYPE(GSYM_PREFIX`'mulredc`'13,`function')
 
@@ -667,7 +667,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc14.asm ecm-6.4.4/x86_64/mulredc14.asm
--- ecm-6.4.4-orig/x86_64/mulredc14.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc14.asm	2015-09-23 21:05:59.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 bytes long
 	GLOBL GSYM_PREFIX`'mulredc14
 	TYPE(GSYM_PREFIX`'mulredc`'14,`function')
 
@@ -711,7 +711,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc15.asm ecm-6.4.4/x86_64/mulredc15.asm
--- ecm-6.4.4-orig/x86_64/mulredc15.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc15.asm	2015-09-23 21:06:02.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 bytes long
 	GLOBL GSYM_PREFIX`'mulredc15
 	TYPE(GSYM_PREFIX`'mulredc`'15,`function')
 
@@ -755,7 +755,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc16.asm ecm-6.4.4/x86_64/mulredc16.asm
--- ecm-6.4.4-orig/x86_64/mulredc16.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc16.asm	2015-09-23 21:06:03.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 bytes long
 	GLOBL GSYM_PREFIX`'mulredc16
 	TYPE(GSYM_PREFIX`'mulredc`'16,`function')
 
@@ -799,7 +799,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc17.asm ecm-6.4.4/x86_64/mulredc17.asm
--- ecm-6.4.4-orig/x86_64/mulredc17.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc17.asm	2015-09-23 21:06:05.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 bytes long
 	GLOBL GSYM_PREFIX`'mulredc17
 	TYPE(GSYM_PREFIX`'mulredc`'17,`function')
 
@@ -843,7 +843,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc18.asm ecm-6.4.4/x86_64/mulredc18.asm
--- ecm-6.4.4-orig/x86_64/mulredc18.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc18.asm	2015-09-23 21:06:07.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 bytes long
 	GLOBL GSYM_PREFIX`'mulredc18
 	TYPE(GSYM_PREFIX`'mulredc`'18,`function')
 
@@ -887,7 +887,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc19.asm ecm-6.4.4/x86_64/mulredc19.asm
--- ecm-6.4.4-orig/x86_64/mulredc19.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc19.asm	2015-09-23 21:06:10.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 bytes long
 	GLOBL GSYM_PREFIX`'mulredc19
 	TYPE(GSYM_PREFIX`'mulredc`'19,`function')
 
@@ -931,7 +931,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_10.asm ecm-6.4.4/x86_64/mulredc1_10.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_10.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_10.asm	2015-09-23 20:52:08.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 bytes long
 	GLOBL GSYM_PREFIX`'mulredc1_10
 	TYPE(GSYM_PREFIX`'mulredc1_`'10,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_11.asm ecm-6.4.4/x86_64/mulredc1_11.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_11.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_11.asm	2015-09-23 21:03:24.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_11
 	TYPE(GSYM_PREFIX`'mulredc1_`'11,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_12.asm ecm-6.4.4/x86_64/mulredc1_12.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_12.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_12.asm	2015-09-23 21:03:25.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_12
 	TYPE(GSYM_PREFIX`'mulredc1_`'12,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_13.asm ecm-6.4.4/x86_64/mulredc1_13.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_13.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_13.asm	2015-09-23 21:03:25.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_13
 	TYPE(GSYM_PREFIX`'mulredc1_`'13,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_14.asm ecm-6.4.4/x86_64/mulredc1_14.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_14.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_14.asm	2015-09-23 21:03:25.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_14
 	TYPE(GSYM_PREFIX`'mulredc1_`'14,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_15.asm ecm-6.4.4/x86_64/mulredc1_15.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_15.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_15.asm	2015-09-23 21:03:25.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_15
 	TYPE(GSYM_PREFIX`'mulredc1_`'15,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_16.asm ecm-6.4.4/x86_64/mulredc1_16.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_16.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_16.asm	2015-09-23 21:03:26.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_16
 	TYPE(GSYM_PREFIX`'mulredc1_`'16,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_17.asm ecm-6.4.4/x86_64/mulredc1_17.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_17.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_17.asm	2015-09-23 21:03:26.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_17
 	TYPE(GSYM_PREFIX`'mulredc1_`'17,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_18.asm ecm-6.4.4/x86_64/mulredc1_18.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_18.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_18.asm	2015-09-23 21:03:26.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_18
 	TYPE(GSYM_PREFIX`'mulredc1_`'18,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_19.asm ecm-6.4.4/x86_64/mulredc1_19.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_19.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_19.asm	2015-09-23 21:03:26.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_19
 	TYPE(GSYM_PREFIX`'mulredc1_`'19,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_2.asm ecm-6.4.4/x86_64/mulredc1_2.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_2.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_2.asm	2015-09-23 21:03:26.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_2
 	TYPE(GSYM_PREFIX`'mulredc1_`'2,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_20.asm ecm-6.4.4/x86_64/mulredc1_20.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_20.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_20.asm	2015-09-23 21:03:27.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_20
 	TYPE(GSYM_PREFIX`'mulredc1_`'20,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_3.asm ecm-6.4.4/x86_64/mulredc1_3.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_3.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_3.asm	2015-09-23 21:03:27.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_3
 	TYPE(GSYM_PREFIX`'mulredc1_`'3,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_4.asm ecm-6.4.4/x86_64/mulredc1_4.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_4.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_4.asm	2015-09-23 21:03:28.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_4
 	TYPE(GSYM_PREFIX`'mulredc1_`'4,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_5.asm ecm-6.4.4/x86_64/mulredc1_5.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_5.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_5.asm	2015-09-23 21:03:28.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_5
 	TYPE(GSYM_PREFIX`'mulredc1_`'5,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_6.asm ecm-6.4.4/x86_64/mulredc1_6.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_6.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_6.asm	2015-09-23 21:03:28.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_6
 	TYPE(GSYM_PREFIX`'mulredc1_`'6,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_7.asm ecm-6.4.4/x86_64/mulredc1_7.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_7.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_7.asm	2015-09-23 21:03:28.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_7
 	TYPE(GSYM_PREFIX`'mulredc1_`'7,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_8.asm ecm-6.4.4/x86_64/mulredc1_8.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_8.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_8.asm	2015-09-23 21:03:28.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_8
 	TYPE(GSYM_PREFIX`'mulredc1_`'8,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc1_9.asm ecm-6.4.4/x86_64/mulredc1_9.asm
--- ecm-6.4.4-orig/x86_64/mulredc1_9.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc1_9.asm	2015-09-23 21:03:29.000000000 -0700
@@ -18,7 +18,7 @@
 define(`INVM_PARAM',`%r8')dnl'
 )dnl
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc1_9
 	TYPE(GSYM_PREFIX`'mulredc1_`'9,`function')
 
diff -r -u ecm-6.4.4-orig/x86_64/mulredc2.asm ecm-6.4.4/x86_64/mulredc2.asm
--- ecm-6.4.4-orig/x86_64/mulredc2.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc2.asm	2015-09-23 21:04:57.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc2
 	TYPE(GSYM_PREFIX`'mulredc`'2,`function')
 
@@ -183,7 +183,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc20.asm ecm-6.4.4/x86_64/mulredc20.asm
--- ecm-6.4.4-orig/x86_64/mulredc20.asm	2013-02-27 07:17:53.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc20.asm	2015-09-23 21:05:17.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc20
 	TYPE(GSYM_PREFIX`'mulredc`'20,`function')
 
@@ -975,7 +975,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc3.asm ecm-6.4.4/x86_64/mulredc3.asm
--- ecm-6.4.4-orig/x86_64/mulredc3.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc3.asm	2015-09-23 21:06:14.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc3
 	TYPE(GSYM_PREFIX`'mulredc`'3,`function')
 
@@ -227,7 +227,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc4.asm ecm-6.4.4/x86_64/mulredc4.asm
--- ecm-6.4.4-orig/x86_64/mulredc4.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc4.asm	2015-09-23 21:06:18.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc4
 	TYPE(GSYM_PREFIX`'mulredc`'4,`function')
 
@@ -271,7 +271,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc5.asm ecm-6.4.4/x86_64/mulredc5.asm
--- ecm-6.4.4-orig/x86_64/mulredc5.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc5.asm	2015-09-23 21:06:23.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc5
 	TYPE(GSYM_PREFIX`'mulredc`'5,`function')
 
@@ -315,7 +315,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc6.asm ecm-6.4.4/x86_64/mulredc6.asm
--- ecm-6.4.4-orig/x86_64/mulredc6.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc6.asm	2015-09-23 21:06:27.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc6
 	TYPE(GSYM_PREFIX`'mulredc`'6,`function')
 
@@ -359,7 +359,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc7.asm ecm-6.4.4/x86_64/mulredc7.asm
--- ecm-6.4.4-orig/x86_64/mulredc7.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc7.asm	2015-09-23 21:06:31.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc7
 	TYPE(GSYM_PREFIX`'mulredc`'7,`function')
 
@@ -403,7 +403,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc8.asm ecm-6.4.4/x86_64/mulredc8.asm
--- ecm-6.4.4-orig/x86_64/mulredc8.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc8.asm	2015-09-23 21:06:34.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc8
 	TYPE(GSYM_PREFIX`'mulredc`'8,`function')
 
@@ -447,7 +447,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
diff -r -u ecm-6.4.4-orig/x86_64/mulredc9.asm ecm-6.4.4/x86_64/mulredc9.asm
--- ecm-6.4.4-orig/x86_64/mulredc9.asm	2013-02-27 07:17:52.000000000 -0800
+++ ecm-6.4.4/x86_64/mulredc9.asm	2015-09-23 21:06:38.000000000 -0700
@@ -23,7 +23,7 @@
 include(`config.m4')
 
 	TEXT
-.align 64 # Opteron L1 code cache line is 64 bytes long
+.p2align 6 # Opteron L1 code cache line is 64 5,,4
 	GLOBL GSYM_PREFIX`'mulredc9
 	TYPE(GSYM_PREFIX`'mulredc`'9,`function')
 
@@ -491,7 +491,7 @@
 # i > 0 passes
 #########################################################################
 
-.align 32,,16
+.p2align 5,,4
 LABEL_SUFFIX(1)
 
 # register values at loop entry: %TP = tmp, %I = i, %YP = y, %MP = m
