--- src/test/nuss-test.c	2008-09-19 17:37:47.000000000 +0200
+++ src/test/nuss-test.c	2013-05-24 08:53:44.133597843 +0200
@@ -55,7 +55,7 @@
    // compare target implementation against reference implementation
    ref_zn_array_negamul (ref, buf1, buf2, n, mod);
    nuss_mul (res, buf1, buf2, vec1, vec2);
-   ulong x = nuss_mul_fudge (lgL, 0, mod);
+   ulong x = nuss_mul_fudge (lgL, sqr, mod);
    ref_zn_array_scalar_mul (res, res, n, x, mod);
    int success = !zn_array_cmp (ref, res, n);
    
