--- makemakefile.py.orig	2008-08-19 17:13:51.000000000 -0700
+++ makemakefile.py	2008-08-19 17:14:37.000000000 -0700
@@ -167,6 +167,9 @@
 print "libzn_poly.dylib: $(LIBOBJS)"
 print "\t$(CC) -single_module -fPIC -dynamiclib -o libzn_poly.dylib $(LIBOBJS) $(LIBS)"
 print
+print "libzn_poly.dylib64: $(LIBOBJS)"
+print "\t$(CC) -m64 -single_module -fPIC -dynamiclib -o libzn_poly.dylib $(LIBOBJS) $(LIBS)"
+print
 print "libzn_poly.so: $(LIBOBJS)"
 print "\t$(CC) -shared -o libzn_poly.so $(LIBOBJS) $(LIBS)"
 
