--- src/tune/mpn_mulmid-tune.c	2010-04-26 23:47:46.000000000 +0200
+++ patches/mpn_mulmid-tune.c	2010-04-26 23:52:16.000000000 +0200
@@ -46,6 +46,9 @@
    ZNP_mpn_smp_kara_thresh = SIZE_MAX;
 
    size_t thresh;
+   const int max_intervals = 200;
+   size_t points[max_intervals + 1];
+   double score[max_intervals + 1];
    double result[2];
    profile_info_t info;
 
@@ -75,9 +78,6 @@
    // subdivide [2, upper] into intervals and sample at each endpoint
    double lower = 2.0;
    double ratio = (double) upper / lower;
-   const int max_intervals = 200;
-   size_t points[max_intervals + 1];
-   double score[max_intervals + 1];
    unsigned i;
    for (i = 0; i <= max_intervals; i++)
    {
@@ -135,6 +135,9 @@
    const double speed = 0.0001;
 
    size_t thresh;
+   const int max_intervals = 30;
+   size_t points[max_intervals + 1];
+   double score[max_intervals + 1];
    double result[2];
    profile_info_t info;
 
@@ -166,9 +169,6 @@
    // each endpoint
    double lower = ZNP_mpn_smp_kara_thresh;
    double ratio = (double) upper / lower;
-   const int max_intervals = 30;
-   size_t points[max_intervals + 1];
-   double score[max_intervals + 1];
    unsigned i;
    for (i = 0; i <= max_intervals; i++)
    {
