--- src/test/mpn_mulmid-test.c.orig	2013-02-13 13:52:42.942882000 +1300
+++ src/test/mpn_mulmid-test.c	2013-02-14 21:22:17.106157000 +1300
@@ -121,7 +121,10 @@
    // now a few larger problems too
    for (trial = 0; trial < (quick ? 100 : 3000) && success; trial++)
    {
-      n = random_ulong (3 * ZNP_mpn_smp_kara_thresh) + 2;
+      if (ZNP_mpn_smp_kara_thresh == SIZE_MAX)
+         n = random_ulong (100) + 2;
+      else
+         n = random_ulong (3 * ZNP_mpn_smp_kara_thresh) + 2; 
       success = success && testcase_mpn_smp_kara (n);
    }
 
