--- src/profile/profiler.c	2008-09-19 17:37:47.000000000 +0200
+++ patches/profiler.c	2012-04-20 03:44:42.854102915 +0200
@@ -20,13 +20,13 @@
 
 */
 
+/* Include system headers *before* 'ulong' gets (re)defined: */ 
 #include <math.h>
-#include "profiler.h"
-
-
 #include <sys/time.h>
 #include <sys/resource.h>
 
+#include "profiler.h"	/* Includes zn_poly.h, which defines 'ulong'. */
+
 
 double cycle_scale_factor;
 
