diff --git a/makemakefile.py b/makemakefile.py
index acb6084..9c0fb96 100644
--- a/makemakefile.py
+++ b/makemakefile.py
@@ -151,7 +151,7 @@ now = time.strftime("%a, %d %b %Y %H:%M:%S +0000", time.gmtime())
 print(
 """#
 # Do not edit directly -- this file was auto-generated
-# by {} on {}
+# by {0} on {1}
 #
 # (makemakefile.py patched for Sage, 04/2012)
 """.format(__file__, now))
@@ -161,11 +161,11 @@ print(
 """
 CC ?= gcc
 CPP ?= cpp
-CFLAGS = {}
-CPPFLAGS = {}
-LDFLAGS = {}
-INCLUDES = {}  # These are options to the C preprocessor.
-LIBS = {}  # These are linker options passed to the compiler.
+CFLAGS = {0}
+CPPFLAGS = {1}
+LDFLAGS = {2}
+INCLUDES = {3}  # These are options to the C preprocessor.
+LIBS = {4}  # These are linker options passed to the compiler.
 
 AR ?= ar
 RANLIB ?= ranlib
@@ -177,19 +177,19 @@ SONAME_FLAG ?= -soname  # '-h' for the Sun/Solaris linker
 
 print(
 """CXX ?= g++  # The C++ compiler.
-CXXFLAGS = {}  # Options passed to the C++ compiler.
-CPP_INCLUDES = {}
-CPP_LIBS = {}
+CXXFLAGS = {0}  # Options passed to the C++ compiler.
+CPP_INCLUDES = {1}
+CPP_LIBS = {2}
 """.format(cxxflags, cpp_includes, cpp_libs))
 
 
 print(
-"""HEADERS = {}
-LIBOBJS = {}
-TESTOBJS = {}
-PROFOBJS = {}
-CPP_PROFOBJS = {}
-TUNEOBJS = {}""".format(
+"""HEADERS = {0}
+LIBOBJS = {1}
+TESTOBJS = {2}
+PROFOBJS = {3}
+CPP_PROFOBJS = {4}
+TUNEOBJS = {5}""".format(
     " ".join(install_headers + other_headers),
     " ".join([x + ".o" for x in lib_modules]),
     " ".join([x + "-DEBUG.o" for x in
@@ -205,9 +205,9 @@ TUNEOBJS = {}""".format(
 
 
 print(
-"""ZN_POLY_TUNING = {}
-ZN_POLY_VERSION = {}
-ZN_POLY_ABI_VERSION = {}
+"""ZN_POLY_TUNING = {0}
+ZN_POLY_VERSION = {1}
+ZN_POLY_ABI_VERSION = {2}
 """.format(int(zn_poly_tuning), version, abi_version))
 
 
