--- src/include/zn_poly.h	2008-09-19 19:51:09.000000000 +0200
+++ patches/zn_poly.h	2012-04-20 02:40:11.274100880 +0200
@@ -57,9 +57,9 @@
 /*
    ULONG_BITS = number of bits per unsigned long
 */
-#if ULONG_MAX == 4294967295U
+#if ULONG_MAX == 4294967295UL
 #define ULONG_BITS 32
-#elif ULONG_MAX == 18446744073709551615U
+#elif ULONG_MAX == 18446744073709551615UL
 #define ULONG_BITS 64
 #else
 #error zn_poly requires that unsigned long is either 32 bits or 64 bits
