diff --git a/acb/acos.c b/acb/acos.c
index 31650a01..ae42708c 100644
--- a/acb/acos.c
+++ b/acb/acos.c
@@ -22,6 +22,22 @@ acb_acos(acb_t res, const acb_t z, slong prec)
     {
         acb_t t;
         acb_init(t);
+        if (arb_is_zero(acb_imagref(z)))
+        {
+            arb_t one;
+            arb_init(one);
+            arb_one(one);
+            if (arb_gt(acb_realref(z), one))
+            {
+                acb_asin(res, z, prec);
+                acb_neg(res, res);
+                arb_zero(acb_realref(res));
+                arb_clear(one);
+                arb_clear(t);
+                return;
+            }
+            arb_clear(one);
+        }
         acb_asin(res, z, prec);
         acb_const_pi(t, prec);
         acb_mul_2exp_si(t, t, -1);
diff --git a/acb/acosh.c b/acb/acosh.c
index e0c12306..cc807e45 100644
--- a/acb/acosh.c
+++ b/acb/acosh.c
@@ -30,10 +30,28 @@ acb_acosh(acb_t res, const acb_t z, slong prec)
         acb_sqrt(u, u, prec);
         acb_mul(t, t, u, prec);
         acb_add(t, t, z, prec);
-        acb_log(res, t, prec);
 
+        if (!arb_is_zero(acb_imagref(z)))
+        {
+            acb_log(res, t, prec);
+            acb_clear(t);
+            acb_clear(u);
+            return;
+        }
+        arb_t one, abs;
+        arb_init(one);
+        arb_init(abs);
+        arb_one(one);
+        arb_abs(abs, acb_realref(z));
+        acb_log(res, t, prec);
+        if (arb_lt(abs, one))
+        {
+            arb_zero(acb_realref(res));
+        }
         acb_clear(t);
         acb_clear(u);
+        arb_clear(one);
+        arb_clear(abs);
     }
 }
 
