diff -ru src/Makefile.in src.patched/Makefile.in
--- src/Makefile.in	2010-04-20 06:12:21.000000000 +0200
+++ src.patched/Makefile.in	2011-12-09 23:52:49.000000000 +0100
@@ -136,7 +136,7 @@
 	$(CC) $(SFLAGS) -DPIC -c -o objs/$*.o $<
 	-@mv objs/$*.o $@
 
-$(SHAREDLIBV): $(PIC_OBJS)
+$(SHAREDLIBV): $(PIC_OBJS) $(STATICLIB)
 	$(LDSHARED) $(SFLAGS) -o $@ $(PIC_OBJS) $(LDSHAREDLIBC) $(LDFLAGS)
 	rm -f $(SHAREDLIB) $(SHAREDLIBM)
 	ln -s $@ $(SHAREDLIB)
