Declare array sizes in header file, use 1024 as size

diff -ru meataxe-2.4.24//src/args.c b/src/args.c
--- meataxe-2.4.24//src/args.c	2009-05-23 21:35:19.000000000 +0200
+++ b/src/args.c	2017-02-22 10:32:01.324405681 +0100
@@ -125,7 +125,7 @@
  ** directory, which was selected when building the MeatAxe, is used.
  **/
 
-char MtxBinDir[250] = MTXBIN;
+char MtxBinDir[] = MTXBIN;
 
 
 /**
@@ -139,7 +139,7 @@
  ** @see  MtxBinDir 
  **/
 
-char MtxLibDir[250] = MTXLIB;
+char MtxLibDir[] = MTXLIB;
 
 
 
diff -ru meataxe-2.4.24//src/meataxe.h b/src/meataxe.h
--- meataxe-2.4.24//src/meataxe.h	2011-08-08 00:15:19.000000000 +0200
+++ b/src/meataxe.h	2017-02-22 10:31:52.685290030 +0100
@@ -246,8 +246,8 @@
    Application framework
    ------------------------------------------------------------------ */
 
-extern char MtxBinDir[];	/* MeatAxe library directory */
-extern char MtxLibDir[];	/* MeatAxe program directory */
+extern char MtxBinDir[1024];	/* MeatAxe library directory */
+extern char MtxLibDir[1024];	/* MeatAxe program directory */
 
 #define APP_MAX_ARGS 50
 
