diff -ur a/Makefile b/Makefile
--- a/Makefile	2000-04-07 00:46:28.000000000 -0700
+++ b/Makefile	2007-05-20 23:10:43.000000000 -0700
@@ -88,7 +88,7 @@
 	$(CC) -ansi -c -o gapdate.o date.c
 
 Lie.exe: date.o
-	$(CC) -o Lie.exe $(objects) date.o static/*.o box/*.o -lreadline
+	$(CC) -o Lie.exe $(objects) date.o static/*.o box/*.o -lreadline -lncurses
 	chmod g+w Lie.exe
 Liegap.exe: gapdate.o
 	$(CC) -o Liegap.exe $(GAP_objects) gapdate.o static/*.o box/*.o
