diff -ru old/Makefile.am b/Makefile.am
--- old/Makefile.am	2012-12-18 16:04:09.000000000 +0100
+++ b/Makefile.am	2013-04-26 08:28:12.000000000 +0200
@@ -7,6 +7,8 @@
 liblrcalc_la_LIBADD = mathlib/libmath.la lrcoef/libsymfcn.la schubert/libschub.la
 
 lrcalcincludedir = $(includedir)/lrcalc
-lrcalcinclude_HEADERS=mathlib/*.h lrcoef/*.h schubert/*.h
+lrcalcinclude_HEADERS = mathlib/alloc.h mathlib/claim.h mathlib/hashtab.h mathlib/list.h mathlib/set.h mathlib/vectarg.h mathlib/vector.h lrcoef/maple.h lrcoef/symfcn.h schubert/lincomb.h schubert/schublib.h
 
-TESTS=testsuite
+TESTS = ./testsuite
+
+EXTRA_DIST = LICENSE testsuite lrcalc
diff -ru old/lrcoef/Makefile.am b/lrcoef/Makefile.am
--- old/lrcoef/Makefile.am	2012-01-09 14:34:07.000000000 +0100
+++ b/lrcoef/Makefile.am	2013-04-26 08:21:46.000000000 +0200
@@ -1,11 +1,12 @@
-AM_CFLAGS = -I .. -I ../mathlib
+AM_CPPFLAGS = -I @top_srcdir@/mathlib
 
 noinst_LTLIBRARIES = libsymfcn.la
 libsymfcn_la_SOURCES = symfcn.c maple.c
 
-bin_PROGRAMS = coprod lrcoef lrskew mult skew
+bin_PROGRAMS = coprod lrcoef lrskew mult skew sat
 coprod_LDADD = ../mathlib/libmath.la libsymfcn.la
 lrskew_LDADD = ../mathlib/libmath.la libsymfcn.la
 lrcoef_LDADD = ../mathlib/libmath.la libsymfcn.la
 mult_LDADD = ../mathlib/libmath.la libsymfcn.la
 skew_LDADD   = ../mathlib/libmath.la libsymfcn.la
+sat_LDADD   = ../mathlib/libmath.la libsymfcn.la
diff -ru old/schubert/Makefile.am b/schubert/Makefile.am
--- old/schubert/Makefile.am	2012-01-09 15:40:23.000000000 +0100
+++ b/schubert/Makefile.am	2013-04-26 08:27:48.000000000 +0200
@@ -1,4 +1,4 @@
-AM_CFLAGS = -I .. -I ../mathlib -I ../lrcoef
+AM_CPPFLAGS = -I @top_srcdir@/mathlib -I @top_srcdir@/lrcoef
 
 bin_PROGRAMS = schubmult
 
@@ -6,3 +6,6 @@
 libschub_la_SOURCES = schublib.c lincomb.c
 
 schubmult_LDADD = ../mathlib/libmath.la ../lrcoef/libsymfcn.la libschub.la
+
+# Unused?
+EXTRA_DIST = maple.c
