No longer require adding an -I option to gcc

diff -ru a/src/symfcn.h b/src/symfcn.h
--- a/src/symfcn.h	2014-07-17 17:54:21.000000000 +0200
+++ b/src/symfcn.h	2015-11-29 13:58:22.652567261 +0100
@@ -1,8 +1,8 @@
 #ifndef _SYMFCN_H
 #define _SYMFCN_H
 
-#include <hashtab.h>
-#include <vector.h>
+#include "hashtab.h"
+#include "vector.h"
 
 int part_itr_sz(vector *part);
 int part_itr_sub(vector *part, vector *outer);
--- schublib.h	2014-07-17 10:54:21.000000000 -0500
+++ schublib.h	2015-11-29 10:12:28.051670497 -0600
@@ -1,7 +1,7 @@
 #ifndef _SCHUBLIB_H
 #define _SCHUBLIB_H
 
-#include <hashtab.h>
+#include "hashtab.h"
 
 hashtab *trans(vector *w, int vars, hashtab *res);
 hashtab *monk(int i, hashtab *slc, int rank);
