diff -Naur lrcalc-sage-1.1.7-orig/lrcoef/coprod.c lrcalc-sage-1.1.7/lrcoef/coprod.c
--- lrcalc-sage-1.1.7-orig/lrcoef/coprod.c	2013-06-09 16:12:42.000000000 +1200
+++ lrcalc-sage-1.1.7/lrcoef/coprod.c	2014-06-29 09:51:37.000000000 +1200
@@ -11,7 +11,7 @@
 #include <vectarg.h>
 
 #include "symfcn.h"
-
+#include "lrcalc_jump.h"
 
 void print_usage()
 {
diff -Naur lrcalc-sage-1.1.7-orig/lrcoef/lrcalc.c lrcalc-sage-1.1.7/lrcoef/lrcalc.c
--- lrcalc-sage-1.1.7-orig/lrcoef/lrcalc.c	2013-06-09 16:12:42.000000000 +1200
+++ lrcalc-sage-1.1.7/lrcoef/lrcalc.c	2014-06-29 09:52:25.000000000 +1200
@@ -12,7 +12,7 @@
 
 #include "symfcn.h"
 #include "maple.h"
-
+#include "lrcalc_jump.h"
 
 #define MULT_USAGE \
 "lrcalc mult [-mz] [-r rows] [-q rows,cols] [-f rows,level] part1 - part2\n"
diff -Naur lrcalc-sage-1.1.7-orig/lrcoef/lrcoef.c lrcalc-sage-1.1.7/lrcoef/lrcoef.c
--- lrcalc-sage-1.1.7-orig/lrcoef/lrcoef.c	2013-06-09 16:12:42.000000000 +1200
+++ lrcalc-sage-1.1.7/lrcoef/lrcoef.c	2014-06-29 09:52:50.000000000 +1200
@@ -14,7 +14,7 @@
 
 #include "symfcn.h"
 #include "maple.h"
-
+#include "lrcalc_jump.h"
 
 void print_usage()
 {
diff -Naur lrcalc-sage-1.1.7-orig/lrcoef/lrskew.c lrcalc-sage-1.1.7/lrcoef/lrskew.c
--- lrcalc-sage-1.1.7-orig/lrcoef/lrskew.c	2013-06-09 16:12:42.000000000 +1200
+++ lrcalc-sage-1.1.7/lrcoef/lrskew.c	2014-06-29 09:53:16.000000000 +1200
@@ -11,7 +11,7 @@
 #include <vectarg.h>
 
 #include "symfcn.h"
-
+#include "lrcalc_jump.h"
 
 void print_usage()
 {
diff -Naur lrcalc-sage-1.1.7-orig/lrcoef/mult.c lrcalc-sage-1.1.7/lrcoef/mult.c
--- lrcalc-sage-1.1.7-orig/lrcoef/mult.c	2013-06-09 16:12:42.000000000 +1200
+++ lrcalc-sage-1.1.7/lrcoef/mult.c	2014-06-29 09:53:45.000000000 +1200
@@ -12,7 +12,7 @@
 
 #include "symfcn.h"
 #include "maple.h"
-
+#include "lrcalc_jump.h"
 
 void print_usage()
 {
diff -Naur lrcalc-sage-1.1.7-orig/lrcoef/sat.c lrcalc-sage-1.1.7/lrcoef/sat.c
--- lrcalc-sage-1.1.7-orig/lrcoef/sat.c	2013-06-09 16:12:42.000000000 +1200
+++ lrcalc-sage-1.1.7/lrcoef/sat.c	2014-06-29 09:54:08.000000000 +1200
@@ -10,7 +10,7 @@
 #include <vector.h>
 
 #include "symfcn.h"
-
+#include "lrcalc_jump.h"
 
 #define NUM_PRIMES	50
 
diff -Naur lrcalc-sage-1.1.7-orig/lrcoef/skew.c lrcalc-sage-1.1.7/lrcoef/skew.c
--- lrcalc-sage-1.1.7-orig/lrcoef/skew.c	2013-06-09 16:12:42.000000000 +1200
+++ lrcalc-sage-1.1.7/lrcoef/skew.c	2014-06-29 09:54:40.000000000 +1200
@@ -13,7 +13,7 @@
 
 #include "symfcn.h"
 #include "maple.h"
-
+#include "lrcalc_jump.h"
 
 void print_usage()
 {
diff -Naur lrcalc-sage-1.1.7-orig/mathlib/alloc.c lrcalc-sage-1.1.7/mathlib/alloc.c
--- lrcalc-sage-1.1.7-orig/mathlib/alloc.c	2013-06-09 16:12:17.000000000 +1200
+++ lrcalc-sage-1.1.7/mathlib/alloc.c	2014-06-29 09:55:08.000000000 +1200
@@ -8,6 +8,7 @@
 #include <string.h>
 
 #include "alloc.h"
+#include "lrcalc_jump.h"
 
 #if 0
 #define DEBUG_MEMORY_PRINT
diff -Naur lrcalc-sage-1.1.7-orig/mathlib/alloc.h lrcalc-sage-1.1.7/mathlib/alloc.h
--- lrcalc-sage-1.1.7-orig/mathlib/alloc.h	2013-06-09 16:12:17.000000000 +1200
+++ lrcalc-sage-1.1.7/mathlib/alloc.h	2014-06-29 09:50:13.000000000 +1200
@@ -2,13 +2,6 @@
 #define _ALLOC_H
 
 #include <stdlib.h>
-#include <setjmp.h>
-
-/*  Programs using the lrcalc library should set lrcalc_panic_frame
- *  with setjmp(lrcalc_panic_frame).  The lrcalc library will call
- *  longjmp(lrcalc_panic_frame, 1) if an "out of memory" event occurs.
- */
-jmp_buf lrcalc_panic_frame;
 
 void *amalloc(size_t size);
 void *acalloc(size_t num, size_t size);
diff -Naur lrcalc-sage-1.1.7-orig/mathlib/lrcalc_jump.h lrcalc-sage-1.1.7/mathlib/lrcalc_jump.h
--- lrcalc-sage-1.1.7-orig/mathlib/lrcalc_jump.h	1970-01-01 12:00:00.000000000 +1200
+++ lrcalc-sage-1.1.7/mathlib/lrcalc_jump.h	2014-06-29 09:49:42.000000000 +1200
@@ -0,0 +1,11 @@
+#ifndef _JUMP_H
+
+#include <setjmp.h>
+
+/*  Programs using the lrcalc library should set lrcalc_panic_frame
+ *  with setjmp(lrcalc_panic_frame).  The lrcalc library will call
+ *  longjmp(lrcalc_panic_frame, 1) if an "out of memory" event occurs.
+ */
+jmp_buf lrcalc_panic_frame;
+
+#endif
diff -Naur lrcalc-sage-1.1.7-orig/mathlib/salloc.c lrcalc-sage-1.1.7/mathlib/salloc.c
--- lrcalc-sage-1.1.7-orig/mathlib/salloc.c	2013-06-09 16:12:17.000000000 +1200
+++ lrcalc-sage-1.1.7/mathlib/salloc.c	2014-06-29 09:55:34.000000000 +1200
@@ -13,6 +13,7 @@
 #include <string.h>
 
 #include "alloc.h"
+#include "lrcalc_jump.h"
 
 typedef struct mlink {
   struct mlink *next;
