--- archinfo_linux.c.orig	2009-02-20 11:01:34.000000000 -0800
+++ archinfo_linux.c	2009-02-20 11:02:27.000000000 -0800
@@ -97,12 +97,12 @@
       break;
    case AFIA64:
       res[0] = '\0';
-      ierr = CmndOneLine(NULL, "fgrep 'Itanium' /proc/cpuinfo", res);
+      ierr = CmndOneLine(NULL, "fgrep 'IA-64' /proc/cpuinfo", res);
       if (ierr || res[0] == '\0')
          ierr = CmndOneLine(NULL, "fgrep \"model name\" /proc/cpuinfo", res);
       if (!ierr && res[0] != '\0')
       {
-         if (strstr(res, "Itanium 2") || strstr(res, "McKinley"))
+         if (strstr(res, "IA-64") || strstr(res, "McKinley"))
             mach = IA64Itan2;
          else if (strstr(res, "Itanium")) mach = IA64Itan;
       }
