--- archinfo_linux.c.orig	2009-01-01 19:34:07.000000000 -0800
+++ archinfo_linux.c	2009-01-01 19:33:56.000000000 -0800
@@ -72,6 +72,7 @@
          if (strstr(res, "G4")) mach = PPCG4;
          else if (strstr(res, "7400")) mach = PPCG4;
          else if (strstr(res, "7410")) mach = PPCG4;
+         else if (strstr(res, "7447")) mach = PPCG4;
          else if (strstr(res, "7455")) mach = PPCG4;
          else if (strstr(res, "PPC970FX")) mach = PPCG5;
          else if (strstr(res, "POWER5")) mach = IbmPwr5;
