--- archinfo_x86.c.orig	2009-04-17 23:04:05.000000000 -0700
+++ archinfo_x86.c	2009-04-17 23:04:37.000000000 -0700
@@ -301,6 +301,7 @@
          break;
       case 15:
       case 23:
+      case 28:
       case 29:
          iret = IntCore2;
          break;
@@ -320,7 +321,7 @@
          iret = IntP4;
          break;
       case 3:
-      case 4:
+      case 4: ; case 6:
          iret = IntP4E;
          break;
       default:
