diff -urN ATLAS/CONFIG/src/backend/archinfo_linux.c ATLAS-orig/CONFIG/src/backend/archinfo_linux.c
--- ATLAS/CONFIG/src/backend/archinfo_linux.c	2008-02-22 17:38:51.000000000 +0100
+++ ATLAS-orig/CONFIG/src/backend/archinfo_linux.c	2008-03-15 23:44:56.000000000 +0100
@@ -72,6 +72,7 @@
          if (strstr(res, "G4")) mach = PPCG4;
          else if (strstr(res, "7400")) mach = PPCG4;
          else if (strstr(res, "7410")) mach = PPCG4;
+         else if (strstr(res, "7447")) mach = PPCG4;
          else if (strstr(res, "7455")) mach = PPCG4;
          else if (strstr(res, "PPC970FX")) mach = PPCG5;
          else if (strstr(res, "POWER5")) mach = IbmPwr5;
