--- CONFIG/src/probe_comp.c.orig	2008-04-18 22:54:22.000000000 -0400
+++ CONFIG/src/probe_comp.c	2008-04-18 23:21:30.000000000 -0400
@@ -532,6 +532,8 @@
       return(sp);
    else if (OS == OSAIX)
       sp = (ptrbits == 64) ? "-maix64" : "-maix32";
+   else if (arch == IA64Itan2)
+      printf("Itanium2 - not setting -m64"); // -m64 is not supported on RHEL 5/Itanium
    else if (ptrbits == 64)
      sp = "-m64";
    else if (ptrbits == 32)
