--- CONFIG/src/backend/archinfo_linux.c.orig	2008-04-18 23:24:25.000000000 -0400
+++ CONFIG/src/backend/archinfo_linux.c	2008-04-18 22:52:23.000000000 -0400
@@ -96,12 +96,12 @@
       break;
    case AFIA64:
       res[0] = '\0';
-      ierr = CmndOneLine(NULL, "fgrep 'Itanium' /proc/cpuinfo", res);
+      ierr = CmndOneLine(NULL, "fgrep 'IA-64' /proc/cpuinfo", res);
       if (ierr || res[0] == '\0')
          ierr = CmndOneLine(NULL, "fgrep model /proc/cpuinfo", res);
       if (!ierr && res[0] != '\0')
       {
-         if (strstr(res, "Itanium 2")) mach = IA64Itan2;
+         if (strstr(res, "IA-64")) mach = IA64Itan2;
          else if (strstr(res, "Itanium")) mach = IA64Itan;
       }
       break;
