diff -ru ATLAS/CONFIG/include/atlconf.h ATLAS.new/CONFIG/include/atlconf.h
--- ATLAS/CONFIG/include/atlconf.h	2007-11-29 13:10:04.000000000 +0100
+++ ATLAS.new/CONFIG/include/atlconf.h	2008-03-16 11:57:29.000000000 +0100
@@ -16,17 +16,17 @@
 
 enum ARCHFAM {AFOther=0, AFPPC, AFSPARC, AFALPHA, AFX86, AFIA64, AFMIPS};
 
-#define NMACH 32
+#define NMACH 33
 static char *machnam[NMACH] =
    {"UNKNOWN", "POWER3", "POWER4", "POWER5", "PPCG4", "PPCG5",
     "P5", "P5MMX", "PPRO", "PII", "PIII", "PM", "CoreSolo",
-    "CoreDuo", "Core2Solo", "Core2Duo", "P4", "P4E", "Efficeon", "K7",
+    "CoreDuo", "Core2Solo", "Core2Duo", "P4", "P4D", "P4E", "Efficeon", "K7",
     "HAMMER", "AMD64K10h", "UNKNOWNx86", "IA64Itan", "IA64Itan2",
     "USI", "USII", "USIII", "USIV", "UnknownUS", "MIPSR1xK", "MIPSICE9"};
 enum MACHTYPE {MACHOther, IbmPwr3, IbmPwr4, IbmPwr5, PPCG4, PPCG5,
                IntP5, IntP5MMX, IntPPRO, IntPII, IntPIII, IntPM, IntCoreS,
-               IntCoreDuo, IntCore2Solo, IntCore2Duo, IntP4, IntP4E, TMEff,
-               AmdAthlon, AmdHammer, Amd64K10h, x86X, IA64Itan, IA64Itan2,
+               IntCoreDuo, IntCore2Solo, IntCore2Duo, IntP4, IntP4D, IntP4E,
+               TMEff, AmdAthlon, AmdHammer, Amd64K10h, x86X, IA64Itan, IA64Itan2,
                SunUSI, SunUSII, SunUSIII, SunUSIV, SunUSX,
                MIPSR1xK, /* includes R10K, R12K, R14K, R16K */
                MIPSICE9   /* SiCortex ICE9 -- like MIPS5K */
diff -ru ATLAS/CONFIG/src/atlcomp.txt ATLAS.new/CONFIG/src/atlcomp.txt
--- ATLAS/CONFIG/src/atlcomp.txt	2007-11-29 13:10:04.000000000 +0100
+++ ATLAS.new/CONFIG/src/atlcomp.txt	2008-03-16 11:35:01.000000000 +0100
@@ -39,7 +39,7 @@
 # Level 1 & 2, but must confirm)
 #
 # Core2Solo and Core2Duo
-MACH=Core2Solo,Core2Duo OS=ALL LVL=1000 COMPS=smc,dmc,skc,dkc,icc
+MACH=P4D,Core2Solo,Core2Duo OS=ALL LVL=1000 COMPS=smc,dmc,skc,dkc,icc
    'gcc' '-fomit-frame-pointer -mfpmath=sse -msse3 -O2'
 # PPRO,PII,PIII,P4,P4E,K7,HAMMER
 MACH=PPRO,PII,PIII,P4,P4E,K7,HAMMER OS=ALL LVL=1000 COMPS=f77
diff -ru ATLAS/CONFIG/src/backend/archinfo_linux.c ATLAS.new/CONFIG/src/backend/archinfo_linux.c
--- ATLAS/CONFIG/src/backend/archinfo_linux.c	2007-11-29 13:10:04.000000000 +0100
+++ ATLAS.new/CONFIG/src/backend/archinfo_linux.c	2008-03-16 11:56:49.000000000 +0100
@@ -118,6 +118,7 @@
             else if (strstr(res, " II ")) mach = IntPII;
             else if (strstr(res, "Pro")) mach = IntPPRO;
             else if (strstr(res, "MMX")) mach = IntP5MMX;
+            else if (strstr(res, " D ")) mach = IntP4D;
             else if (strstr(res, " 4 "))
             {
                ierr = CmndOneLine(NULL,
Only in ATLAS.new: CONFIG.tar.bz2
