Let ATLAS use Cygwin thread functions on Cygwin64.
diff -druN ATLAS.orig/ATLAS/src/threads/ATL_thread_start.c ATLAS.new/ATLAS/src/threads/ATL_thread_start.c
--- ATLAS.orig/ATLAS/src/threads/ATL_thread_start.c	2014-07-10 09:22:06.000000000 -0700
+++ ATLAS.new/ATLAS/src/threads/ATL_thread_start.c	2014-11-18 07:17:39.207997205 -0800
@@ -14,14 +14,14 @@
  */
 {
 #ifdef ATL_WINTHREADS
-   #ifdef ATL_WIN32THREADS
+   #if defined(ATL_WIN32THREADS) || defined(__CYGWIN__)
       DWORD thrID;
    #else
       unsigned thrID;
    #endif
 
    #ifdef ATL_NOAFFINITY
-      #ifdef ATL_WIN32THREADS
+      #if defined(ATL_WIN32THREADS) || defined(__CYGWIN__)
          thr->thrH = CreateThread(NULL, 0, rout, arg, 0, &thrID);
       #else
          thr->thrH = (HANDLE)_beginthreadex(NULL, 0, rout, arg, 0, &thrID);
@@ -29,7 +29,7 @@
       ATL_assert(thr->thrH);
    #else
       thr->rank = proc;
-      #ifdef ATL_WIN32THREADS
+      #if defined(ATL_WIN32THREADS) || defined(__CYGWIN__)
          thr->thrH = CreateThread(NULL, 0, rout, arg, CREATE_SUSPENDED, &thrID);
       #else
          thr->thrH = (HANDLE)_beginthreadex(NULL, 0, rout, arg,
