diff -druN ATLAS.orig/CONFIG/include/atlconf.h ATLAS/CONFIG/include/atlconf.h
--- ATLAS.orig/CONFIG/include/atlconf.h	2013-01-08 10:15:42.000000000 -0800
+++ ATLAS/CONFIG/include/atlconf.h	2014-02-10 04:31:19.992981182 -0800
@@ -18,7 +18,7 @@
 enum ARCHFAM {AFOther=0, AFPPC, AFSPARC, AFALPHA, AFX86, AFIA64, AFMIPS,
               AFARM, AFS390};
 
-#define NMACH 52
+#define NMACH 53
 static char *machnam[NMACH] =
    {"UNKNOWN", "POWER3", "POWER4", "POWER5", "PPCG4", "PPCG5",
     "POWER6", "POWER7", "POWERe6500", "IBMz9", "IBMz10", "IBMz196",
@@ -29,7 +29,7 @@
     "Efficeon", "K7", "HAMMER", "AMD64K10h", "AMDLLANO", "AMDDOZER","AMDDRIVER",
     "UNKNOWNx86", "IA64Itan", "IA64Itan2",
     "USI", "USII", "USIII", "USIV", "UST1", "UST2", "UnknownUS",
-    "MIPSR1xK", "MIPSICE9", "ARMv7"};
+    "MIPSR1xK", "MIPSICE9", "ARMv6", "ARMv7"};
 enum MACHTYPE {MACHOther, IbmPwr3, IbmPwr4, IbmPwr5, PPCG4, PPCG5,
                IbmPwr6, IbmPwr7, Pwre6500,
                IbmZ9, IbmZ10, IbmZ196,  /* s390(x) in Linux */
@@ -42,6 +42,7 @@
                SunUSI, SunUSII, SunUSIII, SunUSIV, SunUST1, SunUST2, SunUSX,
                MIPSR1xK, /* includes R10K, R12K, R14K, R16K */
                MIPSICE9,  /* SiCortex ICE9 -- like MIPS5K */
+               ARMv6, /* includes Raspberry Pi */
                ARMv7      /* includes Cortex A8, A9 */
                };
 #define MachIsX86(mach_) \
@@ -60,7 +61,7 @@
 #define MachIsPPC(mach_) \
    ( (mach_) >= PPCG4 && (mach_) <= PPCG5 )
 #define MachIsARM(mach_) \
-   ( (mach_) == ARMv7 )
+   ( (mach_) >= ARMv6 && (mach_) <= ARMv7 )
 #define MachIsS390(mach_) \
    ( (mach_) >= IbmZ9 && (mach_) <= IbmZ196 )
 
