--- src/CONFIG/src/probe_comp.c	2012-06-23 17:27:27.000000000 +0100
+++ new/CONFIG/src/probe_comp.c	2012-06-24 03:20:46.118154762 +0100
@@ -607,6 +607,8 @@
       return(sp);
    else if (OS == OSAIX)
       sp = (ptrbits == 64) ? "-maix64" : "-maix32";
+   else if (arch == IA64Itan2)
+      printf("Itanium2 - not setting -m64"); // -m64 is not supported on RHEL 5/Itanium
    else if ((MachIsX86(arch) || MachIsPPC(arch) || MachIsUS(arch)) ||
             arch == IbmPwr7 || arch == IbmPwr6)
    {
