diff -dNru src.orig/makes/Make.Flp src/makes/Make.Flp
--- src.orig/makes/Make.Flp	2013-01-08 19:15:41.000000000 +0100
+++ src/makes/Make.Flp	2013-03-27 18:01:17.224597923 +0100
@@ -756,7 +756,7 @@
 
 $(FLAdir)/lapack_$(ARCH).a : $(FLAdir)/make.inc
 	cd $(FLAdir) ; $(PMAKE) lib
-	if [ -e $(FLAdir)/liblapack.a ]; then \
+	if [ -f $(FLAdir)/liblapack.a ]; then \
            ln -s $(FLAdir)/liblapack.a $(FLAdir)/lapack_$(ARCH).a ; \
         fi
 
diff -dNru src.orig/makes/Make.ttune src/makes/Make.ttune
--- src.orig/makes/Make.ttune	2013-01-08 19:15:42.000000000 +0100
+++ src/makes/Make.ttune	2013-03-27 18:01:17.020597929 +0100
@@ -26,7 +26,7 @@
 	- rm -f $(sumf)
 	echo "#ifndef ATLAS_SUMM_H" > $(sumf)
 	echo "   #define ATLAS_SUMM_H" >> $(sumf)
-	if [ -e res/aff.h ]; then \
+	if [ -f res/aff.h ]; then \
            cat res/aff.h >> $(sumf) ; \
         else \
            echo "#define ATL_TAFFINITY 0" >> $(sumf) ; \
