diff -drNu src.orig/bin/atlas_install.c src/bin/atlas_install.c
--- src.orig/bin/atlas_install.c	2013-01-08 19:15:41.000000000 +0100
+++ src/bin/atlas_install.c	2013-05-17 18:16:01.004809341 +0200
@@ -1075,8 +1075,13 @@
    PrintStartStop(stdout, fpsum, 3, 1, 5, 1, 0, "FINAL STATIC LIBRARY UPDATE");
    sprintf(ln2, "INSTALL_LOG/LIBUPDATE.LOG");
    PrintBanner(ln2, 1, 5, 1, 1);
-   sprintf(ln, "%s IBuildLibs IBuildPtlibs0 %s %s\n", fmake, redir, ln2);
+   sprintf(ln, "%s IBuildLibs %s %s\n", fmake, redir, ln2);
    fprintf(stdout, ln);
+#ifdef ATL_NCPU
+   PrintBanner(ln2, 1, 5, 1, 2);
+   sprintf(ln, "%s IBuildPtlibs0 %s %s\n", fmake, redir, ln2);
+   fprintf(stdout, ln);
+#endif
    ATL_Cassert(system(ln)==0, "STATIC LIBRARY UPDATE", ln2);
    PrintBanner(ln2, 0, 5, 1, 1);
    PrintStartStop(stdout, fpsum, 3, 0, 5, 1, 0, "FINAL STATIC LIBRARY UPDATE");
