diff -ur src/Makefile.in src.spkg/Makefile.in
--- src/Makefile.in	1995-08-17 02:54:29.000000000 +0200
+++ src.spkg/Makefile.in	2005-12-12 05:39:42.000000000 +0100
@@ -46,7 +46,10 @@
 # so compilers besides gcc can find it by default.
 # If it is empty or not defined, termcap.h will only be installed in
 # includedir. 
-oldincludedir = /usr/include
+
+# I commented this, since a Sage install should work as non-root.
+# -- William Stein
+#oldincludedir = /usr/include
 
 # Directory in which to install the documentation info files.
 infodir = $(prefix)/info
@@ -66,7 +69,7 @@
 termcap.src termcap.texi termcap.info* \
 texinfo.tex Makefile.in configure configure.in mkinstalldirs install-sh
 
-all:	libtermcap.a info
+all:	libtermcap.a 
 
 .c.o:
 	$(CC) -c $(CPPFLAGS) $(DEFS) -I. -I$(srcdir) $(CFLAGS) $<
@@ -77,8 +80,6 @@
 	cd $(srcdir); $(INSTALL_DATA) termcap.h $(includedir)/termcap.h
 	-cd $(srcdir); test -z "$(oldincludedir)" || \
 	  $(INSTALL_DATA) termcap.h $(oldincludedir)/termcap.h
-	cd $(srcdir); for f in termcap.info*; \
-	do $(INSTALL_DATA) $$f $(infodir)/$$f; done
 
 uninstall: @uninstalldata@
 	rm -f $(libdir)/libtermcap.a $(includedir)/termcap.h
