diff -Naur src/buildutils/detect.py src-patched/buildutils/detect.py
--- src/buildutils/detect.py	2014-05-07 20:24:59.000000000 +0200
+++ src-patched/buildutils/detect.py	2014-06-07 16:40:54.923796430 +0200
@@ -45,17 +45,7 @@
 
     cpreargs = lpreargs = None
     if sys.platform == 'darwin':
-        # use appropriate arch for compiler
-        if platform.architecture()[0]=='32bit':
-            if platform.processor() == 'powerpc':
-                cpu = 'ppc'
-            else:
-                cpu = 'i386'
-            cpreargs = ['-arch', cpu]
-            lpreargs = ['-arch', cpu, '-undefined', 'dynamic_lookup']
-        else:
-            # allow for missing UB arch, since it will still work:
-            lpreargs = ['-undefined', 'dynamic_lookup']
+        lpreargs = ['-undefined', 'dynamic_lookup']
     if sys.platform == 'sunos5':
         if platform.architecture()[0]=='32bit':
             lpreargs = ['-m32']
