--- ../src/makefile	2004-08-22 06:01:51.000000000 -0700
+++ makefile	2010-02-15 23:51:13.535565200 -0800
@@ -30,7 +30,7 @@
 LIBMCFILES = $(SFILES) $(DFILES) $(LFILES)
 
 # mtstf and mtstl test the float and long double functions.
-all: libmc.a mtst cvect dccalc mtstf mtstl whitebxf whitebx
+all: libmc.a mtst cvect dccalc mtstf whitebxf whitebx
 
 mtst: mtst.o $(DFILES) stubs.o
 	$(CC) -v -o mtst mtst.o $(DFILES) $(LIBS)
@@ -89,7 +89,6 @@
 	-./mtstf
 	-./cvect
 	-./mtst
-	-./mtstl
 	-./whitebxf
 	-./whitebx
 	echo For additional tests, make -f celefunt.mak
