diff -urN gf2x-1.2/ChangeLog gf2x-test/ChangeLog
--- gf2x-1.2/ChangeLog	2017-07-03 16:27:03.000000000 +0200
+++ gf2x-test/ChangeLog	2017-07-04 00:55:55.000000000 +0200
@@ -1,3 +1,9 @@
+commit fe124f625f9037c6f9ed22aad50221b87db7101c
+Author: Emmanuel Thomé <Emmanuel.Thome@inria.fr>
+Date:   Tue Jul 4 00:55:31 2017 +0200
+
+    forcibly disable all the _supports_XXX macros when --disable-hardware-specific-code is used
+
 commit 6a40623ffd93e7b412ffceab80184ef39c3426a7
 Author: Emmanuel Thomé <Emmanuel.Thome@inria.fr>
 Date:   Thu Jun 29 17:17:11 2017 +0200
diff -urN gf2x-1.2/configure gf2x-test/configure
--- gf2x-1.2/configure	2017-07-03 16:26:52.000000000 +0200
+++ gf2x-test/configure	2017-07-04 00:55:47.000000000 +0200
@@ -17508,6 +17508,13 @@
 
 if test "x${enable_hardware_specific_code}" = xno ; then
  echo "Not using hardware-specific code due to --disable-hardware-specific-code flag"
+ gf2x_cv_cc_supports_sse2=no
+ gf2x_cv_cc_supports_sse3=no
+ gf2x_cv_cc_supports_ssse3=no
+ gf2x_cv_cc_supports_sse41=no
+ gf2x_cv_cc_supports_pclmul=no
+ gf2x_cv_cc_supports_march_native=no
+ gf2x_cv_cc_supports_mtune_native=no
 else
  # It is necessary to make all tests. We do encounter in the wild
  # binutils (openbsd binutils 2.15, namely) which are buggy with ssse3,
diff -urN gf2x-1.2/configure.ac gf2x-test/configure.ac
--- gf2x-1.2/configure.ac	2017-06-29 10:56:09.000000000 +0200
+++ gf2x-test/configure.ac	2017-07-04 00:54:59.000000000 +0200
@@ -114,6 +114,13 @@
 
 if test "x${enable_hardware_specific_code}" = xno ; then
  echo "Not using hardware-specific code due to --disable-hardware-specific-code flag"
+ gf2x_cv_cc_supports_sse2=no
+ gf2x_cv_cc_supports_sse3=no
+ gf2x_cv_cc_supports_ssse3=no
+ gf2x_cv_cc_supports_sse41=no
+ gf2x_cv_cc_supports_pclmul=no
+ gf2x_cv_cc_supports_march_native=no
+ gf2x_cv_cc_supports_mtune_native=no
 else
  # It is necessary to make all tests. We do encounter in the wild
  # binutils (openbsd binutils 2.15, namely) which are buggy with ssse3,
