From 8aed4cdb611f8e95f2ce08ff354e3b2dbcf9980f Mon Sep 17 00:00:00 2001
From: Jean-Pierre Flori <jean-pierre.flori@ssi.gouv.fr>
Date: Tue, 17 Jun 2014 14:48:11 +0200
Subject: [PATCH 1/5] Trac #15014: Let gf2x build a shared library on Cygwin.

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 725ca20..ad6d6d5 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -44,7 +44,7 @@ nobase_nodist_include_HEADERS=	\
 		gf2x/gf2x_mul9.h
 
 
-libgf2x_la_LDFLAGS=-version-info @gf2x_lib_version@
+libgf2x_la_LDFLAGS=-version-info @gf2x_lib_version@ -no-undefined
 
 if ENABLE_NTL_CHECKS
     MAYBE_APPS=apps
-- 
2.0.0

