diff --git a/src/mul9clk2.c b/src/mul9clk2.c
index bf7ad4a..5c6b980 100644
--- a/src/mul9clk2.c
+++ b/src/mul9clk2.c
@@ -149,7 +149,7 @@ void gf2x_mul9 (unsigned long *c, const unsigned long *a, const unsigned long *b
   e = p2e[1]; h=l; l=p2o[1];
   _mm_storeu_si128((__v2di*)(c+14), e ^ _mm_slli_si128(l, 8) ^ _mm_srli_si128(h, 8));
 
-  e = p2e[2];
+  e = p2e[2]; h=l;
   _mm_storeu_si128((__v2di*)(c+16), e ^ _mm_srli_si128(h, 8));
 
   /*
