diff -druN src.orig/Makefile.am src/Makefile.am
--- src.orig/Makefile.am	2012-05-31 22:53:47.000000000 +0200
+++ src/Makefile.am	2013-08-06 23:37:13.652085927 +0200
@@ -44,7 +44,7 @@
 		gf2x/gf2x_mul9.h
 
 
-libgf2x_la_LDFLAGS=-version-info @gf2x_lib_version@
+libgf2x_la_LDFLAGS=-version-info @gf2x_lib_version@ -no-undefined
 
 if ENABLE_NTL_CHECKS
     MAYBE_APPS=apps
diff -druN src.orig/Makefile.in src/Makefile.in
--- src.orig/Makefile.in	2012-05-31 22:54:02.000000000 +0200
+++ src/Makefile.in	2013-08-06 23:37:31.055964200 +0200
@@ -325,7 +325,7 @@
 		gf2x/gf2x_mul8.h		\
 		gf2x/gf2x_mul9.h
 
-libgf2x_la_LDFLAGS = -version-info @gf2x_lib_version@
+libgf2x_la_LDFLAGS = -version-info @gf2x_lib_version@ -no-undefined
 @ENABLE_NTL_CHECKS_TRUE@MAYBE_APPS = apps
 SUBDIRS = . tests src $(MAYBE_APPS)
 DIST_SUBDIRS = tests src apps
