diff --git a/src/toom2.c b/src/toom2.c
index 371d5a8..5ea6eb3 100644
--- a/toom2.c
+++ b/toom2.c
@@ -496,7 +496,7 @@ void gf2x_mul_tc3(unsigned long *c, const unsigned long *a,
 void gf2x_mul_tc3u(unsigned long * c, const unsigned long * a, long sa,
 	      const unsigned long * b, unsigned long * stk)
 {
-    ASSERT(sa >= GF2X_MUL_TOOMU_THRESHOLD);	// n should be at least 6 for internal
+    ASSERT(sa >= 33);	// n should be at least 6 for internal
     // reasons and 17 so calls to Toom
     // have size at least 9, so need
     // sa >= 33.
