diff -druN gf2x-1.1.orig/configure.ac gf2x-1.1/configure.ac
--- gf2x-1.1.orig/configure.ac	2014-01-14 03:13:43.839238178 -0800
+++ gf2x-1.1/configure.ac	2014-01-14 03:19:28.939272688 -0800
@@ -126,6 +126,12 @@
  fi
 else
  AC_MSG_NOTICE([Using supplied hwdir value $hwdir])
+ if test "$enable_hardware_specific_code" != "yes" ;then
+  if test "$hwdir" = "x86_64" ;then
+   AC_MSG_NOTICE([Assuming SSE2 is supported on x86_64])
+   AC_DEFINE([HAVE_SSE2_SUPPORT],[1],[Define if sse-2 code as present in the source tree is supported by the compiler])
+  fi
+ fi
 fi
 
 AC_MSG_NOTICE([using ABI="$ABI"])
diff -druN gf2x-1.1.orig/configure gf2x-1.1/configure
--- gf2x-1.1.orig/configure	2014-01-14 03:13:43.829238177 -0800
+++ gf2x-1.1/configure	2014-01-14 03:19:41.479273942 -0800
@@ -16913,6 +16913,15 @@
 else
  { $as_echo "$as_me:${as_lineno-$LINENO}: Using supplied hwdir value $hwdir" >&5
 $as_echo "$as_me: Using supplied hwdir value $hwdir" >&6;}
+ if test "$enable_hardware_specific_code" != "yes" ;then
+  if test "$hwdir" = "x86_64" ;then
+   { $as_echo "$as_me:${as_lineno-$LINENO}: Assuming SSE2 is supported on x86_64" >&5
+$as_echo "$as_me: Assuming SSE2 is supported on x86_64" >&6;}
+
+$as_echo "#define HAVE_SSE2_SUPPORT 1" >>confdefs.h
+
+  fi
+ fi
 fi
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: using ABI=\"$ABI\"" >&5
