diff --git a/Cudd/cudd/Makefile.in b/Cudd/cudd/Makefile.in
index 645eb7d..760278b
--- a/Cudd/cudd/Makefile.in
+++ b/Cudd/cudd/Makefile.in
@@ -350,7 +350,7 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/libpolybori/include

 AM_CFLAGS = -std=c99
-AM_CXXFLAGS = -std=c++98 -ftemplate-depth-100
+AM_CXXFLAGS = -std=gnu++98 -ftemplate-depth-100
 AM_DEFAULT_SOURCE_EXT = .cc
 noinst_LTLIBRARIES = libcudd.la
 libcudd_la_SOURCES = \
diff --git a/common.mk b/common.mk
index 7915587..0afda02 100755
--- a/common.mk
+++ b/common.mk
@@ -3,6 +3,6 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/libpolybori/include

 AM_CFLAGS = -std=c99
-AM_CXXFLAGS = -std=c++98 -ftemplate-depth-100
+AM_CXXFLAGS = -std=gnu++98 -ftemplate-depth-100

 AM_DEFAULT_SOURCE_EXT = .cc
diff --git a/groebner/src/Makefile.in b/groebner/src/Makefile.in
index ef7882c..cb3b04f
--- a/groebner/src/Makefile.in
+++ b/groebner/src/Makefile.in
@@ -372,7 +372,7 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/libpolybori/include

 AM_CFLAGS = -std=c99
-AM_CXXFLAGS = -std=c++98 -ftemplate-depth-100
+AM_CXXFLAGS = -std=gnu++98 -ftemplate-depth-100
 AM_DEFAULT_SOURCE_EXT = .cc
 lib_LTLIBRARIES = libpolybori_groebner.la
 libpolybori_groebner_la_CXXFLAGS = $(AM_CXXFLAGS) $(SIMMD_CFLAGS)
diff --git a/libpolybori/src/Makefile.in b/libpolybori/src/Makefile.in
index 4e4e60a..4d110c2
--- a/libpolybori/src/Makefile.in
+++ b/libpolybori/src/Makefile.in
@@ -323,7 +323,7 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/libpolybori/include

 AM_CFLAGS = -std=c99
-AM_CXXFLAGS = -std=c++98 -ftemplate-depth-100
+AM_CXXFLAGS = -std=gnu++98 -ftemplate-depth-100
 AM_DEFAULT_SOURCE_EXT = .cc
 noinst_LTLIBRARIES = libpolybori_base.la
 libpolybori_base_la_SOURCES = \
