--- ../matplotlib-src-svn-6821/lib/matplotlib/cbook.py	2009-01-23 13:48:57.000000000 -0800
+++ patches-new/cbook.py	2009-01-23 14:15:31.000000000 -0800
@@ -14,7 +14,10 @@
 
 
 # on some systems, locale.getpreferredencoding returns None, which can break unicode
-preferredencoding = locale.getpreferredencoding()
+try:
+    preferredencoding = locale.getpreferredencoding()
+except:
+    preferredencoding = None
 
 def unicode_safe(s):
     if preferredencoding is None: return unicode(s)
