diff --git a/lib/matplotlib/sphinxext/plot_directive.py b/lib/matplotlib/sphinxext/plot_directive.py
index d59d9bdd71bf..d40e0c58b4a5 100644
--- a/lib/matplotlib/sphinxext/plot_directive.py
+++ b/lib/matplotlib/sphinxext/plot_directive.py
@@ -266,6 +266,7 @@ def _copy_css_file(app, exc):
     if exc is None and app.builder.format == 'html':
         src = cbook._get_data_path('plot_directive/plot_directive.css')
         dst = app.outdir / Path('_static')
+        dst.mkdir(exist_ok=True)
         shutil.copy(src, dst)
