--- src/lib/matplotlib/finance.py	2011-01-10 21:47:07.000000000 -0800
+++ src/lib/matplotlib/finance.py	2011-08-13 09:09:16.000000000 -0700
@@ -4,7 +4,7 @@
 
 """
 #from __future__ import division
-import os, warnings
+import os, warnings, errno
 from urllib2 import urlopen
 
 try:
@@ -173,8 +173,13 @@
         fh = file(cachename)
         verbose.report('Using cachefile %s for %s'%(cachename, ticker))
     else:
-        if not os.path.isdir(cachedir):
+        try:
             os.mkdir(cachedir)
+        except OSError, e:
+            if e.errno == errno.EEXIST:
+                pass
+            else:
+                raise
         urlfh = urlopen(url)
 
         fh = file(cachename, 'w')
