--- patches.py.orig	2009-02-14 06:14:40.000000000 -0600
+++ patches.py	2009-02-14 06:15:08.000000000 -0600
@@ -2287,15 +2287,21 @@
                 x, y = path.vertices[0]
                 insideA = inside_circle(x, y, shrinkA)
 
-                left, right = split_path_inout(path, insideA)
-                path = right
+                try:
+                    left, right = split_path_inout(path, insideA)
+                    path = right
+                except ValueError:
+                    pass
 
             if shrinkB:
                 x, y = path.vertices[-1]
                 insideB = inside_circle(x, y, shrinkB)
 
-                left, right = split_path_inout(path, insideB)
-                path = left
+                try:
+                    left, right = split_path_inout(path, insideB)
+                    path = left
+                except ValueError:
+                    pass
 
             return path
 
