--- src/lib/matplotlib/tri/_tri.cpp	2013-10-10 05:42:48.000000000 -0700
+++ src/lib/matplotlib/tri/_tri.cpp	2014-02-28 13:32:15.000000000 -0800
@@ -2177,14 +2177,14 @@
 
 
 RandomNumberGenerator::RandomNumberGenerator(unsigned long seed)
-    : _M(21870), _A(1291), _C(4621), _seed(seed % _M)
+    : _m(21870), _a(1291), _c(4621), _seed(seed % _m)
 {}
 
 unsigned long
 RandomNumberGenerator::operator()(unsigned long max_value)
 {
-    _seed = (_seed*_A + _C) % _M;
-    return (_seed*max_value) / _M;
+    _seed = (_seed*_a + _c) % _m;
+    return (_seed*max_value) / _m;
 }
 
 
--- src/lib/matplotlib/tri/_tri.h	2013-10-10 05:42:48.000000000 -0700
+++ src/lib/matplotlib/tri/_tri.h	2014-02-28 13:32:29.000000000 -0800
@@ -818,7 +818,7 @@
     unsigned long operator()(unsigned long max_value);
 
 private:
-    const unsigned long _M, _A, _C;
+    const unsigned long _m, _a, _c;
     unsigned long _seed;
 };
 
