--- src/setupext.py	2011-10-06 09:50:23.000000000 -0500
+++ src/setupext.py	2011-12-14 22:42:14.000000000 -0600
@@ -608,7 +608,17 @@
     module.include_dirs.append(numpy.get_include())
 
 def add_png_flags(module):
-    try_pkgconfig(module, 'libpng', 'png')
+    # 'command -v pkg-config' returns 0 iff pkg-config is found, so
+    # here 'call' evaluates to True iff pkg-config is not found.
+    # (This code is taken from sage/misc/sage_ostools.py.)
+    try:
+        if subprocess.call('command -v pkg-config', shell=True,
+                           stdout=subprocess.PIPE, stderr=subprocess.PIPE):
+            try_pkgconfig(module, 'libpng12', 'png12')
+        else:
+            try_pkgconfig(module, 'libpng', 'png')
+    except OSError: # shouldn't be necessary, but just in case
+        try_pkgconfig(module, 'libpng12', 'png12')
     add_base_flags(module)
     add_numpy_flags(module)
     module.libraries.append('z')
