--- src/setupext.py	2009-01-23 17:35:47.000000000 -0800
+++ patches-new/setupext.py	2009-01-23 17:40:08.000000000 -0800
@@ -44,14 +44,16 @@
 import os
 import re
 import subprocess
+### FOR SAGE
+sage_inc = os.environ['SAGE_LOCAL'] + '/include/'
+sage_lib = os.environ['SAGE_LOCAL'] + '/lib/'
 
 basedir = {
     'win32'  : ['win32_static',],
-    'linux2' : ['/usr/local', '/usr'],
+    'linux2' : [sage_lib],
     'linux'  : ['/usr/local', '/usr',],
     'cygwin' : ['/usr/local', '/usr',],
-    'darwin' : ['/sw/lib/freetype2', '/sw/lib/freetype219', '/usr/local',
-                '/usr', '/sw', '/usr/X11R6'],
+    'darwin' : [sage_lib],
     'freebsd4' : ['/usr/local', '/usr'],
     'freebsd5' : ['/usr/local', '/usr'],
     'freebsd6' : ['/usr/local', '/usr'],
@@ -322,7 +324,9 @@
 
     module.include_dirs.extend(incdirs)
     module.include_dirs.append('.')
+    module.include_dirs.append(sage_inc)    	
     module.library_dirs.extend(libdirs)
+    module.library_dirs.extend([sage_lib])
 
 def getoutput(s):
     'get the output of a system command'
@@ -1027,6 +1031,7 @@
         #      of distros.
 
         # Query Tcl/Tk system for library paths and version string
+        tk_ver = ''
         try:
             tcl_lib_dir, tk_lib_dir, tk_ver = query_tcltk()
         except:
