--- src/setupext.py	2013-10-10 05:42:48.000000000 -0700
+++ src/setupext.py	2013-12-19 13:39:17.000000000 -0800
@@ -922,8 +920,12 @@
             'src/_png.cpp', 'src/mplutils.cpp'
             ]
         ext = make_extension('matplotlib._png', sources)
+        if self.check().find('Using unknown') == -1:
+            defaults = ['png', 'z']
+        else:
+            defaults = ['png12', 'z']
         pkg_config.setup_extension(
-            ext, 'libpng', default_libraries=['png', 'z'])
+            ext, 'libpng', default_libraries=defaults)
         Numpy().add_flags(ext)
         CXX().add_flags(ext)
         return ext
