--- a/setup.py
+++ b/setup.py
@@ -85,8 +85,8 @@ mpl_packages = [
     'Optional subpackages',
     setupext.SampleData(),
     setupext.Toolkits(),
-    setupext.Tests(),
-    setupext.Toolkits_Tests(),
+    # setupext.Tests(),
+    # setupext.Toolkits_Tests(),
     'Optional backend extensions',
     # These backends are listed in order of preference, the first
     # being the most preferred.  The first one that looks like it will
