--- src/setupext.py	2010-03-09 09:20:20.000000000 -0500
+++ patches/setupext.py	2010-06-10 07:35:47.000000000 -0400
@@ -326,6 +326,8 @@
     return True
 
 def add_base_flags(module):
+    module.include_dirs.append(os.environ['SAGE_LOCAL'] + '/include/')       
+    module.library_dirs.append(os.environ['SAGE_LOCAL'] + '/lib/') 
     incdirs = filter(os.path.exists,
                      [os.path.join(p, 'include') for p in basedir[sys.platform] ])
     libdirs = filter(os.path.exists,
@@ -519,7 +521,7 @@
     module.include_dirs.append(numpy.get_include())
 
 def add_png_flags(module):
-    try_pkgconfig(module, 'libpng', 'png')
+    try_pkgconfig(module, 'libpng12', 'png12')
     add_base_flags(module)
     add_numpy_flags(module)
     module.libraries.append('z')
