--- src/setupext.py	2010-06-11 10:50:55.000000000 -0500
+++ patches/setupext.py	2010-06-11 14:42:15.000000000 -0500
@@ -538,7 +538,7 @@
     module.include_dirs.append(numpy.get_include())
 
 def add_png_flags(module):
-    try_pkgconfig(module, 'libpng', 'png')
+    try_pkgconfig(module, 'libpng12', 'png12')
     add_base_flags(module)
     add_numpy_flags(module)
     module.libraries.append('z')
