--- src/setupext.py	2009-08-23 00:27:32.000000000 -0500
+++ patches/setupext.py	2009-09-22 14:21:04.000000000 -0500
@@ -44,10 +44,13 @@
 import os
 import re
 import subprocess
+### FOR SAGE
+sage_inc = os.environ['SAGE_LOCAL'] + '/include/'
+sage_lib = os.environ['SAGE_LOCAL'] + '/lib/'
 
 basedir = {
     'win32'  : ['win32_static',],
-    'linux2' : ['/usr/local', '/usr'],
+    'linux2' : [sage_lib],
     'linux'  : ['/usr/local', '/usr',],
     'cygwin' : ['/usr/local', '/usr',],
     '_darwin' : ['/sw/lib/freetype2', '/sw/lib/freetype219', '/usr/local',
@@ -57,7 +60,7 @@
     # people to :
     #   make -f make.osx fetch deps mpl_build mpl_install
 
-    'darwin' : [],
+    'darwin' : [sage_lib],
 
     'freebsd4' : ['/usr/local', '/usr'],
     'freebsd5' : ['/usr/local', '/usr'],
@@ -328,7 +331,9 @@
 
     module.include_dirs.extend(incdirs)
     module.include_dirs.append('.')
+    module.include_dirs.append(sage_inc)    	
     module.library_dirs.extend(libdirs)
+    module.library_dirs.extend([sage_lib])
 
 def getoutput(s):
     'get the output of a system command'
@@ -512,7 +517,7 @@
     module.include_dirs.append(numpy.get_include())
 
 def add_png_flags(module):
-    try_pkgconfig(module, 'libpng', 'png')
+    try_pkgconfig(module, 'libpng12', 'png12')
     add_base_flags(module)
     add_numpy_flags(module)
     module.libraries.append('z')
