--- src/setupext.py	2009-08-01 12:15:24.000000000 -0700
+++ patches/setupext.py	2009-08-08 23:43:29.000000000 -0700
@@ -44,14 +44,16 @@
 import os
 import re
 import subprocess
+### FOR SAGE
+sage_inc = os.environ['SAGE_LOCAL'] + '/include/'
+sage_lib = os.environ['SAGE_LOCAL'] + '/lib/'
 
 basedir = {
     'win32'  : ['win32_static',],
-    'linux2' : ['/usr/local', '/usr'],
+    'linux2' : [sage_lib],
     'linux'  : ['/usr/local', '/usr',],
     'cygwin' : ['/usr/local', '/usr',],
-    'darwin' : ['/sw/lib/freetype2', '/sw/lib/freetype219', '/usr/local',
-                '/usr', '/sw'],
+    'darwin' : [sage_lib],
     'freebsd4' : ['/usr/local', '/usr'],
     'freebsd5' : ['/usr/local', '/usr'],
     'freebsd6' : ['/usr/local', '/usr'],
@@ -321,7 +323,9 @@
 
     module.include_dirs.extend(incdirs)
     module.include_dirs.append('.')
+    module.include_dirs.append(sage_inc)    	
     module.library_dirs.extend(libdirs)
+    module.library_dirs.extend([sage_lib])
 
 def getoutput(s):
     'get the output of a system command'
@@ -505,7 +509,7 @@
     module.include_dirs.append(numpy.get_include())
 
 def add_png_flags(module):
-    try_pkgconfig(module, 'libpng', 'png')
+    try_pkgconfig(module, 'libpng12', 'png12')
     add_base_flags(module)
     add_numpy_flags(module)
     module.libraries.append('z')
@@ -1034,6 +1038,7 @@
         #      of distros.
 
         # Query Tcl/Tk system for library paths and version string
+        tk_ver = ''
         try:
             tcl_lib_dir, tk_lib_dir, tk_ver = query_tcltk()
         except:
