--- src/lib/matplotlib/texmanager.py	2011-01-10 21:47:12.000000000 -0800
+++ src/lib/matplotlib/texmanager.py	2011-08-13 09:11:25.000000000 -0700
@@ -33,7 +33,7 @@
 
 """
 
-import copy, glob, os, shutil, sys, warnings
+import copy, glob, os, shutil, sys, warnings, errno
 from subprocess import Popen, PIPE, STDOUT
 
 try:
@@ -88,8 +88,13 @@
 WARNING: found a TeX cache dir in the deprecated location "%s".
   Moving it to the new default location "%s"."""%(oldcache, texcache)
         shutil.move(oldcache, texcache)
-    if not os.path.exists(texcache):
+    try:
         os.mkdir(texcache)
+    except OSError, e:
+        if e.errno == errno.EEXIST:
+            pass
+        else:
+            raise
 
     _dvipng_hack_alpha = None
     #_dvipng_hack_alpha = dvipng_hack_alpha()
@@ -130,8 +135,14 @@
 
     def __init__(self):
 
-        if not os.path.isdir(self.texcache):
+        try:
             os.mkdir(self.texcache)
+        except OSError, e:
+            if e.errno == errno.EEXIST:
+                pass
+            else:
+                raise
+
         ff = rcParams['font.family'].lower()
         if ff in self.font_families:
             self.font_family = ff
