diff --git a/src/dot2tex-2.8.6/dot2tex/dot2tex.py b/src/dot2tex-2.8.6/dot2tex/dot2tex.py
--- a/src/dot2tex/dot2tex.py
+++ b/src/dot2tex/dot2tex.py
@@ -204,7 +204,6 @@ def create_xdot(dotdata,prog='dot'):
     log.debug('Creating xdot data with: %s',cmd)
     stdin, stdout, stderr = os.popen3(cmd,'t')
     stdin.close()
-    stderr.close()
 
     # I'm not quite sure why this is necessary, but some files
     # produces data with line endings that confuses pydot/pyparser.
@@ -213,6 +212,7 @@ def create_xdot(dotdata,prog='dot'):
     #data = "".join(data)
 
     stdout.close()
+    stderr.close()
     os.unlink(tmp_name)
 
     return data
