--- src/tests/test_buildexamples.py	2009-09-30 19:08:30.000000000 +0200
+++ src/tests/test_buildexamples.py	2013-05-21 03:35:45.180551641 +0200
@@ -163,18 +163,23 @@
 
 if __name__ == "__main__":
     filelist = []
+    filelist2 = []
+    to_be_built_last = [ "graphofgraphs.dot" ] # includes PDFs of others
     exdir= EXAMPLES_DIR
     os.chdir(exdir)
     for f in glob.glob("*.dot"):
         if not f.find('dot2tex-fig') > 0:
             src = os.path.join(exdir,f)
             dst = normpath(join(DEST_DIR,basename(src)))
-            filelist.append(dst)
+            if f in to_be_built_last:
+                filelist2.append(dst)
+            else:
+                filelist.append(dst)
             copyifnewer(src,dst)
 
 
 
-    failedfiles = build_gallery(filelist)
+    failedfiles = build_gallery(filelist+filelist2)
     if failedfiles:
         log.warning('Failed files: %s',failedfiles);
         sys.exit(1)
