--- a/dot2tex/dotparsing.py	2014-03-29 17:44:24.182778200 +0000
+++ b/dot2tex/dotparsing.py	2014-03-29 17:59:43.414164266 +0000
@@ -26,7 +26,7 @@
 from pyparsing import  (Literal, CaselessLiteral, Word, Upcase, OneOrMore, ZeroOrMore,
                         Forward, NotAny, delimitedList, oneOf, Group, Optional, Combine, alphas, nums,
                         restOfLine, cStyleComment, nums, alphanums, printables, empty, quotedString,
-                        ParseException, ParseResults, CharsNotIn, _noncomma, dblQuotedString, QuotedString, ParserElement,
+                        ParseException, ParseResults, CharsNotIn, dblQuotedString, QuotedString, ParserElement,
                         Suppress, Regex, removeQuotes)
 
 dot_keywords = ['graph', 'subgraph', 'digraph', 'node', 'edge', 'strict']
