diff -ur a/dot2tex/dotparsing.py b/dot2tex/dotparsing.py
--- a/dot2tex/dotparsing.py	2014-03-29 19:08:59.000000000 +0000
+++ b/dot2tex/dotparsing.py	2014-03-29 20:29:30.028437938 +0000
@@ -478,13 +478,13 @@
         subgraph = (Optional(subgraph_, '') + Optional(ID, '') + Group(graph_stmt)).setName("subgraph").setResultsName(
             'ssubgraph')
 
-        edge_point << (subgraph | graph_stmt | node_id )
+        edge_point <<= (subgraph | graph_stmt | node_id )
 
         node_stmt = (node_id + Optional(attr_list) + Optional(semi)).setName("node_stmt")
 
         assignment = (ID + equals + righthand_id).setName("assignment")
         stmt = (assignment | edge_stmt | attr_stmt | subgraph | graph_stmt | node_stmt).setName("stmt")
-        stmt_list << OneOrMore(stmt + Optional(semi))
+        stmt_list <<= OneOrMore(stmt + Optional(semi))
 
         graphparser = ( (Optional(strict_, 'notstrict') + ((graph_ | digraph_)) +
                          Optional(ID, '') + lbrace + Group(Optional(stmt_list)) + rbrace).setResultsName("graph") )
