--- a/Singular/normaliz.lib	2012-06-29 11:08:48.000000000 -0600
+++ b/Singular/normaliz.lib	2012-12-02 17:58:34.273516924 -0700
@@ -1371,7 +1374,8 @@
 
     //adjust size of input matrix
     if (!last_comp) { // remove last component
-        intmat tmp = expo_vecs[1..nrows(expo_vecs),1..(ncols(expo_vecs)-1)];
+        intmat tmp[nrows(expo_vecs)][ncols(expo_vecs)-1]
+               = expo_vecs[1..nrows(expo_vecs),1..(ncols(expo_vecs)-1)];
         expo_vecs = tmp;
     }
     intmat nmz_data=runNormaliz(expo_vecs,nmz_mode,prepareGrading(#));
