diff -druN src/source/libnormaliz/cone.cpp patches/source/libnormaliz/cone.cpp
--- src/source/libnormaliz/cone.cpp	2017-02-22 17:59:08.000000000 +0100
+++ patches/source/libnormaliz/cone.cpp	2017-04-26 16:33:57.460366489 +0200
@@ -3223,7 +3223,7 @@
     Deg1Elements=Matrix<Integer>(0,dim);
     ModuleGenerators=Matrix<Integer>(0,dim);
     
-    Matrix<Integer> Raw=ApproxCone.getDeg1ElementsMatrix();
+    const Matrix<Integer>& Raw=ApproxCone.getDeg1ElementsMatrix();
     Matrix<Integer> Result(0,dim);
     Matrix<Integer> Eq=BasisChange.getEquations();
     Matrix<Integer> Cong=BasisChange.getCongruences();
