--- src/source/libnormaliz/sublattice_representation.h	2018-03-05 08:44:58.532530287 +1000
+++ src/source/libnormaliz/sublattice_representation.h	2018-03-05 08:46:07.541587773 +1000
@@ -320,9 +320,9 @@
     assert(Supps.nr_of_rows()>0);
     size_t EmbDim=Supps.nr_of_columns();
     
-    if(Vertices.nr_of_rows()==0 || Vertices.rank()<EmbDim-1){ // use Supps for LLL coordinates    
+    if(Vertices.nr_of_rows()==0 || Vertices.rank()<EmbDim){ // use Supps for LLL coordinates    
         Matrix<nmz_float> SuppHelp=Supps.nmz_float_without_first_column();
-        if(Supps.rank()<EmbDim)
+        if(Supps.rank()<EmbDim-1)
             return;
         Sublattice_Representation<Integer> HelpCoord=LLL_coordinates_dual<Integer,nmz_float>(SuppHelp);
         convert(HelpA,HelpCoord.getEmbeddingMatrix()); convert(HelpB,HelpCoord.getProjectionMatrix()); convert(HelpC,HelpCoord.getAnnihilator());
