diff -ru src/configure b/configure
--- src/configure	2012-11-09 21:42:47.000000000 +0100
+++ b/configure	2013-05-08 12:12:08.233238871 +0200
@@ -262,13 +262,7 @@
 #various tuning parameters
 
 if [ -z "$FLINT_TUNE" ]; then
-   if [ "$KERNEL" = "Linux" -a "$ARCH" = "x86_64" ]; then
-      FLINT_TUNE="-funroll-loops"
-   elif [ "$KERNEL" = "Darwin" -a "$ARCH" = "ppc" ]; then
-      FLINT_TUNE=" -funroll-loops"
-   elif [ "`uname -p`" = "powerpc" ]; then
-      FLINT_TUNE="-m64 -mcpu=970 -mtune=970 -mpowerpc64 -falign-loops=16 -falign-functions=16 -falign-labels=16 -falign-jumps=16"
-   elif [ "$ARCH" = "ia64" ]; then
+   if [ "$ARCH" = "ia64" ]; then
       # -funroll-loops crashes the build on itanium under GCC-4.2.1, as reported by
       # Kate Minola.
       FLINT_TUNE=""
@@ -325,6 +319,11 @@
    CFLAGS="-O2 -g -ansi -pedantic -Wall $POPCNT_FLAG $ABI_FLAG"
 fi
 
+# Old Darwin versions (OS X 10.4) require -fno-common
+if [ "$KERNEL" = Darwin ]; then
+   CFLAGS="-fno-common $CFLAGS"
+fi
+
 #add tuning parameters to CFLAGS
 
 CFLAGS="$CFLAGS $FLINT_TUNE"
