diff -druN src.orig/configure src/configure
--- src.orig/configure	2012-11-09 21:42:47.000000000 +0100
+++ src/configure	2013-03-06 18:21:03.644037750 +0100
@@ -225,12 +225,7 @@
 if [ -z "$FLINT_LIB" ]; then
    case $OS in
       Darwin)
-         case $MACHINE in
-            *64)
-               FLINT_LIB="libflint.dylib64";;
-            *)
-               FLINT_LIB="libflint.dylib";;
-         esac;;
+         FLINT_LIB="libflint.dylib";;
       CYGWIN | MINGW*)
          FLINT_LIB="libflint.dll";;
       *)
