diff -Naur flint-2.4.5.orig/Makefile.in flint-2.4.5/Makefile.in
--- flint-2.4.5.orig/Makefile.in	2015-05-15 19:05:59.034143372 +1200
+++ flint-2.4.5/Makefile.in	2015-05-15 19:06:26.204204225 +1200
@@ -212,7 +212,7 @@
 	$(QUIET_CXX) $(CXX) $(CFLAGS) $(INCS) -c $< -o $@;
 
 build/interfaces/test/t-NTL-interface$(EXEEXT): interfaces/test/t-NTL-interface.cpp
-	$(QUIET_CXX) $(CXX) $(CFLAGS) $(INCS) $< build/interfaces/NTL-interface.o -o $@ $(LIBS);
+	$(QUIET_CXX) $(CXX) $(CFLAGS) $(INCS) $< build/interfaces/NTL-interface.lo -o $@ $(LIBS);
 
 print-%:
 	@echo '$*=$($*)'
