--- flint-1.5.0/fmpz-test.c	2011-08-25 04:15:09.000000000 +0200
+++ flint-1.5.2/fmpz-test.c	2010-04-09 01:29:21.000000000 +0200
@@ -1367,7 +1367,11 @@
    fmpz_t fnum1;
    unsigned long limbs, limbs2;
    int result = 1;
-   
+
+   gmp_randstate_t state;
+
+   gmp_randinit_default(state);
+
    mpz_init(num1);
    mpz_init(num2);
    
@@ -1376,7 +1380,7 @@
        limbs = random_ulong(1000);
        limbs2 = random_ulong(1000);
        fnum1 = fmpz_init(limbs+limbs2);
-       mpz_random(num1, limbs);
+       mpz_urandomb(num1, state, limbs*FLINT_BITS);
 #if SIGNS
        if (random_ulong(2)) mpz_neg(num1, num1);
 #endif
@@ -1397,7 +1401,8 @@
    
    mpz_clear(num1);
    mpz_clear(num2);
-   
+   gmp_randclear(state);
+
    return result;
 }
 
