commit 1e998bc74049ecb054dc86cf66a6f81e0d365d94
Author: Jean-Pierre Flori <jean-pierre.flori@ssi.gouv.fr>
Date:   Thu Jan 2 14:05:41 2014 +0100

    Fix NTL interface tests. Reported by Francois Bissey.

diff --git a/interfaces/test/t-NTL-interface.cpp b/interfaces/test/t-NTL-interface.cpp
index 9e25cd6..1beffc3 100644
--- a/interfaces/test/t-NTL-interface.cpp
+++ b/interfaces/test/t-NTL-interface.cpp
@@ -273,7 +273,7 @@ int test_ZZ_pE_to_fq()
         fmpz_mod_poly_init(fmod, p);
         fmpz_mod_poly_set_ZZ_pX(fmod, mod);
 
-        fq_ctx_init_modulus(ctx, p, d, fmod, "a");
+        fq_ctx_init_modulus(ctx, fmod, "a");
 
         ZZ_pE zzpe;
 
@@ -345,7 +345,7 @@ int test_ZZ_pEX_to_fq_poly()
         fmpz_mod_poly_init(fmod, p);
         fmpz_mod_poly_set_ZZ_pX(fmod, mod);
 
-        fq_ctx_init_modulus(ctx, p, d, fmod, "a");
+        fq_ctx_init_modulus(ctx, fmod, "a");
 
         ZZ_pEX zzpex;
 
@@ -417,7 +417,7 @@ int test_zz_pE_to_fq()
         fmpz_mod_poly_init(fmod, p);
         fmpz_mod_poly_set_zz_pX(fmod, mod);
 
-        fq_ctx_init_modulus(ctx, p, d, fmod, "a");
+        fq_ctx_init_modulus(ctx, fmod, "a");
 
         zz_pE zzpe;
 
@@ -487,7 +487,7 @@ int test_zz_pEX_to_fq_poly()
         fmpz_mod_poly_init(fmod, p);
         fmpz_mod_poly_set_zz_pX(fmod, mod);
 
-        fq_ctx_init_modulus(ctx, p, d, fmod, "a");
+        fq_ctx_init_modulus(ctx, fmod, "a");
 
         zz_pEX zzpex;
 
