diff -ru src/de.c b/de.c
--- src/de.c	2007-12-06 17:30:00.000000000 +0100
+++ b/de.c	2013-10-19 18:36:11.098758179 +0200
@@ -24,7 +24,7 @@
 OP cons_negeins;/* global INTEGER variable -1 */
 OP cons_null;   /* global INTEGER variable 0 */
 FILE *texout;   /* global variable for texoutput */
-INT no_banner = FALSE; /* AK 281293 */
+INT no_banner = TRUE; /* AK 281293 */
 INT no_mem_check=TRUE; /* AK 100893 */
 INT english_tableau=FALSE; /* AK 290995 */
 
