--- macro.h.orig	2009-05-15 04:31:26.000000000 -0700
+++ macro.h	2007-12-07 00:40:20.000000000 -0800
@@ -171,7 +170,7 @@
 
 
 
-extern INT partition_speichersize,partition_speicherindex,mem_counter_part;
+extern long partition_speichersize,partition_speicherindex,mem_counter_part;
 extern struct partition **partition_speicher; 
 
 #define FREEPARTITION(d)\
@@ -266,7 +265,7 @@
       C_MO_K(c,b); \
 } while(0)
 
-extern INT monom_speicherindex,mem_counter_monom,monom_speichersize;
+extern long monom_speicherindex,mem_counter_monom,monom_speichersize;
 extern struct monom **monom_speicher;
 
 #define FREEMONOM(v) \
@@ -554,10 +553,10 @@
 
 /* for longint */
 
-extern INT loc_index, loc_size,loc_counter;
+extern long loc_index, loc_size,loc_counter;
 extern struct loc **loc_speicher;
 
-extern INT longint_speicherindex,mem_counter_loc,longint_speichersize;
+extern long longint_speicherindex,mem_counter_loc,longint_speichersize;
 extern struct longint **longint_speicher;
 
 
