diff -ru src/part.c b/part.c
--- src/part.c	Thu May 14 06:01:00 2009 -0400
+++ b/part.c 	Thu May 14 06:01:00 2009 -0400
@@ -1767,8 +1767,8 @@
 /* to compute number of partitions */
 {
     INT erg = OK;
-    if (ni<0) return;
-    if (not EMPTYP(S_V_I(vec,ni))) return;
+    if (ni<0) return 0;
+    if (not EMPTYP(S_V_I(vec,ni))) return 0;
     else if (ni<=1) M_I_I(1,S_V_I(vec,ni));
     else {
     
