diff --git a/pGroupCohomology-3.3.2/pGroupCohomology/cohomology.pyx b/pGroupCohomology-3.3.2/pGroupCohomology/cohomology.pyx
index 8e37be9..828b6a2 100644
--- a/pGroupCohomology-3.3.2/pGroupCohomology/cohomology.pyx
+++ b/pGroupCohomology-3.3.2/pGroupCohomology/cohomology.pyx
@@ -3108,7 +3108,7 @@ class COHO(Ring):
         if self.Resl.coef()!=2: # non-commutative case
             singular.LIB('ncall.lib')
         singular.LIB('general.lib')
-        singular.LIB('poly.lib')
+        singular.LIB('polylib.lib')
         singular.LIB('dickson.lib')
         singular.eval('option(redSB)');
         singular.eval('int i')
@@ -3877,7 +3877,7 @@ class COHO(Ring):
             if p!=2:
                 singular.LIB("ncall.lib")
             singular.LIB('general.lib')
-            singular.LIB('poly.lib')
+            singular.LIB('polylib.lib')
             singular.LIB('dickson.lib')
             if singular.eval('defined(i)')=='0':
                 singular.eval('int i')
@@ -10062,7 +10062,7 @@ is an error. Please inform the author!""")
         if p!=2:
             singular.LIB("ncall.lib")
         singular.LIB('general.lib')
-        singular.LIB('poly.lib')
+        singular.LIB('polylib.lib')
         singular.LIB('dickson.lib')
         from pGroupCohomology.cochain import MODCOCH
         if singular.eval('defined(i)')=='0':
diff --git a/pGroupCohomology-3.3.2/pGroupCohomology/isomorphism_test.py b/pGroupCohomology-3.3.2/pGroupCohomology/isomorphism_test.py
index 37627c4..dd2bb85 100644
--- a/pGroupCohomology-3.3.2/pGroupCohomology/isomorphism_test.py
+++ b/pGroupCohomology-3.3.2/pGroupCohomology/isomorphism_test.py
@@ -218,7 +218,7 @@ class IsomorphismTest:
         i = self.singular._next_var_name()
         self.singular.eval('for (int %s=1; %s<=size(%s[2]); %s++){%s[2][%s]="@"+%s[2][%s];} kill %s;'%(i,i,L.name(),i,L.name(), i,L.name(),i,i))
         if self._domain.base_ring().characteristic()%2:
-            self.singular.LIB("poly.lib")
+            self.singular.LIB("polylib.lib")
             self._dim_command = 'GKdim'
             #self._R_elim = self.singular('gcRingSum(ring(%s))+%s'%(L.name(),self._SD.name()))
         else:
diff --git a/singular_helper/dickson.lib b/singular_helper/dickson.lib
index 3284920..dc1c69f 100644
--- a/singular_helper/dickson.lib
+++ b/singular_helper/dickson.lib
@@ -21,7 +21,7 @@
 //*****************************************************************************
 
 LIB "general.lib";
-LIB "poly.lib";
+LIB "polylib.lib";
 //LIB "ring.lib"; // for changeord
 
 // Auxiliar procedure, since NF in non-commutative rings now removes zeroes from ideals:
