--- src/factory/assert.h	2009-11-02 10:12:22.000000000 +0000
+++ src/factory/assert.h	2010-07-10 14:26:29.949490152 +0100
@@ -127,3 +127,13 @@
 #define PVIRT_CHARCC(msg) = 0
 #endif /* NOASSERT */
 /* }}} */
+
+
+
+/* SAGE hack: This is *critically* needed to compile on some systems
+   for which the logic above is broken.   Obviously it defeates the
+   purpose of assert here, but at least it means that the code compiles.
+*/
+#ifndef assert
+#define assert
+#endif
