diff -druN src.orig/Singular/Makefile.in src/Singular/Makefile.in
--- src.orig/Singular/Makefile.in	2012-07-11 12:00:13.000000000 +0200
+++ src/Singular/Makefile.in	2013-02-04 23:43:26.565194514 +0100
@@ -121,9 +121,9 @@
 else
 ## for cones & fans if under Win OS:
 ## put "-lgfan -lcddgmp " before "-Xlinker" in definition of LIBS and LIBSG
-LIBS		= -lsingfac -lsingcf -lntl  -static -lreadline -lhtmlhelp -Xlinker -Bdynamic -lgmp -lomalloc_ndebug -lncurses
+LIBS		= @NEED_LIBS@
 ## -lpython_module -lpython2.4 /usr/local/lib/libboost_python-gcc-d-1_32.dll
-LIBSG		= -lsingfac -lsingcf -lntl  -static -lreadline -lhtmlhelp -Xlinker -Bdynamic -lgmp -lncurses
+LIBSG		= @NEED_LIBSG@
 ## with cdd and gfan: LIBSG		= -lsingfac -lsingcf -lntl  -static -lreadline -lhtmlhelp -lgfan -lcddgmp -Xlinker -Bdynamic -lgmp -lncurses
 endif
 MP_LIBS		= @MP_LIBS@
