diff -druN singular-3.1.6.orig/latest/omalloc/configure.in singular-3.1.6/latest/omalloc/configure.in
--- latest/omalloc/configure.in	2012-12-19 13:01:16.000000000 -0800
+++ latest/omalloc/configure.in	2014-01-18 13:36:09.877532545 -0800
@@ -487,8 +487,12 @@
 int main()
 {
    int i;
+#if defined(__CYGWIN__) && defined(__x86_64__)
+   i = 0;
+#else
    omInitGetBackTrace();
    i = test_backtrace();
+#endif
    if (i == 1) exit(0);
    else exit(i+1);
 }
diff -druN singular-3.1.6.orig/latest/omalloc/configure singular-3.1.6/latest/omalloc/configure
--- latest/omalloc/configure	2012-12-19 13:01:16.000000000 -0800
+++ latest/omalloc/configure	2014-01-18 13:40:54.607561018 -0800
@@ -2330,8 +2330,12 @@
 int main()
 {
    int i;
+#if defined(__CYGWIN__) && defined(__x86_64__)
+   i = 0;
+#else
    omInitGetBackTrace();
    i = test_backtrace();
+#endif
    if (i == 1) exit(0);
    else exit(i+1);
 }
diff -druN singular-3.1.6.orig/latest/Singular/Makefile.in singular-3.1.6/latest/Singular/Makefile.in
--- latest/Singular/Makefile.in	2012-12-19 13:01:16.000000000 -0800
+++ latest/Singular/Makefile.in	2014-01-18 13:16:43.947415952 -0800
@@ -174,6 +174,13 @@
 LIBSINGULAR_LIBS = -lsingfac -lsingcf -lntl -lreadline -lgmp -lomalloc  -lhtmlhelp
 endif
 
+ifeq ($(SINGUNAME),x86_64-Win)
+SO_SUFFIX = dll
+MODULE_SUFFIX    = dll
+LIBSINGULAR_FLAGS = -shared
+LIBSINGULAR_LIBS = -lsingfac -lsingcf -lntl -lreadline -lgmp -lomalloc 
+endif
+
 ifeq ($(SINGUNAME),ix86-SunOS)
 SO_SUFFIX  = so
 MODULE_SUFFIX    = so
diff -druN singular-3.1.6.orig/latest/singuname.sh singular-3.1.6/latest/singuname.sh
--- latest/singuname.sh	2012-12-19 13:01:16.000000000 -0800
+++ latest/singuname.sh	2014-01-18 13:15:48.907410448 -0800
@@ -96,7 +96,12 @@
 elif (echo $uname_m | $egrep "x86_64" > $devnull)
 then
     prefix=x86_64
-    if (echo $uname_a | $egrep "Linux" > $devnull)
+    # Win ################
+    if (echo $uname_a | $egrep "CYGWIN" > $devnull)
+    then
+        echo ${prefix}-Win
+        exit 0
+    elif (echo $uname_a | $egrep "Linux" > $devnull)
     then
         echo ${prefix}-Linux
         exit 0
