--- src/factory/GNUmakefile.in	2011-09-01 15:40:30.000000000 +0200
+++ src/factory/GNUmakefile.in	2012-03-17 09:10:44.136311788 +0100
@@ -452,7 +452,7 @@
 # - pattern rules.
 #
 
-# how to create library objets
+# how to create library objects
 %.o: %.cc config.h factoryconf.h
 	$(CXX) -c $< $(LIBCXXFLAGS) -o $@
 %.o: %.c config.h
@@ -524,7 +524,7 @@
 $(srcdir)/readcf.cc: readcf.y
 		@./bin/gen-readcf "${BISON}"
 
-# note that these are non-source file which are distributed!
+# note that these are non-source files which are distributed!
 # We do not specify any dependencies except the most important ones.
 $(srcdir)/winnt/factory.h: factory.template
 		$(MAKEHEADER) $(MAKEHEADERFLAGS) $< $@
@@ -799,6 +799,6 @@
 # create cf_gmp.h
 cf_gmp.h: GNUmakefile gen_cf_gmp.template
 	echo "#!/bin/sh" >gen_cf_gmp.sh 
-	echo "GMP_H_T=\"`g++ $(LIBCXXFLAGS) -M gen_cf_gmp.cc|grep gmp.h` \"" >>gen_cf_gmp.sh 
+	echo "GMP_H_T=\"`$(CXX) $(LIBCXXFLAGS) -M gen_cf_gmp.cc|grep gmp.h` \"" >>gen_cf_gmp.sh 
 	cat gen_cf_gmp.template >>gen_cf_gmp.sh
 	$(SHELL) gen_cf_gmp.sh
