diff -druN a/latest/Singular/configure b/latest/Singular/configure
--- a/latest/Singular/configure	2014-08-06 08:59:15.000000000 -0700
+++ b/latest/Singular/configure	2014-10-21 02:26:41.030021194 -0700
@@ -9939,6 +9939,7 @@
 
 if test "x$flint_found" = "xyes"; then
   LDFLAGS="${LDFLAGS} ${FLINT_LIBS}"
+  SLDFLAGS="${SLDFLAGS} ${FLINT_LIBS}"
   NEED_LIBS="-lflint -lmpfr ${NEED_LIBS}"
 fi
 
diff -druN a/latest/Singular/configure.in b/latest/Singular/configure.in
--- a/latest/Singular/configure.in	2014-08-06 08:59:15.000000000 -0700
+++ b/latest/Singular/configure.in	2014-10-21 02:26:41.030021194 -0700
@@ -1431,6 +1431,7 @@
 
 if test "x$flint_found" = "xyes"; then
   LDFLAGS="${LDFLAGS} ${FLINT_LIBS}"
+  SLDFLAGS="${SLDFLAGS} ${FLINT_LIBS}"
   NEED_LIBS="-lflint -lmpfr ${NEED_LIBS}"
 fi
 
