diff -druN latest.orig/Singular/configure latest/Singular/configure
--- latest.orig/Singular/configure	2012-12-19 13:01:16.000000000 -0800
+++ latest/Singular/configure	2014-02-19 05:46:21.061190243 -0800
@@ -8989,6 +8989,7 @@
 if test "x$flint_found" = "xyes"; then
   includedir= "${FLINT_HOME}/include ${includedir}"
   LDFLAGS="${LDFLAGS} ${FLINT_LIBS}"
+  SLDFLAGS="${SLDFLAGS} ${FLINT_LIBS}"
   NEED_LIBS="-lflint -lmpfr ${NEED_LIBS}"
 fi
 
diff -druN latest.orig/Singular/configure.in latest/Singular/configure.in
--- latest.orig/Singular/configure.in	2012-12-19 13:01:16.000000000 -0800
+++ latest/Singular/configure.in	2014-02-19 05:45:48.911187028 -0800
@@ -1406,6 +1406,7 @@
 if test "x$flint_found" = "xyes"; then
   includedir= "${FLINT_HOME}/include ${includedir}"
   LDFLAGS="${LDFLAGS} ${FLINT_LIBS}"
+  SLDFLAGS="${SLDFLAGS} ${FLINT_LIBS}"
   NEED_LIBS="-lflint -lmpfr ${NEED_LIBS}"
 fi
 
