commit 1a13c7c54e8a386019b0bbf897243b395e01ebe5
Author: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date:   Sat Feb 3 09:21:08 2018 +0100

    Avoid some floor() calls

diff --git a/Singular/walk.cc b/Singular/walk.cc
index 08b7658..bf692d9 100644
--- a/Singular/walk.cc
+++ b/Singular/walk.cc
@@ -2302,7 +2302,7 @@ static intvec* MwalkNextWeightCC(intvec* curr_weight, intvec* target_weight,
       }
       for(j=0; j<nRing; j++)
       {
-        (*diff_weight1)[j] = floor(0.1*(*diff_weight1)[j] + 0.5);
+        (*diff_weight1)[j] = ((*diff_weight1)[j] + 5) / 10;
       }
     }
 
@@ -2317,7 +2317,7 @@ static intvec* MwalkNextWeightCC(intvec* curr_weight, intvec* target_weight,
       {
         (*curr_weight)[j] = (*diff_weight1)[j];
         j = MivAbsMaxArg(diff_weight1);
-        (*diff_weight1)[j] = floor(0.1*(*diff_weight1)[j] + 0.5);
+        (*diff_weight1)[j] = ((*diff_weight1)[j] + 5) / 10;
       }
     }
 
@@ -4560,7 +4560,7 @@ static intvec* MWalkRandomNextWeight(ideal G, intvec* orig_M, intvec* target_wei
           (*next_weight2)[i] = rand() % 60000 - 30000;
           weight_norm = weight_norm + (*next_weight2)[i]*(*next_weight2)[i];
         }
-        weight_norm = 1 + floor(sqrt(weight_norm));
+        weight_norm = 1 + static_cast<int>(sqrt(double(weight_norm)));
       }
       for(i=0; i<nV; i++)
       {
