diff -dru latest.orig/kernel/fast_maps.cc latest/kernel/fast_maps.cc
--- latest.orig/kernel/fast_maps.cc	2014-08-06 08:59:15.000000000 -0700
+++ latest/kernel/fast_maps.cc	2014-10-20 04:37:32.066824137 -0700
@@ -434,12 +434,17 @@
 static int maPoly_Substitute(macoeff c, poly p, ring dest_r)
 {
   // substitute the monomial: go through macoeff
-  int len=pLength(p);
+  int len;
+  BOOLEAN zero_div= (rField_is_Ring(dest_r) && !rField_is_Domain(dest_r));
+  if (!zero_div) len=pLength(p);
   int done=0;
   while (c!=NULL)
   {
     done++;
     poly t=pp_Mult_nn(p,c->n,dest_r);
+    #ifdef HAVE_RINGS
+    if (zero_div) len=pLength(t);
+    #endif
     sBucket_Add_p(c->bucket, t, len);
     c=c->next;
   }
diff -dru latest.orig/kernel/longrat.cc latest/kernel/longrat.cc
--- latest.orig/kernel/longrat.cc	2014-08-06 08:59:15.000000000 -0700
+++ latest/kernel/longrat.cc	2014-10-20 04:38:28.926958120 -0700
@@ -704,8 +704,6 @@
 #if defined(LDEBUG)
   u->debug=123456;
 #endif
-  assume(a->s==3);
-  assume(b->s==3);
   mpz_init_set(u->z,aa);
   /* u=u/b */
   u->s = 3;
diff -dru latest.orig/kernel/p_Mult_q.cc latest/kernel/p_Mult_q.cc
--- latest.orig/kernel/p_Mult_q.cc	2014-08-06 08:59:15.000000000 -0700
+++ latest/kernel/p_Mult_q.cc	2014-10-20 04:37:32.076824161 -0700
@@ -69,7 +69,7 @@
   assume(p != NULL && pNext(p) != NULL && q != NULL && pNext(q) != NULL);
   pAssume1(! pHaveCommonMonoms(p, q));
 #ifdef HAVE_RINGS
-  assume(!rField_is_Ring(currRing));
+  assume(!rField_is_Ring(currRing) || rField_is_Domain(currRing));
 #endif
   assume(lp >= 1 && lq >= 1);
   p_Test(p, r);
