--- src/Singular/Makefile.in    2012-07-11 11:00:13.000000000 +0100
+++ src/Singular/Makefile.in    2012-08-27 16:22:26.013159361 +0100
@@ -599,7 +599,7 @@
 	  ${INSTALL_PROGRAM}  $$file ${libdir}; \
 	done
 	${INSTALL_PROGRAM} libsingular.h ${includedir}
-	for file in subexpr.h tok.h grammar.h ipid.h lists.h ipshell.h attrib.h; do \
+	for file in subexpr.h tok.h grammar.h ipid.h lists.h ipshell.h attrib.h table.h; do \
 	sed -e "s:<kernel/:<singular/:" < $$file |sed -e "s:<Singular/:<singular/:"|sed -e "s:<omalloc/:<:"|sed -e "s:<factory/:<:" >${includedir}/singular/$$file ;\
 	done
