Continue reading input after an interrupt

diff -ru a/latest/kernel/oswrapper/feread.cc b/latest/kernel/oswrapper/feread.cc
--- a/latest/kernel/oswrapper/feread.cc	2016-08-19 16:56:16.000000000 +0200
+++ b/latest/kernel/oswrapper/feread.cc	2016-09-30 17:19:33.345251619 +0200
@@ -314,10 +314,25 @@
     fprintf(stdout,"%s",pr);
   }
   mflush();
+  errno = 0;  /* To detect errors from fgets() */
   char *line=fgets(s,size,stdin);
   if (line!=NULL)
+  {
     for (int i=strlen(line)-1;i>=0;i--) line[i]=line[i]&127;
-  return line;
+    return line;
+  }
+
+  /* NULL can mean various things... */
+  switch (errno)
+  {
+    case 0:         /* EOF */
+      return NULL;
+    case EINTR:     /* CTRL-C */
+      return strcpy(s, "\n");
+    default:        /* Other error */
+      fprintf(stderr, "fgets() failed");
+      return NULL;
+  }
 }
 
 /* ===================================================================*/
