--- ../src/IntegerProgramming/Makefile.in	2010-01-29 09:10:49.000000000 +0000
+++ IntegerProgramming-Makefile.in	2010-07-10 14:27:16.550740818 +0100
@@ -58,11 +58,11 @@
 
 install: $(MAIN1) $(MAIN2) $(MAIN3) $(MAIN4) $(LLL)
 	${MKINSTALLDIRS} ${bindir}
-	${INSTALL_PROGRAM} $(MAIN1) ${bindir}
-	${INSTALL_PROGRAM} $(MAIN2) ${bindir}
-	${INSTALL_PROGRAM} $(MAIN3) ${bindir}
-	${INSTALL_PROGRAM} $(MAIN4) ${bindir}
-	${INSTALL_PROGRAM} $(LLL) ${bindir}
+	${INSTALL_PROGRAM} $(MAIN1).exe ${bindir}
+	${INSTALL_PROGRAM} $(MAIN2).exe ${bindir}
+	${INSTALL_PROGRAM} $(MAIN3).exe ${bindir}
+	${INSTALL_PROGRAM} $(MAIN4).exe ${bindir}
+	${INSTALL_PROGRAM} $(LLL).exe ${bindir}
 
 install-bindist: $(MAIN1) $(MAIN2) $(MAIN3) $(MAIN4)
 	${MKINSTALLDIRS} ${install_bindir}
