--- src/kernel/Makefile.in	2011-09-28 17:34:37.000000000 +0100
+++ patches/kernel.Makefile.in	2011-09-28 20:34:57.060439241 +0100
@@ -49,7 +49,7 @@
 CXXFLAGS	= @CXXFLAGS@ ${PIPE} 
 CXXTEMPLFLAGS	= @CXXTEMPLFLAGS@
 CPPFLAGS	= -I${srcdir} -I.. -I@prefix@  @CPPFLAGS@ 
-DEFS		= -DNDEBUG -DOM_NDEBUG -D@SING_UNAME@ @DEFS@
+DEFS		= -DOM_NDEBUG -D@SING_UNAME@ @DEFS@
 LDFLAGS		= @LDFLAGS@
 LD_DYN_FLAGS	= @LD_DYN_FLAGS@
 SFLAGS		= @SFLAGS@
@@ -64,7 +64,7 @@
 LIBS		= -lm @NEED_LIBS@ 
 else
 # for the 2-0-* versions under Windows, we don't need gdbm, readline and ncurses
-LIBS		= -lsingfac -lsingcf -lntl -lgmp -lreadline -lncurses -lomalloc_ndebug
+LIBS		= -lsingfac -lsingcf -lntl -lgmp -lreadline -lncurses -lomalloc
 #LIBS		= -lsingfac -lsingcf -lgmp
 endif
 MP_LIBS		= @MP_LIBS@
