--- libfac/Makefile.in.orig	2010-09-19 13:54:38.489217458 +1200
+++ libfac/Makefile.in	2010-09-19 13:55:22.653711324 +1200
@@ -122,11 +122,11 @@
 OPTOBJ/tmpl_inst.og: factor/tmpl_inst.cc 
 	$(CXX) -g $(CPPFLAGS) $(DEFS) -c $< -o $@
 
-libfac:		libfac.a
+libfac:		OPTOBJ libfac.a
 
-libsingfac:	libsingfac.a
+libsingfac:	OPTOBJ libsingfac.a
 
-libsingfac_g:	libsingfac_g.a
+libsingfac_g:	OPTOBJ libsingfac_g.a
 
 libfac.a:	$(OPTFACTOROBJ) $(OPTCHARSETOBJ)
 		$(AR) ${LIBFACNAME}.a $(OPTFACTOROBJ) $(OPTCHARSETOBJ)
