--- Makefile.in.orig	2010-08-13 22:57:16.547892094 +1200
+++ Makefile.in	2010-08-13 23:04:02.824894172 +1200
@@ -75,12 +75,12 @@
 	echo "* do not forget to install Singular-@SINGULAR_VERSION@-share.tar.gz" 
 
 libsingular:
-	cd Singular; make libsingular
+	cd Singular; ${MAKE} libsingular
 
 install-libsingular: libsingular
-	(cd Singular && make install-libsingular);
-	(cd kernel && make install-libsingular);
-	(cd omalloc && make install-libsingular);
+	(cd Singular && ${MAKE} install-libsingular);
+	(cd kernel && ${MAKE} install-libsingular);
+	(cd omalloc && ${MAKE} install-libsingular);
 
 maintainer-install uninstall:
 	@ for SUBDIR in ${BUILD_SUBDIRS}; \
@@ -174,9 +174,9 @@
 test:
 	@echo "make Tst for the official Singular test suite"
 	@echo "this part here are unittest and still under development"
-	(cd Singular; make libsingular.a)
+	(cd Singular; ${MAKE} libsingular.a)
 	mkdir -p build
-	(cd build && cmake .. && make && make test)
+	(cd build && cmake .. && ${MAKE} && ${MAKE} test)
 
 ##############################################################
 # distribution targets
