diff -ur orig/Singular/configure latest/Singular/configure
--- orig/Singular/configure	2014-08-02 09:16:46.762463059 -0400
+++ latest/Singular/configure	2014-08-02 09:22:26.284787562 -0400
@@ -4745,15 +4745,6 @@
 # expand includedir
 test "x$includedir" = 'x${prefix}/include' && includedir="${exec_prefix}/include"
 
-# add to -I to CPPFLAGS and and -L to LDFLAGS, just to be sure
-if test "$ac_cv_c_cpp" = gcc3 ; then
-  LDFLAGS="-L${libdir} ${LDFLAGS}"
-  CPPFLAGS="-I${includedir} ${CPPFLAGS}"
-else
-  LDFLAGS="-L${libdir} ${LDFLAGS} -L/usr/local/lib"
-  CPPFLAGS="-I${includedir} ${CPPFLAGS} -I/usr/local/include"
-fi
-
 # this is a work-around to include the right term.h
 if test "$ac_cv_singuname" = SunOS-5; then
 CPPFLAGS="-I/usr/xpg4/include ${CPPFLAGS}"
diff -ur orig/Singular/configure.in latest/Singular/configure.in
--- orig/Singular/configure.in	2014-08-02 09:16:46.762463059 -0400
+++ latest/Singular/configure.in	2014-08-02 09:22:11.013528822 -0400
@@ -681,15 +681,6 @@
 # expand includedir
 test "x$includedir" = 'x${prefix}/include' && includedir="${exec_prefix}/include"
 
-# add to -I to CPPFLAGS and and -L to LDFLAGS, just to be sure
-if test "$ac_cv_c_cpp" = gcc3 ; then
-  LDFLAGS="-L${libdir} ${LDFLAGS}"
-  CPPFLAGS="-I${includedir} ${CPPFLAGS}"
-else
-  LDFLAGS="-L${libdir} ${LDFLAGS} -L/usr/local/lib"
-  CPPFLAGS="-I${includedir} ${CPPFLAGS} -I/usr/local/include"
-fi
-
 # this is a work-around to include the right term.h
 if test "$ac_cv_singuname" = SunOS-5; then
 CPPFLAGS="-I/usr/xpg4/include ${CPPFLAGS}"
