From 861b7899904e3bbe3e8fe49fc0dd11ef9f5f238e Mon Sep 17 00:00:00 2001
From: Hans Schoenemann <hannes@mathematik.uni-kl.de>
Date: Wed, 19 Oct 2016 12:20:04 +0200
Subject: [PATCH] fix: NTL <-> factory: port to NTL 10.1

---
 factory/NTLconvert.cc | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/latest/factory/NTLconvert.cc b/latest/factory/NTLconvert.cc
index 5e095d3..b48df10 100644
--- a/latest/factory/NTLconvert.cc
+++ b/latest/factory/NTLconvert.cc
@@ -505,8 +505,10 @@ convertZZ2CF (const ZZ & a)
     const long * rep =
 #if NTL_MAJOR_VERSION <= 6
       static_cast<long *>( a.rep );
-#else
+#elif NTL_MAJOR_VERSION <=9
       static_cast<long *>( a.rep.rep ); // what about NTL7?
+#else
+      (long*)( a.rep.rep );
 #endif
     long sizeofrep= rep[1];
     bool lessZero= false;
@@ -547,7 +549,6 @@ convertZZ2CF (const ZZ & a)
 
     result= CanonicalForm (cf_stringtemp2, 16);
     delete [] cf_stringtemp2;
-    return result;
   }
   return result;
 }
