diff -ru src/factory/cf_chinese.cc b/factory/cf_chinese.cc
--- src/factory/cf_chinese.cc	2012-06-15 20:00:10.000000000 +0200
+++ b/factory/cf_chinese.cc	2012-07-25 15:48:19.893359649 +0200
@@ -211,12 +211,12 @@
             ZZ NTLc= convertFacCF2NTLZZ (c);
             bool lessZero= (sign (NTLc) == -1);
             if (lessZero)
-              negate (NTLc, NTLc);
+              NTL::negate (NTLc, NTLc);
             ZZ NTLnum, NTLden;
             if (ReconstructRational (NTLnum, NTLden, NTLc, NTLq, bound, bound))
             {
               if (lessZero)
-                negate (NTLnum, NTLnum);
+                NTL::negate (NTLnum, NTLnum);
               CanonicalForm num= convertNTLZZX2CF (to_ZZX (NTLnum), Variable (1));
               CanonicalForm den= convertNTLZZX2CF (to_ZZX (NTLden), Variable (1));
               On (SW_RATIONAL);
