diff -ru src/Singular/Makefile.in b/Singular/Makefile.in
--- src/Singular/Makefile.in	2012-07-11 12:00:13.000000000 +0200
+++ b/Singular/Makefile.in	2013-04-05 10:49:00.627513150 +0200
@@ -158,8 +158,7 @@
 MACOSX_DEPLOYMENT_TARGET=10.4
 SO_SUFFIX        = dylib
 MODULE_SUFFIX    = bundle
-LIBSINGULAR_FLAGS = -single_module
-LIBSINGULAR_LD = $(LD)
+LIBSINGULAR_FLAGS = -shared -single_module
 endif
 
 ifeq ($(SINGUNAME),ix86-Win)
