commit eb4ae817a42927ba2ac7a0273e69d1d14a295809
Author: Hans Schoenemann <hannes@mathematik.uni-kl.de>
Date:   Fri Feb 2 12:24:48 2018 +0100

    fix: use pow(double,double) by explicit casting

diff --git a/libpolys/coeffs/ffields.cc b/libpolys/coeffs/ffields.cc
index 61d8eff..612ed2d 100644
--- a/libpolys/coeffs/ffields.cc
+++ b/libpolys/coeffs/ffields.cc
@@ -806,7 +806,7 @@ static BOOLEAN nfCoeffIsEqual (const coeffs r, n_coeffType n, void * parameter)
 {
   if (n==n_GF) {
     GFInfo* p = (GFInfo *)(parameter);
-    int c = pow (p->GFChar, p->GFDegree);
+    int c = (int)pow ((double)p->GFChar, (double)p->GFDegree);
     if ((c == r->m_nfCharQ) && (strcmp(n_ParameterNames(r)[0], p->GFPar_name) == 0))
       return TRUE;
   }
@@ -927,7 +927,7 @@ BOOLEAN nfInitChar(coeffs r,  void * parameter)
     return TRUE;
   }
 
-  int c = pow (p->GFChar, p->GFDegree);
+  int c = (int)pow ((double)p->GFChar, (double)p->GFDegree);
 
   nfReadTable(c, r);
 
