--- a/kernel/bigintmat.cc	2012-12-19 22:01:16.000000000 +0100
+++ b/kernel/bigintmat.cc	2013-08-26 22:21:06.839607851 +0200
@@ -341,7 +341,7 @@
     if (nl > colwid[cj])
     {
       StringSetS("");
-      int ci = floor(i/col);
+      int ci = floor((double)i/col);
       StringAppend("[%d,%d]", ci+1, cj+1);
       char *tmp = StringAppendS("");
       char * ph = omStrDup(tmp);
@@ -420,7 +420,7 @@
     int index = cols*i+j;
     if ((a[index] > sndlong) && (a[index] < l))
     {
-      int min = (int)(floor(log10(cols))+floor(log10(rows)))+5;
+      int min = (int)(floor(log10((double)cols))+floor(log10((double)rows)))+5;
       if ((a[index] < min) && (min < l))
         sndlong = min;
       else
@@ -429,7 +429,7 @@
   }
   if (sndlong == 0)
   {
-    int min = (int)(floor(log10(cols))+floor(log10(rows)))+5;
+    int min = (int)(floor(log10((double)cols))+floor(log10((double)rows)))+5;
     if (min < l)
       sndlong = min;
     else
@@ -501,7 +501,7 @@
       if (nl > colwid[cj])
       {
         StringSetS("");
-        int ci = floor(i/col);
+        int ci = floor((double)i/col); // actually the same as ci = i/col
         StringAppend("[%d,%d]", ci+1, cj+1);
         char *tmp = StringAppendS("");
         char * ph = omStrDup(tmp);
