--- src/kernel/bigintmat.cc	2012-06-20 17:00:07.000000000 +0200
+++ src/kernel/bigintmat.cc	2013-03-18 16:18:26.281905612 +0100
@@ -344,7 +344,7 @@
     int index = cols*i+j;
     if ((a[index] > sndlong) && (a[index] < l))
     {
-      int min = floor(log10(cols))+floor(log10(rows))+5;
+      int min = floor(log10((double)cols))+floor(log10((double)rows))+5;
       if ((a[index] < min) && (min < l))
         sndlong = min;
       else
@@ -353,7 +353,7 @@
   }
   if (sndlong == 0)
   {
-    int min = floor(log10(cols))+floor(log10(rows))+5;
+    int min = floor(log10((double)cols))+floor(log10((double)rows))+5;
     if (min < l)
       sndlong = min;
     else
@@ -424,7 +424,7 @@
       if (nl > colwid[cj])
       {
         StringSetS("");
-        int ci = floor(i/col);
+        int ci = floor((double)(i/col)); // actually the same as ci = i/col
         StringAppend("[%d,%d]", ci+1, cj+1);
         char *tmp = StringAppendS("");
         char * ph = omStrDup(tmp);
