From 7902222c6e73d452b3d75576bd9bc0140f5e827a Mon Sep 17 00:00:00 2001
From: Alexander Dreyer <alexander.dreyer@itwm.fraunhofer.de>
Date: Sat, 4 Aug 2012 21:53:32 +0200
Subject: [PATCH] fix: apply LDFLAGS fixes from svn#13210/git#b1dfafe to cygwin-specific flags (Sage's trac #12089; reported by Jean-Pierre Flori)

---
 Singular/Makefile.in |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Singular/Makefile.in b/Singular/Makefile.in
index c4a6e78..b907085 100644
--- a/Singular/Makefile.in
+++ b/Singular/Makefile.in
@@ -90,9 +90,9 @@ LDFLAGSG	= -L@prefix@/kernel -L../kernel -lkernel_g @LD_DYN_FLAGS1@ @LDFLAGS@
 LDFLAGSP	= -L@prefix@/kernel -L../kernel -lkernel_p @LD_DYN_FLAGS1@ @LDFLAGS@ 
 else
 ## -L/usr/local/lib -L@prefix@/modules/python --> PySingular
-LDFLAGS		= -L@prefix@/kernel -L/bin -lkernel @LDFLAGS@ -L/usr/local/lib -L../modules/python 
-LDFLAGSG	= -L@prefix@/kernel -L/bin -lkernel_g @LDFLAGS@ 
-LDFLAGSP	= -L@prefix@/kernel -L/bin -lkernel_p @LDFLAGS@ 
+LDFLAGS		= -L@prefix@/kernel -L../kernel -L/bin -lkernel @LDFLAGS@ -L/usr/local/lib -L../modules/python 
+LDFLAGSG	= -L@prefix@/kernel -L../kernel -L/bin -lkernel_g @LDFLAGS@ 
+LDFLAGSP	= -L@prefix@/kernel -L../kernel -L/bin -lkernel_p @LDFLAGS@ 
 endif
 LD_DYN_FLAGS1	= @LD_DYN_FLAGS1@
 LD_DYN_FLAGS2	= @LD_DYN_FLAGS2@
-- 
1.6.0.2

