diff -Naur Singular-3-1-5.orig/factory/gen_cf_gmp.cc Singular-3-1-5/factory/gen_cf_gmp.cc
--- Singular-3-1-5.orig/factory/gen_cf_gmp.cc	2009-11-02 11:12:22.000000000 +0100
+++ Singular-3-1-5/factory/gen_cf_gmp.cc	1970-01-01 01:00:00.000000000 +0100
@@ -1 +0,0 @@
-#include <gmp.h>
diff -Naur Singular-3-1-5.orig/factory/gen_cf_gmp.template Singular-3-1-5/factory/gen_cf_gmp.template
--- Singular-3-1-5.orig/factory/gen_cf_gmp.template	2009-11-02 11:12:22.000000000 +0100
+++ Singular-3-1-5/factory/gen_cf_gmp.template	1970-01-01 01:00:00.000000000 +0100
@@ -1,3 +0,0 @@
-GMP_H=`echo $GMP_H_T| sed -e 's/^.*gmp.cc//' -e 's/ .$//'`
-echo generating cf_gmp.h from $GMP_H
-cat $GMP_H | grep -v __GMP_DECLSPEC_XX |grep -v std::FILE > cf_gmp.h
diff -Naur Singular-3-1-5.orig/factory/GNUmakefile.in Singular-3-1-5/factory/GNUmakefile.in
--- Singular-3-1-5.orig/factory/GNUmakefile.in	2012-05-11 16:00:15.000000000 +0200
+++ Singular-3-1-5/factory/GNUmakefile.in	2013-06-24 23:08:48.317324421 +0200
@@ -61,6 +61,9 @@
 MKINSTALLDIRS =	@FACTORY_MKINSTALLDIRS@
 MAKEHEADER =	@FACTORY_MAKEHEADER@
 
+GREP =		@GREP@
+EGREP = 	@EGREP@
+
 #
 # - flags.
 #
@@ -363,7 +366,7 @@
 		$(distsrc) $(distincl) $(hdrtemplsrc) \
 		GNUmakefile.in configure.in config.h.in configure \
 		ChangeLog README NEWS INSTALL \
-		factory.cfg gen_cf_gmp.template \
+		factory.cfg \
 		examples/GNUmakefile examples/application.cc \
 		examples/factorize.cc examples/gcd.cc \
 		bin/folding.el bin/fold-docu.el \
@@ -693,9 +696,18 @@
 config.status:	configure
 		./config.status --recheck
 
-# create cf_gmp.h
-cf_gmp.h: GNUmakefile gen_cf_gmp.template
-	echo "#!/bin/sh" >gen_cf_gmp.sh 
-	echo "GMP_H_T=\"`$(CXX) $(LIBCXXFLAGS) -M gen_cf_gmp.cc|grep gmp.h` \"" >>gen_cf_gmp.sh 
-	cat gen_cf_gmp.template >>gen_cf_gmp.sh
-	$(SHELL) gen_cf_gmp.sh
+
+# Doesn't need to be shipped:
+gen_cf_gmp.cc:
+	echo '#include <gmp.h>' > $@
+
+# Create cf_gmp.h:
+# cf_gmp.cc just contains "#include <gmp.h>"; we use '-M' just to get the path
+# to gmp.h in order to create a modified copy of it (i.e., cf_gmp.h) below.
+# (If '--with-gmp=...' was given to 'configure', one could also just use the
+# path determined by that.)
+cf_gmp.h: GNUmakefile gen_cf_gmp.cc
+	GMP_H=`$(CXX) $(LIBCXXFLAGS) -M gen_cf_gmp.cc|tr ' ' '\n'|$(GREP) 'gmp\.h$$'`; \
+	echo "Generating $@ from $${GMP_H}..."; \
+	test -f "$${GMP_H}" || { echo >&2 "Error determining location of gmp.h"; exit 1; }; \
+	$(EGREP) -v '__GMP_DECLSPEC_XX|std::FILE' $${GMP_H} > $@
