From de688442dfe449992dc14a000bca0691ecc7e106 Mon Sep 17 00:00:00 2001
From: Oleksandr Motsak <motsak@mathematik.uni-kl.de>
Date: Sat, 14 Mar 2015 16:58:01 +0100
Subject: [PATCH] Fixing incompatibility with NTL8

---
 factory/NTLconvert.cc | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/latest/factory/NTLconvert.cc b/latest/factory/NTLconvert.cc
index 41ce9b2..e32093d 100644
--- a/latest/factory/NTLconvert.cc
+++ b/latest/factory/NTLconvert.cc
@@ -30,6 +30,7 @@
 #include <NTL/GF2EXFactoring.h>
 #include <NTL/tools.h>
 #include <NTL/mat_ZZ.h>
+#include <NTL/version.h>
 #include "int_int.h"
 #include <limits.h>
 #include "NTLconvert.h"
@@ -500,8 +501,14 @@ convertZZ2CF (const ZZ & a)
     return CanonicalForm(coeff_long);
   }
   else
-  {
-    long sizeofrep= ((long *) a.rep) [1];
+  { 
+    const long * rep =
+#if NTL_MAJOR_VERSION <= 6
+      static_cast<long *>( a.rep );
+#else
+      static_cast<long *>( a.rep.rep ); // what about NTL7?
+#endif
+    long sizeofrep= rep[1];
     bool lessZero= false;
     if (sizeofrep < 0)
     {
@@ -519,7 +526,7 @@ convertZZ2CF (const ZZ & a)
       cf_stringtemp_l= sizeofrep*sizeof(mp_limb_t)*2;
       cf_stringtemp= (unsigned char*) Alloc (cf_stringtemp_l);
     }
-    int cc= mpn_get_str (cf_stringtemp, 16, (mp_limb_t *) (((long *) (a.rep)) + 2), sizeofrep);
+    int cc= mpn_get_str (cf_stringtemp, 16, (mp_limb_t *) ((rep) + 2), sizeofrep);
 
     char* cf_stringtemp2;
     if (lessZero)
