--- src/Singular/configure	2011-09-30 12:10:15.000000000 +0200
+++ patches/Singular.configure	2011-10-02 15:16:49.468526670 +0200
@@ -1800,7 +1800,7 @@
 PYTHON_PLUGIN="module"
 PYTHON_LINKAGE="static"
 
-if test "${with_python+set}" == set; then
+if test "${with_python+set}" = set; then
   case "${with_python}" in
     yes|static*|dynamic*|shared*|module*|embed*)
       python_exe_name="python"
@@ -1971,7 +1971,7 @@
 		py_version=`$PYTHON -c "from distutils.sysconfig import *; \
                         from string import join; \
                         print join(get_config_vars('VERSION'))"`
-		if test "$py_version" == "None"; then
+		if test "$py_version" = "None"; then
 			if test -n "$PYTHON_VERSION"; then
 				py_version=$PYTHON_VERSION
 			else
@@ -2134,7 +2134,7 @@
 
         echo "$ac_t""$pythonstaticexists" 1>&6
 
-        if test "$pythonstaticexists" == yes ; then
+        if test "$pythonstaticexists" = yes ; then
           cat >> confdefs.h <<\EOF
 #define HAVE_STATIC_PYTHON 1
 EOF
