commit a315346ac7424b4a5427320fad1f4eb47dbb22ed
Author: Martin Lee <martinlee84@web.de>
Date:   Mon Jul 30 11:40:42 2012 +0200

    fix: tr 440

diff --git a/factory/facFqBivar.cc b/factory/facFqBivar.cc
index 79b968b..f526883 100644
--- a/factory/facFqBivar.cc
+++ b/factory/facFqBivar.cc
@@ -5928,7 +5928,7 @@ biFactorize (const CanonicalForm& F, const ExtensionInfo& info)
   
   //check trivial case
   if (degree (A) == 1 || degree (A, 1) == 1 || 
-      (size (A) == 2 && gcd (degree (A), degree (A,1)).isOne()))
+      (size (A) == 2 && igcd (degree (A), degree (A,1))==1))
   {
     factors.append (A);
 
commit e5e9d9dc87093af880507e242a15709368fdcf28
Author: Martin Lee <martinlee84@web.de>
Date:   Mon Jul 30 11:40:04 2012 +0200

    chg: new command igcd to compute integer gcd

diff --git a/factory/cf_util.cc b/factory/cf_util.cc
index 02ecb42..91c3303 100644
--- a/factory/cf_util.cc
+++ b/factory/cf_util.cc
@@ -46,6 +46,22 @@ int ilog2 (int a)
   return n;
 }
 
+int igcd( int a, int b )
+{
+    if ( a < 0 ) a = -a;
+    if ( b < 0 ) b = -b;
+
+    int c;
+
+    while ( b != 0 )
+    {
+        c = a % b;
+        a = b;
+        b = c;
+    }
+    return a;
+}
+
 #include<stdio.h>
 #include<stdlib.h>
 
diff --git a/factory/cf_util.h b/factory/cf_util.h
index 17baf35..e7309b0 100644
--- a/factory/cf_util.h
+++ b/factory/cf_util.h
@@ -13,6 +13,7 @@
 
 int ipower ( int b, int n );
 int ilog2 (int a);
+int igcd (int a, int b);
 
 /*BEGINPUBLIC*/
 void factoryError_intern(const char *s);
