commit 53f79f794aabd5451e090764871dde7c6020d5fa
Author: Martin Lee <martinlee84@web.de>
Date:   Mon Jul 30 11:20:00 2012 +0200

    fix: tr 439

diff --git a/factory/facBivar.h b/factory/facBivar.h
index cabdff8..2d8531e 100644
--- a/factory/facBivar.h
+++ b/factory/facBivar.h
@@ -183,7 +183,7 @@ ratBiFactorize (const CanonicalForm & G,         ///< [in] a bivariate poly
       {
         for (CFFListIterator i= result; i.hasItem(); i++)
         {
-          LcF /= bCommonDen (i.getItem().factor());
+          LcF /= power (bCommonDen (i.getItem().factor()), i.getItem().exp());
           i.getItem()= CFFactor (i.getItem().factor()*
                        bCommonDen(i.getItem().factor()), i.getItem().exp());
         }
@@ -215,7 +215,7 @@ ratBiFactorize (const CanonicalForm & G,         ///< [in] a bivariate poly
     {
       for (CFFListIterator i= result; i.hasItem(); i++)
       {
-        LcF /= bCommonDen (i.getItem().factor());
+        LcF /= power (bCommonDen (i.getItem().factor()), i.getItem().exp());
         i.getItem()= CFFactor (i.getItem().factor()*
                      bCommonDen(i.getItem().factor()), i.getItem().exp());
       }
diff --git a/factory/facFactorize.h b/factory/facFactorize.h
index 5ccfe16..95bfa66 100644
--- a/factory/facFactorize.h
+++ b/factory/facFactorize.h
@@ -136,7 +136,7 @@ ratFactorize (const CanonicalForm& G,          ///<[in] a multivariate poly
     {
       for (CFFListIterator i= result; i.hasItem(); i++)
       {
-        LcF /= bCommonDen (i.getItem().factor());
+        LcF /= power (bCommonDen (i.getItem().factor()), i.getItem().exp());
         i.getItem()= CFFactor (i.getItem().factor()*
                      bCommonDen(i.getItem().factor()), i.getItem().exp());
       }
