diff -ru src/Singular/sing_win.cc c/Singular/sing_win.cc
--- src/Singular/sing_win.cc	2012-07-10 12:00:15.000000000 +0200
+++ c/Singular/sing_win.cc	2012-08-01 12:31:32.753384989 +0200
@@ -37,7 +37,12 @@
    link.pszMsgTitle =  NULL ;
    link.pszWindow =    NULL ;
    link.fIndexOnFail = TRUE ;
-  HtmlHelpA(NULL, path, HH_KEYWORD_LOOKUP, (DWORD)&link);
+
+   /* Commented out, since this breaks building on *Cygwin* for SAGE,
+      which evidently isn't setup to open the standard Windows help
+      system.  Since we use Singular as a library anyways, this is
+      reasonable to comment out for now. */
+   /* HtmlHelpA(NULL, path, HH_KEYWORD_LOOKUP, (DWORD)&link); */
 }
 
 void heOpenWinntHlp(const char* keyw, char* helppath )
