--- ../src/Singular/Makefile.in	2010-06-24 14:10:14.000000000 +0100
+++ Singular.Makefile.in	2010-07-10 14:28:06.552750836 +0100
@@ -130,6 +130,12 @@
 LIBSINGULAR_LIBS = -lsingfac -lsingcf -lntl -lreadline -lgmp -lomalloc  -lhtmlhelp
 endif
 
+ifeq ($(SINGUNAME),ix86-SunOS)
+SO_SUFFIX  = so
+LIBSINGULAR_FLAGS = -shared
+LIBSINGULAR_LIBS = -lsingfac -lsingcf -lntl -lreadline -lgmp -lomalloc
+endif
+
 #
 # End libSINGULAR
 #
