--- src/Singular/Makefile.in	2011-09-28 17:36:50.000000000 +0100
+++ patches/Singular.Makefile.in	2011-09-28 20:34:39.588439234 +0100
@@ -82,10 +82,10 @@
 CPPFLAGS	= -I${srcdir} -I.. -I@prefix@ @CPPFLAGS@ @PYTHON_CPPFLAGS@
 ## -I/usr/include/python2.4 -I/usr/local/include -I../modules/python
 endif
-DEFS		= -DNDEBUG -DOM_NDEBUG -D@SING_UNAME@ @DEFS@
+DEFS		= -DOM_NDEBUG -D@SING_UNAME@ @DEFS@
 LDFLAGS2	= @LDFLAGS@
 ifneq ($(SINGUNAME),ix86-Win)
-LDFLAGS		= -L@prefix@/kernel -L../kernel -lkernel @LDFLAGS@ 
+LDFLAGS		= -L@prefix@/kernel -L../kernel -lkernel_g @LDFLAGS@ 
 LDFLAGSG	= -L@prefix@/kernel -L../kernel -lkernel_g @LD_DYN_FLAGS1@ @LDFLAGS@ 
 LDFLAGSP	= -L@prefix@/kernel -L../kernel -lkernel_p @LD_DYN_FLAGS1@ @LDFLAGS@ 
 else
