--- latest/Singular/libparse.l	2012-12-19 22:01:16.000000000 +0100
+++ latest/Singular/libparse.l	2014-05-17 18:24:37.707569492 +0200
@@ -963,7 +963,7 @@
 }
 
 #ifdef STANDALONE_PARSER
-main( int argc, char *argv[] )
+int main( int argc, char *argv[] )
 {
   lib_style_types lib_style;
   main_init(argc, argv);
--- latest/Singular/libparse.cc	2012-12-19 22:01:16.000000000 +0100
+++ latest/Singular/libparse.cc	2014-05-17 18:45:54.543560513 +0200
@@ -3504,7 +3504,7 @@
 }
 
 #ifdef STANDALONE_PARSER
-main( int argc, char *argv[] )
+int main( int argc, char *argv[] )
 {
   lib_style_types lib_style;
   main_init(argc, argv);
