commit 0246ba4f2c356ac24a96ef92762abf72bf775fdd
Author: Martin Lee <martinlee84@web.de>
Date:   Thu Aug 2 10:49:05 2012 +0200

    fix: tr 441

diff --git a/factory/facFqBivar.cc b/factory/facFqBivar.cc
index f526883..6fad188 100644
--- a/factory/facFqBivar.cc
+++ b/factory/facFqBivar.cc
@@ -6102,9 +6102,7 @@ biFactorize (const CanonicalForm& F, const ExtensionInfo& info)
       if (extension)
       {
         CFList source, dest;
-        ExtensionInfo info2= ExtensionInfo (beta, alpha, delta, gamma, k,
-                             info.getGFName(), info.isInExtension());
-        appendMapDown (factors, A, info2, source, dest);
+        appendMapDown (factors, A, info, source, dest);
       }
       else
         factors.append (A);
