diff -ru src/Singular/configure b/Singular/configure
--- src/Singular/configure	2012-07-11 12:00:13.000000000 +0200
+++ b/Singular/configure	2012-08-10 15:33:02.283359555 +0200
@@ -742,12 +742,8 @@
 LIBS
 CPPFLAGS
 CPP
-CPPFLAGS
 CXX
 CXXFLAGS
-LDFLAGS
-LIBS
-CPPFLAGS
 CCC
 CXXCPP'
 
@@ -5413,9 +5409,9 @@
 fi
 
 fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for  in -lnsl" >&5
-$as_echo_n "checking for  in -lnsl... " >&6; }
-if ${ac_cv_lib_nsl_+:} false; then :
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for gethostbyname in -lnsl" >&5
+$as_echo_n "checking for gethostbyname in -lnsl... " >&6; }
+if ${ac_cv_lib_nsl_gethostbyname+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
@@ -5429,27 +5425,27 @@
 #ifdef __cplusplus
 extern "C"
 #endif
-char  ();
+char gethostbyname ();
 int
 main ()
 {
-return  ();
+return gethostbyname ();
   ;
   return 0;
 }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_nsl_=yes
+  ac_cv_lib_nsl_gethostbyname=yes
 else
-  ac_cv_lib_nsl_=no
+  ac_cv_lib_nsl_gethostbyname=no
 fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_nsl_" >&5
-$as_echo "$ac_cv_lib_nsl_" >&6; }
-if test "x$ac_cv_lib_nsl_" = xyes; then :
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_nsl_gethostbyname" >&5
+$as_echo "$ac_cv_lib_nsl_gethostbyname" >&6; }
+if test "x$ac_cv_lib_nsl_gethostbyname" = xyes; then :
   cat >>confdefs.h <<_ACEOF
 #define HAVE_LIBNSL 1
 _ACEOF
@@ -5458,9 +5454,9 @@
 
 fi
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for  in -lsocket" >&5
-$as_echo_n "checking for  in -lsocket... " >&6; }
-if ${ac_cv_lib_socket_+:} false; then :
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for accept in -lsocket" >&5
+$as_echo_n "checking for accept in -lsocket... " >&6; }
+if ${ac_cv_lib_socket_accept+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
@@ -5474,27 +5470,27 @@
 #ifdef __cplusplus
 extern "C"
 #endif
-char  ();
+char accept ();
 int
 main ()
 {
-return  ();
+return accept ();
   ;
   return 0;
 }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_socket_=yes
+  ac_cv_lib_socket_accept=yes
 else
-  ac_cv_lib_socket_=no
+  ac_cv_lib_socket_accept=no
 fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_socket_" >&5
-$as_echo "$ac_cv_lib_socket_" >&6; }
-if test "x$ac_cv_lib_socket_" = xyes; then :
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_socket_accept" >&5
+$as_echo "$ac_cv_lib_socket_accept" >&6; }
+if test "x$ac_cv_lib_socket_accept" = xyes; then :
   cat >>confdefs.h <<_ACEOF
 #define HAVE_LIBSOCKET 1
 _ACEOF
@@ -5503,6 +5499,52 @@
 
 fi
 
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for sem_open in -lrt" >&5
+$as_echo_n "checking for sem_open in -lrt... " >&6; }
+if ${ac_cv_lib_rt_sem_open+:} false; then :
+  $as_echo_n "(cached) " >&6
+else
+  ac_check_lib_save_LIBS=$LIBS
+LIBS="-lrt  $LIBS"
+cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+
+/* Override any GCC internal prototype to avoid an error.
+   Use char because int might match the return type of a GCC
+   builtin and then its argument prototype would still apply.  */
+#ifdef __cplusplus
+extern "C"
+#endif
+char sem_open ();
+int
+main ()
+{
+return sem_open ();
+  ;
+  return 0;
+}
+_ACEOF
+if ac_fn_c_try_link "$LINENO"; then :
+  ac_cv_lib_rt_sem_open=yes
+else
+  ac_cv_lib_rt_sem_open=no
+fi
+rm -f core conftest.err conftest.$ac_objext \
+    conftest$ac_exeext conftest.$ac_ext
+LIBS=$ac_check_lib_save_LIBS
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_rt_sem_open" >&5
+$as_echo "$ac_cv_lib_rt_sem_open" >&6; }
+if test "x$ac_cv_lib_rt_sem_open" = xyes; then :
+  cat >>confdefs.h <<_ACEOF
+#define HAVE_LIBRT 1
+_ACEOF
+
+  LIBS="-lrt $LIBS"
+
+fi
+
+
 SAVE_LIBS=${LIBS}
 LIBS=
 
@@ -8978,55 +9020,6 @@
 $as_echo "no" >&6; }
 fi
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for  in -lrt" >&5
-$as_echo_n "checking for  in -lrt... " >&6; }
-if ${ac_cv_lib_rt_+:} false; then :
-  $as_echo_n "(cached) " >&6
-else
-  ac_check_lib_save_LIBS=$LIBS
-LIBS="-lrt  $LIBS"
-cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-/* Override any GCC internal prototype to avoid an error.
-   Use char because int might match the return type of a GCC
-   builtin and then its argument prototype would still apply.  */
-#ifdef __cplusplus
-extern "C"
-#endif
-char  ();
-int
-main ()
-{
-return  ();
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_rt_=yes
-else
-  ac_cv_lib_rt_=no
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-LIBS=$ac_check_lib_save_LIBS
-fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_rt_" >&5
-$as_echo "$ac_cv_lib_rt_" >&6; }
-if test "x$ac_cv_lib_rt_" = xyes; then :
-  cat >>confdefs.h <<_ACEOF
-#define HAVE_LIBRT 1
-_ACEOF
-
-  LIBS="-lrt $LIBS"
-
-fi
-
-if test "$ac_cv_lib_rt" = yes; then
-  NEED_LIBS="${NEED_LIBS} -lrt"
-fi
-
 NEED_LIBSG="${NEED_LIBS} -lpthread"
 
 if test "$ac_cv_sizeof_voidp" != 4; then
diff -ru src/Singular/configure.in b/Singular/configure.in
--- src/Singular/configure.in	2012-07-11 12:00:13.000000000 +0200
+++ b/Singular/configure.in	2012-08-10 15:32:54.943358842 +0200
@@ -799,8 +799,10 @@
 if test "$ac_cv_singuname" != ix86-Win; then
 AC_CHECK_LIB(m, atof)
 fi
-AC_CHECK_LIB(nsl)
-AC_CHECK_LIB(socket)
+AC_CHECK_LIB(nsl,gethostbyname)
+AC_CHECK_LIB(socket,accept)
+AC_CHECK_LIB(rt,sem_open)
+
 SAVE_LIBS=${LIBS}
 LIBS=
 
@@ -1438,11 +1440,6 @@
   AC_MSG_RESULT(no)
 fi
 
-AC_CHECK_LIB(rt)
-if test "$ac_cv_lib_rt" = yes; then
-  NEED_LIBS="${NEED_LIBS} -lrt"
-fi
-
 NEED_LIBSG="${NEED_LIBS} -lpthread"
 
 if test "$ac_cv_sizeof_voidp" != 4; then
