diff -ru a/IntegerProgramming/Makefile.in b/IntegerProgramming/Makefile.in
--- a/IntegerProgramming/Makefile.in	2012-12-19 22:01:16.000000000 +0100
+++ b/IntegerProgramming/Makefile.in	2013-08-26 22:44:59.909607832 +0200
@@ -5,7 +5,7 @@
 ##
 bindir          = @bindir@
 prefix          = @prefix@
-slibdir         = ${prefix}/LIB
+slibdir         = @datarootdir@/singular
 install_bindir  = ${install_prefix}/${SINGUNAME}
 libdir          = @libdir@
 
diff -ru a/Singular/Makefile.in b/Singular/Makefile.in
--- a/Singular/Makefile.in	2012-12-19 22:01:16.000000000 +0100
+++ b/Singular/Makefile.in	2013-08-26 22:44:59.909607832 +0200
@@ -48,7 +48,7 @@
 bindir		= @bindir@
 # includes are taken from here
 includedir	= @includedir@
-slibdir		= ${prefix}/LIB
+slibdir		= @datarootdir@/singular
 install_bindir  = ${install_prefix}/${SINGUNAME}
 install_slibdir = ${install_prefix}/LIB
 ifeq (${PYOBJECT_MODULE},)
Only in b/Singular: Makefile.in.orig
