diff -druN a/latest/IntegerProgramming/Makefile.in b/latest/IntegerProgramming/Makefile.in
--- a/latest/IntegerProgramming/Makefile.in	2014-10-21 05:59:54.327560770 -0700
+++ b/latest/IntegerProgramming/Makefile.in	2014-10-21 05:59:36.327521758 -0700
@@ -5,7 +5,7 @@
 ##
 bindir          = @bindir@
 prefix          = @prefix@
-slibdir         = ${prefix}/LIB
+slibdir         = @datarootdir@/singular
 install_bindir  = ${install_prefix}/${SINGUNAME}
 libdir          = @libdir@
 
diff -druN a/latest/Singular/Makefile.in b/latest/Singular/Makefile.in
--- a/latest/Singular/Makefile.in	2014-10-21 05:58:50.087421525 -0700
+++ b/latest/Singular/Makefile.in	2014-10-21 05:59:36.327521758 -0700
@@ -48,7 +48,7 @@
 bindir		= @bindir@
 # includes are taken from here
 includedir	= @includedir@
-slibdir		= ${prefix}/LIB
+slibdir		= @datarootdir@/singular
 install_bindir  = ${install_prefix}/${SINGUNAME}
 install_slibdir = ${install_prefix}/LIB
 dir_mod		= MOD
