--- ../Singular-3-1-0/Singular/Makefile.in	2009-04-23 09:46:09.000000000 +0100
+++ src.Singular.Makefile.in	2009-05-12 18:25:18.000000000 +0100
@@ -101,7 +101,7 @@
 #
 # Handle libSINGULAR stuff
 #
-LIBSINGULAR_LIBS =-lsingfac -lsingcf -lntl -lreadline -lgmp -lomalloc_ndebug
+LIBSINGULAR_LIBS =-lsingfac -lsingcf -lntl -lreadline -lgmp -lomalloc
 
 SO_SUFFIX = so
 #LIBSINGULAR_FLAGS = -export-dynamic 
@@ -123,10 +123,17 @@
 
 ifeq ($(SINGUNAME),ix86-Win)
 SO_SUFFIX = dll
-LIBSINGULAR_FLAGS = --shared
+LIBSINGULAR_FLAGS = -shared
 LIBSINGULAR_LIBS = -lsingfac -lsingcf -lntl -lreadline -lgmp -lomalloc_ndebug  -lhtmlhelp
 endif
 
+ifeq ($(SINGUNAME),ix86-SunOS)
+SO_SUFFIX  = so
+LIBSINGULAR_FLAGS = -shared
+LIBSINGULAR_LIBS = -lsingfac -lsingcf -lntl -lreadline -lgmp -lomalloc_ndebug
+endif
+
+
 #
 # End libSINGULAR
 #
