--- ../src/Singular/Makefile.in	2009-05-20 10:44:37.000000000 +0100
+++ src.Singular.Makefile.in	2009-06-11 11:23:38.000000000 +0100
@@ -130,6 +130,12 @@
 LIBSINGULAR_LIBS = -lsingfac -lsingcf -lntl -lreadline -lgmp -lomalloc  -lhtmlhelp
 endif
 
+ifeq ($(SINGUNAME),ix86-SunOS)
+SO_SUFFIX  = so
+LIBSINGULAR_FLAGS = -shared
+LIBSINGULAR_LIBS = -lsingfac -lsingcf -lntl -lreadline -lgmp -lomalloc
+endif
+
 #
 # End libSINGULAR
 #
