From 320ecea8996a062317ac7790eebe17f5f3f95551 Mon Sep 17 00:00:00 2001
From: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date: Tue, 2 Aug 2022 16:20:50 -0700
Subject: [PATCH] Project.platform_tag: Handle the case of 32bit Python on
 x86_64

---
 mesonpy/__init__.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/mesonpy/__init__.py b/mesonpy/__init__.py
index aff3dd1..bdcda25 100644
--- a/mesonpy/__init__.py
+++ b/mesonpy/__init__.py
@@ -684,6 +684,11 @@ def platform_tag(self) -> str:
             # https://github.com/pypa/packaging/issues/578
             parts[1] += '.0'
             platform_ = '-'.join(parts)
+        elif parts[0] == 'linux' and parts[1] == 'x86_64' and sys.maxsize == 2147483647:
+            # 32-bit Python running on an x86_64 host
+            # https://github.com/FFY00/meson-python/issues/123
+            parts[1] = 'i686'
+            platform_ = '-'.join(parts)
         return platform_.replace('-', '_').replace('.', '_')
 
     def _calculate_file_abi_tag_heuristic_windows(self, filename: str) -> Optional[mesonpy._tags.Tag]:
