diff -ur src/lanczos.h b/lanczos.h
--- src/lanczos.h	2007-05-06 00:52:39.000000000 +0200
+++ b/lanczos.h	2012-05-26 06:17:05.784874818 +0200
@@ -21,6 +21,13 @@
 #ifndef LANCZOS_H
 #define LANCZOS_H
 
+#ifdef __sun
+#define u_int32_t unsigned int
+#define u_int64_t unsigned long long
+#endif
+
+#include <sys/types.h> // needed on MacOS X 10.5 for uint type
+
 #include <stdlib.h>
 
 typedef struct {
