diff -ru src/QS.cpp b/QS.cpp
--- src/QS.cpp	2008-04-14 23:16:38.000000000 +0200
+++ b/QS.cpp	2012-05-26 06:30:15.823360044 +0200
@@ -467,7 +467,7 @@
      primeSizes = (unsigned char *) calloc(sizeof(unsigned char),numPrimes);
      for (unsigned long i = 0; i<numPrimes; i++)
      {
-         primeSizes[i]=(unsigned char)floor(log(factorBase[i])/log(2.0)-FUDGE+0.5);
+         primeSizes[i]=(unsigned char)floor(log((double)factorBase[i])/log(2.0)-FUDGE+0.5);
      }
      
      return;
