commit 78e6dcf937c960c51132132e14f86bddbbe5b7d9
Author: Jeroen Demeyer <J.Demeyer@UGent.be>
Date:   Tue Feb 19 11:50:49 2019 +0100

    Use the trashcan for Gen

diff --git a/cypari2/gen.pxd b/cypari2/gen.pxd
index 2ac0669..664d57d 100644
--- a/cypari2/gen.pxd
+++ b/cypari2/gen.pxd
@@ -1,3 +1,4 @@
+cimport cython
 from cpython.object cimport PyObject
 from .types cimport GEN, pari_sp
 
diff --git a/cypari2/gen.pyx b/cypari2/gen.pyx
index d268cf1..687f5b2 100644
--- a/cypari2/gen.pyx
+++ b/cypari2/gen.pyx
@@ -135,6 +135,7 @@ cdef extern from *:
     GEN new_nfeltup(GEN nf, GEN x, GEN zknf)
 
 
+@cython.trashcan(True)
 cdef class Gen(Gen_base):
     """
     Wrapper for a PARI ``GEN`` with memory management.
diff --git a/cypari2/pari_instance.pyx b/cypari2/pari_instance.pyx
index ba82438..d13bc45 100644
--- a/cypari2/pari_instance.pyx
+++ b/cypari2/pari_instance.pyx
@@ -243,6 +243,15 @@ Reset default precision for the following tests:
 
 >>> pari.set_real_precision_bits(53)
 
+Test the trashcan mechanism (without the trashcan, this would cause
+a stack overflow):
+
+>>> pari.allocatemem(2**27, silent=True)
+>>> L = [pari(i) for i in range(2**20)]
+>>> x = pari.Pi()
+>>> del L
+>>> del x
+
 Test that interrupts work properly:
 
 >>> pari.allocatemem(8000000, 2**29)
