--- src/sphinx/ext/autodoc.py	2010-07-24 03:07:36.000000000 -0700
+++ patches/autodoc.py	2010-11-05 04:51:44.248635318 -0700
@@ -828,8 +828,13 @@
     def format_args(self):
         if inspect.isbuiltin(self.object) or \
                inspect.ismethoddescriptor(self.object):
-            # can never get arguments of a C function or method
-            return None
+            # can never get arguments of a C function or method unless
+            # a function to do so is supplied
+            if self.env.config.autodoc_builtin_argspec:
+                argspec = self.env.config.autodoc_builtin_argspec(self.object)
+                return inspect.formatargspec(*argspec)
+            else:
+                return None
         try:
             argspec = inspect.getargspec(self.object)
         except TypeError:
@@ -1026,9 +1031,14 @@
     def format_args(self):
         if inspect.isbuiltin(self.object) or \
                inspect.ismethoddescriptor(self.object):
-            # can never get arguments of a C function or method
-            return None
-        argspec = inspect.getargspec(self.object)
+            # can never get arguments of a C function or method unless
+            # a function to do so is supplied
+            if self.env.config.autodoc_builtin_argspec:
+                argspec = self.env.config.autodoc_builtin_argspec(self.object)
+            else:
+                return None
+        else:
+            argspec = inspect.getargspec(self.object)
         if argspec[0] and argspec[0][0] in ('cls', 'self'):
             del argspec[0][0]
         return inspect.formatargspec(*argspec)
@@ -1201,6 +1211,7 @@
 
     app.add_config_value('autoclass_content', 'class', True)
     app.add_config_value('autodoc_member_order', 'alphabetic', True)
+    app.add_config_value('autodoc_builtin_argspec', None, True)
     app.add_config_value('autodoc_default_flags', [], True)
     app.add_event('autodoc-process-docstring')
     app.add_event('autodoc-process-signature')
