--- ../src/sphinx/ext/autodoc.py	2009-09-03 15:02:12.000000000 -0700
+++ autodoc.py	2009-09-03 16:19:26.000000000 -0700
@@ -645,8 +645,8 @@
         try:
             sig = self.format_signature()
         except Exception, err:
-            self.directive.warn('error while formatting signature for '
-                                '%s: %s' % (self.fullname, err))
+            # self.directive.warn('error while formatting signature for '
+            #                     '%s: %s' % (self.fullname, err))
             sig = ''
 
         # generate the directive header and options, if applicable
