--- ../src/sphinx/ext/autodoc.py	2009-09-04 05:02:12.000000000 +0700
+++ autodoc.py	2009-12-15 16:29:49.000000000 +0700
@@ -798,8 +798,13 @@
     def format_args(self):
         if inspect.isbuiltin(self.object) or \
                inspect.ismethoddescriptor(self.object):
-            # can never get arguments of a C function or method
-            return None
+            # can never get arguments of a C function or method unless
+            # a function to do so is supplied
+            if self.env.config.autodoc_builtin_argspec:
+                argspec = self.env.config.autodoc_builtin_argspec(self.object)
+                return inspect.formatargspec(*argspec)
+            else:
+                return None
         try:
             argspec = inspect.getargspec(self.object)
         except TypeError:
@@ -993,9 +998,14 @@
     def format_args(self):
         if inspect.isbuiltin(self.object) or \
                inspect.ismethoddescriptor(self.object):
-            # can never get arguments of a C function or method
-            return None
-        argspec = inspect.getargspec(self.object)
+            # can never get arguments of a C function or method unless
+            # a function to do so is supplied
+            if self.env.config.autodoc_builtin_argspec:
+                argspec = self.env.config.autodoc_builtin_argspec(self.object)
+            else:
+                return None
+        else:
+            argspec = inspect.getargspec(self.object)
         if argspec[0] and argspec[0][0] in ('cls', 'self'):
             del argspec[0][0]
         return inspect.formatargspec(*argspec)
@@ -1120,6 +1130,7 @@
 
     app.add_config_value('autoclass_content', 'class', True)
     app.add_config_value('autodoc_member_order', 'alphabetic', True)
+    app.add_config_value('autodoc_builtin_argspec', None, True)
     app.add_event('autodoc-process-docstring')
     app.add_event('autodoc-process-signature')
     app.add_event('autodoc-skip-member')
