--- Sphinx-1.2.2/sphinx/ext/autodoc.py.orig	2014-03-02 20:38:09.000000000 +1300
+++ Sphinx-1.2.2/sphinx/ext/autodoc.py	2014-10-19 23:02:09.000000000 +1300
@@ -1043,8 +1043,13 @@
     def format_args(self):
         if inspect.isbuiltin(self.object) or \
                 inspect.ismethoddescriptor(self.object):
-            # cannot introspect arguments of a C function or method
-            return None
+            # can never get arguments of a C function or method unless
+            # a function to do so is supplied
+            if self.env.config.autodoc_builtin_argspec:
+                argspec = self.env.config.autodoc_builtin_argspec(self.object)
+                return inspect.formatargspec(*argspec)
+            else:
+                return None
         try:
             argspec = getargspec(self.object)
         except TypeError:
@@ -1278,9 +1283,14 @@
     def format_args(self):
         if inspect.isbuiltin(self.object) or \
                 inspect.ismethoddescriptor(self.object):
-            # can never get arguments of a C function or method
-            return None
-        argspec = getargspec(self.object)
+            # can never get arguments of a C function or method unless
+            # a function to do so is supplied
+            if self.env.config.autodoc_builtin_argspec:
+                argspec = self.env.config.autodoc_builtin_argspec(self.object)
+            else:
+                return None
+        else:
+            argspec = getargspec(self.object)
         if argspec[0] and argspec[0][0] in ('cls', 'self'):
             del argspec[0][0]
         return inspect.formatargspec(*argspec)
@@ -1518,6 +1528,7 @@
 
     app.add_config_value('autoclass_content', 'class', True)
     app.add_config_value('autodoc_member_order', 'alphabetic', True)
+    app.add_config_value('autodoc_builtin_argspec', None, True)
     app.add_config_value('autodoc_default_flags', [], True)
     app.add_config_value('autodoc_docstring_signature', True, True)
     app.add_event('autodoc-process-docstring')
