diff -ur src/sphinx/environment.py src.patched/sphinx/environment.py
--- src/sphinx/environment.py	2010-08-24 01:19:38.000000000 +0200
+++ src.patched/sphinx/environment.py	2010-11-28 12:02:00.998657620 +0100
@@ -469,10 +469,13 @@
         else:
             # check if a config value was changed that affects how
             # doctrees are read
+            import inspect
             for key, descr in config.values.iteritems():
                 if descr[1] != 'env':
                     continue
                 if self.config[key] != config[key]:
+                    if inspect.isfunction(config[key]):
+                        continue
                     msg = '[config changed] '
                     config_changed = True
                     break
