--- src/sphinx/environment.py	2010-08-23 16:19:38.000000000 -0700
+++ src/sphinx/environment.py	2010-11-05 02:27:34.128635428 -0700
@@ -469,10 +469,13 @@
         else:
             # check if a config value was changed that affects how
             # doctrees are read
+            import inspect
             for key, descr in config.values.iteritems():
                 if descr[1] != 'env':
                     continue
                 if self.config[key] != config[key]:
+                    if inspect.isfunction(config[key]):
+                        continue
                     msg = '[config changed] '
                     config_changed = True
                     break
