--- src/sphinx/writers/latex.py	2010-08-23 09:47:18.000000000 -0700
+++ patches/latex.py	2010-11-06 01:59:01.068638964 -0700
@@ -32,7 +32,6 @@
 \def\sphinxdocclass{%(docclass)s}
 \documentclass[%(papersize)s,%(pointsize)s%(classoptions)s]{%(wrapperclass)s}
 %(inputenc)s
-%(utf8extra)s
 %(fontenc)s
 %(babel)s
 %(fontpkg)s
@@ -129,7 +128,6 @@
         'pointsize':       '10pt',
         'classoptions':    '',
         'inputenc':        '\\usepackage[utf8]{inputenc}',
-        'utf8extra':       '\\DeclareUnicodeCharacter{00A0}{\\nobreakspace}',
         'fontenc':         '\\usepackage[T1]{fontenc}',
         'babel':           '\\usepackage{babel}',
         'fontpkg':         '\\usepackage{times}',
