diff -ur src/sphinx/ext/pngmath.py src.patched/sphinx/ext/pngmath.py
--- src/sphinx/ext/pngmath.py	2010-07-24 12:07:36.000000000 +0200
+++ src.patched/sphinx/ext/pngmath.py	2010-11-28 12:00:17.938636725 +0100
@@ -35,7 +35,7 @@
 
 DOC_HEAD = r'''
 \documentclass[12pt]{article}
-\usepackage[utf8x]{inputenc}
+\usepackage[utf8]{inputenc}
 \usepackage{amsmath}
 \usepackage{amsthm}
 \usepackage{amssymb}
