See https://github.com/sphinx-doc/sphinx/pull/2534

diff --git a/sphinx/util/inspect.py b/sphinx/util/inspect.py
index 5dc2008..dbd578e 100644
--- a/sphinx/util/inspect.py
+++ b/sphinx/util/inspect.py
@@ -20,7 +20,7 @@ from sphinx.util import force_decode
 # relatively import this module
 inspect = __import__('inspect')
 
-memory_address_re = re.compile(r' at 0x[0-9a-f]{8,16}(?=>$)')
+memory_address_re = re.compile(r' at 0x[0-9a-f]{8,16}(?=>)')
 
 
 if PY3:
