commit b1fbd4cd3118a5f8b4c0790ac843ac4da7d09864
Author: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date:   Tue Mar 14 18:56:18 2017 +0100

    Allow running Sphinx without ssl

diff --git a/sphinx/util/requests.py b/sphinx/util/requests.py
index ec5aa23..fcce42f 100644
--- a/sphinx/util/requests.py
+++ b/sphinx/util/requests.py
@@ -36,30 +36,34 @@ except ImportError:
         # for requests < 2.4.0
         InsecureRequestWarning = None
 
-# try to load requests[security]
+# try to load requests[security] (but only if SSL is available)
 try:
-    pkg_resources.require(['requests[security]'])
-except (pkg_resources.DistributionNotFound,
-        pkg_resources.VersionConflict):
     import ssl
-    if not getattr(ssl, 'HAS_SNI', False):
-        # don't complain on each url processed about the SSL issue
-        requests.packages.urllib3.disable_warnings(
-            requests.packages.urllib3.exceptions.InsecurePlatformWarning)
+except ImportError:
+    pass
+else:
+    try:
+        pkg_resources.require(['requests[security]'])
+    except (pkg_resources.DistributionNotFound,
+            pkg_resources.VersionConflict):
+        if not getattr(ssl, 'HAS_SNI', False):
+            # don't complain on each url processed about the SSL issue
+            requests.packages.urllib3.disable_warnings(
+                requests.packages.urllib3.exceptions.InsecurePlatformWarning)
+            warnings.warn(
+                'Some links may return broken results due to being unable to '
+                'check the Server Name Indication (SNI) in the returned SSL cert '
+                'against the hostname in the url requested. Recommended to '
+                'install "requests[security]" as a dependency or upgrade to '
+                'a python version with SNI support (Python 3 and Python 2.7.9+).'
+            )
+    except pkg_resources.UnknownExtra:
         warnings.warn(
             'Some links may return broken results due to being unable to '
             'check the Server Name Indication (SNI) in the returned SSL cert '
             'against the hostname in the url requested. Recommended to '
-            'install "requests[security]" as a dependency or upgrade to '
-            'a python version with SNI support (Python 3 and Python 2.7.9+).'
+            'install requests-2.4.1+.'
         )
-except pkg_resources.UnknownExtra:
-    warnings.warn(
-        'Some links may return broken results due to being unable to '
-        'check the Server Name Indication (SNI) in the returned SSL cert '
-        'against the hostname in the url requested. Recommended to '
-        'install requests-2.4.1+.'
-    )
 
 if False:
     # For type annotation
