commit f4c8115a7e5f1a5d9ba10279b5a7634f63c18526
Author: Jeroen Demeyer <jdemeyer@cage.ugent.be>
Date:   Fri Sep 8 11:42:35 2017 +0200

    Turn any environment unpickling error into IOError

diff --git a/sphinx/environment/__init__.py b/sphinx/environment/__init__.py
index 536aa95..a3a3e3a 100644
--- a/sphinx/environment/__init__.py
+++ b/sphinx/environment/__init__.py
@@ -110,7 +110,12 @@ class BuildEnvironment(object):
     @staticmethod
     def load(f, app=None):
         # type: (IO, Sphinx) -> BuildEnvironment
-        env = pickle.load(f)
+        try:
+            env = pickle.load(f)
+        except Exception as exc:
+            # This can happen for example when the pickle is from a
+            # different version of Sphinx.
+            raise IOError(exc)
         if env.version != ENV_VERSION:
             raise IOError('build environment version not current')
         if app:
