diff --git a/Makefile.am b/Makefile.am
index a194bd5..1038934 100755
--- a/Makefile.am
+++ b/Makefile.am
@@ -8,8 +8,12 @@ libpolybori_la_SOURCES =
 libpolybori_la_LIBADD = \
 	Cudd/cudd/libcudd.la \
 	libpolybori/src/libpolybori_base.la
+libpolybori_la_LDFLAGS = -no-undefined $(AM_LDFLAGS)
 
 EXTRA_DIST = \
 	Cudd/LICENSE \
 	LICENSE \
 	README
+# Dummy C++ source to cause C++ linking.
+nodist_EXTRA_libpolybori_la_SOURCES = dummy.cc
+ 
diff --git a/groebner/src/Makefile.am b/groebner/src/Makefile.am
index 68c7df5..d044ab8 100755
--- a/groebner/src/Makefile.am
+++ b/groebner/src/Makefile.am
@@ -10,6 +10,8 @@ libpolybori_groebner_la_LIBADD = \
 	$(M4RI_LIBS) \
 	$(GD_LIBS)
 
+libpolybori_groebner_la_LDFLAGS = -no-undefined $(AM_LDFLAGS)
+
 libpolybori_groebner_la_SOURCES = \
 	dlex4data.cc \
 	dp_asc4data.cc \

