--- /home/jonathan/Documents/jmol-12.3.27.p0/src/jmol/Jmol.js
+++ /home/jonathan/Documents/jmol-12.3.27.p0/patches/Jmol.js
@@ -81,11 +81,6 @@
 // bh 4/2010  -- added jmolSetMemoryMb(nMb)
 // ah 1/2011  -- wider detection of browsers; more browsers now use the object tag instead of the applet tag; 
 //               fix of object tag (removed classid) accounts for change of behavior in Chrome
-// bh 3/2011  -- added jmolLoadAjax_STOLAF_NIH
-// ah 9/2011  -- Applet is now wrapped in a <span> tag (might break existing user code).
-//               Added jmolSwitchToSignedApplet(); replaces an applet with the signed applet, 
-//               preserving size, model and state; an additional script may be specified. 
-//               Note: as a result, unsigned and signed applets may coexist in a page.
 
 var defaultdir = "."
 var defaultjar = "JmolApplet.jar"
@@ -104,9 +99,6 @@
 // then the user is presented with a warning and asked whether it is OK to change Jar files.
 // The default action, if the user just presses "OK" is to NOT allow the change. 
 // The user must type the word "yes" in the prompt box for the change to be approved.
-
-// For a simple change to the signed applet in the same original directory (if it's available),
-// you can use JMOLJAR=SIGNED on the URL.
 
 // If you don't want people to be able to switch in their own JAR file on your page,
 // simply set this next line to read "var allowJMOLJAR = false".
@@ -135,7 +127,6 @@
         alert("The web page URL was ignored. Continuing using " + _jmol.archivePath + ' in directory "' + codebaseDirectory + '"');
       }
     } else {
-			if (f=="SIGNED") { f=true; }
       fileNameOrUseSignedApplet = f;
     }
   }
@@ -148,6 +139,10 @@
   _jmol.params.doTranslate = ''+TF;
 }
 
+function jmolToSigned(){
+    _jmolGetJarFilename(true);
+}
+    
 function _jmolGetJarFilename(fileNameOrFlag) {
   _jmol.archivePath =
     (typeof(fileNameOrFlag) == "string"  ? fileNameOrFlag : (fileNameOrFlag ?  "JmolAppletSigned" : "JmolApplet") + "0.jar");
@@ -494,26 +489,6 @@
   }
   if (typeof nowOrLater == "string" && nowOrLater.toLowerCase() == "now")
     _jmolCheckBrowser();
-}
-
-var _jmolScriptForSwitching;
-function jmolSwitchToSignedApplet(targetSuffix, additionalScript) {
-  if (!targetSuffix) { targetSuffix = "0"; }
-  if (!additionalScript) { additionalScript = ""; }
-  var s = jmolEvaluate("_signedApplet",targetSuffix);
-  var w = jmolEvaluate("_width",targetSuffix); 
-  var h = jmolEvaluate("_height",targetSuffix); 
-  if (s=="true") { 
-    jmolScript(additionalScript,targetSuffix);
-    return;
-  }
-  var appletParent = document.getElementById("jmolApplet"+targetSuffix).parentNode;
-  _jmolScriptForSwitching = jmolGetPropertyAsString("stateInfo", "", targetSuffix) + additionalScript;
-  appletParent.innerHTML = ""; 
-  _jmolGetJarFilename(true);
-  jmolSetDocument(false);
-  appletParent.innerHTML = jmolApplet([w,h], "javascript jmolScript(_jmolScriptForSwitching," + targetSuffix + ")", targetSuffix);
-  jmolSetDocument(document);
 }
 
 ////////////////////////////////////////////////////////////////
@@ -789,11 +764,11 @@
       params.mayscript = 'true';
       params.codebase = codebase;
       params.code = 'JmolApplet';
-      tHeader = "<span>" +
+      tHeader = 
         "<object name='jmolApplet" + nameSuffix +
         "' id='jmolApplet" + nameSuffix + "' " + appletCssText + "\n" +
 				widthAndHeight + "\n";
-      tFooter = "</object></span>";
+      tFooter = "</object>";
     }
     if (java_arguments)
       params.java_arguments = java_arguments;
@@ -810,14 +785,14 @@
 					Removing the classid parameter seems to be well tolerated by all browsers (even IE!).
 				*/
     } else { // use applet tag
-      tHeader = "<span>" +
+      tHeader = 
         "<applet name='jmolApplet" + nameSuffix +
         "' id='jmolApplet" + nameSuffix + "' " + appletCssText + "\n" +
 				widthAndHeight + "\n" +
         " code='JmolApplet'" +
         " archive='" + archivePath + "' codebase='" + codebase + "'\n" +
         " mayscript='true'>\n";
-      tFooter = "</applet></span>";
+      tFooter = "</applet>";
     }
     var visitJava;
     if (useIEObject || useHtml4Object) {
@@ -1510,29 +1485,6 @@
  return url
 }
 
-
-///////////////auto load NIH CACTVS data -- compound name or SMILES ///////////
-
-function jmolLoadAjax_STOLAF_NIH(compoundid,optionalscript,targetSuffix){
- _jmol.thismodel || (_jmol.thismodel = "aspirin")
- _jmol.serverURL || (_jmol.serverURL="http://fusion.stolaf.edu/chemistry/jmol/getajaxjs.cfm")
- _jmol.defaultURL_NIH || (_jmol.defaultURL_NIH="http://cactus.nci.nih.gov/chemical/structure/FILE/file?format=sdf&get3d=True")
- compoundid || (compoundid=prompt("Enter a compound name or a SMILES string:",_jmol.thismodel))
- if(!compoundid)return ""
- targetSuffix || (targetSuffix="0")
- optionalscript || (optionalscript="")
- var url=_jmol.defaultURL_NIH.replace(/FILE/g,compoundid)
- _jmol.optionalscript=optionalscript
- _jmol.thismodel=compoundid
- _jmol.thistargetsuffix=targetSuffix
- _jmol.thisurl=url
- _jmol.modelArray = []
- url=_jmol.serverURL+"?returnfunction=_jmolLoadModel&returnArray=_jmol.modelArray&id="+targetSuffix+_jmolExtractPostData(url)
- _jmolDomScriptLoad(url)
- return url
-}
-
-
 /////////////// St. Olaf College AJAX server -- ANY URL ///////////
 
 function jmolLoadAjax_STOLAF_ANY(url, userid, optionalscript,targetSuffix){
