--- /home/jonathan/Documents/jmol-12.3.27.p0/src/jmol/jmol
+++ /home/jonathan/Documents/jmol-12.3.27.p0/patches/jmol
@@ -1,16 +1,17 @@
 #!/bin/sh
+#JMOL_HOME=`dirname "$0"`
+JMOL_HOME="$SAGE_LOCAL/share/jmol"
+echo $JMOL_HOME
 
 # Collect -D & -m options as java arguments
 command=java
 while [ `echo $1 | egrep '^-D|^-m' | wc -l` != 0 ]; do
-	command="$command $1"
-	shift
+        command="$command $1"
+        shift
 done
 
 if [ -f ./Jmol.jar ] ; then
   jarpath=./Jmol.jar
-elif [ -f ./build/Jmol.jar ] ; then
-  jarpath=./build/Jmol.jar
 elif [ -f $JMOL_HOME/Jmol.jar ] ; then
   jarpath=$JMOL_HOME/Jmol.jar
 elif [ -f /usr/share/jmol/Jmol.jar ] ; then
