From 1fc5aecb4eca500917407b008c8c8eb8637a9c27 Mon Sep 17 00:00:00 2001
From: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date: Sat, 19 Nov 2022 19:03:37 -0800
Subject: [PATCH] CMakeLists.txt: Do not require boost program_options for the
 library

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7256877..db905aa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,7 +47,7 @@ if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE Release)
 endif()
 
-find_package(Boost ${BOOST_MIN_VERSION} COMPONENTS program_options REQUIRED)
+find_package(Boost ${BOOST_MIN_VERSION} REQUIRED)
 
 if(GMP)
     find_package(GMP)
-- 
2.37.3

