From e34cfbe0b6d6b4c6e8d5d9305046a00821c81828 Mon Sep 17 00:00:00 2001
From: Alexander Hoen <hoen@zib.de>
Date: Wed, 30 Nov 2022 14:05:56 +0100
Subject: [PATCH] import memory in ProbingView.hpp

---
 src/papilo/core/ProbingView.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/papilo/core/ProbingView.hpp b/src/papilo/core/ProbingView.hpp
index f977493b..1425e978 100644
--- a/src/papilo/core/ProbingView.hpp
+++ b/src/papilo/core/ProbingView.hpp
@@ -27,6 +27,7 @@
 #include "papilo/core/Problem.hpp"
 #include "papilo/core/SingleRow.hpp"
 #include "papilo/io/Message.hpp"
+#include <memory>
 #include "papilo/misc/MultiPrecision.hpp"
 #include "papilo/misc/Vec.hpp"
 
