commit 855bd67a64c5c044080d790e5c9fb7e298a61ab7
Author: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date:   Thu Dec 8 18:10:29 2022 -0800

    src/papilo/misc/MultiPrecision.hpp: Add #include <memory>

diff --git a/src/papilo/misc/MultiPrecision.hpp b/src/papilo/misc/MultiPrecision.hpp
index 669014c..44d3e63 100644
--- a/src/papilo/misc/MultiPrecision.hpp
+++ b/src/papilo/misc/MultiPrecision.hpp
@@ -26,6 +26,9 @@
 
 #include "papilo/Config.hpp"
 
+// work around build failure with boost on Fedora 37
+#include <memory>
+
 #include <boost/serialization/split_free.hpp>
 
 #ifdef PAPILO_HAVE_FLOAT128
