--- src/src/Lcommandline_elliptic.cc	2010-01-31 16:16:45.000000000 +0100
+++ src/src/Lcommandline_elliptic.cc	2011-05-10 17:08:10.000000000 +0200
@@ -121,11 +121,11 @@
 
 
     F = cgetg(6, t_VEC);
-    F[1] = lgeti(BIGDEFAULTPREC);
-    F[2] = lgeti(BIGDEFAULTPREC);
-    F[3] = lgeti(BIGDEFAULTPREC);
-    F[4] = lgeti(BIGDEFAULTPREC);
-    F[5] = lgeti(BIGDEFAULTPREC);
+    F[1] = (long)cgeti(BIGDEFAULTPREC);
+    F[2] = (long)cgeti(BIGDEFAULTPREC);
+    F[3] = (long)cgeti(BIGDEFAULTPREC);
+    F[4] = (long)cgeti(BIGDEFAULTPREC);
+    F[5] = (long)cgeti(BIGDEFAULTPREC);
 
     //gaffsg(a1,(GEN) F[1]);
     //gaffsg(a2,(GEN) F[2]);
