--- src/include/Lcommandline_numbertheory.h	2010-01-31 07:16:45.000000000 -0800
+++ src/include/Lcommandline_numbertheory.h	2010-08-31 18:53:18.000000000 -0700
@@ -24,6 +24,7 @@
 #ifndef Lcommandline_numbertheory_H
 #define Lcommandline_numbertheory_H
 
+#include <time.h>
 #include <stdlib.h>     //for things like srand
 #include <iostream>     //for input and output
 #include <iomanip>      //for manipulating output such as setprecision
