diff -ruN a/python/CMakeLists.txt b/python/CMakeLists.txt
--- a/python/CMakeLists.txt	2018-05-16 15:45:51.351070200 +0200
+++ b/python/CMakeLists.txt	2018-05-16 15:46:55.471242500 +0200
@@ -14,7 +14,7 @@
 
 add_custom_target(python_interface ALL DEPENDS ${OUTPUT}/timestamp)
 
-install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ${SETUP_PY} install --record files.txt)")
+install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ${SETUP_PY} install --root=\$ENV{DESTDIR} --record files.txt)")
 
 if (ENABLE_TESTING)
     add_test (NAME pytest
