Index: src/Makefile
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ src/Makefile	2008-02-08 15:22:00.000000000 -0800
@@ -0,0 +1,14 @@
+CFLAGS = -O2 -I/usr/include/pari
+LDFLAGS = -lpari
+CC = gcc
+
+genus2reduction:
+	${CC} ${CFLAGS} ${LDFLAGS} -o genus2reduction genus2reduction.c
+
+install: genus2reduction
+	mkdir -p ${DESTDIR}/usr/lib
+	install -p $< ${DESTDIR}/usr/lib
+
+clean:
+	rm -f genus2reduction
+
