--- src/src/kernel/memory/givaromm.h.orig	2008-04-14 15:04:37.000000000 -0700
+++ src/src/kernel/memory/givaromm.h	2008-04-14 15:05:46.000000000 -0700
@@ -12,6 +12,7 @@
 //   of blocs, the second one implement a reference mecanism on
 //   the bloc. The latter used method of the former.
 
+#include <cstring>            // gcc 4.3
 #include <stddef.h>           // size_t definition 
 #include <iostream>         // ostream definition
 #include <new>
