--- givtablelimits.h.orig	2008-05-18 10:09:46.000000000 -0700
+++ givtablelimits.h	2008-05-16 16:34:17.000000000 -0700
@@ -36,6 +36,11 @@
 #include "givaro/givprimes16.h"
 
 #include <math.h>
+/* SAGE: Added because for some reason this is 
+   missing from /usr/include/math.h, even though
+   it *should* be there.  -- William Stein*/
+extern double logb _PARAMS((double));
+
 #include <stddef.h>
 
   // ---------------------------------------------  class 
