--- gmp++_int.inl.orig	2010-04-30 09:48:10.000000000 +0200
+++ gmp++_int.inl	2010-04-30 09:47:37.000000000 +0200
@@ -314,8 +314,8 @@
 
 #ifdef __GMP_PLUSPLUS__
 inline gmp_randclass& Integer::randstate(long unsigned int seed) {
-	static gmp_randclass randstate(GMP_RAND_ALG_DEFAULT,seed);
-	return static_cast<gmp_randclass&>(randstate);
+	static gmp_randclass* randstate = new gmp_randclass(GMP_RAND_ALG_DEFAULT,seed);
+	return *randstate;
 }
 
 inline void Integer::seeding(long unsigned int s) {
