commit 0c2527842fe4eaed8d9e5107d6a6621b3d6a716f
Author: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date:   Sat Nov 26 17:08:40 2022 -0800

    CMakeLists.txt, src/CMakeLists.txt: Remove hardcoded RPATH settings

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4f8635fc..6f7a6e1b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -64,9 +64,6 @@ if(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE Release)
 endif()
 
-# set the correct rpath for OS X
-set(CMAKE_MACOSX_RPATH ON)
-
 # use C++14 standard
 set(CMAKE_CXX_STANDARD 14)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 27d52f14..fd3705ed 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -3,8 +3,7 @@
 #
 function(setLibProperties targetname outputname)
     set_target_properties(${targetname} PROPERTIES
-        OUTPUT_NAME ${outputname}
-        MACOSX_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+        OUTPUT_NAME ${outputname})
 endfunction(setLibProperties)
 
 include_directories(BEFORE ${CMAKE_CURRENT_SOURCE_DIR})
@@ -200,9 +199,6 @@ endif()
 add_executable(example EXCLUDE_FROM_ALL example.cpp)
 target_link_libraries(example libsoplex)
 
-# set the install rpath to the installed destination
-set_target_properties(soplex PROPERTIES INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
-
 # install the header files of soplex
 install(FILES ${headers} ${PROJECT_BINARY_DIR}/soplex/config.h DESTINATION include/soplex)
 install(FILES soplex.h soplex.hpp soplex_interface.h DESTINATION include)
