diff -ru src/src/nullspace.c b/src/nullspace.c
--- src/src/nullspace.c	2008-06-16 00:26:25.000000000 +0200
+++ b/src/nullspace.c	2013-06-12 14:28:39.382392511 +0200
@@ -109,7 +109,7 @@
 	    for (i = 0; i < m; i++) {
 		for (j = 0; j < m; j++)
 		    mpz_init_set_ui(mp_N[i * m + j], 0);
-		mpz_init_set_ui(mp_N[i * m + i], 1);
+		mpz_set_ui(mp_N[i * m + i], 1);
 	    }
 	    *mp_N_pass = mp_N;
 	} else {		/* r>0 and s>0 */
@@ -253,7 +253,7 @@
 	    for (i = 0; i < m; i++) {
 		for (j = 0; j < m; j++)
 		    mpz_init_set_ui(mp_N[i * m + j], 0);
-		mpz_init_ui(mp_N[i * m + i], 1);
+		mpz_set_ui(mp_N[i * m + i], 1);
 	    }
 	    *mp_N_pass = mp_N;
 	} else {		/* r>0 and s>0 */
@@ -400,7 +400,7 @@
 	    for (i = 0; i < m; i++) {
 		for (j = 0; j < m; j++)
 		    mpz_init_set_ui(mp_N[i * m + j], 0);
-		mpz_init_set_ui(mp_N[i * m + i], 1);
+		mpz_set_ui(mp_N[i * m + i], 1);
 	    }
 	    *mp_N_pass = mp_N;
 	} else {		/* r>0 and s>0 */
