diff -ru src/Makefile.in b/Makefile.in
--- src/Makefile.in	2006-11-26 15:45:52.000000000 +0100
+++ b/Makefile.in	2013-06-11 00:11:55.562405457 +0200
@@ -177,7 +177,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 EXTRA_DIST = bootstrap
-SUBDIRS = repl config src doc tests examples
+SUBDIRS = config src doc tests examples
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
diff -ru src/src/Makefile.in b/src/Makefile.in
--- src/src/Makefile.in	2006-11-26 15:45:53.000000000 +0100
+++ b/src/Makefile.in	2013-06-11 00:12:39.062405430 +0200
@@ -63,7 +63,7 @@
 am__DEPENDENCIES_1 =
 am__DEPENDENCIES_2 = $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 libiml_la_DEPENDENCIES = $(am__DEPENDENCIES_2) \
-	$(top_builddir)/repl/librepl.la
+	
 am_libiml_la_OBJECTS = libiml_la-basisop.lo libiml_la-certsolve.lo \
 	libiml_la-error.lo libiml_la-latreduce.lo \
 	libiml_la-memalloc.lo libiml_la-mtrans.lo \
@@ -221,7 +221,7 @@
 
 libiml_la_CFLAGS = $(AM_CFLAGS)
 libiml_la_LIBADD = $(EXTERNLIB) \
-	 	   $(top_builddir)/repl/librepl.la
+	 	   
 
 all: all-am
 
