diff -ruN a/src/nonsysolve.c b/src/nonsysolve.c
--- a/src/nonsysolve.c	2006-11-23 22:25:23.000000000 +0100
+++ b/src/nonsysolve.c	2013-06-10 23:07:48.882393429 +0200
@@ -188,7 +188,7 @@
 #endif
 
     /* lifting kincr more steps */
-    C1 = lift(solupos, kincr, n, m, basislen, extbasislen, mp_basisprod, \
+    C1 = iml_lift(solupos, kincr, n, m, basislen, extbasislen, mp_basisprod, \
 	      mp_extbasisprod, liftbasis, cmbasis, extbdcoeff, liftbasisInv, \
 	      mp_r, extbasis, AInv, ARNS);
 
@@ -396,7 +396,7 @@
   C = NULL;
   do {
     /* lifting kincr more steps */
-    C1 = lift(solupos, kincr, n, m, basislen, extbasislen, mp_basisprod, \
+    C1 = iml_lift(solupos, kincr, n, m, basislen, extbasislen, mp_basisprod, \
 	      mp_extbasisprod, liftbasis, cmbasis, extbdcoeff, liftbasisInv, \
 	      mp_r, extbasis, AInv, ARNS);
 
@@ -617,7 +617,7 @@
 #endif
 
     /* lifting kincr more steps */
-    C1 = lift(solupos, kincr, n, m, liftbasislen, extbasislen, \
+    C1 = iml_lift(solupos, kincr, n, m, liftbasislen, extbasislen, \
 	      mp_liftbasisprod,  mp_extbasisprod, liftbasis, cmliftbasis, \
 	      extbdcoeff, liftbasisInv, mp_r, extbasis, AInv, AExtRNS);
 
diff -ruN a/src/padiclift.c b/src/padiclift.c
--- a/src/padiclift.c	2006-11-23 22:25:23.000000000 +0100
+++ b/src/padiclift.c	2013-06-10 23:07:48.882393429 +0200
@@ -446,7 +446,7 @@
  */
 
 Double ***
-lift (const enum SOLU_POS solupos, const long k, const long n, \
+iml_lift (const enum SOLU_POS solupos, const long k, const long n, \
       const long m, const long liftbasislen, const long extbasislen, \
       const mpz_t mp_basisprod, const mpz_t mp_extbasisprod, \
       const FiniteField *liftbasis, const FiniteField *cmbasis, \
diff -ruN a/src/padiclift.h b/src/padiclift.h
--- a/src/padiclift.h	2006-11-23 22:25:24.000000000 +0100
+++ b/src/padiclift.h	2013-06-10 23:07:48.882393429 +0200
@@ -95,7 +95,7 @@
 		 Double **AInv, FiniteField ***extbasis, Double ***AExtRNS);
 
 
-Double*** lift(const enum SOLU_POS solupos, const long k, const long n, \
+Double*** iml_lift(const enum SOLU_POS solupos, const long k, const long n, \
 	       const long m, const long liftbasislen, const long extbasislen, \
 	       const mpz_t mp_basisprod, const mpz_t mp_extbasisprod, \
 	       const FiniteField *liftbasis, const FiniteField *cmbasis, \
