diff -dru src.orig/configure src/configure
--- src.orig/configure	2012-12-04 10:54:58.329999299 +0100
+++ src/configure	2012-12-04 11:08:46.105968510 +0100
@@ -4689,12 +4689,10 @@
 
     # 32bit apple darwin doesn't like our PIC format asm code
     case $host in
-	core2-apple-darwin* | penryn-apple-darwin*)		path="x86/applenopic/core2 x86/applenopic" ;;
-	prescott-apple-darwin* | pentium4-apple-darwin*)	path="x86/applenopic" ;;
-	pentium3-apple-darwin* | pentium2-apple-darwin*)	path="x86/applenopic" ;;
-	i686-apple-darwin* | pentiumpro-apple-darwin*)		path="x86/applenopic" ;;
-	core-apple-darwin*)					path="x86/applenopic" ;;
-	*)							;;
+	i[34567]86-apple-darwin* | pentium*-apple-darwin* | prescott-apple-darwin* | core-apple-darwin* )	path="x86/applenopic" ;;
+	# assume Core2 or later
+	*-apple-darwin* )	path="x86/applenopic/core2 x86/applenopic" ;;
+	*)			;;
     esac
 
     # If the user asked for a fat build, override the path set above
diff -dru src.orig/configure.in src/configure.in
--- src.orig/configure.in	2012-12-04 10:54:56.681999359 +0100
+++ src/configure.in	2012-12-04 11:09:04.321968955 +0100
@@ -1122,12 +1122,10 @@
 
     # 32bit apple darwin doesn't like our PIC format asm code
     case $host in
-	core2-apple-darwin* | penryn-apple-darwin*)		path="x86/applenopic/core2 x86/applenopic" ;;
-	prescott-apple-darwin* | pentium4-apple-darwin*)	path="x86/applenopic" ;;
-	pentium3-apple-darwin* | pentium2-apple-darwin*)	path="x86/applenopic" ;;
-	i686-apple-darwin* | pentiumpro-apple-darwin*)		path="x86/applenopic" ;;
-	core-apple-darwin*)					path="x86/applenopic" ;;
-	*)							;;
+	i[34567]86-apple-darwin* | pentium*-apple-darwin* | prescott-apple-darwin* | core-apple-darwin* )	path="x86/applenopic" ;;
+	# assume Core2 or later
+	*-apple-darwin* )	path="x86/applenopic/core2 x86/applenopic" ;;
+	*)			;;
     esac
 
     # If the user asked for a fat build, override the path set above
