diff -ru src/acinclude.m4 b/acinclude.m4
--- src/acinclude.m4	2012-10-23 16:56:46.000000000 +0200
+++ b/acinclude.m4	2013-10-02 09:22:41.958758575 +0200
@@ -480,29 +480,6 @@
 # first see a simple "main()" works, then go on to other checks
 GMP_PROG_CC_WORKS_PART([$1], [])
 
-GMP_PROG_CC_WORKS_PART_MAIN([$1], [gcc-4.3.2 on 64bit is bad , try -O1 or -fno-strict-aliasing for the flags],
-[/* The following aborts with gcc-4.3.2 on a 64bit system which is an unusable compiler */
-#if defined(__GNUC__) && !defined(__INTEL_COMPILER)
-int __attribute__((noinline))
-foo(int i)
-{
-  int *p = __builtin_malloc (4 * sizeof(int));
-  *p = 0;
-  p[i] = 1;
-  return *p;
-}
-extern void abort (void);
-int main()
-{
-   if (foo(0) != 1)
-        abort ();
-        return 0;
-}
-#else
-int main(){return 0;}
-#endif
-])
-
 GMP_PROG_CC_WORKS_PART([$1], [function pointer return],
 [/* The following provokes an internal error from gcc 2.95.2 -mpowerpc64
    (without -maix64), hence detecting an unusable compiler */
@@ -570,6 +547,10 @@
 
 #ifdef __GNUC__
 typedef unsigned long long t1;typedef t1*t2;
+/* Clang fails to link this on XCode 5 unless static is added */
+#ifdef __clang__
+static
+#endif
 __inline__ t1 e(t2 rp,t2 up,int n,t1 v0)
 {t1 c,x,r;int i;if(v0){c=1;for(i=1;i<n;i++){x=up[i];r=x+1;rp[i]=r;}}return c;}
 f(){static const struct{t1 n;t1 src[9];t1 want[9];}d[]={{1,{0},{1}},};t1 got[9];int i;
diff -ru src/configure b/configure
--- src/configure	2012-11-08 23:13:27.000000000 +0100
+++ b/configure	2013-10-02 09:22:56.458758077 +0200
@@ -5131,74 +5131,6 @@
   # remove anything that might look like compiler output to our "||" expression
   rm -f conftest* a.out b.out a.exe a_out.exe
   cat >conftest.c <<EOF
-/* The following aborts with gcc-4.3.2 on a 64bit system which is an unusable compiler */
-#if defined(__GNUC__) && !defined(__INTEL_COMPILER)
-int __attribute__((noinline))
-foo(int i)
-{
-  int *p = __builtin_malloc (4 * sizeof(int));
-  *p = 0;
-  p[i] = 1;
-  return *p;
-}
-extern void abort (void);
-int main()
-{
-   if (foo(0) != 1)
-        abort ();
-        return 0;
-}
-#else
-int main(){return 0;}
-#endif
-
-EOF
-  echo "Test compile: gcc-4.3.2 on 64bit is bad , try -O1 or -fno-strict-aliasing for the flags" >&5
-  gmp_compile="$cc $cflags $cppflags conftest.c >&5"
-  if { { eval echo "\"\$as_me\":${as_lineno-$LINENO}: \"$gmp_compile\""; } >&5
-  (eval $gmp_compile) 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-    cc_works_part=yes
-    if test "$cross_compiling" = no; then
-      if { ac_try='./a.out || ./b.out || ./a.exe || ./a_out.exe || ./conftest'
-  { { eval echo "\"\$as_me\":${as_lineno-$LINENO}: \"$ac_try\""; } >&5
-  (eval $ac_try) 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; }; then :;
-      else
-        cc_works_part=norun
-      fi
-    fi
-  else
-    cc_works_part=no
-  fi
-  if test "$cc_works_part" != yes; then
-    echo "failed program was:" >&5
-    cat conftest.c >&5
-  fi
-  rm -f conftest* a.out b.out a.exe a_out.exe
-  case $cc_works_part in
-    yes)
-
-      ;;
-    no)
-      gmp_prog_cc_works="no, gcc-4.3.2 on 64bit is bad , try -O1 or -fno-strict-aliasing for the flags"
-      ;;
-    norun)
-      gmp_prog_cc_works="no, gcc-4.3.2 on 64bit is bad , try -O1 or -fno-strict-aliasing for the flags, program does not run"
-      ;;
-  esac
-fi
-
-
-
-if test "$gmp_prog_cc_works" = yes; then
-  # remove anything that might look like compiler output to our "||" expression
-  rm -f conftest* a.out b.out a.exe a_out.exe
-  cat >conftest.c <<EOF
 /* The following provokes an internal error from gcc 2.95.2 -mpowerpc64
    (without -maix64), hence detecting an unusable compiler */
 void *g() { return (void *) 0; }
@@ -5590,6 +5522,10 @@
 
 #ifdef __GNUC__
 typedef unsigned long long t1;typedef t1*t2;
+/* Clang fails to link this on XCode 5 unless static is added */
+#ifdef __clang__
+static
+#endif
 __inline__ t1 e(t2 rp,t2 up,int n,t1 v0)
 {t1 c,x,r;int i;if(v0){c=1;for(i=1;i<n;i++){x=up[i];r=x+1;rp[i]=r;}}return c;}
 f(){static const struct{t1 n;t1 src[9];t1 want[9];}d[]={{1,{0},{1}},};t1 got[9];int i;
@@ -6392,74 +6328,6 @@
   # remove anything that might look like compiler output to our "||" expression
   rm -f conftest* a.out b.out a.exe a_out.exe
   cat >conftest.c <<EOF
-/* The following aborts with gcc-4.3.2 on a 64bit system which is an unusable compiler */
-#if defined(__GNUC__) && !defined(__INTEL_COMPILER)
-int __attribute__((noinline))
-foo(int i)
-{
-  int *p = __builtin_malloc (4 * sizeof(int));
-  *p = 0;
-  p[i] = 1;
-  return *p;
-}
-extern void abort (void);
-int main()
-{
-   if (foo(0) != 1)
-        abort ();
-        return 0;
-}
-#else
-int main(){return 0;}
-#endif
-
-EOF
-  echo "Test compile: gcc-4.3.2 on 64bit is bad , try -O1 or -fno-strict-aliasing for the flags" >&5
-  gmp_compile="$cc $cflags $cppflags $flag conftest.c >&5"
-  if { { eval echo "\"\$as_me\":${as_lineno-$LINENO}: \"$gmp_compile\""; } >&5
-  (eval $gmp_compile) 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-    cc_works_part=yes
-    if test "$cross_compiling" = no; then
-      if { ac_try='./a.out || ./b.out || ./a.exe || ./a_out.exe || ./conftest'
-  { { eval echo "\"\$as_me\":${as_lineno-$LINENO}: \"$ac_try\""; } >&5
-  (eval $ac_try) 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; }; then :;
-      else
-        cc_works_part=norun
-      fi
-    fi
-  else
-    cc_works_part=no
-  fi
-  if test "$cc_works_part" != yes; then
-    echo "failed program was:" >&5
-    cat conftest.c >&5
-  fi
-  rm -f conftest* a.out b.out a.exe a_out.exe
-  case $cc_works_part in
-    yes)
-
-      ;;
-    no)
-      gmp_prog_cc_works="no, gcc-4.3.2 on 64bit is bad , try -O1 or -fno-strict-aliasing for the flags"
-      ;;
-    norun)
-      gmp_prog_cc_works="no, gcc-4.3.2 on 64bit is bad , try -O1 or -fno-strict-aliasing for the flags, program does not run"
-      ;;
-  esac
-fi
-
-
-
-if test "$gmp_prog_cc_works" = yes; then
-  # remove anything that might look like compiler output to our "||" expression
-  rm -f conftest* a.out b.out a.exe a_out.exe
-  cat >conftest.c <<EOF
 /* The following provokes an internal error from gcc 2.95.2 -mpowerpc64
    (without -maix64), hence detecting an unusable compiler */
 void *g() { return (void *) 0; }
@@ -6851,6 +6719,10 @@
 
 #ifdef __GNUC__
 typedef unsigned long long t1;typedef t1*t2;
+/* Clang fails to link this on XCode 5 unless static is added */
+#ifdef __clang__
+static
+#endif
 __inline__ t1 e(t2 rp,t2 up,int n,t1 v0)
 {t1 c,x,r;int i;if(v0){c=1;for(i=1;i<n;i++){x=up[i];r=x+1;rp[i]=r;}}return c;}
 f(){static const struct{t1 n;t1 src[9];t1 want[9];}d[]={{1,{0},{1}},};t1 got[9];int i;
