# porting to gcc 4.9; see #18240
--- src/gmp-h.in
+++ src/gmp-h.in
@@ -29,6 +29,8 @@ along with the GNU MP Library.  If not, see http://www.gnu.org/licenses/.  */
 #endif 
 
 #if defined (__cplusplus)
+
+#include <cstddef>     /* for size_t */
 #include <iosfwd>   /* for std::istream, std::ostream, std::string */
 #include <cstdio>
 #endif
@@ -50,9 +52,7 @@ along with the GNU MP Library.  If not, see http://www.gnu.org/licenses/.  */
 #define __GNU_MP__ 4
 
 #define __need_size_t  /* tell gcc stddef.h we only want size_t */
-#if defined (__cplusplus)
-#include <cstddef>     /* for size_t */
-#else
+#if ! defined (__cplusplus)
 #include <stddef.h>    /* for size_t */
 #endif
 #undef __need_size_t
--- src/mpirxx.h
+++ src/mpirxx.h
@@ -32,6 +32,8 @@ along with the GNU MP Library.  If not, see http://www.gnu.org/licenses/.  */
 #ifndef __GMP_PLUSPLUS__
 #define __GMP_PLUSPLUS__
 
+#include <cstddef>     /* for size_t */
+
 #include <iosfwd>
 
 #include <cstring>  /* for strlen */
