--- ../src/gmp-h.in	2010-06-24 03:35:47.000000000 -0700
+++ gmp-h.in	2010-08-16 22:04:40.000000000 -0700
@@ -22,6 +22,11 @@
 
 #ifndef __GMP_H__
 
+#ifdef __SUNPRO_CC    /* See: http://trac.sagemath.org/sage_trac/ticket/7849 */
+#include <stddef.h>   /* This is Bill Hart's fix, but I've applied it only */
+#include <stdarg.h>   /* on Sun Studio */
+#endif 
+
 #if defined (__cplusplus)
 #include <iosfwd>   /* for std::istream, std::ostream, std::string */
 #include <cstdio>
