--- src/gmp-h.in	Mon Nov 16 15:42:38 2009
+++ patches/gmp-h.in	Tue Jan  5 20:28:08 2010
@@ -24,6 +24,11 @@
 
 #ifndef __GMP_H__
 
+#ifdef __SUNPRO_CC    /* See: http://trac.sagemath.org/sage_trac/ticket/7849 */
+#include <stddef.h>   /* This is Bill Hart's fix, but I've applied it only */
+#include <stdarg.h>   /* on Sun Studio */
+#endif 
+
 #if defined (__cplusplus)
 #include <iosfwd>   /* for std::istream, std::ostream, std::string */
 #include <cstdio>
