diff -ru src/gmp-h.in src.patched/gmp-h.in
--- src/gmp-h.in	2011-06-14 07:33:49.000000000 +0200
+++ src.patched/gmp-h.in	2012-04-07 14:32:53.758280009 +0200
@@ -22,6 +22,11 @@
 
 #ifndef __GMP_H__
 
+#ifdef __SUNPRO_CC    /* See: http://trac.sagemath.org/sage_trac/ticket/7849 */
+#include <stddef.h>   /* This is Bill Hart's fix, but I've applied it only */
+#include <stdarg.h>   /* on Sun Studio */
+#endif 
+
 #if defined (__cplusplus)
 #include <iosfwd>   /* for std::istream, std::ostream, std::string */
 #include <cstdio>
