--- mpir-2.6.0/tests/mpz/t-scan.c	2012-10-03 22:07:32.000000000 +0200
+++ mpir-2.6.0/tests/mpz/t-scan.c	2013-04-02 18:39:17.704366985 +0200
@@ -84,7 +84,7 @@
 
               for (isize = 0; isize <= size; isize++)
                 {
-                  for (oindex = 0; oindex <= numberof (offset); oindex++)
+                  for (oindex = 0; oindex < numberof (offset); oindex++)
                     {
                       o = offset[oindex];
                       if ((int) isize*GMP_NUMB_BITS < -o)
