diff --git a/src/set_d.c b/src/set_d.c
index c076ccf..5fc9548 100644
--- a/src/set_d.c
+++ b/src/set_d.c
@@ -118,8 +118,15 @@ __gmpfr_extract_double (mpfr_limb_ptr rp, double d)
 
     d *= MP_BASE_AS_DOUBLE;
 #if GMP_NUMB_BITS >= 64
+#ifndef __clang__
     manl = d;
 #else
+    /* clang produces an invalid exception when d >= 2^63,
+       see https://bugs.llvm.org//show_bug.cgi?id=17686.
+       Since this is always the case, here, we use the following patch. */
+    manl = 0x8000000000000000 + (mp_limb_t) (d - 0x8000000000000000);
+#endif /* __clang__ */
+#else
     manh = (mp_limb_t) d;
     manl = (mp_limb_t) ((d - manh) * MP_BASE_AS_DOUBLE);
 #endif
