--- trunk/mpfr-gmp.h	2008/01/01 03:26:37	5151
+++ trunk/mpfr-gmp.h	2008/01/14 17:22:35	5186
@@ -172,12 +172,6 @@
 #undef  XDEBUG
 #define XDEBUG
 
-/* Temp memory allocate */
-#define TMP_DECL(m)
-#define TMP_ALLOC(x) alloca(x)
-#define TMP_MARK(m)
-#define TMP_FREE(m)
-
 /* For longlong.h */
 #ifdef HAVE_ATTRIBUTE_MODE
 typedef unsigned int UQItype    __attribute__ ((mode (QI)));
@@ -297,6 +291,24 @@
                                                              size_t));
 __MPFR_DECLSPEC void __gmp_default_free _MPFR_PROTO ((void *, size_t));
 
+/* Temp memory allocate */
+
+struct tmp_marker
+{
+  void *ptr;
+  size_t size;
+  struct tmp_marker *next;
+};
+
+__MPFR_DECLSPEC void *mpfr_tmp_allocate _MPFR_PROTO ((struct tmp_marker **,
+                                                      size_t));
+__MPFR_DECLSPEC void mpfr_tmp_free _MPFR_PROTO ((struct tmp_marker *));
+
+#define TMP_DECL(m) struct tmp_marker *tmp_marker
+#define TMP_ALLOC(x) mpfr_tmp_allocate (&tmp_marker, (x))
+#define TMP_MARK(m) tmp_marker = 0;
+#define TMP_FREE(m) mpfr_tmp_free (tmp_marker)
+
 #if defined (__cplusplus)
 }
 #endif
