--- src/src/mpn_exp.c	2011-10-03 10:17:09.000000000 +0200
+++ patches/mpn_exp.c	2012-02-25 15:48:28.000000000 +0100
@@ -70,7 +70,18 @@
   /* allocate space for A and set it to B */
   c = MPFR_TMP_LIMBS_ALLOC (2 * n);
   a [n - 1] = B;
-  MPN_ZERO (a, n - 1);
+  /* The following couple of lines work around a bug seen so far only on
+  a Sun T5240 with a pair of T2+ processors. It only happens with some versions
+  of gcc, so the patch will only be used on sun4v machines by default, but
+  and enviroment variable will allow the patch to be included or not. 
+  There would be an advantage in including it even on sun4u machines, 
+  if it was felt the binary might be used on sun4v.
+
+  The patch to the C code was developed by Paul Zimmermann */
+
+  if (n > 1)
+    MPN_ZERO (a, n - 1);
+
   /* initial exponent for A: invariant is A = {a, n} * 2^f */
   f = h - (n - 1) * GMP_NUMB_BITS;
 
