diff --git a/linbox/matrix/densematrix/blas-transposed-matrix.h b/linbox/matrix/densematrix/blas-transposed-matrix.h
index b81e659..d9c0314 100644
--- a/linbox/matrix/densematrix/blas-transposed-matrix.h
+++ b/linbox/matrix/densematrix/blas-transposed-matrix.h
@@ -67,9 +67,6 @@ namespace LinBox
 	/*! TransposedBlasMatrix.
 	 * NO DOC
 	 */
-#if !defined(__INTEL_COMPILER) && !defined(__CUDACC__) & !defined(__clang__)
-	template <>
-#endif
 	template< class Matrix >
 	class TransposedBlasMatrix< TransposedBlasMatrix< Matrix > > : public Matrix {
 
