diff -ur src.orig/interfaces/sage/linbox-sage.C src/interfaces/sage/linbox-sage.C
--- src.orig/interfaces/sage/linbox-sage.C	2012-08-24 11:43:55.000000000 +0200
+++ src/interfaces/sage/linbox-sage.C	2012-11-25 15:56:49.646931145 +0100
@@ -25,14 +25,11 @@
  */
 
 #include <iostream>
-#include <gmp.h>
 
 #include <cstdlib>
 #include <vector>
 #include <list>
 
-#include "linbox-sage.h"
-
 #include "linbox/util/commentator.h"
 
 #include "linbox/blackbox/sparse.h"
@@ -60,6 +57,10 @@
 #include "linbox/ring/givaro-polynomial.h"
 #include "linbox/field/modular.h"
 
+#include <gmp.h>
+
+#include "linbox-sage.h"
+
 using namespace LinBox;
 
 /*************************************************************************
diff -ur src.orig/interfaces/sage/linbox-sage.h src/interfaces/sage/linbox-sage.h
--- src.orig/interfaces/sage/linbox-sage.h	2012-08-24 11:43:55.000000000 +0200
+++ src/interfaces/sage/linbox-sage.h	2012-11-25 15:56:18.263226724 +0100
@@ -25,9 +25,9 @@
  */
 
 #ifndef __LINBOX_sage_H
-#define __LINBOX_sagE_H
+#define __LINBOX_sage_H
 
-#include<stddef.h>
+#include <stddef.h>
 
 #include <cstdlib>
 #include <vector>
