From e6f686d73778bbc7850df74dd0e14f8707fa845f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Bissey?= <francois.bissey@canterbury.ac.nz>
Date: Thu, 6 Apr 2017 10:29:40 +1200
Subject: [PATCH] Simplification in gf2.h headers. Be more standard conformant

---
 linbox/field/gf2.h | 24 +-----------------------
 1 file changed, 1 insertion(+), 23 deletions(-)

diff --git a/linbox/field/gf2.h b/linbox/field/gf2.h
index 08c14fd..8afc33c 100644
--- a/linbox/field/gf2.h
+++ b/linbox/field/gf2.h
@@ -41,11 +41,7 @@
 #include "linbox/field/field-traits.h"
 // #include "linbox/vector/vector-domain.h"
 
-#if !defined(__PATHCC__) && !(defined(__APPLE__) && defined(__clang__))
-#define stdBitReference std::_Bit_reference
-#else
 #define stdBitReference std::vector<bool>::reference
-#endif
 
 // Namespace in which all LinBox code resides
 namespace LinBox
@@ -1015,25 +1011,7 @@ namespace LinBox
 // Specialization of homomorphism for basefield
 #include "linbox/randiter/gf2.h"
 
-#if __cplusplus >= 201103L
-#if defined( __APPLE__) && defined(__clang__)
-#include <__bit_reference>
-#else
-#include <bits/stl_bvector.h>
-#endif /* __clang__ */
-#else
-// #include <bits/stl_bvector.h>
-{
-	//! @todo JGD 05.11.2009 : it should be in bits/stl_bvector.h  ...
-	inline void swap(stdBitReference __x, stdBitReference __y)
-	{
-		bool __tmp = __x;
-		__x = __y;
-		__y = __tmp;
-	}
-}
-#endif
-
+#include <vector>
 
 #include "linbox/field/gf2.inl"
 
