--- a/linbox/Makefile.in	2017-11-23 14:38:39.000000000 +0100
+++ b/linbox/Makefile.in	2017-12-04 22:10:27.000000000 +0100
@@ -473,6 +473,7 @@
 SUBDIRS = util element field randiter vector matrix blackbox algorithms solutions ring polynomial
 BASIC_HDRS = \
     linbox-config.h \
+    config.h \
     config-blas.h     \
     integer.h         \
     linbox-tags.h
--- a/linbox/Makefile.am	2017-11-17 16:07:54.000000000 +0100
+++ b/linbox/Makefile.am	2017-12-04 22:10:23.000000000 +0100
@@ -30,6 +30,7 @@
 
 BASIC_HDRS =          \
     linbox-config.h \
+    config.h \
     config-blas.h     \
     integer.h         \
     linbox-tags.h
