diff -ru linbox-1.1.6/linbox/field/modular-float.h src/linbox/field/modular-float.h
--- linbox-1.1.6/linbox/field/modular-float.h	2008-09-18 21:40:02.000000000 +0200
+++ src/linbox/field/modular-float.h	2012-03-05 14:25:31.118364037 +0100
@@ -38,6 +38,9 @@
 	template< class Element >
 	class ModularRandIter;
 
+	template< class Field, class RandIter >
+	class NonzeroRandIter;
+
 	template <class Ring>
 	struct ClassifyRing; 
 	template <class Element>
@@ -67,6 +70,7 @@
 			       
 		typedef float Element;
 		typedef ModularRandIter<float> RandIter;
+		typedef NonzeroRandIter<Modular<float>, ModularRandIter<float> > NonZeroRandIter;
 
 		static ClassifyRing<Modular<float> >::categoryTag getCategory() {return ClassifyRing<Modular<float> >::categoryTag();}
 		
