diff --git a/test.sh b/test.sh
index b4c95b1..e51b4bc 100755
--- a/test.sh
+++ b/test.sh
@@ -3354,6 +3354,10 @@ case $compiler_version in
     *clang*)
         COMPILER_TYPE_CLANG=true
         ;;
+    *Xcode.app*)
+# /usr/bin/gcc on OS X which is a front end to clang doesn't have gcc or clang in the first line.
+        COMPILER_TYPE_CLANG=true
+        ;;
     *)
         echo "WARNING: Compiler $COMPILER not supported (version: $compiler_version) -- not running tests" >&2
         exit 0
