diff --git a/config/TeX-AMS_HTML-full.js b/config/TeX-AMS_HTML-full.js
--- a/config/TeX-AMS_HTML-full.js
+++ b/config/TeX-AMS_HTML-full.js
@@ -34,4 +34,12 @@ MathJax.Ajax.Preloading(
 );
 
+//  This extends the limit on the size of the string being processed by MathJax
+//  (whose aim is to avoid infinite loops) up to 50KB, see
+//  http://docs.mathjax.org/en/latest/options/TeX.html#the-tex-input-processor
+//
+//  See trac ticket #18577.
+//
+MathJax.Hub.Config({TeX: {MAXBUFFER: 50*1024,},});
+
 MathJax.Hub.Config({"v1.0-compatible":false});
 
